{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "name": "Text Summarization using Transformers and Beautiful Soup.ipynb",
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "widgets": {
      "application/vnd.jupyter.widget-state+json": {
        "94d7f6443c4f41a08d2ce1b00d2d0239": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_7074724286ab424489436b3bf735343c",
              "IPY_MODEL_28883ec26ca34345b6003e76b4f5700d",
              "IPY_MODEL_4b8e83065d9541649bd1f46773ca032e"
            ],
            "layout": "IPY_MODEL_9fe6089462e041d5b35f0b6af302e2d3"
          }
        },
        "7074724286ab424489436b3bf735343c": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_822af8d84238489fa4b2512bd8f6c136",
            "placeholder": "​",
            "style": "IPY_MODEL_d6cfc91320394043b36545cd3092cc10",
            "value": "Downloading: 100%"
          }
        },
        "28883ec26ca34345b6003e76b4f5700d": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_516b3621258f468ba767f1247db219c4",
            "max": 1802,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_22d652460b804135821f70ee674aa741",
            "value": 1802
          }
        },
        "4b8e83065d9541649bd1f46773ca032e": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_689f574611614f7eab7effaa4c36c690",
            "placeholder": "​",
            "style": "IPY_MODEL_fd29156b19ee4d5a9702497a21e8e129",
            "value": " 1.76k/1.76k [00:00&lt;00:00, 28.6kB/s]"
          }
        },
        "9fe6089462e041d5b35f0b6af302e2d3": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "822af8d84238489fa4b2512bd8f6c136": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "d6cfc91320394043b36545cd3092cc10": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "516b3621258f468ba767f1247db219c4": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "22d652460b804135821f70ee674aa741": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "689f574611614f7eab7effaa4c36c690": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "fd29156b19ee4d5a9702497a21e8e129": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "4d80e3c7704148689d30e46c038e7983": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_7df5e9afcddf46bcbf3f50f1f6803507",
              "IPY_MODEL_e36f1b568aae4e2c804b12ac156814d4",
              "IPY_MODEL_8461a1fc97814070916b23d2723c6d26"
            ],
            "layout": "IPY_MODEL_6e3b6b2b3a5b46a3b78a145ff94be3d8"
          }
        },
        "7df5e9afcddf46bcbf3f50f1f6803507": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_e395464781cb42088bd7ff5aa1288ea2",
            "placeholder": "​",
            "style": "IPY_MODEL_6cca8a98d8e046a0bd397076fd887c8e",
            "value": "Downloading: 100%"
          }
        },
        "e36f1b568aae4e2c804b12ac156814d4": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_a281177b925d4e8998eaeb70a828cc84",
            "max": 1222317369,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_7c671eea484d4090b4f97812aa42705b",
            "value": 1222317369
          }
        },
        "8461a1fc97814070916b23d2723c6d26": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_f508d148c2e845919b4b1958a73bdb7b",
            "placeholder": "​",
            "style": "IPY_MODEL_cf65bea6d5624c23b3358080abea59a9",
            "value": " 1.14G/1.14G [00:27&lt;00:00, 44.1MB/s]"
          }
        },
        "6e3b6b2b3a5b46a3b78a145ff94be3d8": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "e395464781cb42088bd7ff5aa1288ea2": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "6cca8a98d8e046a0bd397076fd887c8e": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "a281177b925d4e8998eaeb70a828cc84": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "7c671eea484d4090b4f97812aa42705b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "f508d148c2e845919b4b1958a73bdb7b": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "cf65bea6d5624c23b3358080abea59a9": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "0b340932a75a4b289d6e12af4e631ed8": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_5068c5cff47d4ccb8a921d5c1a499259",
              "IPY_MODEL_4c387115a6af4520915597d919800368",
              "IPY_MODEL_a9b9cdaf9ac641d8bcc81a8571546f0b"
            ],
            "layout": "IPY_MODEL_8bc3010f91e240fd92dda20a4cd090b9"
          }
        },
        "5068c5cff47d4ccb8a921d5c1a499259": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_19601fd5d3e84fb58e3c125537a8f9c9",
            "placeholder": "​",
            "style": "IPY_MODEL_5c5c9364c6cf41d7ace388db408f7b22",
            "value": "Downloading: 100%"
          }
        },
        "4c387115a6af4520915597d919800368": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_9d056718d69146b895f30ec222c5486f",
            "max": 26,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_2390e5c19fcf44b09066e9fa7b76f4d7",
            "value": 26
          }
        },
        "a9b9cdaf9ac641d8bcc81a8571546f0b": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_91c91ee89df14185b275dacc61e0c9ce",
            "placeholder": "​",
            "style": "IPY_MODEL_8e3d7d26ded54a8099060fa331db81c5",
            "value": " 26.0/26.0 [00:00&lt;00:00, 481B/s]"
          }
        },
        "8bc3010f91e240fd92dda20a4cd090b9": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "19601fd5d3e84fb58e3c125537a8f9c9": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "5c5c9364c6cf41d7ace388db408f7b22": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "9d056718d69146b895f30ec222c5486f": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "2390e5c19fcf44b09066e9fa7b76f4d7": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "91c91ee89df14185b275dacc61e0c9ce": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8e3d7d26ded54a8099060fa331db81c5": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "8342e1f01f4647d18bccd8c8ea283c68": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_6e347a3a757e441981d9338858e3a073",
              "IPY_MODEL_c692cb913af64137adf447afe6c79204",
              "IPY_MODEL_217b7605bec3463e9e895956e226bc20"
            ],
            "layout": "IPY_MODEL_8235f292a7fa4a4994425d2eed230029"
          }
        },
        "6e347a3a757e441981d9338858e3a073": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_8bbb0c144cfb40f68e3751cd67f461eb",
            "placeholder": "​",
            "style": "IPY_MODEL_8a4ca829987d49c7af59d1e544d75481",
            "value": "Downloading: 100%"
          }
        },
        "c692cb913af64137adf447afe6c79204": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_3daa9a7987f54f62ac8903754d5e069a",
            "max": 898822,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_8c4f126be91247f187248b0722e8f979",
            "value": 898822
          }
        },
        "217b7605bec3463e9e895956e226bc20": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_f1373a9285434abea0e8e99d396140df",
            "placeholder": "​",
            "style": "IPY_MODEL_b78e76d8d7ea4762ad87fe0c26ecfcbe",
            "value": " 878k/878k [00:00&lt;00:00, 1.86MB/s]"
          }
        },
        "8235f292a7fa4a4994425d2eed230029": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8bbb0c144cfb40f68e3751cd67f461eb": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8a4ca829987d49c7af59d1e544d75481": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "3daa9a7987f54f62ac8903754d5e069a": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "8c4f126be91247f187248b0722e8f979": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "f1373a9285434abea0e8e99d396140df": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "b78e76d8d7ea4762ad87fe0c26ecfcbe": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "69c13d1bce6140c79e12fec80461b733": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HBoxModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HBoxModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HBoxView",
            "box_style": "",
            "children": [
              "IPY_MODEL_66137c9639da40f2b2ecd52f3d475e97",
              "IPY_MODEL_b0e957985ee7496cbe46f471acff1a20",
              "IPY_MODEL_7104f3be38724ebfa34f1547a5bea9ef"
            ],
            "layout": "IPY_MODEL_24040212d16c4c3a87bd7c71f7f9635d"
          }
        },
        "66137c9639da40f2b2ecd52f3d475e97": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_147e5239d3f2496e8394c388461f478e",
            "placeholder": "​",
            "style": "IPY_MODEL_7d247c6f30514a899242bfe2726ec949",
            "value": "Downloading: 100%"
          }
        },
        "b0e957985ee7496cbe46f471acff1a20": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "FloatProgressModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "FloatProgressModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "ProgressView",
            "bar_style": "success",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_e33909210fcc48af84e5cc67edd2b201",
            "max": 456318,
            "min": 0,
            "orientation": "horizontal",
            "style": "IPY_MODEL_a1f104ac907d4da594b6e08ae7c118fa",
            "value": 456318
          }
        },
        "7104f3be38724ebfa34f1547a5bea9ef": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "HTMLModel",
          "model_module_version": "1.5.0",
          "state": {
            "_dom_classes": [],
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "HTMLModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/controls",
            "_view_module_version": "1.5.0",
            "_view_name": "HTMLView",
            "description": "",
            "description_tooltip": null,
            "layout": "IPY_MODEL_ee8ec733c4f040f9bcd1c931f6e0e75d",
            "placeholder": "​",
            "style": "IPY_MODEL_cf60b34b2eec4d28a305b39a100b4ad9",
            "value": " 446k/446k [00:00&lt;00:00, 3.05MB/s]"
          }
        },
        "24040212d16c4c3a87bd7c71f7f9635d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "147e5239d3f2496e8394c388461f478e": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "7d247c6f30514a899242bfe2726ec949": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        },
        "e33909210fcc48af84e5cc67edd2b201": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "a1f104ac907d4da594b6e08ae7c118fa": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "ProgressStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "ProgressStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "bar_color": null,
            "description_width": ""
          }
        },
        "ee8ec733c4f040f9bcd1c931f6e0e75d": {
          "model_module": "@jupyter-widgets/base",
          "model_name": "LayoutModel",
          "model_module_version": "1.2.0",
          "state": {
            "_model_module": "@jupyter-widgets/base",
            "_model_module_version": "1.2.0",
            "_model_name": "LayoutModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "LayoutView",
            "align_content": null,
            "align_items": null,
            "align_self": null,
            "border": null,
            "bottom": null,
            "display": null,
            "flex": null,
            "flex_flow": null,
            "grid_area": null,
            "grid_auto_columns": null,
            "grid_auto_flow": null,
            "grid_auto_rows": null,
            "grid_column": null,
            "grid_gap": null,
            "grid_row": null,
            "grid_template_areas": null,
            "grid_template_columns": null,
            "grid_template_rows": null,
            "height": null,
            "justify_content": null,
            "justify_items": null,
            "left": null,
            "margin": null,
            "max_height": null,
            "max_width": null,
            "min_height": null,
            "min_width": null,
            "object_fit": null,
            "object_position": null,
            "order": null,
            "overflow": null,
            "overflow_x": null,
            "overflow_y": null,
            "padding": null,
            "right": null,
            "top": null,
            "visibility": null,
            "width": null
          }
        },
        "cf60b34b2eec4d28a305b39a100b4ad9": {
          "model_module": "@jupyter-widgets/controls",
          "model_name": "DescriptionStyleModel",
          "model_module_version": "1.5.0",
          "state": {
            "_model_module": "@jupyter-widgets/controls",
            "_model_module_version": "1.5.0",
            "_model_name": "DescriptionStyleModel",
            "_view_count": null,
            "_view_module": "@jupyter-widgets/base",
            "_view_module_version": "1.2.0",
            "_view_name": "StyleView",
            "description_width": ""
          }
        }
      }
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "Methodology: \n",
        "*   Use Beautiful Soup to web-scrape the article\n",
        "*   Use Huggingface Summarization pipeline\n",
        "*   Summarize the article chunk-wise\n",
        "*   Concatenate that to get a summary of the overall article\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "cJmNh4PMMwmP"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Loading important dependencies"
      ],
      "metadata": {
        "id": "pbufQEdgNaA-"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Mz9h9k5hMTP1",
        "outputId": "ace5259b-85c6-4244-af1e-fe5dd447cb56"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting transformers\n",
            "  Downloading transformers-4.18.0-py3-none-any.whl (4.0 MB)\n",
            "\u001b[K     |████████████████████████████████| 4.0 MB 5.1 MB/s \n",
            "\u001b[?25hCollecting huggingface-hub<1.0,>=0.1.0\n",
            "  Downloading huggingface_hub-0.5.1-py3-none-any.whl (77 kB)\n",
            "\u001b[K     |████████████████████████████████| 77 kB 5.7 MB/s \n",
            "\u001b[?25hRequirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.7/dist-packages (from transformers) (1.21.6)\n",
            "Requirement already satisfied: tqdm>=4.27 in /usr/local/lib/python3.7/dist-packages (from transformers) (4.64.0)\n",
            "Collecting tokenizers!=0.11.3,<0.13,>=0.11.1\n",
            "  Downloading tokenizers-0.12.1-cp37-cp37m-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (6.6 MB)\n",
            "\u001b[K     |████████████████████████████████| 6.6 MB 27.7 MB/s \n",
            "\u001b[?25hRequirement already satisfied: importlib-metadata in /usr/local/lib/python3.7/dist-packages (from transformers) (4.11.3)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.7/dist-packages (from transformers) (2.23.0)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.7/dist-packages (from transformers) (3.6.0)\n",
            "Requirement already satisfied: regex!=2019.12.17 in /usr/local/lib/python3.7/dist-packages (from transformers) (2019.12.20)\n",
            "Collecting sacremoses\n",
            "  Downloading sacremoses-0.0.49-py3-none-any.whl (895 kB)\n",
            "\u001b[K     |████████████████████████████████| 895 kB 63.8 MB/s \n",
            "\u001b[?25hRequirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.7/dist-packages (from transformers) (21.3)\n",
            "Collecting pyyaml>=5.1\n",
            "  Downloading PyYAML-6.0-cp37-cp37m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (596 kB)\n",
            "\u001b[K     |████████████████████████████████| 596 kB 48.4 MB/s \n",
            "\u001b[?25hRequirement already satisfied: typing-extensions>=3.7.4.3 in /usr/local/lib/python3.7/dist-packages (from huggingface-hub<1.0,>=0.1.0->transformers) (4.1.1)\n",
            "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /usr/local/lib/python3.7/dist-packages (from packaging>=20.0->transformers) (3.0.8)\n",
            "Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.7/dist-packages (from importlib-metadata->transformers) (3.8.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests->transformers) (2021.10.8)\n",
            "Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests->transformers) (2.10)\n",
            "Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests->transformers) (3.0.4)\n",
            "Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests->transformers) (1.24.3)\n",
            "Requirement already satisfied: joblib in /usr/local/lib/python3.7/dist-packages (from sacremoses->transformers) (1.1.0)\n",
            "Requirement already satisfied: click in /usr/local/lib/python3.7/dist-packages (from sacremoses->transformers) (7.1.2)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.7/dist-packages (from sacremoses->transformers) (1.15.0)\n",
            "Installing collected packages: pyyaml, tokenizers, sacremoses, huggingface-hub, transformers\n",
            "  Attempting uninstall: pyyaml\n",
            "    Found existing installation: PyYAML 3.13\n",
            "    Uninstalling PyYAML-3.13:\n",
            "      Successfully uninstalled PyYAML-3.13\n",
            "Successfully installed huggingface-hub-0.5.1 pyyaml-6.0 sacremoses-0.0.49 tokenizers-0.12.1 transformers-4.18.0\n"
          ]
        }
      ],
      "source": [
        "!pip install transformers"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from transformers import pipeline\n",
        "from bs4 import BeautifulSoup\n",
        "import requests"
      ],
      "metadata": {
        "id": "y-RfqzxbOWfm"
      },
      "execution_count": 2,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Summarization pipeline from Huggingface"
      ],
      "metadata": {
        "id": "zLb_yHWIQFvW"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "summarizer=pipeline('summarization')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 194,
          "referenced_widgets": [
            "94d7f6443c4f41a08d2ce1b00d2d0239",
            "7074724286ab424489436b3bf735343c",
            "28883ec26ca34345b6003e76b4f5700d",
            "4b8e83065d9541649bd1f46773ca032e",
            "9fe6089462e041d5b35f0b6af302e2d3",
            "822af8d84238489fa4b2512bd8f6c136",
            "d6cfc91320394043b36545cd3092cc10",
            "516b3621258f468ba767f1247db219c4",
            "22d652460b804135821f70ee674aa741",
            "689f574611614f7eab7effaa4c36c690",
            "fd29156b19ee4d5a9702497a21e8e129",
            "4d80e3c7704148689d30e46c038e7983",
            "7df5e9afcddf46bcbf3f50f1f6803507",
            "e36f1b568aae4e2c804b12ac156814d4",
            "8461a1fc97814070916b23d2723c6d26",
            "6e3b6b2b3a5b46a3b78a145ff94be3d8",
            "e395464781cb42088bd7ff5aa1288ea2",
            "6cca8a98d8e046a0bd397076fd887c8e",
            "a281177b925d4e8998eaeb70a828cc84",
            "7c671eea484d4090b4f97812aa42705b",
            "f508d148c2e845919b4b1958a73bdb7b",
            "cf65bea6d5624c23b3358080abea59a9",
            "0b340932a75a4b289d6e12af4e631ed8",
            "5068c5cff47d4ccb8a921d5c1a499259",
            "4c387115a6af4520915597d919800368",
            "a9b9cdaf9ac641d8bcc81a8571546f0b",
            "8bc3010f91e240fd92dda20a4cd090b9",
            "19601fd5d3e84fb58e3c125537a8f9c9",
            "5c5c9364c6cf41d7ace388db408f7b22",
            "9d056718d69146b895f30ec222c5486f",
            "2390e5c19fcf44b09066e9fa7b76f4d7",
            "91c91ee89df14185b275dacc61e0c9ce",
            "8e3d7d26ded54a8099060fa331db81c5",
            "8342e1f01f4647d18bccd8c8ea283c68",
            "6e347a3a757e441981d9338858e3a073",
            "c692cb913af64137adf447afe6c79204",
            "217b7605bec3463e9e895956e226bc20",
            "8235f292a7fa4a4994425d2eed230029",
            "8bbb0c144cfb40f68e3751cd67f461eb",
            "8a4ca829987d49c7af59d1e544d75481",
            "3daa9a7987f54f62ac8903754d5e069a",
            "8c4f126be91247f187248b0722e8f979",
            "f1373a9285434abea0e8e99d396140df",
            "b78e76d8d7ea4762ad87fe0c26ecfcbe",
            "69c13d1bce6140c79e12fec80461b733",
            "66137c9639da40f2b2ecd52f3d475e97",
            "b0e957985ee7496cbe46f471acff1a20",
            "7104f3be38724ebfa34f1547a5bea9ef",
            "24040212d16c4c3a87bd7c71f7f9635d",
            "147e5239d3f2496e8394c388461f478e",
            "7d247c6f30514a899242bfe2726ec949",
            "e33909210fcc48af84e5cc67edd2b201",
            "a1f104ac907d4da594b6e08ae7c118fa",
            "ee8ec733c4f040f9bcd1c931f6e0e75d",
            "cf60b34b2eec4d28a305b39a100b4ad9"
          ]
        },
        "id": "dextpWDJOm1X",
        "outputId": "da435c00-55cf-4cf1-9194-914794dac2ad"
      },
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "No model was supplied, defaulted to sshleifer/distilbart-cnn-12-6 (https://huggingface.co/sshleifer/distilbart-cnn-12-6)\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Downloading:   0%|          | 0.00/1.76k [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "94d7f6443c4f41a08d2ce1b00d2d0239"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Downloading:   0%|          | 0.00/1.14G [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "4d80e3c7704148689d30e46c038e7983"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Downloading:   0%|          | 0.00/26.0 [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "0b340932a75a4b289d6e12af4e631ed8"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Downloading:   0%|          | 0.00/878k [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "8342e1f01f4647d18bccd8c8ea283c68"
            }
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "Downloading:   0%|          | 0.00/446k [00:00<?, ?B/s]"
            ],
            "application/vnd.jupyter.widget-view+json": {
              "version_major": 2,
              "version_minor": 0,
              "model_id": "69c13d1bce6140c79e12fec80461b733"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Using BeautifulSoup to scrape an article"
      ],
      "metadata": {
        "id": "BFQZCzvWj3kN"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "URL='https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3.full' #Strainflow Paper by Dr. Tavpritesh Sethi from IIIT-D"
      ],
      "metadata": {
        "id": "E3-ore6nOm4V"
      },
      "execution_count": 38,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "req_url=requests.get(URL)\n",
        "req_url\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AP_4vLBFOm-G",
        "outputId": "d04d7072-7d33-47e1-ef54-155387e7cc4e"
      },
      "execution_count": 39,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<Response [200]>"
            ]
          },
          "metadata": {},
          "execution_count": 39
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "req_url.text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 122
        },
        "id": "MgotCMswOnQU",
        "outputId": "d2525976-800d-4471-c9fe-07f45be632fd"
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'<!DOCTYPE html>\\n<html lang=\"en\" dir=\"ltr\" \\n  xmlns=\"http://www.w3.org/1999/xhtml\"\\n  xmlns:mml=\"http://www.w3.org/1998/Math/MathML\">\\n  <head prefix=\"og: http://ogp.me/ns# article: http://ogp.me/ns/article# book: http://ogp.me/ns/book#\" >\\n    <!--[if IE]><![endif]-->\\n<link rel=\"dns-prefetch\" href=\"//d33xdlntwy0kbs.cloudfront.net\" />\\n<link rel=\"dns-prefetch\" href=\"//www.google.com\" />\\n<link rel=\"dns-prefetch\" href=\"//scholar.google.com\" />\\n<link rel=\"dns-prefetch\" href=\"//www.google-analytics.com\" />\\n<link rel=\"dns-prefetch\" href=\"//stats.g.doubleclick.net\" />\\n<meta http-equiv=\"Content-Type\" content=\"text/html; charset=utf-8\" />\\n<meta name=\"viewport\" content=\"width=device-width, initial-scale=1\" />\\n<link rel=\"shortcut icon\" href=\"https://www.biorxiv.org/sites/default/files/images/favicon.ico\" type=\"image/vnd.microsoft.icon\" />\\n<link rel=\"alternate\" type=\"application/pdf\" title=\"Full Text (PDF)\" href=\"/content/10.1101/2021.05.25.445601v3.full.pdf\" />\\n<link rel=\"alternate\" type=\"text/plain\" title=\"Full Text (Plain)\" href=\"/content/10.1101/2021.05.25.445601v3.full.txt\" />\\n<link rel=\"alternate\" type=\"application/vnd.ms-powerpoint\" title=\"Powerpoint\" href=\"/content/10.1101/2021.05.25.445601v3.ppt\" />\\n<meta name=\"article_thumbnail\" content=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-10.gif\" />\\n<meta name=\"type\" content=\"article\" />\\n<meta name=\"category\" content=\"article\" />\\n<meta name=\"HW.identifier\" content=\"/biorxiv/early/2021/08/26/2021.05.25.445601.atom\" />\\n<meta name=\"HW.pisa\" content=\"biorxiv;2021.05.25.445601v3\" />\\n<meta name=\"DC.Format\" content=\"text/html\" />\\n<meta name=\"DC.Language\" content=\"en\" />\\n<meta name=\"DC.Title\" content=\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning\" />\\n<meta name=\"DC.Identifier\" content=\"10.1101/2021.05.25.445601\" />\\n<meta name=\"DC.Date\" content=\"2021-08-26\" />\\n<meta name=\"DC.Publisher\" content=\"Cold Spring Harbor Laboratory\" />\\n<meta name=\"DC.Rights\" content=\"© 2021, Posted by Cold Spring Harbor Laboratory. This pre-print is available under a Creative Commons License (Attribution-NonCommercial-NoDerivs 4.0 International), CC BY-NC-ND 4.0, as described at http://creativecommons.org/licenses/by-nc-nd/4.0/\" />\\n<meta name=\"DC.AccessRights\" content=\"restricted\" />\\n<meta name=\"DC.Description\" content=\"The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed Strainflow , to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.\\n\\nHighlights \\n\\n![Figure][1]&lt;/img&gt;\\n\\n### Competing Interest Statement\\n\\nThe authors have declared no competing interest.\\n\\n [1]: pending:yes\" />\\n<meta name=\"DC.Contributor\" content=\"Sargun Nagpal\" />\\n<meta name=\"DC.Contributor\" content=\"Ridam Pal\" />\\n<meta name=\"DC.Contributor\" content=\"Ashima\" />\\n<meta name=\"DC.Contributor\" content=\"Ananya Tyagi\" />\\n<meta name=\"DC.Contributor\" content=\"Sadhana Tripathi\" />\\n<meta name=\"DC.Contributor\" content=\"Aditya Nagori\" />\\n<meta name=\"DC.Contributor\" content=\"Saad Ahmad\" />\\n<meta name=\"DC.Contributor\" content=\"Hara Prasad Mishra\" />\\n<meta name=\"DC.Contributor\" content=\"Rintu Kutum\" />\\n<meta name=\"DC.Contributor\" content=\"Tavpritesh Sethi\" />\\n<meta name=\"article:published_time\" content=\"2021-08-26\" />\\n<meta name=\"article:section\" content=\"New Results\" />\\n<meta name=\"citation_title\" content=\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning\" />\\n<meta name=\"citation_abstract\" lang=\"en\" content=\"&lt;h3&gt;Abstract&lt;/h3&gt;\\n&lt;p&gt;The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed &lt;i&gt;Strainflow&lt;/i&gt;, to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our &lt;i&gt;Strainflow&lt;/i&gt; model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called &lt;i&gt;Dimensions of Concern&lt;/i&gt; (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.&lt;/p&gt;&lt;h3&gt;Highlights&lt;/h3&gt;\\n&lt;p&gt;We developed a genomic surveillance model for SARS-CoV-2 genome sequences, &lt;i&gt;Strainflow&lt;/i&gt;, where sequences were treated as documents with words (codons) to learn the codon context of 0.9 million spike genes using the skip-gram algorithm.&lt;/p&gt;&lt;p&gt;Time series analysis of the information content (Entropy) of the latent dimensions learned by &lt;i&gt;Strainflow&lt;/i&gt; shows a leading relationship with the monthly COVID-19 cases for seven countries (e.g., USA, Japan, India, and others).&lt;/p&gt;&lt;p&gt;Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads.&lt;/p&gt;&lt;p&gt;The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants.&lt;/p&gt;&lt;h3&gt;Graphical abstract&lt;/h3&gt;\\n\" />\\n<meta name=\"citation_journal_title\" content=\"bioRxiv\" />\\n<meta name=\"citation_publisher\" content=\"Cold Spring Harbor Laboratory\" />\\n<meta name=\"citation_publication_date\" content=\"2021/01/01\" />\\n<meta name=\"citation_mjid\" content=\"biorxiv;2021.05.25.445601v3\" />\\n<meta name=\"citation_id\" content=\"2021.05.25.445601v3\" />\\n<meta name=\"citation_public_url\" content=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" />\\n<meta name=\"citation_abstract_html_url\" content=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3.abstract\" />\\n<meta name=\"citation_full_html_url\" content=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3.full\" />\\n<meta name=\"citation_pdf_url\" content=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601.full.pdf\" />\\n<meta name=\"citation_doi\" content=\"10.1101/2021.05.25.445601\" />\\n<meta name=\"citation_num_pages\" content=\"20\" />\\n<meta name=\"citation_article_type\" content=\"Article\" />\\n<meta name=\"citation_section\" content=\"New Results\" />\\n<meta name=\"citation_firstpage\" content=\"2021.05.25.445601\" />\\n<meta name=\"citation_author\" content=\"Sargun Nagpal\" />\\n<meta name=\"citation_author_institution\" content=\"Indraprastha Institute of Information Technology Delhi\" />\\n<meta name=\"citation_author_orcid\" content=\"http://orcid.org/0000-0001-9936-7790\" />\\n<meta name=\"citation_author\" content=\"Ridam Pal\" />\\n<meta name=\"citation_author_institution\" content=\"Indraprastha Institute of Information Technology Delhi\" />\\n<meta name=\"citation_author\" content=\"Ashima\" />\\n<meta name=\"citation_author_institution\" content=\"Indraprastha Institute of Information Technology Delhi\" />\\n<meta name=\"citation_author\" content=\"Ananya Tyagi\" />\\n<meta name=\"citation_author_institution\" content=\"Indraprastha Institute of Information Technology Delhi\" />\\n<meta name=\"citation_author\" content=\"Sadhana Tripathi\" />\\n<meta name=\"citation_author_institution\" content=\"Indraprastha Institute of Information Technology Delhi\" />\\n<meta name=\"citation_author\" content=\"Aditya Nagori\" />\\n<meta name=\"citation_author_institution\" content=\"Indraprastha Institute of Information Technology Delhi\" />\\n<meta name=\"citation_author_orcid\" content=\"http://orcid.org/0000-0002-6389-2179\" />\\n<meta name=\"citation_author\" content=\"Saad Ahmad\" />\\n<meta name=\"citation_author_institution\" content=\"Indraprastha Institute of Information Technology Delhi\" />\\n<meta name=\"citation_author\" content=\"Hara Prasad Mishra\" />\\n<meta name=\"citation_author_institution\" content=\"Indraprastha Institute of Information Technology Delhi\" />\\n<meta name=\"citation_author_orcid\" content=\"http://orcid.org/0000-0001-9405-7822\" />\\n<meta name=\"citation_author\" content=\"Rintu Kutum\" />\\n<meta name=\"citation_author_institution\" content=\"Indraprastha Institute of Information Technology Delhi\" />\\n<meta name=\"citation_author_email\" content=\"tavpriteshsethi@iiitd.ac.in\" />\\n<meta name=\"citation_author_email\" content=\"rintuk@iiitd.ac.in\" />\\n<meta name=\"citation_author_orcid\" content=\"http://orcid.org/0000-0001-8667-6199\" />\\n<meta name=\"citation_author\" content=\"Tavpritesh Sethi\" />\\n<meta name=\"citation_author_institution\" content=\"Indraprastha Institute of Information Technology Delhi\" />\\n<meta name=\"citation_author_institution\" content=\"All India Institute of Medical Sciences\" />\\n<meta name=\"citation_author_email\" content=\"tavpriteshsethi@iiitd.ac.in\" />\\n<meta name=\"citation_author_email\" content=\"rintuk@iiitd.ac.in\" />\\n<meta name=\"citation_author_orcid\" content=\"http://orcid.org/0000-0002-4776-7941\" />\\n<meta name=\"citation_reference\" content=\"A. Liaw and M. Wiener (2002). Classification and Regression by randomForest. R News 2(3), 18--22.\" />\\n<meta name=\"citation_reference\" content=\"Arslan, H. (2021). Machine Learning Methods for COVID-19 Prediction Using Human Genomic Data. Proceedings 74, 20.\" />\\n<meta name=\"citation_reference\" content=\"CDC (2021). SARS-CoV-2 Variant Classifications and Definitions.\" />\\n<meta name=\"citation_reference\" content=\"Hadfield, J., Megill, C., Bell, S.M., Huddleston, J., Potter, B., Callender, C., Sagulenko, P., Bedford, T., and Neher, R.A. (2018). Nextstrain: real-time tracking of pathogen evolution. Bioinformatics 34, 4121–4123.\" />\\n<meta name=\"citation_reference\" content=\"Hie, B., Zhong, E.D., Berger, B., and Bryson, B. (2021). Learning the language of viral evolution and escape. Science 371, 284–288.\" />\\n<meta name=\"citation_reference\" content=\"Khan, M.I., Khan, Z.A., Baig, M.H., Ahmad, I., Farouk, A.-E., Song, Y.G., and Dong, J.-J. (2020). Comparative genome analysis of novel coronavirus (SARS-CoV-2) from different geographical locations and the effect of mutations on major target proteins: An in silico insight. PLoS One 15, e0238344.\" />\\n<meta name=\"citation_reference\" content=\"Kursa, M.B., and Rudnicki, W.R. (2010). Feature Selection with theBorutaPackage. Journal of Statistical Software 36.\" />\\n<meta name=\"citation_reference\" content=\"Kursa, M.B., Jankowski, A., and Rudnicki, W.R. (2010). Boruta - A System for Feature Selection. Fund. Inform. 101, 271–285.\" />\\n<meta name=\"citation_reference\" content=\"Lee, J., Yoon, W., Kim, S., Kim, D., Kim, S., So, C.H., and Kang, J. (2020). BioBERT: a pre-trained biomedical language representation model for biomedical text mining. Bioinformatics 36, 1234–1240.\" />\\n<meta name=\"citation_reference\" content=\"Letunic, I. iTOL: Interactive Tree Of Life.\" />\\n<meta name=\"citation_reference\" content=\"Li, Q., Wu, J., Nie, J., Zhang, L., Hao, H., Liu, S., Zhao, C., Zhang, Q., Liu, H., Nie, L., et al. (2020). The Impact of Mutations in SARS-CoV-2 Spike on Viral Infectivity and Antigenicity. Cell 182, 1284–1294.e9.\" />\\n<meta name=\"citation_reference\" content=\"Linderman, G.C., Rachh, M., Hoskins, J.G., Steinerberger, S., and Kluger, Y. (2019). Fast interpolation-based t-SNE for improved visualization of single-cell RNA-seq data. Nat. Methods 16, 243–245.\" />\\n<meta name=\"citation_reference\" content=\"Liu, B., Liu, K., Zhang, H., Zhang, L., Bian, Y., and Huang, L. (2020). CoV-Seq, a New Tool for SARS-CoV-2 Genome Analysis and Visualization: Development and Usability Study. J. Med. Internet Res. 22, e22299.\" />\\n<meta name=\"citation_reference\" content=\"Lopez-Rincon, A., Perez-Romero, C.A., Tonda, A., Mendoza-Maldonado, L., Claassen, E., Garssen, J., and Kraneveld, A.D. Design of Specific Primer Sets for the Detection of B.1.1.7, B.1.351, P.1, B.1.617.2 and B.1.1.519 Variants of SARS-CoV-2 using Artificial Intelligence.\" />\\n<meta name=\"citation_reference\" content=\"Mandal, S., Roychowdhury, T., and Bhattacharya, A. (2021). Pattern of genomic variation in SARS-CoV-2 (COVID-19) suggests restricted nonrandom changes: Analysis using Shewhart control charts. J. Biosci. 46.\" />\\n<meta name=\"citation_reference\" content=\"Mikolov, T., Chen, K., Corrado, G., and Dean, J. (2013). Efficient Estimation of Word Representations in Vector Space.\" />\\n<meta name=\"citation_reference\" content=\"Mugnai, M.L., Templeton, C., Elber, R., and Thirumalai, D. (2020). Role of Long-range Allosteric Communication in Determining the Stability and Disassembly of SARS-COV-2 in Complex with ACE2. bioRxiv.\" />\\n<meta name=\"citation_reference\" content=\"Mushtaq, R. (2011). Augmented Dickey Fuller Test. SSRN Electronic Journal.\" />\\n<meta name=\"citation_reference\" content=\"Pan, Y.H., Wang, Y.H., Liang, S.F., and Lee, K.T. (2011). Fast computation of sample entropy and approximate entropy in biomedicine. Comput. Methods Programs Biomed. 104.\" />\\n<meta name=\"citation_reference\" content=\"paola (2021). Phylogenetic relationship of SARS-CoV-2 sequences from Amazonas with emerging Brazilian variants harboring mutations E484K and N501Y in the Spike protein.\" />\\n<meta name=\"citation_reference\" content=\"Peacock, T.P., Penrice-Randal, R., Hiscox, J.A., and Barclay, W.S. (2021). SARS-CoV-2 one year on: evidence for ongoing viral adaptation. J. Gen. Virol. 102.\" />\\n<meta name=\"citation_reference\" content=\"Podobnik, B., and Stanley, H.E. (2007). Detrended Cross-Correlation Analysis: A New Method for Analyzing Two Non-stationary Time Series.\" />\\n<meta name=\"citation_reference\" content=\"Prass, T.S., and Pumi, G. (2019). On the behavior of the DFA and DCCA in trend-stationary processes.\" />\\n<meta name=\"citation_reference\" content=\"Radim Rehurek, P.S. (2010). Software Framework for Topic Modelling with Large Corpora. In IN PROCEEDINGS OF THE LREC 2010 WORKSHOP ON NEW CHALLENGES FOR NLP FRAMEWORKS,.\" />\\n<meta name=\"citation_reference\" content=\"Rouchka, E.C., Chariker, J.H., and Chung, D. (2020). Variant analysis of 1,040 SARS-CoV-2 genomes. PLoS One 15, e0241535.\" />\\n<meta name=\"citation_reference\" content=\"Shishir, T.A., Naser, I.B., and Faruque, S.M. (2021). In silico comparative genomics of SARS-CoV-2 to determine the source and diversity of the pathogen in Bangladesh. PLoS One 16, e0245584.\" />\\n<meta name=\"citation_reference\" content=\"Shu, Y., and McCauley, J. (2017). GISAID: Global initiative on sharing all influenza data - from vision to reality. Euro Surveill. 22.\" />\\n<meta name=\"citation_reference\" content=\"Srivastava, S., Banu, S., Singh, P., Sowpati, D.T., and Mishra, R.K. (2021). SARS-CoV-2 genomics: An Indian perspective on sequencing viral variants. J. Biosci. 46.\" />\\n<meta name=\"citation_reference\" content=\"Tomcala, J. (2018). Time Series Entropies [R package TSEntropies version 0.9].\" />\\n<meta name=\"citation_reference\" content=\"Tomčala, J. (2020). New Fast ApEn and SampEn Entropy Algorithms Implementation and Their Application to Supercomputer Power Consumption. Entropy 22, 863.\" />\\n<meta name=\"citation_reference\" content=\"Wang, B., Zhao, D., Lioma, C., Li, Q., Zhang, P., and Simonsen, J.G. (2019). Encoding word order in complex embeddings.\" />\\n<meta name=\"citation_reference\" content=\"Wang, R.Y., Guo, T.Q., Li, L.G., Jiao, J.Y., and Wang, L.Y. (2020). Predictions of COVID-19 Infection Severity Based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 through Machine Learning of Genetic Data. 2020 IEEE 8th International Conference on Computer Science and Network Technology (ICCSNT).\" />\\n<meta name=\"citation_reference\" content=\"WHO Coronavirus (COVID-19) Dashboard.\" />\\n<meta name=\"citation_reference\" content=\"Wolf, T., Debut, L., Sanh, V., Chaumond, J., Delangue, C., Moi, A., Cistac, P., Rault, T., Louf, R., Funtowicz, M., et al. (2020). Transformers: State-of-the-Art Natural Language Processing. Proceedings of the 2020 Conference on Empirical Methods in Natural Language Processing: System Demonstrations.\" />\\n<meta name=\"citation_reference\" content=\"Wu, F., Yang, R., Zhang, C., and Zhang, L. (2021). A deep learning framework combined with word embedding to identify DNA replication origins. Sci. Rep. 11, 844.\" />\\n<meta name=\"citation_reference\" content=\"Yilmaz, A. (2020). Assessment of Mutation Susceptibility in DNA Sequences with Word Vectors. Journal of Intelligent Systems: Theory and Applications 3, 1–6.\" />\\n<meta name=\"citation_reference\" content=\"Yin, Z., and Shen, Y. (2018). On the Dimensionality of Word Embedding. Adv. Neural Inf. Process. Syst. 31.\" />\\n<meta name=\"citation_reference\" content=\"(2020a). Time Series Analysis and Computational Finance [R package tseries version 0.10-48].\" />\\n<meta name=\"citation_reference\" content=\"(2020b). Detrended Fluctuation and Detrended Cross-Correlation Analysis [R package DCCA version 0.1.1].\" />\\n<meta name=\"twitter:title\" content=\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning\" />\\n<meta name=\"twitter:site\" content=\"@biorxivpreprint\" />\\n<meta name=\"twitter:card\" content=\"summary\" />\\n<meta name=\"twitter:image\" content=\"https://www.biorxiv.org/sites/default/files/images/biorxiv_logo_homepage7-5-small.png\" />\\n<meta name=\"twitter:description\" content=\"The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed Strainflow , to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.\\n\\nHighlights \\n\\n![Figure][1]&lt;/img&gt;\\n\\n### Competing Interest Statement\\n\\nThe authors have declared no competing interest.\\n\\n [1]: pending:yes\" />\\n<meta name=\"og-title\" property=\"og:title\" content=\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning\" />\\n<meta name=\"og-url\" property=\"og:url\" content=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" />\\n<meta name=\"og-site-name\" property=\"og:site_name\" content=\"bioRxiv\" />\\n<meta name=\"og-description\" property=\"og:description\" content=\"The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed Strainflow , to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.\\n\\nHighlights \\n\\n![Figure][1]&lt;/img&gt;\\n\\n### Competing Interest Statement\\n\\nThe authors have declared no competing interest.\\n\\n [1]: pending:yes\" />\\n<meta name=\"og-type\" property=\"og:type\" content=\"article\" />\\n<meta name=\"og-image\" property=\"og:image\" content=\"https://www.biorxiv.org/sites/default/files/images/biorxiv_logo_homepage7-5-small.png\" />\\n<meta name=\"citation_date\" content=\"2021-08-26\" />\\n<meta name=\"description\" content=\"bioRxiv - the preprint server for biology, operated by Cold Spring Harbor Laboratory, a research and educational institution\" />\\n<meta name=\"generator\" content=\"Drupal 7 (http://drupal.org)\" />\\n<link rel=\"canonical\" href=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" />\\n<link rel=\"shortlink\" href=\"https://www.biorxiv.org/node/2125191\" />\\n    <title>Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning | bioRxiv</title>  \\n    <link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__GW-q1yS4llZsZXYilALi0SfHZdla2dBZSfgKsyPoQv0__Mq2l818MUdgw7DQFZ42QKliXne5RR1HOY89XjtvuppE__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\\n<link type=\"text/css\" rel=\"stylesheet\" href=\"//cdn.jsdelivr.net/qtip2/2.2.1/jquery.qtip.min.css\" media=\"all\" />\\n<link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__Sjm8qRC6pDZnJVeNrkx67-ZH1Mte_Ys5jCfEFFU1muc__r6-n363p7S972yCeOZaxDruAzBGzgr4Ie4Ci5dXnv0k__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\\n<style type=\"text/css\" media=\"all\">\\n/* <![CDATA[ */\\n.panels-flexible-new .panels-flexible-region{padding:0}.panels-flexible-new .panels-flexible-region-inside{padding-right:.5em;padding-left:.5em}.panels-flexible-new .panels-flexible-region-inside-first{padding-left:0}.panels-flexible-new .panels-flexible-region-inside-last{padding-right:0}.panels-flexible-new .panels-flexible-column{padding:0}.panels-flexible-new .panels-flexible-column-inside{padding-right:.5em;padding-left:.5em}.panels-flexible-new .panels-flexible-column-inside-first{padding-left:0}.panels-flexible-new .panels-flexible-column-inside-last{padding-right:0}.panels-flexible-new .panels-flexible-row{padding:0 0 .5em;margin:0}.panels-flexible-new .panels-flexible-row-last{padding-bottom:0}.panels-flexible-column-new-main{float:left;width:99.0000%}.panels-flexible-new-inside{padding-right:0}.panels-flexible-new{width:auto}.panels-flexible-region-new-center{float:left;width:99.0000%}.panels-flexible-row-new-main-row-inside{padding-right:0}\\n/* ]]> */\\n</style>\\n<link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__PWuQ_RYRTJ4BLEKsbWQeHysPg0gOQ3571ruQa_rXvAo__cj81oxtSGUe-AIO3tlkSBJPNpKP5ZgLdB8f8awBQsWU__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\\n<style type=\"text/css\" media=\"all\">\\n/* <![CDATA[ */\\n#sliding-popup.sliding-popup-bottom,#sliding-popup.sliding-popup-bottom .eu-cookie-withdraw-banner,.eu-cookie-withdraw-tab{background:gray}#sliding-popup.sliding-popup-bottom.eu-cookie-withdraw-wrapper{background:transparent}#sliding-popup .popup-content #popup-text h1,#sliding-popup .popup-content #popup-text h2,#sliding-popup .popup-content #popup-text h3,#sliding-popup .popup-content #popup-text p,.eu-cookie-compliance-secondary-button,.eu-cookie-withdraw-tab{color:#fff !important}.eu-cookie-withdraw-tab{border-color:#fff}.eu-cookie-compliance-more-button{color:#fff !important}\\n/* ]]> */\\n</style>\\n\\n<!--[if lte IE 7]>\\n<link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__ElJr3PIJEvw3qLXc1cnYiLj2G4KgDPSXFOfm6Phf8hw__JdWGm15cDWjsK6KrFlQVXQix9YgNeYysf22XZHj-Y-c__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\\n<![endif]-->\\n<link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__Wnlyen9qEpwh_Qaf9okEu4QdVGM0BDothxeqA6Nbvo8__EJmw6SZD9bYoS8jocCpPYS3JFRURpdzmuvJoAUNiI-g__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\\n<link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__vWH9eagUiEGaWTpZefl9TB2ucrJuL66ajzSCvO73bss__GnHmvNTOB0-xZRDC4PFtbOee_g-W6li7rpWRuc9O4B4__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\\n\\n<!--[if (lt IE 9)&(!IEMobile)]>\\n<link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__XH6bpcI0f2dImc-p674DLCZtWBGb-QwxJK1YexVGtno__vUceGprdo5nIhV6DH93X7fI3r8RcTJbChbas9TQXeW4__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\\n<![endif]-->\\n\\n<!--[if gte IE 9]><!-->\\n<link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__2WBMox6sOrN42ss5lCnH7WWVRdFdJCxtTKnQJYRwTE4__yqNvNYLvMpjy3ffuJrjjm9uW2i-Me1c23KLYuWHaqio__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\\n<!--<![endif]-->\\n<link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__dMP0ngeKhq7e9DSwoh07jSjMdMBpvUxRs4IveUsPUNM__5c1ZiN_pM9tjxvpyA9VpEacScF0S_W4R222pC-s_-Pk__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\\n<link type=\"text/css\" rel=\"stylesheet\" href=\"https://d33xdlntwy0kbs.cloudfront.net/cshl_custom.css\" media=\"all\" />\\n    <script type=\"text/javascript\" src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__-O5AqrChkaQBJNz3KVCNgPD7B2WnlmFaBWhKn3OLnSk__HPUPKsZe95KwbHSnKRhhlLxkHEchbpL1f2QRQnZQX38__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\"></script>\\n<script type=\"text/javascript\" src=\"//cdn.jsdelivr.net/qtip2/2.2.1/jquery.qtip.min.js\"></script>\\n<script type=\"text/javascript\" src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__49Ok9ZaeGdzHaYY0HeAxSZkJaw1DaCwfnGvLBVZ_w1Q__Tcognn9uta6IhrOugHP9-zOAzZitowtsZynB_751i8Q__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\"></script>\\n<script type=\"text/javascript\" src=\"https://www.google.com/recaptcha/api.js?hl=en&amp;render=explicit&amp;onload=drupalRecaptchaOnload\"></script>\\n<script type=\"text/javascript\" src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__6hIesE0QCBXoXlEIn01f-78o39DmDNEoInQ35EHxoXw__qAjUcZEMaB7nsfhw64tJNfG5xuXddaTuPmbIfa6fPMk__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\"></script>\\n<script type=\"text/javascript\" async=\"async\" src=\"https://scholar.google.com/scholar_js/casa.js\"></script>\\n<script type=\"text/javascript\">\\n<!--//--><![CDATA[//><!--\\n/*!\\n * yepnope1.5.4\\n * (c) WTFPL, GPLv2\\n */\\n(function(a,b,c){function d(a){return\"[object Function]\"==o.call(a)}function e(a){return\"string\"==typeof a}function f(){}function g(a){return!a||\"loaded\"==a||\"complete\"==a||\"uninitialized\"==a}function h(){var a=p.shift();q=1,a?a.t?m(function(){(\"c\"==a.t?B.injectCss:B.injectJs)(a.s,0,a.a,a.x,a.e,1)},0):(a(),h()):q=0}function i(a,c,d,e,f,i,j){function k(b){if(!o&&g(l.readyState)&&(u.r=o=1,!q&&h(),l.onload=l.onreadystatechange=null,b)){\"img\"!=a&&m(function(){t.removeChild(l)},50);for(var d in y[c])y[c].hasOwnProperty(d)&&y[c][d].onload()}}var j=j||B.errorTimeout,l=b.createElement(a),o=0,r=0,u={t:d,s:c,e:f,a:i,x:j};1===y[c]&&(r=1,y[c]=[]),\"object\"==a?l.data=c:(l.src=c,l.type=a),l.width=l.height=\"0\",l.onerror=l.onload=l.onreadystatechange=function(){k.call(this,r)},p.splice(e,0,u),\"img\"!=a&&(r||2===y[c]?(t.insertBefore(l,s?null:n),m(k,j)):y[c].push(l))}function j(a,b,c,d,f){return q=0,b=b||\"j\",e(a)?i(\"c\"==b?v:u,a,b,this.i++,c,d,f):(p.splice(this.i++,0,a),1==p.length&&h()),this}function k(){var a=B;return a.loader={load:j,i:0},a}var l=b.documentElement,m=a.setTimeout,n=b.getElementsByTagName(\"script\")[0],o={}.toString,p=[],q=0,r=\"MozAppearance\"in l.style,s=r&&!!b.createRange().compareNode,t=s?l:n.parentNode,l=a.opera&&\"[object Opera]\"==o.call(a.opera),l=!!b.attachEvent&&!l,u=r?\"object\":l?\"script\":\"img\",v=l?\"script\":u,w=Array.isArray||function(a){return\"[object Array]\"==o.call(a)},x=[],y={},z={timeout:function(a,b){return b.length&&(a.timeout=b[0]),a}},A,B;B=function(a){function b(a){var a=a.split(\"!\"),b=x.length,c=a.pop(),d=a.length,c={url:c,origUrl:c,prefixes:a},e,f,g;for(f=0;f<d;f++)g=a[f].split(\"=\"),(e=z[g.shift()])&&(c=e(c,g));for(f=0;f<b;f++)c=x[f](c);return c}function g(a,e,f,g,h){var i=b(a),j=i.autoCallback;i.url.split(\".\").pop().split(\"?\").shift(),i.bypass||(e&&(e=d(e)?e:e[a]||e[g]||e[a.split(\"/\").pop().split(\"?\")[0]]),i.instead?i.instead(a,e,f,g,h):(y[i.url]?i.noexec=!0:y[i.url]=1,f.load(i.url,i.forceCSS||!i.forceJS&&\"css\"==i.url.split(\".\").pop().split(\"?\").shift()?\"c\":c,i.noexec,i.attrs,i.timeout),(d(e)||d(j))&&f.load(function(){k(),e&&e(i.origUrl,h,g),j&&j(i.origUrl,h,g),y[i.url]=2})))}function h(a,b){function c(a,c){if(a){if(e(a))c||(j=function(){var a=[].slice.call(arguments);k.apply(this,a),l()}),g(a,j,b,0,h);else if(Object(a)===a)for(n in m=function(){var b=0,c;for(c in a)a.hasOwnProperty(c)&&b++;return b}(),a)a.hasOwnProperty(n)&&(!c&&!--m&&(d(j)?j=function(){var a=[].slice.call(arguments);k.apply(this,a),l()}:j[n]=function(a){return function(){var b=[].slice.call(arguments);a&&a.apply(this,b),l()}}(k[n])),g(a[n],j,b,n,h))}else!c&&l()}var h=!!a.test,i=a.load||a.both,j=a.callback||f,k=j,l=a.complete||f,m,n;c(h?a.yep:a.nope,!!i),i&&c(i)}var i,j,l=this.yepnope.loader;if(e(a))g(a,0,l,0);else if(w(a))for(i=0;i<a.length;i++)j=a[i],e(j)?g(j,0,l,0):w(j)?B(j):Object(j)===j&&h(j,l);else Object(a)===a&&h(a,l)},B.addPrefix=function(a,b){z[a]=b},B.addFilter=function(a){x.push(a)},B.errorTimeout=1e4,null==b.readyState&&b.addEventListener&&(b.readyState=\"loading\",b.addEventListener(\"DOMContentLoaded\",A=function(){b.removeEventListener(\"DOMContentLoaded\",A,0),b.readyState=\"complete\"},0)),a.yepnope=k(),a.yepnope.executeStack=h,a.yepnope.injectJs=function(a,c,d,e,i,j){var k=b.createElement(\"script\"),l,o,e=e||B.errorTimeout;k.src=a;for(o in d)k.setAttribute(o,d[o]);c=j?h:c||f,k.onreadystatechange=k.onload=function(){!l&&g(k.readyState)&&(l=1,c(),k.onload=k.onreadystatechange=null)},m(function(){l||(l=1,c(1))},e),i?k.onload():n.parentNode.insertBefore(k,n)},a.yepnope.injectCss=function(a,c,d,e,g,i){var e=b.createElement(\"link\"),j,c=i?h:c||f;e.href=a,e.rel=\"stylesheet\",e.type=\"text/css\";for(j in d)e.setAttribute(j,d[j]);g||(n.parentNode.insertBefore(e,n),m(c,0))}})(this,document);\\n\\n//--><!]]>\\n</script>\\n<script type=\"text/javascript\">\\n<!--//--><![CDATA[//><!--\\nyepnope({\\n  test: Modernizr.matchmedia,\\n  nope: \\'/sites/all/libraries/media-match/media.match.min.js\\'\\n});\\n//--><!]]>\\n</script>\\n<script type=\"text/javascript\">\\n<!--//--><![CDATA[//><!--\\nvar _prum=[[\\'id\\', \\'612e3d94173350001100005d\\'], [\\'mark\\', \\'firstbyte\\',\\n      (new Date()).getTime()]]; (function() {\\n      var s=document.getElementsByTagName(\\'script\\')[0],\\n      p=document.createElement(\\'script\\');\\n      p.async=\\'async\\'; p.src=\\'//rum-static.pingdom.net/prum.min.js\\';s.parentNode.insertBefore(p,s);})();\\n//--><!]]>\\n</script>\\n<script type=\"text/javascript\">\\n<!--//--><![CDATA[//><!--\\nif(typeof window.MathJax === \"undefined\") window.MathJax = { menuSettings: { zoom: \"Click\" } };\\n//--><!]]>\\n</script>\\n<script type=\"text/javascript\">\\n<!--//--><![CDATA[//><!--\\n(function(i,s,o,g,r,a,m){i[\"GoogleAnalyticsObject\"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,\"script\",\"//www.google-analytics.com/analytics.js\",\"ga\");ga(\"create\", \"UA-45638026-1\", {\"cookieDomain\":\"auto\"});ga(\"set\", \"page\", location.pathname + location.search + location.hash);ga(\"send\", \"pageview\");ga(\\'create\\', \\'UA-189672-38\\', \\'auto\\', {\\'name\\': \\'hwTracker\\'});\\r\\nga(\\'set\\', \\'anonymizeIp\\', true);\\nga(\\'hwTracker.send\\', \\'pageview\\');\\n//--><!]]>\\n</script>\\n<script type=\"text/javascript\">\\n<!--//--><![CDATA[//><!--\\njQuery.extend(Drupal.settings,{\"basePath\":\"\\\\/\",\"pathPrefix\":\"\",\"ajaxPageState\":{\"theme\":\"jcore_1\",\"theme_token\":\"aTuGQhKIBxC958cyCtwunT2OOmpQqxP4UNn4e5Rh_Ak\"},\"colorbox\":{\"opacity\":\"0.85\",\"current\":\"{current} of {total}\",\"previous\":\"\\\\u00ab Prev\",\"next\":\"Next \\\\u00bb\",\"close\":\"Close\",\"maxWidth\":\"98%\",\"maxHeight\":\"98%\",\"fixed\":true,\"mobiledetect\":true,\"mobiledevicewidth\":\"480px\"},\"highwire\":{\"nid\":\"2125191\",\"apath\":\"\\\\/biorxiv\\\\/early\\\\/2021\\\\/08\\\\/26\\\\/2021.05.25.445601.atom\",\"pisa\":\"biorxiv;2021.05.25.445601v3\",\"processed\":[\"highwire_math\"],\"markup\":[{\"requested\":\"full-text\",\"variant\":\"full-text\",\"view\":\"full\",\"pisa\":\"biorxiv;2021.05.25.445601v3\"}],\"modal_window_width\":\"560\",\"share_modal_width\":\"560\",\"share_modal_title\":\"Share this Article\"},\"jcarousel\":{\"ajaxPath\":\"\\\\/jcarousel\\\\/ajax\\\\/views\"},\"instances\":\"{\\\\u0022highwire_abstract_tooltip\\\\u0022:{\\\\u0022content\\\\u0022:{\\\\u0022text\\\\u0022:\\\\u0022\\\\u0022},\\\\u0022style\\\\u0022:{\\\\u0022tip\\\\u0022:{\\\\u0022width\\\\u0022:20,\\\\u0022height\\\\u0022:20,\\\\u0022border\\\\u0022:1,\\\\u0022offset\\\\u0022:0,\\\\u0022corner\\\\u0022:true},\\\\u0022classes\\\\u0022:\\\\u0022qtip-custom hw-tooltip hw-abstract-tooltip qtip-shadow qtip-rounded\\\\u0022,\\\\u0022classes_custom\\\\u0022:\\\\u0022hw-tooltip hw-abstract-tooltip\\\\u0022},\\\\u0022position\\\\u0022:{\\\\u0022at\\\\u0022:\\\\u0022right center\\\\u0022,\\\\u0022my\\\\u0022:\\\\u0022left center\\\\u0022,\\\\u0022viewport\\\\u0022:true,\\\\u0022adjust\\\\u0022:{\\\\u0022method\\\\u0022:\\\\u0022shift\\\\u0022}},\\\\u0022show\\\\u0022:{\\\\u0022event\\\\u0022:\\\\u0022mouseenter click \\\\u0022,\\\\u0022solo\\\\u0022:true},\\\\u0022hide\\\\u0022:{\\\\u0022event\\\\u0022:\\\\u0022mouseleave \\\\u0022,\\\\u0022fixed\\\\u0022:1,\\\\u0022delay\\\\u0022:\\\\u0022100\\\\u0022}},\\\\u0022highwire_author_tooltip\\\\u0022:{\\\\u0022content\\\\u0022:{\\\\u0022text\\\\u0022:\\\\u0022\\\\u0022},\\\\u0022style\\\\u0022:{\\\\u0022tip\\\\u0022:{\\\\u0022width\\\\u0022:15,\\\\u0022height\\\\u0022:15,\\\\u0022border\\\\u0022:1,\\\\u0022offset\\\\u0022:0,\\\\u0022corner\\\\u0022:true},\\\\u0022classes\\\\u0022:\\\\u0022qtip-custom hw-tooltip hw-author-tooltip qtip-shadow qtip-rounded\\\\u0022,\\\\u0022classes_custom\\\\u0022:\\\\u0022hw-tooltip hw-author-tooltip\\\\u0022},\\\\u0022position\\\\u0022:{\\\\u0022at\\\\u0022:\\\\u0022top center\\\\u0022,\\\\u0022my\\\\u0022:\\\\u0022bottom center\\\\u0022,\\\\u0022viewport\\\\u0022:true,\\\\u0022adjust\\\\u0022:{\\\\u0022method\\\\u0022:\\\\u0022\\\\u0022}},\\\\u0022show\\\\u0022:{\\\\u0022event\\\\u0022:\\\\u0022mouseenter \\\\u0022,\\\\u0022solo\\\\u0022:true},\\\\u0022hide\\\\u0022:{\\\\u0022event\\\\u0022:\\\\u0022mouseleave \\\\u0022,\\\\u0022fixed\\\\u0022:1,\\\\u0022delay\\\\u0022:\\\\u0022100\\\\u0022}},\\\\u0022highwire_reflinks_tooltip\\\\u0022:{\\\\u0022content\\\\u0022:{\\\\u0022text\\\\u0022:\\\\u0022\\\\u0022},\\\\u0022style\\\\u0022:{\\\\u0022tip\\\\u0022:{\\\\u0022width\\\\u0022:15,\\\\u0022height\\\\u0022:15,\\\\u0022border\\\\u0022:1,\\\\u0022mimic\\\\u0022:\\\\u0022top center\\\\u0022,\\\\u0022offset\\\\u0022:0,\\\\u0022corner\\\\u0022:true},\\\\u0022classes\\\\u0022:\\\\u0022qtip-custom hw-tooltip hw-ref-link-tooltip qtip-shadow qtip-rounded\\\\u0022,\\\\u0022classes_custom\\\\u0022:\\\\u0022hw-tooltip hw-ref-link-tooltip\\\\u0022},\\\\u0022position\\\\u0022:{\\\\u0022at\\\\u0022:\\\\u0022bottom left\\\\u0022,\\\\u0022my\\\\u0022:\\\\u0022top left\\\\u0022,\\\\u0022viewport\\\\u0022:true,\\\\u0022adjust\\\\u0022:{\\\\u0022method\\\\u0022:\\\\u0022flip\\\\u0022}},\\\\u0022show\\\\u0022:{\\\\u0022event\\\\u0022:\\\\u0022mouseenter \\\\u0022,\\\\u0022solo\\\\u0022:true},\\\\u0022hide\\\\u0022:{\\\\u0022event\\\\u0022:\\\\u0022mouseleave \\\\u0022,\\\\u0022fixed\\\\u0022:1,\\\\u0022delay\\\\u0022:\\\\u0022100\\\\u0022}}}\",\"qtipDebug\":\"{\\\\u0022leaveElement\\\\u0022:0}\",\"panel_ajax_tab\":{\"path\":\"sites\\\\/all\\\\/modules\\\\/contrib\\\\/panels_ajax_tab\"},\"disqus\":{\"domain\":\"biorxivstage\",\"url\":\"https:\\\\/\\\\/www.biorxiv.org\\\\/content\\\\/10.1101\\\\/2021.05.25.445601v3\",\"title\":\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning\",\"identifier\":\"node\\\\/2125191\"},\"panels_ajax_pane\":{\"new-28877068-b9cb-4641-830b-b6b4638c98bb\":\"{\\\\u0022encrypted\\\\u0022:\\\\u0022{\\\\\\\\\\\\u0022encrypted\\\\\\\\\\\\u0022:\\\\\\\\\\\\u0022YDAGAQlyplmL1K84ipiv+SXHzUxOedjL3cmEwB3AME+J1bbYYBnU4YK+YvYm7FDYUdHVM0qny0j\\\\\\\\\\\\\\\\\\\\\\\\\\\\/O1vqi32LN6lWwmUEd1oM5D1iUwAI7pxoQdrwxZ0q4Ml0yGdKTGNtjbta92Ph5BwIWHGWnO40U9\\\\\\\\\\\\\\\\\\\\\\\\\\\\/ho449ON\\\\\\\\\\\\\\\\\\\\\\\\\\\\/1TjCtdVNnO\\\\\\\\\\\\\\\\\\\\\\\\\\\\/M0hsfBJzYVkxMJcuYuqcB2Z\\\\\\\\\\\\\\\\\\\\\\\\\\\\/GcsQlLy5rTTvLRZ7ulV5oBDMNyp+F29CIDdEdd7jA0zop5veEbcPLtQbrX6RUgwzR615KBDS6IWQoEOtQGjupavTozG2sJTqKTVtkfAj7E0FF2gBvjwE7Q0tUQIAIb6DRVJBxgcq85XtGhytKTgeaS5UJ30QiHaNagWrU8OVtovTaE\\\\\\\\\\\\\\\\\\\\\\\\\\\\/8FeceaO3HyL0x3+HbGDCgOviRAnyTblYSSlP4aWTbmhCvvSzM6lOflkwgeVHKWJcZ6D9K72SI0LVTN4K\\\\\\\\\\\\\\\\\\\\\\\\\\\\/nIXZs8bayjGS4AEbUrsUQHdSz6xW5tFt0O08hosNgV6I\\\\\\\\\\\\\\\\\\\\\\\\\\\\/wsQMcGpdsD68zHKQYBPDLC+wCElX5OMmcS0EycOzPG5084HCKuI5aaD+DmhrQbl3XfnqGwyiZvuGJLEJLMDVg5oGYH5lJoiK2kw2Zywc3frQizJU9NxOgpmYU2FYg\\\\\\\\\\\\\\\\\\\\\\\\\\\\/phlhk1DCLDFWmaJa9bjC4dyVzh8EnCrZedXq6iJv4whr4vgZdrzyALSIZisq\\\\\\\\\\\\\\\\\\\\\\\\\\\\/kNLVz7QxPyAGVyjk95tYxFliaBSbJGIxOQErmhsNjnuPknSFJhT\\\\\\\\\\\\\\\\\\\\\\\\\\\\/uPOxsAwNRaaZlNsyUm1rJLd38E3NyxcuuJ2eQiUgjv5Wy441syY9OZ9+MAA3wu9aK7+LfK\\\\\\\\\\\\\\\\\\\\\\\\\\\\/lx3A99cjmv4FlylCVr\\\\\\\\\\\\\\\\\\\\\\\\\\\\/u0HhqPtKNkw1EUmFyHyMxKBcpEtYmCW\\\\\\\\\\\\\\\\\\\\\\\\\\\\/6G\\\\\\\\\\\\\\\\\\\\\\\\\\\\/hDYzpYVbFgxsFYCDj5q244D14t+sCsiLgcAnRce3r1yxb8FfeDBQUA3poqxzMOVGnOjCem9+vYU3AV4ZtAUtW+\\\\\\\\\\\\\\\\\\\\\\\\\\\\/SccmOM6FLvIU1iRmx0mHMu4R9Gh1QuTDQmbsvtIuIlHe6e0LMWBIh0KPctQ6DuIiRT3rYSpmBQYSZ2JfISXBjkD0AH+epIiDC9IyODqtvuuuok75nk99DijUnS5Q85QGLfeCjhr\\\\\\\\\\\\\\\\\\\\\\\\\\\\/3WpnRi1sXb7trlwk6Dd3Ndc5dCTkGMTPwXbIbUBkHpt005IBBg+igPOHUvgILxsJm4kZT1pFDxg3L2yZ5gWiMBIHE7pcNnvu0+VE+vClDdB1Fzqk\\\\\\\\\\\\\\\\\\\\\\\\\\\\/bp++1Ii7yzTJ826jSj9ZAMXQzv90VUrUe3pO5B5IDpG4JwLptLi6eBhh5jXFN2fywYe3AazAk7olWMmd0aoJJbKski+Q9K2R5p0LvVcz5A8Nho3RL80kKUaKGJEsYGX4wxjPhW4m03YLcYQSKQu0dPCzRJ58254pUi+83wJguxinFcK0KTWJgjAkAUKtVSXA=\\\\\\\\\\\\u0022,\\\\\\\\\\\\u0022iv\\\\\\\\\\\\u0022:\\\\\\\\\\\\u0022+3OHMfVs2dwa72pzkvuKLA==\\\\\\\\\\\\u0022,\\\\\\\\\\\\u0022salt\\\\\\\\\\\\u0022:\\\\\\\\\\\\u0022f9d8845712f72c9fee5db7d9ff111eab\\\\\\\\\\\\u0022}\\\\u0022,\\\\u0022hmac\\\\u0022:\\\\u0022c07ae0a16c3ebb9397b9f2512d47f7b5d004b86cab2a3fb172d5d85ef6a94f72\\\\u0022}\"},\"urlIsAjaxTrusted\":{\"\\\\/content\\\\/10.1101\\\\/2021.05.25.445601v3.full\":true},\"ws_fl\":{\"width\":100,\"height\":21},\"ws_gpo\":{\"size\":\"\",\"annotation\":\"\",\"lang\":\"\",\"callback\":\"\",\"width\":300},\"color\":{\"logo\":\"https:\\\\/\\\\/www.biorxiv.org\\\\/sites\\\\/default\\\\/files\\\\/bioRxiv_article.jpg\"},\"highwire_list_expand\":{\"is_collapsed\":\"1\"},\"highwireResponsive\":{\"enquire_enabled\":1,\"breakpoints_configured\":1,\"breakpoints\":{\"zero\":\"all and (min-width: 0px)\",\"xsmall\":\"all and (min-width: 380px)\",\"narrow\":\"all and (min-width: 768px) and (min-device-width: 768px), (max-device-width: 800px) and (min-width: 768px) and (orientation:landscape)\",\"normal\":\"all and (min-width: 980px) and (min-device-width: 980px), all and (max-device-width: 1024px) and (min-width: 1024px) and (orientation:landscape)\",\"wide\":\"all and (min-width: 1220px)\"}},\"eu_cookie_compliance\":{\"popup_enabled\":1,\"popup_agreed_enabled\":0,\"popup_hide_agreed\":0,\"popup_clicking_confirmation\":1,\"popup_scrolling_confirmation\":false,\"popup_html_info\":\"\\\\u003Cdiv\\\\u003E\\\\n  \\\\u003Cdiv class =\\\\u0022popup-content info\\\\u0022\\\\u003E\\\\n    \\\\u003Cdiv id=\\\\u0022popup-text\\\\u0022\\\\u003E\\\\n      \\\\u003Cp\\\\u003EWe use cookies on this site to enhance your user experience. By clicking any link on this page you are giving your consent for us to set cookies.\\\\u003C\\\\/p\\\\u003E\\\\n    \\\\u003C\\\\/div\\\\u003E\\\\n    \\\\u003Cdiv id=\\\\u0022popup-buttons\\\\u0022\\\\u003E\\\\n      \\\\u003Cbutton type=\\\\u0022button\\\\u0022 class=\\\\u0022agree-button eu-cookie-compliance-default-button\\\\u0022\\\\u003EContinue\\\\u003C\\\\/button\\\\u003E\\\\n                    \\\\u003Cbutton type=\\\\u0022button\\\\u0022 class=\\\\u0022find-more-button eu-cookie-compliance-more-button\\\\u0022\\\\u003EFind out more\\\\u003C\\\\/button\\\\u003E\\\\n          \\\\u003C\\\\/div\\\\u003E\\\\n  \\\\u003C\\\\/div\\\\u003E\\\\n\\\\u003C\\\\/div\\\\u003E\",\"use_mobile_message\":false,\"mobile_popup_html_info\":\"\\\\u003Cdiv\\\\u003E\\\\n  \\\\u003Cdiv class =\\\\u0022popup-content info\\\\u0022\\\\u003E\\\\n    \\\\u003Cdiv id=\\\\u0022popup-text\\\\u0022\\\\u003E\\\\n          \\\\u003C\\\\/div\\\\u003E\\\\n    \\\\u003Cdiv id=\\\\u0022popup-buttons\\\\u0022\\\\u003E\\\\n      \\\\u003Cbutton type=\\\\u0022button\\\\u0022 class=\\\\u0022agree-button eu-cookie-compliance-default-button\\\\u0022\\\\u003EContinue\\\\u003C\\\\/button\\\\u003E\\\\n                    \\\\u003Cbutton type=\\\\u0022button\\\\u0022 class=\\\\u0022find-more-button eu-cookie-compliance-more-button\\\\u0022\\\\u003EFind out more\\\\u003C\\\\/button\\\\u003E\\\\n          \\\\u003C\\\\/div\\\\u003E\\\\n  \\\\u003C\\\\/div\\\\u003E\\\\n\\\\u003C\\\\/div\\\\u003E\\\\n\",\"mobile_breakpoint\":\"768\",\"popup_html_agreed\":\"\\\\u003Cdiv\\\\u003E\\\\n  \\\\u003Cdiv class=\\\\u0022popup-content agreed\\\\u0022\\\\u003E\\\\n    \\\\u003Cdiv id=\\\\u0022popup-text\\\\u0022\\\\u003E\\\\n      \\\\u003Ch2\\\\u003EThank you for accepting cookies\\\\u003C\\\\/h2\\\\u003E\\\\u003Cp\\\\u003EYou can now hide this message or find out more about cookies.\\\\u003C\\\\/p\\\\u003E    \\\\u003C\\\\/div\\\\u003E\\\\n    \\\\u003Cdiv id=\\\\u0022popup-buttons\\\\u0022\\\\u003E\\\\n      \\\\u003Cbutton type=\\\\u0022button\\\\u0022 class=\\\\u0022hide-popup-button eu-cookie-compliance-hide-button\\\\u0022\\\\u003EHide\\\\u003C\\\\/button\\\\u003E\\\\n              \\\\u003Cbutton type=\\\\u0022button\\\\u0022 class=\\\\u0022find-more-button eu-cookie-compliance-more-button-thank-you\\\\u0022 \\\\u003EMore info\\\\u003C\\\\/button\\\\u003E\\\\n          \\\\u003C\\\\/div\\\\u003E\\\\n  \\\\u003C\\\\/div\\\\u003E\\\\n\\\\u003C\\\\/div\\\\u003E\",\"popup_use_bare_css\":false,\"popup_height\":\"auto\",\"popup_width\":\"100%\",\"popup_delay\":1000,\"popup_link\":\"\\\\/help\\\\/cookie-policy\",\"popup_link_new_window\":1,\"popup_position\":null,\"popup_language\":\"en\",\"store_consent\":false,\"better_support_for_screen_readers\":0,\"reload_page\":0,\"domain\":\"\",\"popup_eu_only_js\":0,\"cookie_lifetime\":\"365\",\"cookie_session\":false,\"disagree_do_not_show_popup\":0,\"method\":\"default\",\"whitelisted_cookies\":\"\",\"withdraw_markup\":\"\\\\u003Cbutton type=\\\\u0022button\\\\u0022 class=\\\\u0022eu-cookie-withdraw-tab\\\\u0022\\\\u003EPrivacy settings\\\\u003C\\\\/button\\\\u003E\\\\n\\\\u003Cdiv class=\\\\u0022eu-cookie-withdraw-banner\\\\u0022\\\\u003E\\\\n  \\\\u003Cdiv class=\\\\u0022popup-content info\\\\u0022\\\\u003E\\\\n    \\\\u003Cdiv id=\\\\u0022popup-text\\\\u0022\\\\u003E\\\\n      \\\\u003Cp\\\\u003E\\\\u0026lt;h2\\\\u0026gt;We use cookies on this site to enhance your user experience\\\\u0026lt;\\\\/h2\\\\u0026gt;\\\\u0026lt;p\\\\u0026gt;You have given your consent for us to set cookies.\\\\u0026lt;\\\\/p\\\\u0026gt;\\\\u003C\\\\/p\\\\u003E\\\\n    \\\\u003C\\\\/div\\\\u003E\\\\n    \\\\u003Cdiv id=\\\\u0022popup-buttons\\\\u0022\\\\u003E\\\\n      \\\\u003Cbutton type=\\\\u0022button\\\\u0022 class=\\\\u0022eu-cookie-withdraw-button\\\\u0022\\\\u003EWithdraw consent\\\\u003C\\\\/button\\\\u003E\\\\n    \\\\u003C\\\\/div\\\\u003E\\\\n  \\\\u003C\\\\/div\\\\u003E\\\\n\\\\u003C\\\\/div\\\\u003E\\\\n\",\"withdraw_enabled\":false},\"googleanalytics\":{\"trackOutbound\":1,\"trackMailto\":1,\"trackDownload\":1,\"trackDownloadExtensions\":\"7z|aac|arc|arj|asf|asx|avi|bin|csv|doc(x|m)?|dot(x|m)?|exe|flv|gif|gz|gzip|hqx|jar|jpe?g|js|mp(2|3|4|e?g)|mov(ie)?|msi|msp|pdf|phps|png|ppt(x|m)?|pot(x|m)?|pps(x|m)?|ppam|sld(x|m)?|thmx|qtm?|ra(m|r)?|sea|sit|tar|tgz|torrent|txt|wav|wma|wmv|wpd|xls(x|m|b)?|xlt(x|m)|xlam|xml|z|zip\",\"trackColorbox\":1,\"trackUrlFragments\":1},\"jnl_biorxiv_styles\":{\"defaultJCode\":\"biorxiv\"},\"omega\":{\"layouts\":{\"primary\":\"normal\",\"order\":[\"narrow\",\"normal\",\"wide\"],\"queries\":{\"narrow\":\"all and (min-width: 768px) and (min-device-width: 768px), (max-device-width: 800px) and (min-width: 768px) and (orientation:landscape)\",\"normal\":\"all and (min-width: 980px) and (min-device-width: 980px), all and (max-device-width: 1024px) and (min-width: 1024px) and (orientation:landscape)\",\"wide\":\"all and (min-width: 1220px)\"}}}});\\n//--><!]]>\\n</script>\\n    <!--[if lt IE 9]><script src=\"http://html5shiv.googlecode.com/svn/trunk/html5.js\"></script><![endif]-->\\n  </head>\\n  <body class=\"html not-front not-logged-in page-node page-node- page-node-2125191 node-type-highwire-article context-content hw-default-jcode-biorxiv hw-article-type-article hw-article-category-new-results\">\\n<!-- Google Tag Manager -->\\n<noscript><iframe src=\"//www.googletagmanager.com/ns.html?id=GTM-M677548\" height=\"0\" width=\"0\" style=\"display:none;visibility:hidden\"></iframe></noscript>\\n<script type=\"text/javascript\">(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({\\'gtm.start\\':new Date().getTime(),event:\\'gtm.js\\'});var f=d.getElementsByTagName(s)[0];var j=d.createElement(s);var dl=l!=\\'dataLayer\\'?\\'&l=\\'+l:\\'\\';j.src=\\'//www.googletagmanager.com/gtm.js?id=\\'+i+dl;j.type=\\'text/javascript\\';j.async=true;f.parentNode.insertBefore(j,f);})(window,document,\\'script\\',\\'dataLayer\\',\\'GTM-M677548\\');</script>\\n<!-- End Google Tag Manager -->\\n    <div id=\"skip-link\">\\n      <a href=\"#main-content\" class=\"element-invisible element-focusable\">Skip to main content</a>\\n    </div>\\n        <div class=\"page clearfix page-box-shadows footer-borders panels-page panels-layout-jcore_2col\" id=\"page\">\\n      <header id=\"section-header\" class=\"section section-header\">\\n    \\n  <div id=\"zone-branding\" class=\"zone zone-branding clearfix print-display-block container-30\">\\n    <div class=\"grid-15 prefix-1 region region-branding print-display-block\" id=\"region-branding\">\\n  <div class=\"region-inner region-branding-inner\">\\n        <div class=\"branding-data clearfix\">\\n            <div class=\"logo-img\">\\n        <a href=\"/\" rel=\"home\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\"><img alt=\"bioRxiv\" src=\"https://www.biorxiv.org/sites/default/files/bioRxiv_article.jpg\" /></a>      </div>\\n                </div>\\n          </div>\\n</div><div class=\"grid-11 suffix-1 region region-branding-second print-hidden\" id=\"region-branding-second\">\\n  <div class=\"region-inner region-branding-second-inner\">\\n    <div class=\"block block-system block-menu block-main-menu block-system-main-menu odd block-without-title\" id=\"block-system-main-menu\">\\n  <div class=\"block-inner clearfix\">\\n                \\n    <div class=\"content clearfix\">\\n      <nav class=\"menubar-nav\"><ul class=\"menu\" role=\"menu\"><li class=\"first leaf\" role=\"menuitem\"><a href=\"/\" title=\"\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Home</a></li>\\n<li class=\"leaf\" role=\"menuitem\"><a href=\"/about-biorxiv\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">About</a></li>\\n<li class=\"leaf\" role=\"menuitem\"><a href=\"/submit-a-manuscript\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Submit</a></li>\\n<li class=\"last leaf\" role=\"menuitem\"><a href=\"/content/alertsrss\" title=\"\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">ALERTS / RSS</a></li>\\n</ul></nav>    </div>\\n  </div>\\n</div><div class=\"block block-panels-mini block-biorxiv-search-box block-panels-mini-biorxiv-search-box even block-without-title\" id=\"block-panels-mini-biorxiv-search-box\">\\n  <div class=\"block-inner clearfix\">\\n                \\n    <div class=\"content clearfix\">\\n      <div class=\"panel-display panel-1col clearfix\" id=\"mini-panel-biorxiv_search_box\">\\n  <div class=\"panel-panel panel-col\">\\n    <div><div class=\"panel-pane pane-highwire-seach-quicksearch\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <form class=\"highwire-quicksearch button-style-mini button-style-mini\" action=\"/content/10.1101/2021.05.25.445601v3.full\" method=\"post\" id=\"highwire-search-quicksearch-form-0\" accept-charset=\"UTF-8\"><div><div class=\"form-item form-item-label-invisible form-type-textfield form-item-keywords\">\\n  <label class=\"element-invisible\" for=\"search_rightsidebar_keywords_1492764509\">Search for this keyword </label>\\n <input placeholder=\"Search\" type=\"text\" id=\"search_rightsidebar_keywords_1492764509\" name=\"keywords\" value=\"\" size=\"60\" maxlength=\"128\" class=\"form-text\" />\\n</div>\\n<div class=\"button-wrapper button-mini\"><span class=\"icon-search\"></span><input data-icon-only=\"1\" data-font-icon=\"icon-search\" data-icon-position=\"after\" type=\"submit\" id=\"search_rightsidebar_submit_30200201\" name=\"op\" value=\"Search\" class=\"form-submit\" /></div><input type=\"hidden\" name=\"form_build_id\" value=\"form-ynmanQ3JYnWCGvocrHvLLY0oOoZMiaiz_V5bR_hvUoU\" />\\n<input type=\"hidden\" name=\"form_id\" value=\"highwire_search_quicksearch_form_0\" />\\n</div></form>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-custom pane-2 advanced-search-link\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <a href=\"/search\">Advanced Search</a>  </div>\\n\\n  \\n  </div>\\n</div>\\n  </div>\\n</div>\\n    </div>\\n  </div>\\n</div>  </div>\\n</div>  </div>\\n  \\n  <div id=\"zone-header\" class=\"zone zone-header clearfix container-30\">\\n  \\t      </div>\\n</header>    \\n      <section id=\"section-content\" class=\"section section-content\">\\n    \\n  <div id=\"zone-content\" class=\"zone zone-content clearfix container-30\">    \\n        \\n    <div class=\"grid-28 suffix-1 prefix-1 region region-content\" id=\"region-content\">\\n  <div class=\"region-inner region-content-inner\">\\n    <a id=\"main-content\"></a>\\n                        <div class=\"block block-system block-main block-system-main odd block-without-title\" id=\"block-system-main\">\\n  <div class=\"block-inner clearfix\">\\n                \\n    <div class=\"content clearfix\">\\n      <div class=\"panel-display panels-960-layout jcore-2col-layout\" >\\n\\t  \\n  <div class=\"panel-row-wrapper clearfix\">\\n\\t\\t\\n\\t\\t<div class=\"main-content-wrapper grid-17 suffix-1 alpha\">\\n\\t\\t\\t<div class=\"panel-panel panel-region-content\">\\n\\t\\t\\t  <div class=\"inside\"><div class=\"panel-pane pane-highwire-article-citation\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"highwire-article-citation highwire-citation-type-highwire-article node2125191\" data-node-nid=\"2125191\" id=\"node2125191--21877820458\" data-pisa=\"biorxiv;2021.05.25.445601v3\" data-pisa-master=\"biorxiv;2021.05.25.445601\" data-apath=\"/biorxiv/early/2021/08/26/2021.05.25.445601.atom\" data-hw-author-tooltip-instance=\"highwire_author_tooltip\"><div  class=\"highwire-cite highwire-cite-highwire-article highwire-citation-biorxiv-article-top clearfix has-author-tooltip\" >\\n\\n      <span class=\"biorxiv-article-type\">\\n        New Results    </span>\\n  \\n        <h1 class=\"highwire-cite-title\" id=\"page-title\">Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning</h1>  \\n      <div  class=\"highwire-cite-authors\" ><span  class=\"highwire-citation-authors\"><span class=\"highwire-citation-author first hw-author-orcid-logo-wrapper\" data-delta=\"0\"><a href=\"http://orcid.org/0000-0001-9936-7790\" target=\"_blank\" class=\"hw-author-orcid-logo link-icon-only link-icon\"><span class=\"hw-icon-orcid hw-icon-color-orcid\"></span> <span class=\"title element-invisible\">View ORCID Profile</span></a><span class=\"nlm-given-names\">Sargun</span> <span class=\"nlm-surname\">Nagpal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"1\"><span class=\"nlm-given-names\">Ridam</span> <span class=\"nlm-surname\">Pal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"2\"><span class=\"nlm-surname\">Ashima</span></span>, <span class=\"highwire-citation-author\" data-delta=\"3\"><span class=\"nlm-given-names\">Ananya</span> <span class=\"nlm-surname\">Tyagi</span></span>, <span class=\"highwire-citation-author\" data-delta=\"4\"><span class=\"nlm-given-names\">Sadhana</span> <span class=\"nlm-surname\">Tripathi</span></span>, <span class=\"highwire-citation-author hw-author-orcid-logo-wrapper\" data-delta=\"5\"><a href=\"http://orcid.org/0000-0002-6389-2179\" target=\"_blank\" class=\"hw-author-orcid-logo link-icon-only link-icon\"><span class=\"hw-icon-orcid hw-icon-color-orcid\"></span> <span class=\"title element-invisible\">View ORCID Profile</span></a><span class=\"nlm-given-names\">Aditya</span> <span class=\"nlm-surname\">Nagori</span></span>, <span class=\"highwire-citation-author\" data-delta=\"6\"><span class=\"nlm-given-names\">Saad</span> <span class=\"nlm-surname\">Ahmad</span></span>, <span class=\"highwire-citation-author hw-author-orcid-logo-wrapper\" data-delta=\"7\"><a href=\"http://orcid.org/0000-0001-9405-7822\" target=\"_blank\" class=\"hw-author-orcid-logo link-icon-only link-icon\"><span class=\"hw-icon-orcid hw-icon-color-orcid\"></span> <span class=\"title element-invisible\">View ORCID Profile</span></a><span class=\"nlm-given-names\">Hara Prasad</span> <span class=\"nlm-surname\">Mishra</span></span>, <span class=\"highwire-citation-author hw-author-orcid-logo-wrapper\" data-delta=\"8\"><a href=\"http://orcid.org/0000-0001-8667-6199\" target=\"_blank\" class=\"hw-author-orcid-logo link-icon-only link-icon\"><span class=\"hw-icon-orcid hw-icon-color-orcid\"></span> <span class=\"title element-invisible\">View ORCID Profile</span></a><span class=\"nlm-given-names\">Rintu</span> <span class=\"nlm-surname\">Kutum</span></span>, <span class=\"highwire-citation-author hw-author-orcid-logo-wrapper\" data-delta=\"9\"><a href=\"http://orcid.org/0000-0002-4776-7941\" target=\"_blank\" class=\"hw-author-orcid-logo link-icon-only link-icon\"><span class=\"hw-icon-orcid hw-icon-color-orcid\"></span> <span class=\"title element-invisible\">View ORCID Profile</span></a><span class=\"nlm-given-names\">Tavpritesh</span> <span class=\"nlm-surname\">Sethi</span></span></span></div>\\n  \\n      <div  class=\"highwire-cite-metadata\" ><span  class=\"highwire-cite-metadata-doi highwire-cite-metadata\"><span class=\"label\">doi:</span> https://doi.org/10.1101/2021.05.25.445601 </span></div>\\n  \\n  \\n  </div>\\n<div id=\"hw-article-author-popups-node2125191--21877820458\" style=\"display: none;\"><div class=\"author-tooltip-0\"><div class=\"author-tooltip-name\">Sargun Nagpal </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>1</span><span class=\\'nlm-institution\\'>Indraprastha Institute of Information Technology Delhi</span>, <span class=\\'nlm-country\\'>India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Sargun%2BNagpal%2B\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a href=\"/lookup/external-ref?access_num=Nagpal%20S&amp;link_type=AUTHORSEARCH\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on PubMed</a></li><li class=\"author-site-search-link\"><a href=\"/search/author1%3ASargun%2BNagpal%2B\" rel=\"nofollow\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Search for this author on this site</a></li><li class=\"author-orcid-link last\"><a href=\"http://orcid.org/0000-0001-9936-7790\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">ORCID record for Sargun Nagpal</a></li></ul></div><div class=\"author-tooltip-1\"><div class=\"author-tooltip-name\">Ridam Pal </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>1</span><span class=\\'nlm-institution\\'>Indraprastha Institute of Information Technology Delhi</span>, <span class=\\'nlm-country\\'>India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Ridam%2BPal%2B\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a href=\"/lookup/external-ref?access_num=Pal%20R&amp;link_type=AUTHORSEARCH\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on PubMed</a></li><li class=\"author-site-search-link last\"><a href=\"/search/author1%3ARidam%2BPal%2B\" rel=\"nofollow\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Search for this author on this site</a></li></ul></div><div class=\"author-tooltip-2\"><div class=\"author-tooltip-name\"> Ashima </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>1</span><span class=\\'nlm-institution\\'>Indraprastha Institute of Information Technology Delhi</span>, <span class=\\'nlm-country\\'>India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=%2BAshima%2B\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a href=\"/lookup/external-ref?access_num=Ashima&amp;link_type=AUTHORSEARCH\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on PubMed</a></li><li class=\"author-site-search-link last\"><a href=\"/search/author1%3A%2BAshima%2B\" rel=\"nofollow\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Search for this author on this site</a></li></ul></div><div class=\"author-tooltip-3\"><div class=\"author-tooltip-name\">Ananya Tyagi </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>1</span><span class=\\'nlm-institution\\'>Indraprastha Institute of Information Technology Delhi</span>, <span class=\\'nlm-country\\'>India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Ananya%2BTyagi%2B\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a href=\"/lookup/external-ref?access_num=Tyagi%20A&amp;link_type=AUTHORSEARCH\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on PubMed</a></li><li class=\"author-site-search-link last\"><a href=\"/search/author1%3AAnanya%2BTyagi%2B\" rel=\"nofollow\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Search for this author on this site</a></li></ul></div><div class=\"author-tooltip-4\"><div class=\"author-tooltip-name\">Sadhana Tripathi </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>1</span><span class=\\'nlm-institution\\'>Indraprastha Institute of Information Technology Delhi</span>, <span class=\\'nlm-country\\'>India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Sadhana%2BTripathi%2B\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a href=\"/lookup/external-ref?access_num=Tripathi%20S&amp;link_type=AUTHORSEARCH\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on PubMed</a></li><li class=\"author-site-search-link last\"><a href=\"/search/author1%3ASadhana%2BTripathi%2B\" rel=\"nofollow\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Search for this author on this site</a></li></ul></div><div class=\"author-tooltip-5\"><div class=\"author-tooltip-name\">Aditya Nagori </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>1</span><span class=\\'nlm-institution\\'>Indraprastha Institute of Information Technology Delhi</span>, <span class=\\'nlm-country\\'>India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Aditya%2BNagori%2B\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a href=\"/lookup/external-ref?access_num=Nagori%20A&amp;link_type=AUTHORSEARCH\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on PubMed</a></li><li class=\"author-site-search-link\"><a href=\"/search/author1%3AAditya%2BNagori%2B\" rel=\"nofollow\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Search for this author on this site</a></li><li class=\"author-orcid-link last\"><a href=\"http://orcid.org/0000-0002-6389-2179\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">ORCID record for Aditya Nagori</a></li></ul></div><div class=\"author-tooltip-6\"><div class=\"author-tooltip-name\">Saad Ahmad </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>1</span><span class=\\'nlm-institution\\'>Indraprastha Institute of Information Technology Delhi</span>, <span class=\\'nlm-country\\'>India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Saad%2BAhmad%2B\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a href=\"/lookup/external-ref?access_num=Ahmad%20S&amp;link_type=AUTHORSEARCH\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on PubMed</a></li><li class=\"author-site-search-link last\"><a href=\"/search/author1%3ASaad%2BAhmad%2B\" rel=\"nofollow\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Search for this author on this site</a></li></ul></div><div class=\"author-tooltip-7\"><div class=\"author-tooltip-name\">Hara Prasad Mishra </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>1</span><span class=\\'nlm-institution\\'>Indraprastha Institute of Information Technology Delhi</span>, <span class=\\'nlm-country\\'>India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Hara%2BPrasad%2BMishra%2B\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a href=\"/lookup/external-ref?access_num=Mishra%20HP&amp;link_type=AUTHORSEARCH\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on PubMed</a></li><li class=\"author-site-search-link\"><a href=\"/search/author1%3AHara%2BPrasad%2BMishra%2B\" rel=\"nofollow\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Search for this author on this site</a></li><li class=\"author-orcid-link last\"><a href=\"http://orcid.org/0000-0001-9405-7822\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">ORCID record for Hara Prasad Mishra</a></li></ul></div><div class=\"author-tooltip-8\"><div class=\"author-tooltip-name\">Rintu Kutum </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>1</span><span class=\\'nlm-institution\\'>Indraprastha Institute of Information Technology Delhi</span>, <span class=\\'nlm-country\\'>India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Rintu%2BKutum%2B\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a href=\"/lookup/external-ref?access_num=Kutum%20R&amp;link_type=AUTHORSEARCH\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on PubMed</a></li><li class=\"author-site-search-link\"><a href=\"/search/author1%3ARintu%2BKutum%2B\" rel=\"nofollow\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Search for this author on this site</a></li><li class=\"author-orcid-link\"><a href=\"http://orcid.org/0000-0001-8667-6199\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">ORCID record for Rintu Kutum</a></li><li class=\"author-corresp-email-link last\"><span>For correspondence: \\n<a href=\"mailto:tavpriteshsethi@iiitd.ac.in\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">tavpriteshsethi@iiitd.ac.in</a>\\n<a href=\"mailto:rintuk@iiitd.ac.in\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">rintuk@iiitd.ac.in</a></span></li></ul></div><div class=\"author-tooltip-9\"><div class=\"author-tooltip-name\">Tavpritesh Sethi </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>1</span><span class=\\'nlm-institution\\'>Indraprastha Institute of Information Technology Delhi</span>, <span class=\\'nlm-country\\'>India</span></div><div class=\\'author-affiliation\\'><span class=\\'nlm-sup\\'>2</span><span class=\\'nlm-institution\\'>All India Institute of Medical Sciences</span>, New Delhi, <span class=\\'nlm-country\\'>India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Tavpritesh%2BSethi%2B\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a href=\"/lookup/external-ref?access_num=Sethi%20T&amp;link_type=AUTHORSEARCH\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Find this author on PubMed</a></li><li class=\"author-site-search-link\"><a href=\"/search/author1%3ATavpritesh%2BSethi%2B\" rel=\"nofollow\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Search for this author on this site</a></li><li class=\"author-orcid-link\"><a href=\"http://orcid.org/0000-0002-4776-7941\" target=\"_blank\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">ORCID record for Tavpritesh Sethi</a></li><li class=\"author-corresp-email-link last\"><span>For correspondence: \\n<a href=\"mailto:tavpriteshsethi@iiitd.ac.in\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">tavpriteshsethi@iiitd.ac.in</a>\\n<a href=\"mailto:rintuk@iiitd.ac.in\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">rintuk@iiitd.ac.in</a></span></li></ul></div></div></div>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-panel-tabs pane-panels-ajax-tab-tabs\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"item-list\"><ul class=\"tabs inline panels-ajax-tab\"><li class=\"first\"><a href=\"/content/10.1101/2021.05.25.445601v3\" class=\"panels-ajax-tab-tab\" data-panel-name=\"biorxiv_tab_art\" data-target-id=\"highwire_article_tabs\" data-entity-context=\"node:2125191\" data-trigger=\"\" data-url-enabled=\"1\">Abstract</a><a href=\"/panels_ajax_tab/biorxiv_tab_art/node:2125191/1\" rel=\"nofollow\" style=\"display:none\" class=\"js-crawler-link\"></a></li><li><a href=\"/content/10.1101/2021.05.25.445601v3.full-text\" class=\"panels-ajax-tab-tab\" data-panel-name=\"article_tab_full_text\" data-target-id=\"highwire_article_tabs\" data-entity-context=\"node:2125191\" data-trigger=\"full-text\" data-url-enabled=\"1\">Full Text</a><a href=\"/panels_ajax_tab/article_tab_full_text/node:2125191/1\" rel=\"nofollow\" style=\"display:none\" class=\"js-crawler-link\"></a></li><li><a href=\"/content/10.1101/2021.05.25.445601v3.article-info\" class=\"panels-ajax-tab-tab\" data-panel-name=\"biorxiv_tab_info\" data-target-id=\"highwire_article_tabs\" data-entity-context=\"node:2125191\" data-trigger=\"article-info\" data-url-enabled=\"1\">Info/History</a><a href=\"/panels_ajax_tab/biorxiv_tab_info/node:2125191/1\" rel=\"nofollow\" style=\"display:none\" class=\"js-crawler-link\"></a></li><li><a href=\"/content/10.1101/2021.05.25.445601v3.article-metrics\" class=\"panels-ajax-tab-tab\" data-panel-name=\"article_tab_metrics\" data-target-id=\"highwire_article_tabs\" data-entity-context=\"node:2125191\" data-trigger=\"article-metrics\" data-url-enabled=\"1\">Metrics</a><a href=\"/panels_ajax_tab/article_tab_metrics/node:2125191/1\" rel=\"nofollow\" style=\"display:none\" class=\"js-crawler-link\"></a></li><li><a href=\"/content/10.1101/2021.05.25.445601v3.supplementary-material\" class=\"panels-ajax-tab-tab\" data-panel-name=\"biorxiv_tab_data\" data-target-id=\"highwire_article_tabs\" data-entity-context=\"node:2125191\" data-trigger=\"supplementary-material\" data-url-enabled=\"1\">Supplementary material</a><a href=\"/panels_ajax_tab/biorxiv_tab_data/node:2125191/1\" rel=\"nofollow\" style=\"display:none\" class=\"js-crawler-link\"></a></li><li class=\"last\"><a href=\"/content/10.1101/2021.05.25.445601v3.full.pdf+html\" class=\"panels-ajax-tab-tab\" data-panel-name=\"biorxiv_tab_pdf\" data-target-id=\"highwire_article_tabs\" data-entity-context=\"node:2125191\" data-trigger=\"full.pdf+html\" data-url-enabled=\"1\"><i class=\"icon-file-alt\"></i> Preview PDF</a><a href=\"/panels_ajax_tab/biorxiv_tab_pdf/node:2125191/1\" rel=\"nofollow\" style=\"display:none\" class=\"js-crawler-link\"></a></li></ul></div>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-panel-tabs-container\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div data-panels-ajax-tab-preloaded=\"article_tab_full_text\" id=\"panels-ajax-tab-container-highwire_article_tabs\" class=\"panels-ajax-tab-container\"><div class=\"panels-ajax-tab-loading\" style =\"display:none\"><img class=\"loading\" src=\"https://www.biorxiv.org/sites/all/modules/contrib/panels_ajax_tab/images/loading.gif\" alt=\"Loading\" title=\"Loading\" /></div><div class=\"panels-ajax-tab-wrap-article_tab_full_text\"><div class=\"panel-display panel-1col clearfix\" >\\n  <div class=\"panel-panel panel-col\">\\n    <div><div class=\"panel-pane pane-highwire-markup\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"highwire-markup\"><div xmlns=\"http://www.w3.org/1999/xhtml\" data-highwire-cite-ref-tooltip-instance=\"highwire_reflinks_tooltip\" class=\"content-block-markup\" xmlns:xhtml=\"http://www.w3.org/1999/xhtml\"><div class=\"article fulltext-view \"><span class=\"highwire-journal-article-marker-start\"></span><div class=\"section abstract\" id=\"abstract-1\"><h2 class=\"\">Abstract</h2><p id=\"p-4\">The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed <em>Strainflow</em>, to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our <em>Strainflow</em> model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called <em>Dimensions of Concern</em> (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.</p><div id=\"sec-1\" class=\"subsection\"><p><strong>Highlights</strong></p><ul class=\"list-unord \" id=\"list-1\"><li id=\"list-item-1\"><p id=\"p-5\">We developed a genomic surveillance model for SARS-CoV-2 genome sequences, <em>Strainflow</em>, where sequences were treated as documents with words (codons) to learn the codon context of 0.9 million spike genes using the skip-gram algorithm.</p></li><li id=\"list-item-2\"><p id=\"p-6\">Time series analysis of the information content (Entropy) of the latent dimensions learned by <em>Strainflow</em> shows a leading relationship with the monthly COVID-19 cases for seven countries (e.g., USA, Japan, India, and others).</p></li><li id=\"list-item-3\"><p id=\"p-7\">Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads.</p></li><li id=\"list-item-4\"><p id=\"p-8\">The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants.</p></li></ul></div><div id=\"sec-2\" class=\"subsection\"><div id=\"F1\" class=\"fig pos-float type-figure  odd\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F1.large.jpg?width=800&amp;height=600&amp;carousel=1\" title=\"\" class=\"highwire-fragment fragment-images colorbox-load\" rel=\"gallery-fragment-images-1181327286\" data-figure-caption=\"&lt;div class=&quot;highwire-markup&quot;&gt;&lt;/div&gt;\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"hw-responsive-img\"><img class=\"highwire-fragment fragment-image lazyload\" alt=\"Figure\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F1.medium.gif\" width=\"434\" height=\"440\"/><noscript><img class=\"highwire-fragment fragment-image\" alt=\"Figure\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F1.medium.gif\" width=\"434\" height=\"440\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F1.large.jpg?download=true\" class=\"highwire-figure-link highwire-figure-link-download\" title=\"Download Figure1\" data-icon-position=\"\" data-hide-link-title=\"0\">Download figure</a></li><li class=\"new-tab last\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F1.large.jpg\" class=\"highwire-figure-link highwire-figure-link-newtab\" target=\"_blank\" data-icon-position=\"\" data-hide-link-title=\"0\">Open in new tab</a></li></ul></div></div></div></div><div class=\"section\" id=\"sec-3\"><h2 class=\"\">Introduction</h2><p id=\"p-11\">COVID-19 is reported to have claimed 4.37 million lives as of Aug 17, 2021 (WHO, 2021). A large number of these deaths are attributed to unexpected surges in infections caused by new strains of SARS-CoV-2, prompting international health organizations such as the CDC and WHO to declare these as variants of concerns (<a id=\"xref-ref-6-1\" class=\"xref-bibr\" href=\"#ref-6\">Khan et al., 2020</a>). Such emergence of new variants can seriously undermine the efficacy of global vaccination programs, the need for future booster doses, or cause multiple reinfection waves as new strains escape previously developed antibodies. Several initiatives have focused on providing high-quality tracking information for the strains and lineages as these emerge (<a id=\"xref-ref-4-1\" class=\"xref-bibr\" href=\"#ref-4\">Hadfield et al., 2018</a>). However, early prediction of emerging variants through genomic signals remains an open challenge. Although domain-based, expert-reasoning approaches are the mainstay of our understanding, which often yield retrospective insight rather than proactive predictions. On the other hand, machine learning approaches are likely to be biased by underlying data characteristics and do not explain the biological basis of predictions.</p><p id=\"p-12\">Here, we propose <strong><em>Strainflow</em></strong>, a hybrid architecture of machine learning and language modeling, along with empirical experiments to demonstrate explainable genomic signals for tracking and predicting the spread of SARS-CoV-2 across countries. Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences. Typically, in language models, word embeddings represent the latent space (dimensions) of a corpus of text (<a id=\"xref-ref-16-1\" class=\"xref-bibr\" href=\"#ref-16\">Mikolov et al., 2013</a>) and can capture highly nonlinear and contextual relationships. Codons (tri-nucleotides, 3-mers) translations represent a natural basis for word representations and have been utilized in the past for learning embedding models for modeling various outcomes such as mutation susceptibility (<a id=\"xref-ref-36-1\" class=\"xref-bibr\" href=\"#ref-36\">Yilmaz, 2020</a>) and gene sequence correlations (<a id=\"xref-ref-35-1\" class=\"xref-bibr\" href=\"#ref-35\">Wu et al., 2021</a>). Recently, Hie et. al (<a id=\"xref-ref-5-1\" class=\"xref-bibr\" href=\"#ref-5\">Hie et al., 2021</a>) used machine learning along with word embedding techniques to model the semantics and grammar of amino acids corresponding to antigenic change to predict the mutations which might lead to viral escape.</p><p id=\"p-13\">In this paper, we focus on the semantics of viral DNA sequences to derive Dimensions of Concern (DoCs) and demonstrate their causal potential for increasing epidemiological spread patterns across 7 countries. Our approach, Strainflow, is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance.</p></div><div class=\"section\" id=\"sec-4\"><h2 class=\"\">Results</h2><div id=\"sec-5\" class=\"subsection\"><h3>Genomic sequence-based language modeling captures emerging diversity in the SARS-CoV-2 spike gene</h3><p id=\"p-14\">Following the linguistic idea that the context of a word is characterized by its neighborhood, we leveraged a word2vec language model and developed “<em>Strainflow</em>” to learn the latent representations of codons in spike gene sequences of SARS-CoV-2. Strainflow is trained on 0.3 million high-quality genome sequences from March 1, 2020 to March 31, 2021, collected from the GISAID database (<strong><a id=\"xref-fig-2-1\" class=\"xref-fig\" href=\"#F2\">Fig 1A</a></strong>). To understand the latent space representation (LR) learned by Strainflow, we performed qualitative analysis using tSNE for both global and local (country) levels on a monthly basis. The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 (<strong><a id=\"xref-fig-2-2\" class=\"xref-fig\" href=\"#F2\">Fig 1B</a></strong>) from September, 2020 to March, 2021, along with specific geographic locations (country-level) such as India, UK, USA, and China.</p><div id=\"F2\" class=\"fig pos-float type-figure  odd\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F2.large.jpg?width=800&amp;height=600&amp;carousel=1\" title=\"Latent space of spike gene derived using Strainflow preserves spatiotemporal information of SARS-CoV-2 spread. (A) The implementation framework of Strainflow (details described in the method section) (B) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China. (C) Embeddings estimated or predicted from the Strainflow model for 0.36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021. (D) Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021. For Pakistan, we couldn&#x2019;t estimate the fast sample entropy due to fewer number of strain sequences (\" class=\"highwire-fragment fragment-images colorbox-load\" rel=\"gallery-fragment-images-1181327286\" data-figure-caption=\"&lt;div class=&quot;highwire-markup&quot;&gt;&lt;div xmlns=&quot;http://www.w3.org/1999/xhtml&quot;&gt;&lt;span class=&quot;caption-title&quot;&gt;Latent space of spike gene derived using Strainflow preserves spatiotemporal information of SARS-CoV-2 spread.&lt;/span&gt; (&lt;strong&gt;A&lt;/strong&gt;) The implementation framework of Strainflow (details described in the method section) (&lt;strong&gt;B&lt;/strong&gt;) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China. (&lt;strong&gt;C&lt;/strong&gt;) Embeddings estimated or predicted from the Strainflow model for 0.36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021. &lt;strong&gt;(D)&lt;/strong&gt; Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021. For Pakistan, we couldn&#x2019;t estimate the fast sample entropy due to fewer number of strain sequences (&lt;20).&lt;/div&gt;&lt;/div&gt;\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"hw-responsive-img\"><img class=\"highwire-fragment fragment-image lazyload\" alt=\"Figure 1:\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F2.medium.gif\" width=\"317\" height=\"440\"/><noscript><img class=\"highwire-fragment fragment-image\" alt=\"Figure 1:\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F2.medium.gif\" width=\"317\" height=\"440\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F2.large.jpg?download=true\" class=\"highwire-figure-link highwire-figure-link-download\" title=\"Download Figure 1:\" data-icon-position=\"\" data-hide-link-title=\"0\">Download figure</a></li><li class=\"new-tab last\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F2.large.jpg\" class=\"highwire-figure-link highwire-figure-link-newtab\" target=\"_blank\" data-icon-position=\"\" data-hide-link-title=\"0\">Open in new tab</a></li></ul></div><div class=\"fig-caption\" xmlns:xhtml=\"http://www.w3.org/1999/xhtml\"><span class=\"fig-label\">Figure 1:</span> <span class=\"caption-title\">Latent space of spike gene derived using Strainflow preserves spatiotemporal information of SARS-CoV-2 spread.</span><p id=\"p-15\" class=\"first-child\">(<strong>A</strong>) The implementation framework of Strainflow (details described in the method section) (<strong>B</strong>) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China. (<strong>C</strong>) Embeddings estimated or predicted from the Strainflow model for 0.36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021. <strong>(D)</strong> Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021. For Pakistan, we couldn’t estimate the fast sample entropy due to fewer number of strain sequences (&lt;20).</p><div class=\"sb-div caption-clear\"></div></div></div><p id=\"p-16\">To investigate the information content in the latent dimensions or space (LD or LS) of the <em>Strainflow</em> model, we performed qualitative and quantitative analysis on 0.9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021. Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (<a id=\"xref-ref-12-1\" class=\"xref-bibr\" href=\"#ref-12\">Linderman et al., 2019</a>). We compared the 2D t-SNE plot of the world with four countries (India, UK, USA, China) from September, 2020 to June, 2021, which clearly highlights the dynamic changes in the spike genes across countries in different months (<strong><a id=\"xref-fig-2-3\" class=\"xref-fig\" href=\"#F2\">Fig 1B</a></strong> and <strong><a id=\"xref-fig-2-4\" class=\"xref-fig\" href=\"#F2\">Fig 1C</a></strong>). Additionally, the quantitative analysis of the latent space was performed by calculating the fast sample entropy (<a id=\"xref-ref-19-1\" class=\"xref-bibr\" href=\"#ref-19\">Pan et al., 2011</a>) for each LD. To compare the monthly entropy of the LS per country, mean entropy was calculated and normalized across the months for each country. We observed the highest entropy (information content) for India, UK, USA and China in the months of February-2021, July-2020, August-2020, and January-2021 respectively (<strong><a id=\"xref-fig-2-5\" class=\"xref-fig\" href=\"#F2\">Fig 1D</a></strong>). Interestingly, we observed high entropy for 4 months from August, 2020 to November, 2020 in the USA (<strong><a id=\"xref-fig-2-6\" class=\"xref-fig\" href=\"#F2\">Fig 1D</a></strong>). This highlights that the spike protein latent space representation learned by the Strainflow model could be used as a proxy to capture the spatiotemporal entropy or diversity in the emerging SARS-CoV-2 strains across different countries.</p></div><div id=\"sec-6\" class=\"subsection\"><h3>Preservation of spatiotemporal information for SARS-CoV-2 spread depicted through phylogenetic analysis</h3><p id=\"p-17\">Sequence-level embeddings were obtained from the codon embeddings and investigated for the presence of genomically meaningful characteristics. The phylogenetic tree derived from the embeddings for the United Kingdom (<strong><a id=\"xref-fig-3-1\" class=\"xref-fig\" href=\"#F3\">Fig 2A</a></strong>) shows two clear temporally split clusters for 2020 and 2021 sequences, which may be indicative of different strains in these time periods. The temporality of the collected sequences was found to be preserved in the two clusters, although the model was trained only on genome sequences.</p><div id=\"F3\" class=\"fig pos-float type-figure  odd\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F3.large.jpg?width=800&amp;height=600&amp;carousel=1\" title=\"Phylogenetic trees constructed using cosine similarities between 400 randomly sampled sequence embeddings. (A) Dendrogram for strains from the U.K.: Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021. (B) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.\" class=\"highwire-fragment fragment-images colorbox-load\" rel=\"gallery-fragment-images-1181327286\" data-figure-caption=\"&lt;div class=&quot;highwire-markup&quot;&gt;&lt;div xmlns=&quot;http://www.w3.org/1999/xhtml&quot;&gt;&lt;span class=&quot;caption-title&quot;&gt;Phylogenetic trees constructed using cosine similarities between 400 randomly sampled sequence embeddings.&lt;/span&gt; (&lt;strong&gt;A&lt;/strong&gt;) Dendrogram for strains from the U.K.: Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021. (&lt;strong&gt;B&lt;/strong&gt;) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.&lt;/div&gt;&lt;/div&gt;\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"hw-responsive-img\"><img class=\"highwire-fragment fragment-image lazyload\" alt=\"Figure 2:\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F3.medium.gif\" width=\"440\" height=\"227\"/><noscript><img class=\"highwire-fragment fragment-image\" alt=\"Figure 2:\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F3.medium.gif\" width=\"440\" height=\"227\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F3.large.jpg?download=true\" class=\"highwire-figure-link highwire-figure-link-download\" title=\"Download Figure 2:\" data-icon-position=\"\" data-hide-link-title=\"0\">Download figure</a></li><li class=\"new-tab last\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F3.large.jpg\" class=\"highwire-figure-link highwire-figure-link-newtab\" target=\"_blank\" data-icon-position=\"\" data-hide-link-title=\"0\">Open in new tab</a></li></ul></div><div class=\"fig-caption\"><span class=\"fig-label\">Figure 2:</span> <span class=\"caption-title\">Phylogenetic trees constructed using cosine similarities between 400 randomly sampled sequence embeddings.</span><p id=\"p-18\" class=\"first-child\">(<strong>A</strong>) Dendrogram for strains from the U.K.: Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021. (<strong>B</strong>) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.</p><div class=\"sb-div caption-clear\"></div></div></div><p id=\"p-19\">The phylogenetic tree with globally collected sequences (<strong><a id=\"xref-fig-3-2\" class=\"xref-fig\" href=\"#F3\">Fig 2B</a></strong>) demonstrates that geospatial information is also preserved in the sequence embeddings. Strains from countries like China, Australia, Mexico, and England were found in distinct clusters, while those from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India were found to be dispersed with other countries. This behavior is expected because viral strains spread from one country to another, and our samples had different collection dates for each country. Overall, Strainflow captures the temporal emergence of strains and geographic information in a country-specific manner.</p></div><div id=\"sec-7\" class=\"subsection\"><h3>Entropy in the latent space dimensions captures variability in the spike gene</h3><p id=\"p-20\">Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time. We attributed these changes to mutations in the spike protein of the genome sequence. This hypothesis was based on the fact that the embedding of a word (codon) represents the context it occurs in. A mutation results in a codon that is known to be found in a different context. This alters the semantic sense of the sequence, and this modification of meaning is reflected as a change in values across different latent dimensions. The changed values cause a change in the degree of disorder or the entropy of the latent space.</p><p id=\"p-21\">To compare different geographical regions, the sum of sample entropy was computed for each latent dimension across all the months. This revealed that certain geospatial regions such as France and Germany (<strong><a id=\"xref-fig-4-1\" class=\"xref-fig\" href=\"#F4\">Fig 3A</a></strong>) and USA and Canada (<strong><a id=\"xref-fig-4-2\" class=\"xref-fig\" href=\"#F4\">Fig 3B</a></strong>) have similar total entropies across the latent dimensions, indicating that these regions have been accumulating similar changes.</p><div id=\"F4\" class=\"fig pos-float type-figure  odd\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F4.large.jpg?width=800&amp;height=600&amp;carousel=1\" title=\"Sum of sample entropy for each latent dimension for different countries. Country pairs (A) - France and Germany, (B) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.\" class=\"highwire-fragment fragment-images colorbox-load\" rel=\"gallery-fragment-images-1181327286\" data-figure-caption=\"&lt;div class=&quot;highwire-markup&quot;&gt;&lt;div xmlns=&quot;http://www.w3.org/1999/xhtml&quot;&gt;&lt;span class=&quot;caption-title&quot;&gt;Sum of sample entropy for each latent dimension for different countries.&lt;/span&gt; Country pairs (&lt;strong&gt;A&lt;/strong&gt;) - France and Germany, (&lt;strong&gt;B&lt;/strong&gt;) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.&lt;/div&gt;&lt;/div&gt;\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"hw-responsive-img\"><img class=\"highwire-fragment fragment-image lazyload\" alt=\"Figure 3:\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F4.medium.gif\" width=\"440\" height=\"136\"/><noscript><img class=\"highwire-fragment fragment-image\" alt=\"Figure 3:\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F4.medium.gif\" width=\"440\" height=\"136\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F4.large.jpg?download=true\" class=\"highwire-figure-link highwire-figure-link-download\" title=\"Download Figure 3:\" data-icon-position=\"\" data-hide-link-title=\"0\">Download figure</a></li><li class=\"new-tab last\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F4.large.jpg\" class=\"highwire-figure-link highwire-figure-link-newtab\" target=\"_blank\" data-icon-position=\"\" data-hide-link-title=\"0\">Open in new tab</a></li></ul></div><div class=\"fig-caption\"><span class=\"fig-label\">Figure 3:</span> <span class=\"caption-title\">Sum of sample entropy for each latent dimension for different countries.</span><p id=\"p-22\" class=\"first-child\">Country pairs (<strong>A</strong>) - France and Germany, (<strong>B</strong>) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.</p><div class=\"sb-div caption-clear\"></div></div></div></div><div id=\"sec-8\" class=\"subsection\"><h3>Entropy dimensions are predictive of new COVID-19 caseloads</h3><p id=\"p-23\">We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries. Detrended cross-correlation coefficient was calculated at different lag values, which revealed that entropy dimensions have a leading relationship with new cases (<strong><a id=\"xref-fig-5-1\" class=\"xref-fig\" href=\"#F5\">Fig 4A, 4B</a></strong>). This suggests that the genome sequence data in a given month can be used to predict new cases in subsequent months. A lead period of two months was chosen and Boruta algorithm was employed to assign feature importance scores to different dimensions, which revealed that dimension 32 is the most significant predictor of new cases <strong>(<a id=\"xref-fig-5-2\" class=\"xref-fig\" href=\"#F5\">Fig 4C</a>)</strong>. Significant dimensions from Boruta analysis were termed Dimensions of Concern (DoCs). Random forest based regression modeling on the DoCs achieved a total R-squared of 73% on the test set. The predicted cases were found to be highly correlated with the actual cases (<strong><a id=\"xref-table-wrap-1-1\" class=\"xref-table\" href=\"#T1\">Table 1</a></strong>), which suggests that our model can indicate the directional change of cases for different countries. Further, the predicted relative change in cases between successive months was found to be correlated to the actual relative changes (<strong>Table S1</strong>), which suggests that our model can also indicate the magnitude of change that we expect to observe in the cases.</p><div id=\"T1\" class=\"table pos-float\"><div class=\"table-inline table-callout-links\"><div class=\"callout\"><span>View this table:</span><ul class=\"callout-links\"><li class=\"view-inline first\"><a href=\"##\" class=\"table-expand-inline\" data-table-url=\"/highwire/markup/2128021/expansion?postprocessors=highwire_tables%2Chighwire_reclass%2Chighwire_figures%2Chighwire_math%2Chighwire_inline_linked_media%2Chighwire_embed&amp;table-expand-inline=1\" data-icon-position=\"\" data-hide-link-title=\"0\">View inline</a></li><li class=\"view-popup\"><a href=\"/highwire/markup/2128021/expansion?width=1000&amp;height=500&amp;iframe=true&amp;postprocessors=highwire_tables%2Chighwire_reclass%2Chighwire_figures%2Chighwire_math%2Chighwire_inline_linked_media%2Chighwire_embed\" class=\"colorbox colorbox-load table-expand-popup\" rel=\"gallery-fragment-tables\" data-icon-position=\"\" data-hide-link-title=\"0\">View popup</a></li><li class=\"download-ppt last\"><a href=\"/highwire/powerpoint/2128021\" class=\"highwire-figure-link highwire-figure-link-ppt\" data-icon-position=\"\" data-hide-link-title=\"0\">Download powerpoint</a></li></ul></div></div><div class=\"table-caption\"><span class=\"table-label\">Table 1:</span> <span class=\"caption-title\">Pearson and Spearman’s Correlation coefficients between predicted and actual cases in different countries.</span><div class=\"sb-div caption-clear\"></div></div></div><div id=\"F5\" class=\"fig pos-float type-figure  odd\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F5.large.jpg?width=800&amp;height=600&amp;carousel=1\" title=\"Relationship of the entropy of latent space dimensions with COVID-19 caseloads. (A) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA. High values are observed for a lead of 1 and 2 months. (B) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases. (C) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.\" class=\"highwire-fragment fragment-images colorbox-load\" rel=\"gallery-fragment-images-1181327286\" data-figure-caption=\"&lt;div class=&quot;highwire-markup&quot;&gt;&lt;div xmlns=&quot;http://www.w3.org/1999/xhtml&quot;&gt;&lt;span class=&quot;caption-title&quot;&gt;Relationship of the entropy of latent space dimensions with COVID-19 caseloads.&lt;/span&gt; (&lt;strong&gt;A&lt;/strong&gt;) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA. High values are observed for a lead of 1 and 2 months. (&lt;strong&gt;B&lt;/strong&gt;) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases. (&lt;strong&gt;C&lt;/strong&gt;) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.&lt;/div&gt;&lt;/div&gt;\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"hw-responsive-img\"><img class=\"highwire-fragment fragment-image lazyload\" alt=\"Figure 4:\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F5.medium.gif\" width=\"440\" height=\"378\"/><noscript><img class=\"highwire-fragment fragment-image\" alt=\"Figure 4:\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F5.medium.gif\" width=\"440\" height=\"378\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F5.large.jpg?download=true\" class=\"highwire-figure-link highwire-figure-link-download\" title=\"Download Figure 4:\" data-icon-position=\"\" data-hide-link-title=\"0\">Download figure</a></li><li class=\"new-tab last\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F5.large.jpg\" class=\"highwire-figure-link highwire-figure-link-newtab\" target=\"_blank\" data-icon-position=\"\" data-hide-link-title=\"0\">Open in new tab</a></li></ul></div><div class=\"fig-caption\"><span class=\"fig-label\">Figure 4:</span> <span class=\"caption-title\">Relationship of the entropy of latent space dimensions with COVID-19 caseloads.</span><p id=\"p-24\" class=\"first-child\">(<strong>A</strong>) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA. High values are observed for a lead of 1 and 2 months. (<strong>B</strong>) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases. (<strong>C</strong>) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.</p><div class=\"sb-div caption-clear\"></div></div></div><p id=\"p-25\">Our model can be therefore used to predict the COVID-19 caseloads in several countries. Both USA (<strong><a id=\"xref-fig-6-1\" class=\"xref-fig\" href=\"#F6\">Fig 5A</a></strong>) and Japan (<strong><a id=\"xref-fig-6-2\" class=\"xref-fig\" href=\"#F6\">Fig 5C</a></strong>) show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries. Our model predicts new caseloads with a two-month lead time, which strongly predicts a spike in new cases both in USA (<strong><a id=\"xref-fig-6-3\" class=\"xref-fig\" href=\"#F6\">Fig 5B</a></strong>) and Japan (<strong><a id=\"xref-fig-6-4\" class=\"xref-fig\" href=\"#F6\">Fig 5D</a></strong>) in the months of July and August, 2021. Therefore our model may be used as an epidemiological early warning system to predict new caseloads.</p><div id=\"F6\" class=\"fig pos-float type-figure  odd\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F6.large.jpg?width=800&amp;height=600&amp;carousel=1\" title=\"Prediction of new COVID-19 cases with Sample Entropy values of the latent dimensions. (A) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA. (B) Actual and predicted cases based on the entropy values of DoCs for the USA. The model predicts a rise in cases for July and August 2021. (C) Entropy of DoCs and new cases for Japan. (D) Actual and predicted cases for Japan. A spike in cases is predicted for July and August 2021.\" class=\"highwire-fragment fragment-images colorbox-load\" rel=\"gallery-fragment-images-1181327286\" data-figure-caption=\"&lt;div class=&quot;highwire-markup&quot;&gt;&lt;div xmlns=&quot;http://www.w3.org/1999/xhtml&quot;&gt;&lt;span class=&quot;caption-title&quot;&gt;Prediction of new COVID-19 cases with Sample Entropy values of the latent dimensions.&lt;/span&gt; (&lt;strong&gt;A&lt;/strong&gt;) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA. (&lt;strong&gt;B&lt;/strong&gt;) Actual and predicted cases based on the entropy values of DoCs for the USA. The model predicts a rise in cases for July and August 2021. (&lt;strong&gt;C&lt;/strong&gt;) Entropy of DoCs and new cases for Japan. (&lt;strong&gt;D&lt;/strong&gt;) Actual and predicted cases for Japan. A spike in cases is predicted for July and August 2021.&lt;/div&gt;&lt;/div&gt;\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"hw-responsive-img\"><img class=\"highwire-fragment fragment-image lazyload\" alt=\"Figure 5:\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F6.medium.gif\" width=\"440\" height=\"284\"/><noscript><img class=\"highwire-fragment fragment-image\" alt=\"Figure 5:\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F6.medium.gif\" width=\"440\" height=\"284\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F6.large.jpg?download=true\" class=\"highwire-figure-link highwire-figure-link-download\" title=\"Download Figure 5:\" data-icon-position=\"\" data-hide-link-title=\"0\">Download figure</a></li><li class=\"new-tab last\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F6.large.jpg\" class=\"highwire-figure-link highwire-figure-link-newtab\" target=\"_blank\" data-icon-position=\"\" data-hide-link-title=\"0\">Open in new tab</a></li></ul></div><div class=\"fig-caption\"><span class=\"fig-label\">Figure 5:</span> <span class=\"caption-title\">Prediction of new COVID-19 cases with Sample Entropy values of the latent dimensions.</span><p id=\"p-26\" class=\"first-child\">(<strong>A</strong>) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA. (<strong>B</strong>) Actual and predicted cases based on the entropy values of DoCs for the USA. The model predicts a rise in cases for July and August 2021. (<strong>C</strong>) Entropy of DoCs and new cases for Japan. (<strong>D</strong>) Actual and predicted cases for Japan. A spike in cases is predicted for July and August 2021.</p><div class=\"sb-div caption-clear\"></div></div></div></div><div id=\"sec-9\" class=\"subsection\"><h3>Codons associated with Dimensions of Concerns (DoCs) could be linked to SARS-CoV-2 variants</h3><p id=\"p-27\">We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension (<strong>Table S2</strong>). The intuition behind this idea is that the codons with high weights in a given dimension, when mutated in the viral sequence, are likely to cause a significant change in the entropy of the associated dimensions. Therefore each DoC can be linked to codons, which can further be mapped to Variants of concern (VOCs) and Variants of Interest (VOIs) (<strong>Table S3</strong>). Despite the fact that our model cannot directly capture the SARS-CoV-2 variants, it was observed that dimension-32 captures the CTG, CGG codons (ranks 5 and 8 respectively), known to occur in the T19R variant. Similarly, dimension 3 captures three codons (ACG, CGG, CAC) that are associated with multiple variants such as K417T, L452R, and D1118H, causing increased infectivity, pathogenicity, and spread. Dimension 30 captures codons CAT and CAC associated with Δ69 and D1118H respectively which are linked to B.1.1.7 lineage.</p><p id=\"p-28\">Codon weights (<strong>Table S2</strong>) of a given DoC provide an opportunity to associate with specific Variants of concern (VOCs) and Variants of Interest (VOIs); and to predict emerging SARS-CoV-2 variants and take preventive measures in advance. Distinct dimensions capture country-specific changes and may be surveilled to monitor the spread of the pandemic. This approach was back-validated with several real-world examples. For instance, dimension 32 captures the codons CGG (R) and CAC (R), which are found in B.1.429 lineage (L425R mutation). Dimension 3 captures CGG which is seen in L452R (associated with lineage B.1.617.1), which was first observed in India in December 2020 and was found to have increased infectivity and transmissibility.</p></div><div id=\"sec-10\" class=\"subsection\"><h3>Temporal tracking of entropy in Dimensions of Concern (DoCs) may be used as a surrogate to track the spread of various SARS-CoV-2 variants</h3><p id=\"p-29\">To investigate the potential of DoCs to track the spread of SARS-CoV-2, we used the codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs. We selected DoCs (dimension 3, 4, 12, 13, 15, 16, 25, 28, 30, 32) with high absolute weights for the codons related to the delta variants (<strong><a id=\"xref-fig-7-1\" class=\"xref-fig\" href=\"#F7\">Fig 6A</a></strong>). The entropy of the DoCs was contrasted with the caseloads in England (<strong><a id=\"xref-fig-7-2\" class=\"xref-fig\" href=\"#F7\">Fig 6B</a></strong>), India (<strong><a id=\"xref-fig-7-3\" class=\"xref-fig\" href=\"#F7\">Fig 6C</a></strong>), and USA (<strong><a id=\"xref-fig-7-4\" class=\"xref-fig\" href=\"#F7\">Fig 6D</a></strong>). Overall, the temporal tracking of the entropy in Dimensions of Concern (DoCs) may be used as a surrogate to track the spread of various SAR-CoV-2 variants.</p><div id=\"F7\" class=\"fig pos-float type-figure  odd\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F7.large.jpg?width=800&amp;height=600&amp;carousel=1\" title=\"Potential association of codons observed in SARS-CoV-2 Delta variant (lineage B.1.617.2) with their corresponding DoCs, and the trend of caseloads with the entropy of the DoCs. (A) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling. Line plots showing the entropies of DoCs and cases in countries, (B) England, (C) India, and (D) USA. The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.\" class=\"highwire-fragment fragment-images colorbox-load\" rel=\"gallery-fragment-images-1181327286\" data-figure-caption=\"&lt;div class=&quot;highwire-markup&quot;&gt;&lt;div xmlns=&quot;http://www.w3.org/1999/xhtml&quot;&gt;&lt;span class=&quot;caption-title&quot;&gt;Potential association of codons observed in SARS-CoV-2 Delta variant (lineage B.&lt;strong&gt;1&lt;/strong&gt;.&lt;strong&gt;617&lt;/strong&gt;.&lt;strong&gt;2&lt;/strong&gt;) with their corresponding DoCs, and the trend of caseloads with the entropy of the DoCs.&lt;/span&gt; (&lt;strong&gt;A&lt;/strong&gt;) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling. Line plots showing the entropies of DoCs and cases in countries, (&lt;strong&gt;B&lt;/strong&gt;) England, (&lt;strong&gt;C&lt;/strong&gt;) India, and (&lt;strong&gt;D&lt;/strong&gt;) USA. The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.&lt;/div&gt;&lt;/div&gt;\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"hw-responsive-img\"><img class=\"highwire-fragment fragment-image lazyload\" alt=\"Fig 6.\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F7.medium.gif\" width=\"440\" height=\"432\"/><noscript><img class=\"highwire-fragment fragment-image\" alt=\"Fig 6.\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F7.medium.gif\" width=\"440\" height=\"432\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F7.large.jpg?download=true\" class=\"highwire-figure-link highwire-figure-link-download\" title=\"Download Fig 6.\" data-icon-position=\"\" data-hide-link-title=\"0\">Download figure</a></li><li class=\"new-tab last\"><a href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F7.large.jpg\" class=\"highwire-figure-link highwire-figure-link-newtab\" target=\"_blank\" data-icon-position=\"\" data-hide-link-title=\"0\">Open in new tab</a></li></ul></div><div class=\"fig-caption\"><span class=\"fig-label\">Fig 6.</span> <span class=\"caption-title\">Potential association of codons observed in SARS-CoV-2 Delta variant (lineage B.<strong>1</strong>.<strong>617</strong>.<strong>2</strong>) with their corresponding DoCs, and the trend of caseloads with the entropy of the DoCs.</span><p id=\"p-30\" class=\"first-child\">(<strong>A</strong>) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling. Line plots showing the entropies of DoCs and cases in countries, (<strong>B</strong>) England, (<strong>C</strong>) India, and (<strong>D</strong>) USA. The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.</p><div class=\"sb-div caption-clear\"></div></div></div></div></div><div class=\"section\" id=\"sec-11\"><h2 class=\"\">Discussion</h2><p id=\"p-31\">We have implemented an approach for analyzing the emerging strains based on the latent space of spike protein coding nucleotide sequences. We chose the nucleotide sequences instead of proteins in order to capture and track the variations that may not have immediate functional consequences. Our approach has two main underlying tenets: (i) long-range interactions are known to modulate the functional (<a id=\"xref-ref-17-1\" class=\"xref-bibr\" href=\"#ref-17\">Mugnai et al., 2020</a>) interaction between receptor binding domain and ACE2 receptors, hence may be captured in the NLP models that capture 3-mer changes and context, and (ii) latent dimensions may be differentially correlated with indicators of spread, thus providing a data-driven handle for tracking and predicting variants of concern and variants of interest. The pipeline takes advantage of temporal changes in the semantics of mutating sequences. Preservation of phylogenetic structure based upon the similarity matrix obtained using the embeddings validated that the latent dimensions capture spatio-temporal information. Analyzing the dynamic patterns and underlying correlations in the 30,000 base pair long sequence of SARS-CoV-2 (<a id=\"xref-ref-26-1\" class=\"xref-bibr\" href=\"#ref-26\">Shishir et al., 2021</a>) is important to highlight the mechanistic understanding of mutations. SARS-CoV-2 seems to show a particularly high frequency of recombinations arising due to the absence of a proof-reading mechanism and sequence diversity, which calls for urgency in studying its transmission pattern (<a id=\"xref-ref-15-1\" class=\"xref-bibr\" href=\"#ref-15\">Mandal et al., 2021</a>)(<a id=\"xref-ref-25-1\" class=\"xref-bibr\" href=\"#ref-25\">Rouchka et al., 2020</a>). Therefore predicting mutations in the spike protein, which binds to ACE2 receptors can help us estimate the spread of disease and the efficacy of therapeutic treatments and vaccines (<a id=\"xref-ref-28-1\" class=\"xref-bibr\" href=\"#ref-28\">Srivastava et al., 2021</a>)(<a id=\"xref-ref-11-1\" class=\"xref-bibr\" href=\"#ref-11\">Li et al., 2020</a>).</p><p id=\"p-32\">Entropy is a measure of the disorder of a system. We hypothesized that mutations increase the chaotic dynamics in the latent space of spike genes. To calculate entropy, we used the accelerated versions of the Approximate Entropy and Sample Entropy algorithms, called Fast Approximate Entropy and Fast Sample Entropy (<a id=\"xref-ref-19-2\" class=\"xref-bibr\" href=\"#ref-19\">Pan et al., 2011</a>). Both algorithms aim to quantify how often different patterns of data are found in a time series. Fast Approximate Entropy, however, is a biased statistic and depends on the length of the series. Since we could have different counts of genome sequences collected each month, we preferred Sample Entropy, which is independent of the length of the series. Entropy values were calculated for each latent dimension in each month. Thereafter, Detrended Cross-Correlation Analysis (DCCA) (<a id=\"xref-ref-22-1\" class=\"xref-bibr\" href=\"#ref-22\">Podobnik and Stanley, 2007</a>) was performed between the entropy dimensions and the new cases. DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series. High cross-correlation for different lead periods revealed that the entropy values in a given month could be used to predict the new cases in different countries in subsequent months. Different countries had different lead times at which the highest cross-correlation was observed between the entropy dimensions and the cases, ranging from 1-6 months. Overall, a lead time of two months was chosen to model the new cases.</p><p id=\"p-33\">A similar analysis was done with daily values of entropy and new cases. Entropy was calculated in rolling windows, and cross-correlation analysis was performed between entropy and new cases at different lead periods. Although the cross-correlation values were found to be significant, the values were low and ranged between -0.1 to 0.1. Therefore, we decided to use the monthly entropy values for the modeling exercise.</p><p id=\"p-34\">We also experimented with “blips” as a feature to predict the new cases. Blips are sudden changes in the values of the latent dimensions. These changes may be caused by a mutation, which changes the words (codons) in a given genome sequence. This hypothesis was validated in simulation experiments in synthetic datasets. Each dimension of the spike gene embeddings for a country was analyzed for the presence of temporal anomalies. Countries having a minimum of 20 samples in any given month were selected and the same number of records (minimum samples in any given month for that country) were sampled without replacement from each month. These records were used to define control limits of ±1 standard deviation from the mean value for each dimension, and all values in the full dataset outside those limits were categorized as ‘Blip’ points. Blip counts in each month were normalized by calculating the number of blips per sample collected in a month for each dimension (normalized blips). The embedding dimensions were then compared in terms of the total normalized blips for each country to observe the significant dimensions and dis(similarity) in trends among different countries. Cumulative counts of normalized blips were analyzed to understand the temporal accumulation of blips in each dimension. Similar to entropy, blips were found to have a leading relationship with the cases. However, regression modeling results with sample entropy were found to be better.</p><p id=\"p-35\">To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data. With sample entropy, the R-squared value achieved was 73%, while with approximate entropy, the value was only 10%. Therefore the model trained on sample entropy was selected. The predictions from the model were found to be highly correlated with the actual cases, indicating that our model can be used for preemptive warning signals for the rise in cases in different countries. Further, the actual and the predicted difference in the number of cases in consecutive months was found to be correlated, which suggests that the relative change in the cases in consecutive months predicted by our model is linked to the relative change in the number of cases. Overall, we recommend that our model be used to predict dangerous trends and not the actual number of cases. Further, the mapping from latent dimensions to Variants of Concern (VOCs) and Variants of Interest (VOIs) may help us track the country-specific spread of different variants.</p><p id=\"p-36\">Prediction models can be used for training the genomic sequence for predicting infection severity based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 (<a id=\"xref-ref-32-1\" class=\"xref-bibr\" href=\"#ref-32\">Wang et al., 2020</a>). The machine learning models can also be trained on genomic sequences for COVID-19 classification (<a id=\"xref-ref-2-1\" class=\"xref-bibr\" href=\"#ref-2\">Arslan, 2021</a>). Although the variance explained by our model is low, however, we were able to compute the variability associated with spike protein mutations. So our method showed a potential way to estimate the new cases variability associated with spike protein mutations. Our methods can be incorporated with the epidemic projections model to better predict the epidemic trajectories. The latent dimensions may further be employed to predict the clinical consequences of emerging strains. The currently available vaccines are intended for early SARS-CoV-2 strains, but with new emerging variants, immune responses triggered by these vaccines may be weaker and short-lived. As seen in the devastating second wave of the pandemic in India, newer SARS-CoV-2 variants have acquired an increased pathogenic potential resulting in rapid clinical progression and overwhelmed health systems. Mitigating such events in the future will require stronger surveillance systems in place. Our study offers a promising solution in this direction and lays the foundation for proactive genomic surveillance of COVID-19.</p><div id=\"sec-12\" class=\"subsection\"><h3>Limitations of our study</h3><p id=\"p-37\">Our approach of codon (3-mers) embeddings does not indicate the position where the codon change may have happened in the spike gene. This is because low-dimensional embeddings do not preserve the positional encoding of words. However, we are investigating advanced approaches such as complex-valued word embeddings with positional encodings (<a id=\"xref-ref-31-1\" class=\"xref-bibr\" href=\"#ref-31\">Wang et al., 2019</a>) and transformer models such as BERT (<a id=\"xref-ref-34-1\" class=\"xref-bibr\" href=\"#ref-34\">Wolf et al., 2020</a>)(<a id=\"xref-ref-9-1\" class=\"xref-bibr\" href=\"#ref-9\">Lee et al., 2020</a>) to overcome our current limitation. The latter are considered expensive and data-hungry models and it will remain to be evaluated if the gain of positional information may be countered by the loss of prediction accuracy for forecasting new cases in the future. However, we believe that the availability of sequences for a wide variety of viral pathogens presents an exciting opportunity to train data-hungry models that may be able to transfer insights across pathogens and yet remain interpretable.</p><p id=\"p-38\">Further, our Strainflow model is trained only on the spike gene of the viral genome, which does not represent the complete variation spectrum of the virus. To mitigate this shortcoming, we are currently developing a genome-level Strainflow model (version 2.0) for SARS-CoV-2.</p><p id=\"p-39\">Another limitation of our study is the relatively small number of samples that were used to construct the supervised predictive model for case prediction. As more data becomes available in subsequent months, we can produce more confident case predictions. Nonetheless, the unsupervised embeddings and temporal cross-correlations were learned upon the full datasets and these presented clear patterns in DoCs and significant cross-correlations with caseloads. However, it is important for our models to receive at least 30 samples per month for getting reliable prediction results for the future months. This also underscores the need for a more reliable and agile approach to deposit country-level datasets on repositories such as GISAID. We make an appeal to the countries to facilitate the sharing of such data in order to be prepared for any future waves of the current pandemic and for preventing the new emergence of strains. We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance.</p></div></div><div class=\"section\" id=\"sec-13\"><h2 class=\"\">Methods</h2><div id=\"sec-14\" class=\"subsection\"><h3>Key resources table</h3><div id=\"T2\" class=\"table pos-float\"><div class=\"table-inline table-callout-links\"><div class=\"callout\"><span>View this table:</span><ul class=\"callout-links\"><li class=\"view-inline first\"><a href=\"##\" class=\"table-expand-inline\" data-table-url=\"/highwire/markup/2128022/expansion?postprocessors=highwire_tables%2Chighwire_reclass%2Chighwire_figures%2Chighwire_math%2Chighwire_inline_linked_media%2Chighwire_embed&amp;table-expand-inline=1\" data-icon-position=\"\" data-hide-link-title=\"0\">View inline</a></li><li class=\"view-popup\"><a href=\"/highwire/markup/2128022/expansion?width=1000&amp;height=500&amp;iframe=true&amp;postprocessors=highwire_tables%2Chighwire_reclass%2Chighwire_figures%2Chighwire_math%2Chighwire_inline_linked_media%2Chighwire_embed\" class=\"colorbox colorbox-load table-expand-popup\" rel=\"gallery-fragment-tables\" data-icon-position=\"\" data-hide-link-title=\"0\">View popup</a></li><li class=\"download-ppt last\"><a href=\"/highwire/powerpoint/2128022\" class=\"highwire-figure-link highwire-figure-link-ppt\" data-icon-position=\"\" data-hide-link-title=\"0\">Download powerpoint</a></li></ul></div></div></div></div><div id=\"sec-15\" class=\"subsection\"><h3>Datasets</h3><div id=\"sec-16\" class=\"subsection\"><h4>Training dataset</h4><p id=\"p-40\">The dataset was downloaded from GISAID EpiCoV (<a id=\"xref-ref-27-1\" class=\"xref-bibr\" href=\"#ref-27\">Shu and McCauley, 2017</a>) (April 8, 2021 release). 0.63 million genome sequences with high nucleotide completeness, coverage, complete temporal information, and presence of less than 5% non-identified nucleotide bases (N) were downloaded. The sequences included 63 countries, including India, United Kingdom, USA, Australia, New Zealand, Germany, Russia, Italy, France, Mexico, Canada, China, Japan, Pakistan, Bangladesh, Iran, Iraq, the continent of South America, and Africa. Duplicate samples were removed, and whole genome sequences were parsed using CoV-Seq (<a id=\"xref-ref-13-1\" class=\"xref-bibr\" href=\"#ref-13\">Liu et al., 2020</a>) to extract nucleotide sequences corresponding to each of the 12 CDS. Accession IDs that did not cover 12 coding regions were discarded, yielding 0.31 million high-quality SARS-CoV-2 genome sequences for language modelling. We downloaded country-wise COVID-19 data for new cases from a publicly available repository maintained by Johns Hopkins University Center for Systems Science and Engineering (JHU CSSE).</p></div><div id=\"sec-17\" class=\"subsection\"><h4>Test dataset</h4><p id=\"p-41\">We downloaded around 0.6 million genome sequences submitted to GISAID from April 2021 to June 2021. We have used 0.36 million sequences to predict their latent space using the <em>Strainflow</em> model for the months of April 2021 to June 2021.</p></div></div><div id=\"sec-18\" class=\"subsection\"><h3>Word Embeddings in Strainflow model</h3><p id=\"p-42\">In our Strainflow model, we have adopted a word2vec model (<a id=\"xref-ref-16-2\" class=\"xref-bibr\" href=\"#ref-16\">Mikolov et al., 2013</a>). A low dimensional representation for the genome sequences was learned using the word2vec model. Non-overlapping sequences of 3-mers (codons) were considered as words for training the word2vec model, implemented in Gensim (<a id=\"xref-ref-24-1\" class=\"xref-bibr\" href=\"#ref-24\">Radim Rehurek, 2010</a>). The word2vec model was trained using the skip-gram algorithm, with a fixed window size of twenty and vector size of thirty-six. For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise. The mean was calculated by summing across all the k-mers over each dimension and then dividing it by the total number of codons present in the sequence. For selecting the vector size of the word2vec model, we calculated PIP (Pairwise Inner Product) loss (<a id=\"xref-ref-37-1\" class=\"xref-bibr\" href=\"#ref-37\">Yin and Shen, 2018</a>). PIP loss is a metric used for calculating the dissimilarity between two word embedding matrices. For the embedding matrix of strains (E), the PIP matrix is defined as the dot product of the embedding matrix with its transpose (<em>E. E</em> <sup><em>T</em></sup>). The PIP loss between two embedding matrices is defined as the norm of the difference between their PIP matrices.\\n<span class=\"disp-formula\" id=\"disp-formula-1\"><span class=\"highwire-responsive-lazyload\"><img src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" class=\"highwire-embed lazyload\" alt=\"Embedded Image\" data-src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-10.gif\"/><noscript><img class=\"highwire-embed\" alt=\"Embedded Image\" src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-10.gif\"/></noscript></span>\\n\\n</span>\\n</p><p id=\"p-43\">Various word2vec models were trained on the dataset with different vector sizes varying in multiples of three. Based on the PIP loss calculations, we found out that word embeddings with 36 dimensions showed a differential dent in the curve (change in straight line), due to which we selected this to be the dimension of the word embeddings (<strong>Fig. S1</strong>).</p></div><div id=\"sec-19\" class=\"subsection\"><h3>Phylogenetic analysis using the latent dimensions of the spike genes</h3><p id=\"p-44\">To evaluate the phylogenetic properties based on the latent dimensions of the spike gene, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions. The pairwise distance was further used for hierarchical clustering using the ‘hclust’ function in R statistical programming language. This analysis was performed using 400 random sequences of spike genes from 16 countries. The visualization of the phylogenetic tree derived based on the latent dimensions was done using ‘iTOL’ (Letunic) software (<strong><a id=\"xref-fig-3-3\" class=\"xref-fig\" href=\"#F3\">Fig 2</a></strong>).</p></div><div id=\"sec-20\" class=\"subsection\"><h3>Entropy of the latent dimensions</h3><p id=\"p-45\">To quantify the properties of latent dimensions, we have used a well-known information theory based algorithm suitable for time series datasets, called ‘Fast Sample Entropy’ (<a id=\"xref-ref-30-1\" class=\"xref-bibr\" href=\"#ref-30\">Tomčala, 2020</a>). To compute Fast Sample Entropy, we have used the ‘FastSampEn’ function in the ‘TSEntropies’ package in R (<a id=\"xref-ref-29-1\" class=\"xref-bibr\" href=\"#ref-29\">Tomcala, 2018</a>). Fast Sample Entropy can be computed as follows.\\n<span class=\"disp-formula\" id=\"disp-formula-2\"><span class=\"highwire-responsive-lazyload\"><img src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" class=\"highwire-embed lazyload\" alt=\"Embedded Image\" data-src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-11.gif\"/><noscript><img class=\"highwire-embed\" alt=\"Embedded Image\" src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-11.gif\"/></noscript></span>\\n\\n</span>\\nwhere,\\n<span class=\"disp-formula\" id=\"disp-formula-3\"><span class=\"highwire-responsive-lazyload\"><img src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" class=\"highwire-embed lazyload\" alt=\"Embedded Image\" data-src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-12.gif\"/><noscript><img class=\"highwire-embed\" alt=\"Embedded Image\" src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-12.gif\"/></noscript></span>\\n\\n</span>\\n</p><p id=\"p-46\"><em>S</em><sub><em>i,mm</em></sub> is a set of sub-sequences of length m belonging to the i-th neighborhood, and <em>N</em><sub>m</sub> is the number of these neighborhoods.</p><p id=\"p-47\">In our case, ‘x’ is the latent dimension of the spike genes of the SARS-CoV-2 strains per month for a given country with default values of ‘m’ and ‘r’. Entropy was computed for each latent dimension on a monthly basis for each country. The minimum number of SARS-CoV-2 spike genes was 20. To compare across months, we used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country (<strong><a id=\"xref-fig-2-7\" class=\"xref-fig\" href=\"#F2\">Fig 1D</a></strong>).</p><p id=\"p-48\">To compare the entropy of the latent dimensions among countries, we used the total entropy of the country for each dimension and visualized it with line graphs (<strong><a id=\"xref-fig-4-3\" class=\"xref-fig\" href=\"#F4\">Fig 3</a></strong>).</p></div><div id=\"sec-21\" class=\"subsection\"><h3>Detrended Cross Correlations Analysis (DCCA)</h3><p id=\"p-49\">To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used the Detrended Cross Correlation Analysis (<a id=\"xref-ref-23-1\" class=\"xref-bibr\" href=\"#ref-23\">Prass and Pumi, 2019</a>). Here, DCCA captures the long-range cross correlation between time series (entropy of the months and caseloads for a given country). We tested both time series for stationarity using Augmented Dickey-Fuller (ADF) test (<a id=\"xref-ref-18-1\" class=\"xref-bibr\" href=\"#ref-18\">Mushtaq, 2011</a>). The ADF test was implemented using the function ‘adf.test’ available in the ‘tseries’ (2020a) package in R. Due to the non-stationary distribution of the estimated entropies and the caseloads for a given country, we used the ‘DCCA’ R package (2020b). Cross-correlation was calculated between the entropy dimensions at time t+h and new cases at time t, where h = 0, ±1, ±2, ±3 … ±10.</p></div><div id=\"sec-22\" class=\"subsection\"><h3>Machine Learning based identification of Dimensions of Concern (DoCs)</h3><p id=\"p-50\"><em>Dimensions of Concern (DoCs)</em> were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis and predictive of new case count in the respective countries. Country-wise monthly total new cases data was taken at the end of each month. Total new cases data for each month was merged to monthly entropy dimensions data from March, 2020 to June, 2021. We used a regression based machine learning approach called ‘Boruta’ (<a id=\"xref-ref-7-1\" class=\"xref-bibr\" href=\"#ref-7\">Kursa and Rudnicki, 2010</a>), a wrapper algorithm around random forest algorithm (<a id=\"xref-ref-8-1\" class=\"xref-bibr\" href=\"#ref-8\">Kursa et al., 2010</a>) to select the most relevant entropy dimensions for the prediction of subsequent two months’ new cases. We used the default parameters with the modification of the maximum runs as 1000. We selected the confirmed entropy dimensions as the most relevant features for the prediction of new cases.</p></div><div id=\"sec-23\" class=\"subsection\"><h3>Model development and evaluation for prediction of new cases in subsequent months</h3><p id=\"p-51\">To predict the new cases in the next to next months, we used a regression based random forest model (Breiman 2001) using the most relevant DoCs based on Boruta. The model training was performed using the data from March, 2020 to April, 2021; and the fitted model was tested on data from May, 2021 to June, 2021. The regression modeling was performed using 1000 decision trees using the ‘randomForest’ package in R (A. Liaw et al., 2002).</p></div><div id=\"sec-24\" class=\"subsection\"><h3>Top codons associated with DoCs</h3><p id=\"p-52\">To find the top codons associated with the DoCs, we extracted the absolute weights of each codon for a given DoC. The top 10 codons having the highest absolute weights (contribution) were identified corresponding to each of the DoCs to compare our DoCs with SARS-CoV-2 variants. We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (<a id=\"xref-ref-3-1\" class=\"xref-bibr\" href=\"#ref-3\">CDC, 2021</a>); and a list of codons associated with VOIs and VOCs was curated <strong>(Table S4)</strong> (Lopez-Rincon et al.; <a id=\"xref-ref-20-1\" class=\"xref-bibr\" href=\"#ref-20\">paola, 2021</a>; <a id=\"xref-ref-21-1\" class=\"xref-bibr\" href=\"#ref-21\">Peacock et al., 2021</a>; <a id=\"xref-ref-28-2\" class=\"xref-bibr\" href=\"#ref-28\">Srivastava et al., 2021</a>). The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant (<a id=\"xref-ref-3-2\" class=\"xref-bibr\" href=\"#ref-3\">CDC, 2021</a>)</p></div></div><div class=\"section ack\" id=\"ack-1\"><h2 class=\"\">Acknowledgements</h2><p id=\"p-53\">This work was supported by the Delhi Cluster-Delhi Research Implementation and Innovation (DRIIV) Project supported by the Principal Scientific Advisor Office, Prn.SA/Delhi/ Hub/2018(C) and the Center of Excellence in Healthcare supported by Delhi Knowledge Development Foundation (DKDF) at IIIT-Delhi. We also thank Dr. Chitra Pattabiraman (NIMHANS) for her valuable inputs on viral sequences, and Harleen Kaur, Nishkarsh Saxena, and Anjali for their contribution to the dashboard visualizations.</p></div><div class=\"section fn-group\" id=\"fn-group-1\"><h2>Footnotes</h2><ul><li class=\"fn-update fn-group-summary-of-updates\" id=\"fn-3\"><p id=\"p-10\">In this revised version, we have implemented fast tSNE for qualitative inspection of the latent dimensions (LD) of 0.9 million SARS-CoV-2 spike genes. Quantitative analyses of the LDs were performed using the fast sample entropy method. Also, we have used fast sample entropy instead of \\'blips\\' to model COVID-19 caseloads.</p></li></ul></div><div class=\"section ref-list\" id=\"ref-list-1\"><h2 class=\"\">References</h2><ol class=\"cit-list ref-use-labels\"><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.1\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"> <span class=\"cit-name-given-names\">A.</span> <span class=\"cit-name-surname\">Liaw</span></span> and <span class=\"cit-auth\"> <span class=\"cit-name-given-names\">M.</span> <span class=\"cit-name-surname\">Wiener</span></span> (<span class=\"cit-pub-date\">2002</span>). <span class=\"cit-article-title\">Classification and Regression by randomForest</span>. <abbr class=\"cit-jnl-abbrev\">R News</abbr> <span class=\"cit-vol\">2</span>(<span class=\"cit-issue\">3</span>), <span class=\"cit-fpage\">18</span>--22.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DR%2BNews%26rft.volume%253D2%26rft.spage%253D18%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-2-1\" title=\"View reference  in text\" id=\"ref-2\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.2\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Arslan</span>,  <span class=\"cit-name-given-names\">H.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">Machine Learning Methods for COVID-19 Prediction Using Human Genomic Data</span>. <abbr class=\"cit-jnl-abbrev\">Proceedings</abbr> <span class=\"cit-vol\">74</span>, <span class=\"cit-fpage\">20</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DProceedings%26rft.volume%253D74%26rft.spage%253D20%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-3-1\" title=\"View reference  in text\" id=\"ref-3\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.3\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth cit-collab\">CDC</span> (<span class=\"cit-pub-date\">2021</span>). <abbr class=\"cit-jnl-abbrev\">SARS-CoV-2 Variant Classifications and Definitions</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-4-1\" title=\"View reference  in text\" id=\"ref-4\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.4\" data-doi=\"10/gdkbqx\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Hadfield</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Megill</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Bell</span>,  <span class=\"cit-name-given-names\">S.M.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Huddleston</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Potter</span>,  <span class=\"cit-name-given-names\">B.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Callender</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Sagulenko</span>,  <span class=\"cit-name-given-names\">P.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Bedford</span>,  <span class=\"cit-name-given-names\">T.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Neher</span>,  <span class=\"cit-name-given-names\">R.A.</span></span> (<span class=\"cit-pub-date\">2018</span>). <span class=\"cit-article-title\">Nextstrain: real-time tracking of pathogen evolution</span>. <abbr class=\"cit-jnl-abbrev\">Bioinformatics</abbr> <span class=\"cit-vol\">34</span>, <span class=\"cit-fpage\">4121</span>–<span class=\"cit-lpage\">4123</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DBioinformatics%26rft.volume%253D34%26rft.spage%253D4121%26rft_id%253Dinfo%253Adoi%252F10%252Fgdkbqx%26rft_id%253Dinfo%253Apmid%252Fhttp%253A%252F%252Fwww.n%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a><a href=\"/lookup/external-ref?access_num=10/gdkbqx&amp;link_type=DOI\" class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\"><span>CrossRef</span></a><a href=\"/lookup/external-ref?access_num=http://www.n&amp;link_type=MED&amp;atom=%2Fbiorxiv%2Fearly%2F2021%2F08%2F26%2F2021.05.25.445601.atom\" class=\"cit-ref-sprinkles cit-ref-sprinkles-medline\"><span>PubMed</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-5-1\" title=\"View reference  in text\" id=\"ref-5\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.5\" data-doi=\"10.1126/science.abd7331\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Hie</span>,  <span class=\"cit-name-given-names\">B.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhong</span>,  <span class=\"cit-name-given-names\">E.D.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Berger</span>,  <span class=\"cit-name-given-names\">B.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Bryson</span>,  <span class=\"cit-name-given-names\">B.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">Learning the language of viral evolution and escape</span>. <abbr class=\"cit-jnl-abbrev\">Science</abbr> <span class=\"cit-vol\">371</span>, <span class=\"cit-fpage\">284</span>–<span class=\"cit-lpage\">288</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DScience%26rft.stitle%253DScience%26rft.aulast%253DHie%26rft.auinit1%253DB.%26rft.volume%253D371%26rft.issue%253D6526%26rft.spage%253D284%26rft.epage%253D288%26rft.atitle%253DLearning%2Bthe%2Blanguage%2Bof%2Bviral%2Bevolution%2Band%2Bescape%26rft_id%253Dinfo%253Adoi%252F10.1126%252Fscience.abd7331%26rft_id%253Dinfo%253Apmid%252F33446556%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a><a href=\"/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6Mzoic2NpIjtzOjU6InJlc2lkIjtzOjEyOiIzNzEvNjUyNi8yODQiO3M6NDoiYXRvbSI7czo0ODoiL2Jpb3J4aXYvZWFybHkvMjAyMS8wOC8yNi8yMDIxLjA1LjI1LjQ0NTYwMS5hdG9tIjt9czo4OiJmcmFnbWVudCI7czowOiIiO30=\" class=\"cit-ref-sprinkles cit-ref-sprinkles-ijlink\"><span><span class=\"cit-reflinks-abstract\">Abstract</span><span class=\"cit-sep cit-reflinks-variant-name-sep\">/</span><span class=\"cit-reflinks-full-text\"><span class=\"free-full-text\">FREE </span>Full Text</span></span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-6-1\" title=\"View reference  in text\" id=\"ref-6\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.6\" data-doi=\"10.1371/journal.pone.0238344\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Khan</span>,  <span class=\"cit-name-given-names\">M.I.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Khan</span>,  <span class=\"cit-name-given-names\">Z.A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Baig</span>,  <span class=\"cit-name-given-names\">M.H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Ahmad</span>,  <span class=\"cit-name-given-names\">I.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Farouk</span>,  <span class=\"cit-name-given-names\">A.-E.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Song</span>,  <span class=\"cit-name-given-names\">Y.G.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Dong</span>,  <span class=\"cit-name-given-names\">J.-J.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Comparative genome analysis of novel coronavirus (SARS-CoV-2) from different geographical locations and the effect of mutations on major target proteins: An in silico insight</span>. <abbr class=\"cit-jnl-abbrev\">PLoS One</abbr> <span class=\"cit-vol\">15</span>, <span class=\"cit-fpage\">e0238344</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DPLoS%2BOne%26rft.volume%253D15%26rft.spage%253De0238344%26rft_id%253Dinfo%253Adoi%252F10.1371%252Fjournal.pone.0238344%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a><a href=\"/lookup/external-ref?access_num=10.1371/journal.pone.0238344&amp;link_type=DOI\" class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\"><span>CrossRef</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-7-1\" title=\"View reference  in text\" id=\"ref-7\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.7\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Kursa</span>,  <span class=\"cit-name-given-names\">M.B.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Rudnicki</span>,  <span class=\"cit-name-given-names\">W.R.</span></span> (<span class=\"cit-pub-date\">2010</span>). <span class=\"cit-article-title\">Feature Selection with theBorutaPackage</span>. <abbr class=\"cit-jnl-abbrev\">Journal of Statistical Software</abbr> <span class=\"cit-vol\">36</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-8-1\" title=\"View reference  in text\" id=\"ref-8\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.8\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Kursa</span>,  <span class=\"cit-name-given-names\">M.B.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Jankowski</span>,  <span class=\"cit-name-given-names\">A.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Rudnicki</span>,  <span class=\"cit-name-given-names\">W.R.</span></span> (<span class=\"cit-pub-date\">2010</span>). <span class=\"cit-article-title\">Boruta - A System for Feature Selection</span>. <abbr class=\"cit-jnl-abbrev\">Fund. Inform</abbr>. <span class=\"cit-vol\">101</span>, <span class=\"cit-fpage\">271</span>–<span class=\"cit-lpage\">285</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DFund.%2BInform%26rft.volume%253D101%26rft.spage%253D271%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-9-1\" title=\"View reference  in text\" id=\"ref-9\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.9\" data-doi=\"10.1093/bioinformatics/btz682\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Lee</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Yoon</span>,  <span class=\"cit-name-given-names\">W.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kim</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kim</span>,  <span class=\"cit-name-given-names\">D.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kim</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">So</span>,  <span class=\"cit-name-given-names\">C.H.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kang</span>,  <span class=\"cit-name-given-names\">J.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">BioBERT: a pre-trained biomedical language representation model for biomedical text mining</span>. <abbr class=\"cit-jnl-abbrev\">Bioinformatics</abbr> <span class=\"cit-vol\">36</span>, <span class=\"cit-fpage\">1234</span>–<span class=\"cit-lpage\">1240</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DBioinformatics%26rft.volume%253D36%26rft.spage%253D1234%26rft_id%253Dinfo%253Adoi%252F10.1093%252Fbioinformatics%252Fbtz682%26rft_id%253Dinfo%253Apmid%252F31501885%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a><a href=\"/lookup/external-ref?access_num=10.1093/bioinformatics/btz682&amp;link_type=DOI\" class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\"><span>CrossRef</span></a><a href=\"/lookup/external-ref?access_num=31501885&amp;link_type=MED&amp;atom=%2Fbiorxiv%2Fearly%2F2021%2F08%2F26%2F2021.05.25.445601.atom\" class=\"cit-ref-sprinkles cit-ref-sprinkles-medline\"><span>PubMed</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.10\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Letunic</span>,  <span class=\"cit-name-given-names\">I.</span></span> <abbr class=\"cit-jnl-abbrev\">iTOL: Interactive Tree Of Life</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-11-1\" title=\"View reference  in text\" id=\"ref-11\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.11\" data-doi=\"S0092-8674(20)30877-1\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Li</span>,  <span class=\"cit-name-given-names\">Q.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Wu</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Nie</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">L.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Hao</span>,  <span class=\"cit-name-given-names\">H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Liu</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhao</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">Q.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Liu</span>,  <span class=\"cit-name-given-names\">H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Nie</span>,  <span class=\"cit-name-given-names\">L.</span></span>, <span class=\"cit-etal\">et al.</span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">The Impact of Mutations in SARS-CoV-2 Spike on Viral Infectivity and Antigenicity</span>. <abbr class=\"cit-jnl-abbrev\">Cell</abbr> <span class=\"cit-vol\">182</span>, <span class=\"cit-fpage\">1284</span>–<span class=\"cit-lpage\">1294</span>.e9.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DCell%26rft.volume%253D182%26rft.spage%253D1284%26rft_id%253Dinfo%253Adoi%252FS0092-8674%252820%252930877-1%26rft_id%253Dinfo%253Apmid%252FWOS%253A00057144%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a><a href=\"/lookup/external-ref?access_num=S0092-8674(20)30877-1&amp;link_type=DOI\" class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\"><span>CrossRef</span></a><a href=\"/lookup/external-ref?access_num=WOS:00057144&amp;link_type=MED&amp;atom=%2Fbiorxiv%2Fearly%2F2021%2F08%2F26%2F2021.05.25.445601.atom\" class=\"cit-ref-sprinkles cit-ref-sprinkles-medline\"><span>PubMed</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-12-1\" title=\"View reference  in text\" id=\"ref-12\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.12\" data-doi=\"10.1038/s41592-018-0308-4\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Linderman</span>,  <span class=\"cit-name-given-names\">G.C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Rachh</span>,  <span class=\"cit-name-given-names\">M.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Hoskins</span>,  <span class=\"cit-name-given-names\">J.G.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Steinerberger</span>,  <span class=\"cit-name-given-names\">S.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kluger</span>,  <span class=\"cit-name-given-names\">Y.</span></span> (<span class=\"cit-pub-date\">2019</span>). <span class=\"cit-article-title\">Fast interpolation-based t-SNE for improved visualization of single-cell RNA-seq data</span>. <abbr class=\"cit-jnl-abbrev\">Nat. Methods</abbr> <span class=\"cit-vol\">16</span>, <span class=\"cit-fpage\">243</span>–<span class=\"cit-lpage\">245</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DNat.%2BMethods%26rft.volume%253D16%26rft.spage%253D243%26rft_id%253Dinfo%253Adoi%252F10.1038%252Fs41592-018-0308-4%26rft_id%253Dinfo%253Apmid%252F30742040%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a><a href=\"/lookup/external-ref?access_num=10.1038/s41592-018-0308-4&amp;link_type=DOI\" class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\"><span>CrossRef</span></a><a href=\"/lookup/external-ref?access_num=30742040&amp;link_type=MED&amp;atom=%2Fbiorxiv%2Fearly%2F2021%2F08%2F26%2F2021.05.25.445601.atom\" class=\"cit-ref-sprinkles cit-ref-sprinkles-medline\"><span>PubMed</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-13-1\" title=\"View reference  in text\" id=\"ref-13\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.13\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Liu</span>,  <span class=\"cit-name-given-names\">B.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Liu</span>,  <span class=\"cit-name-given-names\">K.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">L.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Bian</span>,  <span class=\"cit-name-given-names\">Y.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Huang</span>,  <span class=\"cit-name-given-names\">L.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">CoV-Seq, a New Tool for SARS-CoV-2 Genome Analysis and Visualization: Development and Usability Study</span>. <abbr class=\"cit-jnl-abbrev\">J. Med. Internet Res</abbr>. <span class=\"cit-vol\">22</span>, <span class=\"cit-fpage\">e22299</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DJ.%2BMed.%2BInternet%2BRes%26rft.volume%253D22%26rft.spage%253De22299%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.14\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Lopez-Rincon</span>,  <span class=\"cit-name-given-names\">A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Perez-Romero</span>,  <span class=\"cit-name-given-names\">C.A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Tonda</span>,  <span class=\"cit-name-given-names\">A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Mendoza-Maldonado</span>,  <span class=\"cit-name-given-names\">L.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Claassen</span>,  <span class=\"cit-name-given-names\">E.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Garssen</span>,  <span class=\"cit-name-given-names\">J.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kraneveld</span>,  <span class=\"cit-name-given-names\">A.D.</span></span> <abbr class=\"cit-jnl-abbrev\">Design of Specific Primer Sets for the Detection of B.1.1.7, B.1.351, P.1, B.1.617.2 and B.1.1.519 Variants of SARS-CoV-2 using Artificial Intelligence</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-15-1\" title=\"View reference  in text\" id=\"ref-15\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.15\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Mandal</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Roychowdhury</span>,  <span class=\"cit-name-given-names\">T.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Bhattacharya</span>,  <span class=\"cit-name-given-names\">A.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">Pattern of genomic variation in SARS-CoV-2 (COVID-19) suggests restricted nonrandom changes: Analysis using Shewhart control charts</span>. <abbr class=\"cit-jnl-abbrev\">J. Biosci</abbr>. <span class=\"cit-vol\">46</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-16-1\" title=\"View reference  in text\" id=\"ref-16\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.16\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Mikolov</span>,  <span class=\"cit-name-given-names\">T.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Chen</span>,  <span class=\"cit-name-given-names\">K.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Corrado</span>,  <span class=\"cit-name-given-names\">G.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Dean</span>,  <span class=\"cit-name-given-names\">J.</span></span> (<span class=\"cit-pub-date\">2013</span>). <abbr class=\"cit-jnl-abbrev\">Efficient Estimation of Word Representations in Vector Space</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-17-1\" title=\"View reference  in text\" id=\"ref-17\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.17\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Mugnai</span>,  <span class=\"cit-name-given-names\">M.L.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Templeton</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Elber</span>,  <span class=\"cit-name-given-names\">R.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Thirumalai</span>,  <span class=\"cit-name-given-names\">D.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Role of Long-range Allosteric Communication in Determining the Stability and Disassembly of SARS-COV-2 in Complex with ACE2</span>. <abbr class=\"cit-jnl-abbrev\">bioRxiv</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-18-1\" title=\"View reference  in text\" id=\"ref-18\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.18\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Mushtaq</span>,  <span class=\"cit-name-given-names\">R.</span></span> (<span class=\"cit-pub-date\">2011</span>). <span class=\"cit-article-title\">Augmented Dickey Fuller Test</span>. <abbr class=\"cit-jnl-abbrev\">SSRN Electronic Journal</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-19-1\" title=\"View reference  in text\" id=\"ref-19\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.19\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Pan</span>,  <span class=\"cit-name-given-names\">Y.H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Wang</span>,  <span class=\"cit-name-given-names\">Y.H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Liang</span>,  <span class=\"cit-name-given-names\">S.F.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Lee</span>,  <span class=\"cit-name-given-names\">K.T.</span></span> (<span class=\"cit-pub-date\">2011</span>). <span class=\"cit-article-title\">Fast computation of sample entropy and approximate entropy in biomedicine</span>. <abbr class=\"cit-jnl-abbrev\">Comput. Methods Programs Biomed</abbr>. <span class=\"cit-vol\">104</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-20-1\" title=\"View reference  in text\" id=\"ref-20\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.20\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth cit-collab\">paola</span> (<span class=\"cit-pub-date\">2021</span>). <abbr class=\"cit-jnl-abbrev\">Phylogenetic relationship of SARS-CoV-2 sequences from Amazonas with emerging Brazilian variants harboring mutations E484K and N501Y in the Spike protein</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-21-1\" title=\"View reference  in text\" id=\"ref-21\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.21\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Peacock</span>,  <span class=\"cit-name-given-names\">T.P.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Penrice-Randal</span>,  <span class=\"cit-name-given-names\">R.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Hiscox</span>,  <span class=\"cit-name-given-names\">J.A.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Barclay</span>,  <span class=\"cit-name-given-names\">W.S.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">SARS-CoV-2 one year on: evidence for ongoing viral adaptation</span>. <abbr class=\"cit-jnl-abbrev\">J. Gen. Virol</abbr>. <span class=\"cit-vol\">102</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-22-1\" title=\"View reference  in text\" id=\"ref-22\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.22\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Podobnik</span>,  <span class=\"cit-name-given-names\">B.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Stanley</span>,  <span class=\"cit-name-given-names\">H.E.</span></span> (<span class=\"cit-pub-date\">2007</span>). <abbr class=\"cit-jnl-abbrev\">Detrended Cross-Correlation Analysis: A New Method for Analyzing Two Non-stationary Time Series</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-23-1\" title=\"View reference  in text\" id=\"ref-23\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.23\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Prass</span>,  <span class=\"cit-name-given-names\">T.S.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Pumi</span>,  <span class=\"cit-name-given-names\">G.</span></span> (<span class=\"cit-pub-date\">2019</span>). <abbr class=\"cit-jnl-abbrev\">On the behavior of the DFA and DCCA in trend-stationary processes</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-24-1\" title=\"View reference  in text\" id=\"ref-24\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.24\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Radim Rehurek</span>,  <span class=\"cit-name-given-names\">P.S.</span></span> (<span class=\"cit-pub-date\">2010</span>). <span class=\"cit-article-title\">Software Framework for Topic Modelling with Large Corpora</span>. <abbr class=\"cit-jnl-abbrev\">In IN PROCEEDINGS OF THE LREC 2010 WORKSHOP ON NEW CHALLENGES FOR NLP FRAMEWORKS</abbr>,.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-25-1\" title=\"View reference  in text\" id=\"ref-25\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.25\" data-doi=\"10.1371/journal.pone.0241535\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Rouchka</span>,  <span class=\"cit-name-given-names\">E.C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Chariker</span>,  <span class=\"cit-name-given-names\">J.H.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Chung</span>,  <span class=\"cit-name-given-names\">D.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Variant analysis of 1,040 SARS-CoV-2 genomes</span>. <abbr class=\"cit-jnl-abbrev\">PLoS One</abbr> <span class=\"cit-vol\">15</span>, <span class=\"cit-fpage\">e0241535</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DPLoS%2BOne%26rft.volume%253D15%26rft.spage%253De0241535%26rft_id%253Dinfo%253Adoi%252F10.1371%252Fjournal.pone.0241535%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a><a href=\"/lookup/external-ref?access_num=10.1371/journal.pone.0241535&amp;link_type=DOI\" class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\"><span>CrossRef</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-26-1\" title=\"View reference  in text\" id=\"ref-26\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.26\" data-doi=\"10.1371/journal.pone.0245584\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Shishir</span>,  <span class=\"cit-name-given-names\">T.A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Naser</span>,  <span class=\"cit-name-given-names\">I.B.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Faruque</span>,  <span class=\"cit-name-given-names\">S.M.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">In silico comparative genomics of SARS-CoV-2 to determine the source and diversity of the pathogen in Bangladesh</span>. <abbr class=\"cit-jnl-abbrev\">PLoS One</abbr> <span class=\"cit-vol\">16</span>, <span class=\"cit-fpage\">e0245584</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DPLoS%2BOne%26rft.volume%253D16%26rft.spage%253De0245584%26rft_id%253Dinfo%253Adoi%252F10.1371%252Fjournal.pone.0245584%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a><a href=\"/lookup/external-ref?access_num=10.1371/journal.pone.0245584&amp;link_type=DOI\" class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\"><span>CrossRef</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-27-1\" title=\"View reference  in text\" id=\"ref-27\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.27\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Shu</span>,  <span class=\"cit-name-given-names\">Y.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">McCauley</span>,  <span class=\"cit-name-given-names\">J.</span></span> (<span class=\"cit-pub-date\">2017</span>). <span class=\"cit-article-title\">GISAID: Global initiative on sharing all influenza data - from vision to reality</span>. <abbr class=\"cit-jnl-abbrev\">Euro Surveill</abbr>. <span class=\"cit-vol\">22</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-28-1\" title=\"View reference  in text\" id=\"ref-28\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.28\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Srivastava</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Banu</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Singh</span>,  <span class=\"cit-name-given-names\">P.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Sowpati</span>,  <span class=\"cit-name-given-names\">D.T.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Mishra</span>,  <span class=\"cit-name-given-names\">R.K.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">SARS-CoV-2 genomics: An Indian perspective on sequencing viral variants</span>. <abbr class=\"cit-jnl-abbrev\">J. Biosci</abbr>. <span class=\"cit-vol\">46</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-29-1\" title=\"View reference  in text\" id=\"ref-29\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.29\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Tomcala</span>,  <span class=\"cit-name-given-names\">J.</span></span> (<span class=\"cit-pub-date\">2018</span>). <span class=\"cit-article-title\">Time Series Entropies</span> <abbr class=\"cit-jnl-abbrev\">[R package TSEntropies version 0.9]</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-30-1\" title=\"View reference  in text\" id=\"ref-30\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.30\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Tomčala</span>,  <span class=\"cit-name-given-names\">J.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">New Fast ApEn and SampEn Entropy Algorithms Implementation and Their Application to Supercomputer Power Consumption</span>. <abbr class=\"cit-jnl-abbrev\">Entropy</abbr> <span class=\"cit-vol\">22</span>, <span class=\"cit-fpage\">863</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DEntropy%26rft.volume%253D22%26rft.spage%253D863%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-31-1\" title=\"View reference  in text\" id=\"ref-31\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.31\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Wang</span>,  <span class=\"cit-name-given-names\">B.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhao</span>,  <span class=\"cit-name-given-names\">D.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Lioma</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Li</span>,  <span class=\"cit-name-given-names\">Q.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">P.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Simonsen</span>,  <span class=\"cit-name-given-names\">J.G.</span></span> (<span class=\"cit-pub-date\">2019</span>). <abbr class=\"cit-jnl-abbrev\">Encoding word order in complex embeddings</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-32-1\" title=\"View reference  in text\" id=\"ref-32\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.32\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Wang</span>,  <span class=\"cit-name-given-names\">R.Y.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Guo</span>,  <span class=\"cit-name-given-names\">T.Q.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Li</span>,  <span class=\"cit-name-given-names\">L.G.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Jiao</span>,  <span class=\"cit-name-given-names\">J.Y.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Wang</span>,  <span class=\"cit-name-given-names\">L.Y.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Predictions of COVID-19 Infection Severity Based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 through Machine Learning of Genetic Data</span>. <abbr class=\"cit-jnl-abbrev\">2020 IEEE 8th International Conference on Computer Science and Network Technology (ICCSNT)</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.33\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth cit-collab\">WHO Coronavirus (COVID-19) Dashboard</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-34-1\" title=\"View reference  in text\" id=\"ref-34\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.34\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Wolf</span>,  <span class=\"cit-name-given-names\">T.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Debut</span>,  <span class=\"cit-name-given-names\">L.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Sanh</span>,  <span class=\"cit-name-given-names\">V.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Chaumond</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Delangue</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Moi</span>,  <span class=\"cit-name-given-names\">A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Cistac</span>,  <span class=\"cit-name-given-names\">P.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Rault</span>,  <span class=\"cit-name-given-names\">T.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Louf</span>,  <span class=\"cit-name-given-names\">R.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Funtowicz</span>,  <span class=\"cit-name-given-names\">M.</span></span>, <span class=\"cit-etal\">et al.</span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Transformers: State-of-the-Art Natural Language Processing</span>. <abbr class=\"cit-jnl-abbrev\">Proceedings of the 2020 Conference on Empirical Methods in Natural Language Processing: System Demonstrations</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-35-1\" title=\"View reference  in text\" id=\"ref-35\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.35\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Wu</span>,  <span class=\"cit-name-given-names\">F.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Yang</span>,  <span class=\"cit-name-given-names\">R.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">C.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">L.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">A deep learning framework combined with word embedding to identify DNA replication origins</span>. <abbr class=\"cit-jnl-abbrev\">Sci. Rep</abbr>. <span class=\"cit-vol\">11</span>, <span class=\"cit-fpage\">844</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DSci.%2BRep%26rft.volume%253D11%26rft.spage%253D844%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-36-1\" title=\"View reference  in text\" id=\"ref-36\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.36\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Yilmaz</span>,  <span class=\"cit-name-given-names\">A.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Assessment of Mutation Susceptibility in DNA Sequences with Word Vectors</span>. <abbr class=\"cit-jnl-abbrev\">Journal of Intelligent Systems: Theory and Applications</abbr> <span class=\"cit-vol\">3</span>, <span class=\"cit-fpage\">1</span>–<span class=\"cit-lpage\">6</span>.</cite></div><div class=\"cit-extra\"><a href=\"{openurl}?query=rft.jtitle%253DJournal%2Bof%2BIntelligent%2BSystems%253A%2BTheory%2Band%2BApplications%26rft.volume%253D3%26rft.spage%253D1%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\" class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-37-1\" title=\"View reference  in text\" id=\"ref-37\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.37\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Yin</span>,  <span class=\"cit-name-given-names\">Z.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Shen</span>,  <span class=\"cit-name-given-names\">Y.</span></span> (<span class=\"cit-pub-date\">2018</span>). <span class=\"cit-article-title\">On the Dimensionality of Word Embedding</span>. <abbr class=\"cit-jnl-abbrev\">Adv. Neural Inf. Process. Syst</abbr>. <span class=\"cit-vol\">31</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.38\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth cit-collab\">(2020a)</span>. <span class=\"cit-article-title\">Time Series Analysis and Computational Finance</span> <abbr class=\"cit-jnl-abbrev\">[R package tseries version 0.10-48]</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.39\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth cit-collab\">(2020b)</span>. <span class=\"cit-article-title\">Detrended Fluctuation and Detrended Cross-Correlation Analysis</span> <abbr class=\"cit-jnl-abbrev\">[R package DCCA version 0.1.1]</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li></ol></div><span class=\"highwire-journal-article-marker-end\"></span></div><span class=\"related-urls\"></span></div></div>  </div>\\n\\n  \\n  </div>\\n</div>\\n  </div>\\n</div>\\n</div></div>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-disqus-comment\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div id=\"disqus_thread\"><noscript><p><a href=\"http://biorxivstage.disqus.com/?url=https%3A%2F%2Fwww.biorxiv.org%2Fcontent%2F10.1101%2F2021.05.25.445601v3\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">View the discussion thread.</a></p></noscript></div>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-back-to-top\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <a href=\"#page\" class=\"back-to-top\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"icon-chevron-up\"></span> Back to top</a>  </div>\\n\\n  \\n  </div>\\n</div>\\n\\t\\t\\t</div>\\n\\t\\t</div>\\n\\t\\t\\n\\t\\t<div class=\"sidebar-right-wrapper grid-10 omega\">\\n\\t\\t\\t<div class=\"panel-panel panel-region-sidebar-right\">\\n\\t\\t\\t  <div class=\"inside\"><div class=\"panel-pane pane-highwire-node-pager\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"pager highwire-pager pager-mini clearfix highwire-node-pager highwire-article-pager\"><span class=\"pager-prev\"><a href=\"/content/10.1101/2021.05.25.445523v2\" title=\"Impact of glycosylation on a broad-spectrum vaccine against SARS-CoV-2\" rel=\"prev\" class=\"pager-link-prev link-icon\"><span class=\"icon-circle-arrow-left\"></span> <span class=\"title\">Previous</span></a></span><span class=\"pager-next\"><a href=\"/content/10.1101/2021.08.25.457533v1\" title=\"The ultrafast snap of a finger is mediated by skin friction\" rel=\"next\" class=\"pager-link-next link-icon-right link-icon\"><span class=\"title\">Next</span> <span class=\"icon-circle-arrow-right\"></span></a></span></div>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-custom pane-1\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    Posted&nbsp;August 26, 2021.  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-panels-mini pane-biorxiv-art-tools\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div id=\"mini-panel-biorxiv_art_tools\" class=\"highwire-2col-stacked panel-display\">\\n\\t  \\n  <div class=\"panel-row-wrapper clearfix\">\\n\\t\\t<div class=\"content-left-wrapper content-column\">\\n          <div class=\"panel-panel panel-region-content-left\">\\n            <div class=\"inside\"><div class=\"panel-pane pane-highwire-variant-link\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <a href=\"/content/10.1101/2021.05.25.445601v3.full.pdf\" target=\"_self\" class=\"article-dl-pdf-link link-icon\"><span class=\"icon-external-link-sign\"></span> <span class=\"title\">Download PDF</span></a>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-biorxiv-supplementary-fragment\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <p><a href=\"/content/10.1101/2021.05.25.445601v3.supplementary-material?versioned=true\"><i class=\"icon-file\"></i>Supplementary Material </a></p>  </div>\\n\\n  \\n  </div>\\n</div>\\n          </div>\\n        </div>\\n        \\n        <div class=\"content-right-wrapper content-column\">\\n          <div class=\"panel-panel panel-region-content-right\">\\n            <div class=\"inside\"><div class=\"panel-pane pane-minipanel-dialog-link pane-biorxiv-art-email\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\\'minipanel-dialog-wrapper\\'><div class=\\'minipanel-dialog-link-link\\'><a href=\"/\" oncontextmenu=\"javascript: return false;\" class=\"minipanel-dialog-link-trigger\" title=\"Email this Article\" data-icon-position=\"\" data-hide-link-title=\"0\"><i class = \\'icon-envelope\\'></i> Email</a></div><div class=\\'minipanel-dialog-link-mini\\' style=\\'display:none\\'><div class=\"panel-display panel-1col clearfix\" id=\"mini-panel-biorxiv_art_email\">\\n  <div class=\"panel-panel panel-col\">\\n    <div><div class=\"panel-pane pane-block pane-forward-form pane-forward\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <form action=\"/content/10.1101/2021.05.25.445601v3.full\" method=\"post\" id=\"forward-form\" accept-charset=\"UTF-8\"><div><div id=\"edit-instructions\" class=\"form-item form-item-label-before form-type-item\">\\n <p>Thank you for your interest in spreading the word about bioRxiv.</p><p>NOTE: Your email address is requested solely to identify you as the sender of this article.</p>\\n</div>\\n<div class=\"form-item form-item-label-before form-type-textfield form-item-email\">\\n  <label for=\"edit-email\">Your Email <span class=\"form-required\" title=\"This field is required.\">*</span></label>\\n <input type=\"text\" id=\"edit-email\" name=\"email\" value=\"\" size=\"58\" maxlength=\"256\" class=\"form-text required\" />\\n</div>\\n<div class=\"form-item form-item-label-before form-type-textfield form-item-name\">\\n  <label for=\"edit-name\">Your Name <span class=\"form-required\" title=\"This field is required.\">*</span></label>\\n <input type=\"text\" id=\"edit-name\" name=\"name\" value=\"\" size=\"58\" maxlength=\"256\" class=\"form-text required\" />\\n</div>\\n<div class=\"form-item form-item-label-before form-type-textarea form-item-recipients\">\\n  <label for=\"edit-recipients\">Send To <span class=\"form-required\" title=\"This field is required.\">*</span></label>\\n <div class=\"form-textarea-wrapper resizable\"><textarea id=\"edit-recipients\" name=\"recipients\" cols=\"50\" rows=\"5\" class=\"form-textarea required\"></textarea></div>\\n<div class=\"description\">Enter multiple addresses on separate lines or separate them with commas.</div>\\n</div>\\n<div id=\"edit-page\" class=\"form-item form-item-label-before form-type-item\">\\n  <label for=\"edit-page\">You are going to email the following </label>\\n <a href=\"/content/10.1101/2021.05.25.445601v3\" class=\"active\" data-icon-position=\"\" data-hide-link-title=\"0\">Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning</a>\\n</div>\\n<div id=\"edit-subject\" class=\"form-item form-item-label-before form-type-item\">\\n  <label for=\"edit-subject\">Message Subject </label>\\n (Your Name) has forwarded a page to you from bioRxiv\\n</div>\\n<div id=\"edit-body\" class=\"form-item form-item-label-before form-type-item\">\\n  <label for=\"edit-body\">Message Body </label>\\n (Your Name) thought you would like to see this page from the bioRxiv website.\\n</div>\\n<div class=\"form-item form-item-label-before form-type-textarea form-item-message\">\\n  <label for=\"edit-message--2\">Your Personal Message </label>\\n <div class=\"form-textarea-wrapper resizable\"><textarea id=\"edit-message--2\" name=\"message\" cols=\"50\" rows=\"10\" class=\"form-textarea\"></textarea></div>\\n</div>\\n<input type=\"hidden\" name=\"path\" value=\"node/2125191\" />\\n<input type=\"hidden\" name=\"path_cid\" value=\"\" />\\n<input type=\"hidden\" name=\"forward_footer\" value=\" \" />\\n<input type=\"hidden\" name=\"form_build_id\" value=\"form-M42UdW_KPHtrmp0JJEJt9TMaOECXs8X-JzC357FpLlA\" />\\n<input type=\"hidden\" name=\"form_id\" value=\"forward_form\" />\\n<fieldset class=\"captcha form-wrapper\"><legend><span class=\"fieldset-legend\">CAPTCHA</span></legend><div class=\"fieldset-wrapper\"><div class=\"fieldset-description\">This question is for testing whether or not you are a human visitor and to prevent automated spam submissions.</div><input type=\"hidden\" name=\"captcha_sid\" value=\"345930021\" />\\n<input type=\"hidden\" name=\"captcha_token\" value=\"be6f1827934e713c99e2ba0c3456dccf\" />\\n<input type=\"hidden\" name=\"captcha_response\" value=\"Google no captcha\" />\\n<div class=\"g-recaptcha\" data-sitekey=\"6LfnJVIUAAAAAE-bUOMg0MJGki4lqSvDmhJp19fN\" data-theme=\"light\" data-type=\"image\"></div></div></fieldset>\\n<div class=\"form-actions form-wrapper\" id=\"edit-actions\"><input type=\"submit\" id=\"edit-submit\" name=\"op\" value=\"Send Message\" class=\"form-submit\" /></div></div></form>  </div>\\n\\n  \\n  </div>\\n</div>\\n  </div>\\n</div>\\n</div></div>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-share-link highwire_clipboard_link_ajax\"  id=\"shareit\">\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <a href=\"/\" class=\"link-icon\"><span class=\"icon-share-alt\"></span> <span class=\"title\">Share</span></a>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-panels-mini pane-biorxiv-share highwire_clipboard_form_ajax_shareit\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    \\n<div class=\"panel-display omega-12-onecol\" id=\"mini-panel-biorxiv_share\">\\n\\n\\t\\t  <div class=\"panel-panel grid-12 panel-region-preface\">\\n\\t\\t  <div class=\"inside\"><div class=\"panel-pane pane-highwire-article-citation\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"highwire-article-citation highwire-citation-type-highwire-article node2125191--3\" data-node-nid=\"2125191\" id=\"node2125191--4826694706\" data-pisa=\"biorxiv;2021.05.25.445601v3\" data-pisa-master=\"biorxiv;2021.05.25.445601\" data-seqnum=\"2125191\" data-apath=\"/biorxiv/early/2021/08/26/2021.05.25.445601.atom\"><div  class=\"highwire-cite highwire-cite-highwire-article highwire-citation-biorxiv-article-pap-list clearfix\" >\\n\\n      <div class=\"highwire-cite-title\" >\\n      <div class=\"highwire-cite-title\">Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning</div>    </div>\\n  \\n      <div  class=\"highwire-cite-authors\" ><span  class=\"highwire-citation-authors\"><span class=\"highwire-citation-author first\" data-delta=\"0\"><span class=\"nlm-given-names\">Sargun</span> <span class=\"nlm-surname\">Nagpal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"1\"><span class=\"nlm-given-names\">Ridam</span> <span class=\"nlm-surname\">Pal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"2\"><span class=\"nlm-surname\">Ashima</span></span>, <span class=\"highwire-citation-author\" data-delta=\"3\"><span class=\"nlm-given-names\">Ananya</span> <span class=\"nlm-surname\">Tyagi</span></span>, <span class=\"highwire-citation-author\" data-delta=\"4\"><span class=\"nlm-given-names\">Sadhana</span> <span class=\"nlm-surname\">Tripathi</span></span>, <span class=\"highwire-citation-author\" data-delta=\"5\"><span class=\"nlm-given-names\">Aditya</span> <span class=\"nlm-surname\">Nagori</span></span>, <span class=\"highwire-citation-author\" data-delta=\"6\"><span class=\"nlm-given-names\">Saad</span> <span class=\"nlm-surname\">Ahmad</span></span>, <span class=\"highwire-citation-author\" data-delta=\"7\"><span class=\"nlm-given-names\">Hara Prasad</span> <span class=\"nlm-surname\">Mishra</span></span>, <span class=\"highwire-citation-author\" data-delta=\"8\"><span class=\"nlm-given-names\">Rintu</span> <span class=\"nlm-surname\">Kutum</span></span>, <span class=\"highwire-citation-author\" data-delta=\"9\"><span class=\"nlm-given-names\">Tavpritesh</span> <span class=\"nlm-surname\">Sethi</span></span></span></div>\\n  \\n      <div  class=\"highwire-cite-metadata\" ><span  class=\"highwire-cite-metadata-journal highwire-cite-metadata\">bioRxiv </span><span  class=\"highwire-cite-metadata-pages highwire-cite-metadata\">2021.05.25.445601; </span><span  class=\"highwire-cite-metadata-doi highwire-cite-metadata\"><span class=\"doi_label\">doi:</span> https://doi.org/10.1101/2021.05.25.445601 </span></div>\\n  \\n  \\n  </div>\\n</div>  </div>\\n\\n  \\n  </div>\\n</div>\\n\\t  </div>\\n\\t\\n  <div class=\"panel-panel grid-12 panel-region-content\">\\n  \\t<div class=\"inside\"><div class=\"panel-pane pane-highwire-article-clipboard-copy\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class = \"clipboard-copy\">\\n  <span class=\"label-url\">\\n    <label for=\"dynamic\">Share This Article:</label>\\n  </span>\\n  <span class=\"input-text-url\">\\n    <input type=\"text\" id=\"dynamic\" value=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" size=\"50\"/>\\n  </span>\\n  <span class=\"copy-button button\">\\n    <button id=\"copy-dynamic\" class=\"clipboardjs-button\" data-clipboard-target=\"#dynamic\" data-clipboard-alert-style=\"tooltip\" data-clipboard-alert-text=\"Copied!\">Copy</button>\\n  </span>\\n</div>\\n  </div>\\n\\n  \\n  </div>\\n</div>\\n  </div>\\n\\t\\n\\t\\t  <div class=\"panel-panel grid-12 panel-region-postscript\">\\n\\t    <div class=\"inside\"><div class=\"panel-pane pane-service-links\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"service-links\"><a href=\"http://digg.com/submit?phase=2&amp;url=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;title=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"digg\" title=\"Digg this post on digg.com\" class=\"service-links-digg\" rel=\"nofollow\" data-icon-position=\"\" data-hide-link-title=\"0\"><img src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/digg.png\" alt=\"Digg logo\" /></a> <a href=\"http://reddit.com/submit?url=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;title=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"reddit\" title=\"Submit this post on reddit.com\" class=\"service-links-reddit\" rel=\"nofollow\" data-icon-position=\"\" data-hide-link-title=\"0\"><img src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/reddit.png\" alt=\"Reddit logo\" /></a> <a href=\"http://twitter.com/share?url=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;text=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"twitter\" title=\"Share this on Twitter\" class=\"service-links-twitter\" rel=\"nofollow\" data-icon-position=\"\" data-hide-link-title=\"0\"><img src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/twitter.png\" alt=\"Twitter logo\" /></a> <a href=\"http://www.facebook.com/sharer.php?u=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;t=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"facebook\" title=\"Share on Facebook\" class=\"service-links-facebook\" rel=\"nofollow\" data-icon-position=\"\" data-hide-link-title=\"0\"><img src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/fb-blue.png\" alt=\"Facebook logo\" /></a> <a href=\"http://www.google.com/bookmarks/mark?op=add&amp;bkmk=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;title=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"google\" title=\"Bookmark this post on Google\" class=\"service-links-google\" rel=\"nofollow\" data-icon-position=\"\" data-hide-link-title=\"0\"><img src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/google-32.png\" alt=\"Google logo\" /></a> <a href=\"http://www.mendeley.com/import/?url=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;title=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"mendeley\" title=\"Share on Mendeley\" class=\"service-links-mendeley\" rel=\"nofollow\" data-icon-position=\"\" data-hide-link-title=\"0\"><img src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/mendeley.png\" alt=\"Mendeley logo\" /></a></div>  </div>\\n\\n  \\n  </div>\\n</div>\\n\\t  </div>\\n\\t\\n</div>\\n  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-minipanel-dialog-link pane-biorxiv-cite-tool\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\\'minipanel-dialog-wrapper\\'><div class=\\'minipanel-dialog-link-link\\'><a href=\"/\" oncontextmenu=\"javascript: return false;\" class=\"minipanel-dialog-link-trigger link-icon\" title=\"Citation Tools\"><span class=\"icon-globe\"></span> <span class=\"title\">Citation Tools</span></a></div><div class=\\'minipanel-dialog-link-mini\\' style=\\'display:none\\'><div class=\"panel-display panel-1col clearfix\" id=\"mini-panel-biorxiv_cite_tool\">\\n  <div class=\"panel-panel panel-col\">\\n    <div><div class=\"panel-pane pane-highwire-citation-export\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"highwire-citation-export\">  \\n  <div class=\"highwire-citation-info\">\\n      <div class=\"highwire-article-citation highwire-citation-type-highwire-article cite-tool-node2125191--5\" data-node-nid=\"2125191\" id=\"cite-tool-node2125191--6600659653\" data-pisa=\"biorxiv;2021.05.25.445601v3\" data-pisa-master=\"biorxiv;2021.05.25.445601\" data-seqnum=\"2125191\" data-apath=\"/biorxiv/early/2021/08/26/2021.05.25.445601.atom\"><div  class=\"highwire-cite highwire-cite-highwire-article highwire-citation-biorxiv-article-pap-list clearfix\" >\\n\\n      <div class=\"highwire-cite-title\" >\\n      <div class=\"highwire-cite-title\">Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning</div>    </div>\\n  \\n      <div  class=\"highwire-cite-authors\" ><span  class=\"highwire-citation-authors\"><span class=\"highwire-citation-author first\" data-delta=\"0\"><span class=\"nlm-given-names\">Sargun</span> <span class=\"nlm-surname\">Nagpal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"1\"><span class=\"nlm-given-names\">Ridam</span> <span class=\"nlm-surname\">Pal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"2\"><span class=\"nlm-surname\">Ashima</span></span>, <span class=\"highwire-citation-author\" data-delta=\"3\"><span class=\"nlm-given-names\">Ananya</span> <span class=\"nlm-surname\">Tyagi</span></span>, <span class=\"highwire-citation-author\" data-delta=\"4\"><span class=\"nlm-given-names\">Sadhana</span> <span class=\"nlm-surname\">Tripathi</span></span>, <span class=\"highwire-citation-author\" data-delta=\"5\"><span class=\"nlm-given-names\">Aditya</span> <span class=\"nlm-surname\">Nagori</span></span>, <span class=\"highwire-citation-author\" data-delta=\"6\"><span class=\"nlm-given-names\">Saad</span> <span class=\"nlm-surname\">Ahmad</span></span>, <span class=\"highwire-citation-author\" data-delta=\"7\"><span class=\"nlm-given-names\">Hara Prasad</span> <span class=\"nlm-surname\">Mishra</span></span>, <span class=\"highwire-citation-author\" data-delta=\"8\"><span class=\"nlm-given-names\">Rintu</span> <span class=\"nlm-surname\">Kutum</span></span>, <span class=\"highwire-citation-author\" data-delta=\"9\"><span class=\"nlm-given-names\">Tavpritesh</span> <span class=\"nlm-surname\">Sethi</span></span></span></div>\\n  \\n      <div  class=\"highwire-cite-metadata\" ><span  class=\"highwire-cite-metadata-journal highwire-cite-metadata\">bioRxiv </span><span  class=\"highwire-cite-metadata-pages highwire-cite-metadata\">2021.05.25.445601; </span><span  class=\"highwire-cite-metadata-doi highwire-cite-metadata\"><span class=\"doi_label\">doi:</span> https://doi.org/10.1101/2021.05.25.445601 </span></div>\\n  \\n  \\n  </div>\\n</div>  </div>\\n  <div class=\"highwire-citation-formats\">\\n  \\t      <h2>Citation Manager Formats</h2>\\n        <div class=\"highwire-citation-formats-links\">\\n      <span class=\"Z3988\" title=\"ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.spage&amp;rft.epage&amp;rft.atitle=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning&amp;rft.volume&amp;rft.issue&amp;rft.date=2021-01-01%2000%3A00%3A00&amp;rft.stitle&amp;rft.jtitle=bioRxiv&amp;rft.au=Nagpal%2C+Sargun&amp;rft.au=Pal%2C+Ridam&amp;rft.au=Ashima%2C+&amp;rft.au=Tyagi%2C+Ananya&amp;rft.au=Tripathi%2C+Sadhana&amp;rft.au=Nagori%2C+Aditya&amp;rft.au=Ahmad%2C+Saad&amp;rft.au=Mishra%2C+Hara+Prasad&amp;rft.au=Kutum%2C+Rintu&amp;rft.au=Sethi%2C+Tavpritesh\"></span><ul class=\"hw-citation-links inline button button-alt button-grid clearfix\"><li class=\"bibtext first\"><a href=\"/highwire/citation/2125191/bibtext\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">BibTeX</a></li><li class=\"bookends\"><a href=\"/highwire/citation/2125191/bookends\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">Bookends</a></li><li class=\"easybib\"><a href=\"/highwire/citation/2125191/easybib\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">EasyBib</a></li><li class=\"endnote-tagged\"><a href=\"/highwire/citation/2125191/endnote-tagged\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">EndNote (tagged)</a></li><li class=\"endnote-8-xml\"><a href=\"/highwire/citation/2125191/endnote-8-xml\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">EndNote 8 (xml)</a></li><li class=\"medlars\"><a href=\"/highwire/citation/2125191/medlars\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">Medlars</a></li><li class=\"mendeley\"><a href=\"/highwire/citation/2125191/mendeley\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">Mendeley</a></li><li class=\"papers\"><a href=\"/highwire/citation/2125191/papers\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">Papers</a></li><li class=\"refworks-tagged\"><a href=\"/highwire/citation/2125191/refworks-tagged\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">RefWorks Tagged</a></li><li class=\"reference-manager\"><a href=\"/highwire/citation/2125191/reference-manager\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">Ref Manager</a></li><li class=\"ris\"><a href=\"/highwire/citation/2125191/ris\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">RIS</a></li><li class=\"zotero last\"><a href=\"/highwire/citation/2125191/zotero\" class=\"hw-download-citation-link\" data-icon-position=\"\" data-hide-link-title=\"0\">Zotero</a></li></ul>    </div>\\n  </div>\\n</div>\\n  </div>\\n\\n  \\n  </div>\\n</div>\\n  </div>\\n</div>\\n</div></div>  </div>\\n\\n  \\n  </div>\\n</div>\\n          </div>\\n        </div>\\n\\t</div> <!-- /.panel-row-wrapper -->\\t\\n\\t\\n\\t</div>\\n\\n  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-service-links\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"service-links\"><div class=\"item-list\"><ul><li class=\"first\"><a href=\"http://twitter.com/share?url=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;count=horizontal&amp;via=&amp;text=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning&amp;counturl=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" class=\"twitter-share-button service-links-twitter-widget\" id=\"twitter_widget\" title=\"Tweet This\" rel=\"nofollow\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"element-invisible\">Tweet Widget</span></a></li><li><a href=\"http://www.facebook.com/plugins/like.php?href=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;layout=button_count&amp;show_faces=false&amp;action=like&amp;colorscheme=light&amp;width=100&amp;height=21&amp;font=&amp;locale=\" id=\"facebook_like\" title=\"I Like it\" class=\"service-links-facebook-like\" rel=\"nofollow\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"element-invisible\">Facebook Like</span></a></li><li class=\"last\"><a href=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" id=\"google_plus_one\" title=\"Plus it\" class=\"service-links-google-plus-one\" rel=\"nofollow\" data-icon-position=\"\" data-hide-link-title=\"0\"><span class=\"element-invisible\">Google Plus One</span></a></li></ul></div></div>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-article-collections\" >\\n  \\n        <h2 class=\"pane-title\">Subject Area</h2>\\n    \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"highwire-list-wrapper highwire-article-collections\"><div class=\"highwire-list\"><ul class=\"highwire-article-collection-term-list\"><li class=\"first last odd\"><span class=\"highwire-article-collection-term\"><a href=\"/collection/bioinformatics\" class=\"highlight\" data-icon-position=\"\" data-hide-link-title=\"0\">Bioinformatics<i class=\"icon-caret-right\"></i>\\n</a></span></li></ul></div></div>  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-panels-mini pane-biorxiv-subject-collections block-style-col2\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"panel-flexible panels-flexible-new clearfix\" id=\"mini-panel-biorxiv_subject_collections\">\\n<div class=\"panel-flexible-inside panels-flexible-new-inside\">\\n<div class=\"panels-flexible-region panels-flexible-region-new-center panels-flexible-region-first panels-flexible-region-last\">\\n  <div class=\"inside panels-flexible-region-inside panels-flexible-region-new-center-inside panels-flexible-region-inside-first panels-flexible-region-inside-last\">\\n<div class=\"panel-pane pane-snippet\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"snippet biorxiv-subject-areas-table-title\" id=\"biorxiv-subject-areas-table-title\">\\n  \\n      \\n  <div class=\"snippet-content\">\\n    <b>Subject Areas</b>  </div>\\n\\n</div>\\n  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-snippet\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <div class=\"snippet biorxiv-subject-areas-view-papers\" id=\"biorxiv-subject-areas-view-papers\">\\n  \\n      \\n  <div class=\"snippet-content\">\\n    <a href=\"/content/early/recent\"><strong>All Articles</strong></a>  </div>\\n\\n</div>\\n  </div>\\n\\n  \\n  </div>\\n<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-subject-collections\" >\\n  \\n      \\n  \\n  <div class=\"pane-content\">\\n    <ul id=\"collection\" class=\"collection highwire-list-expand\"><li class=\"outer collection depth-2  child first\"><div class = \"data-wrapper\"><a href=\"/collection/animal-behavior-and-cognition\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Animal Behavior and Cognition</a> <span class = \"article-count\">(3426)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/biochemistry\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Biochemistry</a> <span class = \"article-count\">(7191)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/bioengineering\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Bioengineering</a> <span class = \"article-count\">(5204)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/bioinformatics\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Bioinformatics</a> <span class = \"article-count\">(19894)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/biophysics\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Biophysics</a> <span class = \"article-count\">(9797)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/cancer-biology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Cancer Biology</a> <span class = \"article-count\">(7530)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/cell-biology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Cell Biology</a> <span class = \"article-count\">(11031)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/clinical-trials\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Clinical Trials</a> <span class = \"article-count\">(138)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/developmental-biology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Developmental Biology</a> <span class = \"article-count\">(6299)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/ecology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Ecology</a> <span class = \"article-count\">(9765)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/epidemiology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Epidemiology</a> <span class = \"article-count\">(2064)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/evolutionary-biology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Evolutionary Biology</a> <span class = \"article-count\">(13113)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/genetics\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Genetics</a> <span class = \"article-count\">(9270)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/genomics\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Genomics</a> <span class = \"article-count\">(12382)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/immunology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Immunology</a> <span class = \"article-count\">(7523)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/microbiology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Microbiology</a> <span class = \"article-count\">(18640)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/molecular-biology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Molecular Biology</a> <span class = \"article-count\">(7295)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/neuroscience\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Neuroscience</a> <span class = \"article-count\">(40254)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/paleontology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Paleontology</a> <span class = \"article-count\">(292)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/pathology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Pathology</a> <span class = \"article-count\">(1209)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/pharmacology-and-toxicology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Pharmacology and Toxicology</a> <span class = \"article-count\">(2093)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/physiology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Physiology</a> <span class = \"article-count\">(3089)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/plant-biology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Plant Biology</a> <span class = \"article-count\">(6723)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/scientific-communication-and-education\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Scientific Communication and Education</a> <span class = \"article-count\">(1255)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/synthetic-biology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Synthetic Biology</a> <span class = \"article-count\">(1865)</span></div></li>\\n<li class=\"outer collection depth-2  child\"><div class = \"data-wrapper\"><a href=\"/collection/systems-biology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Systems Biology</a> <span class = \"article-count\">(5235)</span></div></li>\\n<li class=\"outer collection depth-2  child last\"><div class = \"data-wrapper\"><a href=\"/collection/zoology\" class=\"\" data-icon-position=\"\" data-hide-link-title=\"0\">Zoology</a> <span class = \"article-count\">(1068)</span></div></li>\\n</ul>  </div>\\n\\n  \\n  </div>\\n  </div>\\n</div>\\n</div>\\n</div>\\n  </div>\\n\\n  \\n  </div>\\n</div>\\n\\t\\t\\t</div>\\n\\t\\t</div>\\n\\t\\n\\t</div> <!-- /.panel-row-wrapper -->\\t\\n\\t\\n\\t</div>\\n\\n    </div>\\n  </div>\\n</div>      </div>\\n</div>  </div>\\n</section>    \\n  \\n  </div>    <div class=\"region region-page-bottom\" id=\"region-page-bottom\">\\n  <div class=\"region-inner region-page-bottom-inner\">\\n      </div>\\n</div><script type=\"text/javascript\" src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__VNH5GD8Zz7g6_hGCOZjVjdMndxxC6naiExSSJKX3k_A__kzmEdtbqCsJx5Z9yg2Qy0PptB__ufXm-TxUOl0KZzUw__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\"></script>\\n<script type=\"text/javascript\" src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__2WRbxlwOW0MEUc_hSWU5MBepQg6Lch6O5SZwefpJ6IE__HCL0YQJqLkOhrLPZZYGqosGvtFsEHMGghHIkSx4y9vA__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\" defer=\"defer\"></script>\\n<script type=\"text/javascript\" src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__N7ERJBYsOWyRnJgyoM125_Aiez2MOJGaUofG1JdWWBg__2cpwCQ7-xzTVeVvg_KOzwA1jka23oWApDPpgjoZKDCY__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\"></script>\\n<script type=\"text/javascript\" src=\"//d33xdlntwy0kbs.cloudfront.net/cshl_custom.js\"></script>\\n<script type=\"text/javascript\" src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__BmqjBnkz3MgYCAoc25s1lDRMEjLhC3mEPVonUFIHi08__Unwv5-ZIuHBfFwytsjEx1niBVJ7n1T4lPws7VrkHXM4__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\"></script>\\n<script type=\"text/javascript\">\\n<!--//--><![CDATA[//><!--\\nfunction euCookieComplianceLoadScripts() {}\\n//--><!]]>\\n</script>\\n<script type=\"text/javascript\">\\n<!--//--><![CDATA[//><!--\\nvar eu_cookie_compliance_cookie_name = \"\";\\n//--><!]]>\\n</script>\\n  </body>\\n</html>\\n'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 40
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "soup = BeautifulSoup(req_url.text, 'html.parser')\n",
        "soup"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "InQEj51WW0WF",
        "outputId": "78ab9b9d-66c0-443c-8efb-2fcd7310f560"
      },
      "execution_count": 41,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<!DOCTYPE html>\n",
              "\n",
              "<html dir=\"ltr\" lang=\"en\" xmlns=\"http://www.w3.org/1999/xhtml\" xmlns:mml=\"http://www.w3.org/1998/Math/MathML\">\n",
              "<head prefix=\"og: http://ogp.me/ns# article: http://ogp.me/ns/article# book: http://ogp.me/ns/book#\">\n",
              "<!--[if IE]><![endif]-->\n",
              "<link href=\"//d33xdlntwy0kbs.cloudfront.net\" rel=\"dns-prefetch\"/>\n",
              "<link href=\"//www.google.com\" rel=\"dns-prefetch\"/>\n",
              "<link href=\"//scholar.google.com\" rel=\"dns-prefetch\"/>\n",
              "<link href=\"//www.google-analytics.com\" rel=\"dns-prefetch\"/>\n",
              "<link href=\"//stats.g.doubleclick.net\" rel=\"dns-prefetch\"/>\n",
              "<meta content=\"text/html; charset=utf-8\" http-equiv=\"Content-Type\"/>\n",
              "<meta content=\"width=device-width, initial-scale=1\" name=\"viewport\"/>\n",
              "<link href=\"https://www.biorxiv.org/sites/default/files/images/favicon.ico\" rel=\"shortcut icon\" type=\"image/vnd.microsoft.icon\"/>\n",
              "<link href=\"/content/10.1101/2021.05.25.445601v3.full.pdf\" rel=\"alternate\" title=\"Full Text (PDF)\" type=\"application/pdf\"/>\n",
              "<link href=\"/content/10.1101/2021.05.25.445601v3.full.txt\" rel=\"alternate\" title=\"Full Text (Plain)\" type=\"text/plain\"/>\n",
              "<link href=\"/content/10.1101/2021.05.25.445601v3.ppt\" rel=\"alternate\" title=\"Powerpoint\" type=\"application/vnd.ms-powerpoint\"/>\n",
              "<meta content=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-10.gif\" name=\"article_thumbnail\"/>\n",
              "<meta content=\"article\" name=\"type\"/>\n",
              "<meta content=\"article\" name=\"category\"/>\n",
              "<meta content=\"/biorxiv/early/2021/08/26/2021.05.25.445601.atom\" name=\"HW.identifier\"/>\n",
              "<meta content=\"biorxiv;2021.05.25.445601v3\" name=\"HW.pisa\"/>\n",
              "<meta content=\"text/html\" name=\"DC.Format\"/>\n",
              "<meta content=\"en\" name=\"DC.Language\"/>\n",
              "<meta content=\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning\" name=\"DC.Title\"/>\n",
              "<meta content=\"10.1101/2021.05.25.445601\" name=\"DC.Identifier\"/>\n",
              "<meta content=\"2021-08-26\" name=\"DC.Date\"/>\n",
              "<meta content=\"Cold Spring Harbor Laboratory\" name=\"DC.Publisher\"/>\n",
              "<meta content=\"© 2021, Posted by Cold Spring Harbor Laboratory. This pre-print is available under a Creative Commons License (Attribution-NonCommercial-NoDerivs 4.0 International), CC BY-NC-ND 4.0, as described at http://creativecommons.org/licenses/by-nc-nd/4.0/\" name=\"DC.Rights\"/>\n",
              "<meta content=\"restricted\" name=\"DC.AccessRights\"/>\n",
              "<meta content=\"The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed Strainflow , to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.\n",
              "\n",
              "Highlights \n",
              "\n",
              "![Figure][1]&lt;/img&gt;\n",
              "\n",
              "### Competing Interest Statement\n",
              "\n",
              "The authors have declared no competing interest.\n",
              "\n",
              " [1]: pending:yes\" name=\"DC.Description\"/>\n",
              "<meta content=\"Sargun Nagpal\" name=\"DC.Contributor\"/>\n",
              "<meta content=\"Ridam Pal\" name=\"DC.Contributor\"/>\n",
              "<meta content=\"Ashima\" name=\"DC.Contributor\"/>\n",
              "<meta content=\"Ananya Tyagi\" name=\"DC.Contributor\"/>\n",
              "<meta content=\"Sadhana Tripathi\" name=\"DC.Contributor\"/>\n",
              "<meta content=\"Aditya Nagori\" name=\"DC.Contributor\"/>\n",
              "<meta content=\"Saad Ahmad\" name=\"DC.Contributor\"/>\n",
              "<meta content=\"Hara Prasad Mishra\" name=\"DC.Contributor\"/>\n",
              "<meta content=\"Rintu Kutum\" name=\"DC.Contributor\"/>\n",
              "<meta content=\"Tavpritesh Sethi\" name=\"DC.Contributor\"/>\n",
              "<meta content=\"2021-08-26\" name=\"article:published_time\"/>\n",
              "<meta content=\"New Results\" name=\"article:section\"/>\n",
              "<meta content=\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning\" name=\"citation_title\"/>\n",
              "<meta content=\"&lt;h3&gt;Abstract&lt;/h3&gt;\n",
              "&lt;p&gt;The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed &lt;i&gt;Strainflow&lt;/i&gt;, to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our &lt;i&gt;Strainflow&lt;/i&gt; model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called &lt;i&gt;Dimensions of Concern&lt;/i&gt; (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.&lt;/p&gt;&lt;h3&gt;Highlights&lt;/h3&gt;\n",
              "&lt;p&gt;We developed a genomic surveillance model for SARS-CoV-2 genome sequences, &lt;i&gt;Strainflow&lt;/i&gt;, where sequences were treated as documents with words (codons) to learn the codon context of 0.9 million spike genes using the skip-gram algorithm.&lt;/p&gt;&lt;p&gt;Time series analysis of the information content (Entropy) of the latent dimensions learned by &lt;i&gt;Strainflow&lt;/i&gt; shows a leading relationship with the monthly COVID-19 cases for seven countries (e.g., USA, Japan, India, and others).&lt;/p&gt;&lt;p&gt;Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads.&lt;/p&gt;&lt;p&gt;The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants.&lt;/p&gt;&lt;h3&gt;Graphical abstract&lt;/h3&gt;\n",
              "\" lang=\"en\" name=\"citation_abstract\"/>\n",
              "<meta content=\"bioRxiv\" name=\"citation_journal_title\"/>\n",
              "<meta content=\"Cold Spring Harbor Laboratory\" name=\"citation_publisher\"/>\n",
              "<meta content=\"2021/01/01\" name=\"citation_publication_date\"/>\n",
              "<meta content=\"biorxiv;2021.05.25.445601v3\" name=\"citation_mjid\"/>\n",
              "<meta content=\"2021.05.25.445601v3\" name=\"citation_id\"/>\n",
              "<meta content=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" name=\"citation_public_url\"/>\n",
              "<meta content=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3.abstract\" name=\"citation_abstract_html_url\"/>\n",
              "<meta content=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3.full\" name=\"citation_full_html_url\"/>\n",
              "<meta content=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601.full.pdf\" name=\"citation_pdf_url\"/>\n",
              "<meta content=\"10.1101/2021.05.25.445601\" name=\"citation_doi\"/>\n",
              "<meta content=\"20\" name=\"citation_num_pages\"/>\n",
              "<meta content=\"Article\" name=\"citation_article_type\"/>\n",
              "<meta content=\"New Results\" name=\"citation_section\"/>\n",
              "<meta content=\"2021.05.25.445601\" name=\"citation_firstpage\"/>\n",
              "<meta content=\"Sargun Nagpal\" name=\"citation_author\"/>\n",
              "<meta content=\"Indraprastha Institute of Information Technology Delhi\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"http://orcid.org/0000-0001-9936-7790\" name=\"citation_author_orcid\"/>\n",
              "<meta content=\"Ridam Pal\" name=\"citation_author\"/>\n",
              "<meta content=\"Indraprastha Institute of Information Technology Delhi\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"Ashima\" name=\"citation_author\"/>\n",
              "<meta content=\"Indraprastha Institute of Information Technology Delhi\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"Ananya Tyagi\" name=\"citation_author\"/>\n",
              "<meta content=\"Indraprastha Institute of Information Technology Delhi\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"Sadhana Tripathi\" name=\"citation_author\"/>\n",
              "<meta content=\"Indraprastha Institute of Information Technology Delhi\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"Aditya Nagori\" name=\"citation_author\"/>\n",
              "<meta content=\"Indraprastha Institute of Information Technology Delhi\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"http://orcid.org/0000-0002-6389-2179\" name=\"citation_author_orcid\"/>\n",
              "<meta content=\"Saad Ahmad\" name=\"citation_author\"/>\n",
              "<meta content=\"Indraprastha Institute of Information Technology Delhi\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"Hara Prasad Mishra\" name=\"citation_author\"/>\n",
              "<meta content=\"Indraprastha Institute of Information Technology Delhi\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"http://orcid.org/0000-0001-9405-7822\" name=\"citation_author_orcid\"/>\n",
              "<meta content=\"Rintu Kutum\" name=\"citation_author\"/>\n",
              "<meta content=\"Indraprastha Institute of Information Technology Delhi\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"tavpriteshsethi@iiitd.ac.in\" name=\"citation_author_email\"/>\n",
              "<meta content=\"rintuk@iiitd.ac.in\" name=\"citation_author_email\"/>\n",
              "<meta content=\"http://orcid.org/0000-0001-8667-6199\" name=\"citation_author_orcid\"/>\n",
              "<meta content=\"Tavpritesh Sethi\" name=\"citation_author\"/>\n",
              "<meta content=\"Indraprastha Institute of Information Technology Delhi\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"All India Institute of Medical Sciences\" name=\"citation_author_institution\"/>\n",
              "<meta content=\"tavpriteshsethi@iiitd.ac.in\" name=\"citation_author_email\"/>\n",
              "<meta content=\"rintuk@iiitd.ac.in\" name=\"citation_author_email\"/>\n",
              "<meta content=\"http://orcid.org/0000-0002-4776-7941\" name=\"citation_author_orcid\"/>\n",
              "<meta content=\"A. Liaw and M. Wiener (2002). Classification and Regression by randomForest. R News 2(3), 18--22.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Arslan, H. (2021). Machine Learning Methods for COVID-19 Prediction Using Human Genomic Data. Proceedings 74, 20.\" name=\"citation_reference\"/>\n",
              "<meta content=\"CDC (2021). SARS-CoV-2 Variant Classifications and Definitions.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Hadfield, J., Megill, C., Bell, S.M., Huddleston, J., Potter, B., Callender, C., Sagulenko, P., Bedford, T., and Neher, R.A. (2018). Nextstrain: real-time tracking of pathogen evolution. Bioinformatics 34, 4121–4123.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Hie, B., Zhong, E.D., Berger, B., and Bryson, B. (2021). Learning the language of viral evolution and escape. Science 371, 284–288.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Khan, M.I., Khan, Z.A., Baig, M.H., Ahmad, I., Farouk, A.-E., Song, Y.G., and Dong, J.-J. (2020). Comparative genome analysis of novel coronavirus (SARS-CoV-2) from different geographical locations and the effect of mutations on major target proteins: An in silico insight. PLoS One 15, e0238344.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Kursa, M.B., and Rudnicki, W.R. (2010). Feature Selection with theBorutaPackage. Journal of Statistical Software 36.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Kursa, M.B., Jankowski, A., and Rudnicki, W.R. (2010). Boruta - A System for Feature Selection. Fund. Inform. 101, 271–285.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Lee, J., Yoon, W., Kim, S., Kim, D., Kim, S., So, C.H., and Kang, J. (2020). BioBERT: a pre-trained biomedical language representation model for biomedical text mining. Bioinformatics 36, 1234–1240.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Letunic, I. iTOL: Interactive Tree Of Life.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Li, Q., Wu, J., Nie, J., Zhang, L., Hao, H., Liu, S., Zhao, C., Zhang, Q., Liu, H., Nie, L., et al. (2020). The Impact of Mutations in SARS-CoV-2 Spike on Viral Infectivity and Antigenicity. Cell 182, 1284–1294.e9.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Linderman, G.C., Rachh, M., Hoskins, J.G., Steinerberger, S., and Kluger, Y. (2019). Fast interpolation-based t-SNE for improved visualization of single-cell RNA-seq data. Nat. Methods 16, 243–245.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Liu, B., Liu, K., Zhang, H., Zhang, L., Bian, Y., and Huang, L. (2020). CoV-Seq, a New Tool for SARS-CoV-2 Genome Analysis and Visualization: Development and Usability Study. J. Med. Internet Res. 22, e22299.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Lopez-Rincon, A., Perez-Romero, C.A., Tonda, A., Mendoza-Maldonado, L., Claassen, E., Garssen, J., and Kraneveld, A.D. Design of Specific Primer Sets for the Detection of B.1.1.7, B.1.351, P.1, B.1.617.2 and B.1.1.519 Variants of SARS-CoV-2 using Artificial Intelligence.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Mandal, S., Roychowdhury, T., and Bhattacharya, A. (2021). Pattern of genomic variation in SARS-CoV-2 (COVID-19) suggests restricted nonrandom changes: Analysis using Shewhart control charts. J. Biosci. 46.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Mikolov, T., Chen, K., Corrado, G., and Dean, J. (2013). Efficient Estimation of Word Representations in Vector Space.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Mugnai, M.L., Templeton, C., Elber, R., and Thirumalai, D. (2020). Role of Long-range Allosteric Communication in Determining the Stability and Disassembly of SARS-COV-2 in Complex with ACE2. bioRxiv.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Mushtaq, R. (2011). Augmented Dickey Fuller Test. SSRN Electronic Journal.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Pan, Y.H., Wang, Y.H., Liang, S.F., and Lee, K.T. (2011). Fast computation of sample entropy and approximate entropy in biomedicine. Comput. Methods Programs Biomed. 104.\" name=\"citation_reference\"/>\n",
              "<meta content=\"paola (2021). Phylogenetic relationship of SARS-CoV-2 sequences from Amazonas with emerging Brazilian variants harboring mutations E484K and N501Y in the Spike protein.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Peacock, T.P., Penrice-Randal, R., Hiscox, J.A., and Barclay, W.S. (2021). SARS-CoV-2 one year on: evidence for ongoing viral adaptation. J. Gen. Virol. 102.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Podobnik, B., and Stanley, H.E. (2007). Detrended Cross-Correlation Analysis: A New Method for Analyzing Two Non-stationary Time Series.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Prass, T.S., and Pumi, G. (2019). On the behavior of the DFA and DCCA in trend-stationary processes.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Radim Rehurek, P.S. (2010). Software Framework for Topic Modelling with Large Corpora. In IN PROCEEDINGS OF THE LREC 2010 WORKSHOP ON NEW CHALLENGES FOR NLP FRAMEWORKS,.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Rouchka, E.C., Chariker, J.H., and Chung, D. (2020). Variant analysis of 1,040 SARS-CoV-2 genomes. PLoS One 15, e0241535.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Shishir, T.A., Naser, I.B., and Faruque, S.M. (2021). In silico comparative genomics of SARS-CoV-2 to determine the source and diversity of the pathogen in Bangladesh. PLoS One 16, e0245584.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Shu, Y., and McCauley, J. (2017). GISAID: Global initiative on sharing all influenza data - from vision to reality. Euro Surveill. 22.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Srivastava, S., Banu, S., Singh, P., Sowpati, D.T., and Mishra, R.K. (2021). SARS-CoV-2 genomics: An Indian perspective on sequencing viral variants. J. Biosci. 46.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Tomcala, J. (2018). Time Series Entropies [R package TSEntropies version 0.9].\" name=\"citation_reference\"/>\n",
              "<meta content=\"Tomčala, J. (2020). New Fast ApEn and SampEn Entropy Algorithms Implementation and Their Application to Supercomputer Power Consumption. Entropy 22, 863.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Wang, B., Zhao, D., Lioma, C., Li, Q., Zhang, P., and Simonsen, J.G. (2019). Encoding word order in complex embeddings.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Wang, R.Y., Guo, T.Q., Li, L.G., Jiao, J.Y., and Wang, L.Y. (2020). Predictions of COVID-19 Infection Severity Based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 through Machine Learning of Genetic Data. 2020 IEEE 8th International Conference on Computer Science and Network Technology (ICCSNT).\" name=\"citation_reference\"/>\n",
              "<meta content=\"WHO Coronavirus (COVID-19) Dashboard.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Wolf, T., Debut, L., Sanh, V., Chaumond, J., Delangue, C., Moi, A., Cistac, P., Rault, T., Louf, R., Funtowicz, M., et al. (2020). Transformers: State-of-the-Art Natural Language Processing. Proceedings of the 2020 Conference on Empirical Methods in Natural Language Processing: System Demonstrations.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Wu, F., Yang, R., Zhang, C., and Zhang, L. (2021). A deep learning framework combined with word embedding to identify DNA replication origins. Sci. Rep. 11, 844.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Yilmaz, A. (2020). Assessment of Mutation Susceptibility in DNA Sequences with Word Vectors. Journal of Intelligent Systems: Theory and Applications 3, 1–6.\" name=\"citation_reference\"/>\n",
              "<meta content=\"Yin, Z., and Shen, Y. (2018). On the Dimensionality of Word Embedding. Adv. Neural Inf. Process. Syst. 31.\" name=\"citation_reference\"/>\n",
              "<meta content=\"(2020a). Time Series Analysis and Computational Finance [R package tseries version 0.10-48].\" name=\"citation_reference\"/>\n",
              "<meta content=\"(2020b). Detrended Fluctuation and Detrended Cross-Correlation Analysis [R package DCCA version 0.1.1].\" name=\"citation_reference\"/>\n",
              "<meta content=\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning\" name=\"twitter:title\"/>\n",
              "<meta content=\"@biorxivpreprint\" name=\"twitter:site\"/>\n",
              "<meta content=\"summary\" name=\"twitter:card\"/>\n",
              "<meta content=\"https://www.biorxiv.org/sites/default/files/images/biorxiv_logo_homepage7-5-small.png\" name=\"twitter:image\"/>\n",
              "<meta content=\"The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed Strainflow , to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.\n",
              "\n",
              "Highlights \n",
              "\n",
              "![Figure][1]&lt;/img&gt;\n",
              "\n",
              "### Competing Interest Statement\n",
              "\n",
              "The authors have declared no competing interest.\n",
              "\n",
              " [1]: pending:yes\" name=\"twitter:description\"/>\n",
              "<meta content=\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning\" name=\"og-title\" property=\"og:title\"/>\n",
              "<meta content=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" name=\"og-url\" property=\"og:url\"/>\n",
              "<meta content=\"bioRxiv\" name=\"og-site-name\" property=\"og:site_name\"/>\n",
              "<meta content=\"The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed Strainflow , to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.\n",
              "\n",
              "Highlights \n",
              "\n",
              "![Figure][1]&lt;/img&gt;\n",
              "\n",
              "### Competing Interest Statement\n",
              "\n",
              "The authors have declared no competing interest.\n",
              "\n",
              " [1]: pending:yes\" name=\"og-description\" property=\"og:description\"/>\n",
              "<meta content=\"article\" name=\"og-type\" property=\"og:type\"/>\n",
              "<meta content=\"https://www.biorxiv.org/sites/default/files/images/biorxiv_logo_homepage7-5-small.png\" name=\"og-image\" property=\"og:image\"/>\n",
              "<meta content=\"2021-08-26\" name=\"citation_date\"/>\n",
              "<meta content=\"bioRxiv - the preprint server for biology, operated by Cold Spring Harbor Laboratory, a research and educational institution\" name=\"description\"/>\n",
              "<meta content=\"Drupal 7 (http://drupal.org)\" name=\"generator\"/>\n",
              "<link href=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" rel=\"canonical\"/>\n",
              "<link href=\"https://www.biorxiv.org/node/2125191\" rel=\"shortlink\"/>\n",
              "<title>Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning | bioRxiv</title>\n",
              "<link href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__GW-q1yS4llZsZXYilALi0SfHZdla2dBZSfgKsyPoQv0__Mq2l818MUdgw7DQFZ42QKliXne5RR1HOY89XjtvuppE__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" rel=\"stylesheet\" type=\"text/css\"/>\n",
              "<link href=\"//cdn.jsdelivr.net/qtip2/2.2.1/jquery.qtip.min.css\" media=\"all\" rel=\"stylesheet\" type=\"text/css\"/>\n",
              "<link href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__Sjm8qRC6pDZnJVeNrkx67-ZH1Mte_Ys5jCfEFFU1muc__r6-n363p7S972yCeOZaxDruAzBGzgr4Ie4Ci5dXnv0k__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" rel=\"stylesheet\" type=\"text/css\"/>\n",
              "<style media=\"all\" type=\"text/css\">\n",
              "/* <![CDATA[ */\n",
              ".panels-flexible-new .panels-flexible-region{padding:0}.panels-flexible-new .panels-flexible-region-inside{padding-right:.5em;padding-left:.5em}.panels-flexible-new .panels-flexible-region-inside-first{padding-left:0}.panels-flexible-new .panels-flexible-region-inside-last{padding-right:0}.panels-flexible-new .panels-flexible-column{padding:0}.panels-flexible-new .panels-flexible-column-inside{padding-right:.5em;padding-left:.5em}.panels-flexible-new .panels-flexible-column-inside-first{padding-left:0}.panels-flexible-new .panels-flexible-column-inside-last{padding-right:0}.panels-flexible-new .panels-flexible-row{padding:0 0 .5em;margin:0}.panels-flexible-new .panels-flexible-row-last{padding-bottom:0}.panels-flexible-column-new-main{float:left;width:99.0000%}.panels-flexible-new-inside{padding-right:0}.panels-flexible-new{width:auto}.panels-flexible-region-new-center{float:left;width:99.0000%}.panels-flexible-row-new-main-row-inside{padding-right:0}\n",
              "/* ]]> */\n",
              "</style>\n",
              "<link href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__PWuQ_RYRTJ4BLEKsbWQeHysPg0gOQ3571ruQa_rXvAo__cj81oxtSGUe-AIO3tlkSBJPNpKP5ZgLdB8f8awBQsWU__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" rel=\"stylesheet\" type=\"text/css\"/>\n",
              "<style media=\"all\" type=\"text/css\">\n",
              "/* <![CDATA[ */\n",
              "#sliding-popup.sliding-popup-bottom,#sliding-popup.sliding-popup-bottom .eu-cookie-withdraw-banner,.eu-cookie-withdraw-tab{background:gray}#sliding-popup.sliding-popup-bottom.eu-cookie-withdraw-wrapper{background:transparent}#sliding-popup .popup-content #popup-text h1,#sliding-popup .popup-content #popup-text h2,#sliding-popup .popup-content #popup-text h3,#sliding-popup .popup-content #popup-text p,.eu-cookie-compliance-secondary-button,.eu-cookie-withdraw-tab{color:#fff !important}.eu-cookie-withdraw-tab{border-color:#fff}.eu-cookie-compliance-more-button{color:#fff !important}\n",
              "/* ]]> */\n",
              "</style>\n",
              "<!--[if lte IE 7]>\n",
              "<link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__ElJr3PIJEvw3qLXc1cnYiLj2G4KgDPSXFOfm6Phf8hw__JdWGm15cDWjsK6KrFlQVXQix9YgNeYysf22XZHj-Y-c__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\n",
              "<![endif]-->\n",
              "<link href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__Wnlyen9qEpwh_Qaf9okEu4QdVGM0BDothxeqA6Nbvo8__EJmw6SZD9bYoS8jocCpPYS3JFRURpdzmuvJoAUNiI-g__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" rel=\"stylesheet\" type=\"text/css\"/>\n",
              "<link href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__vWH9eagUiEGaWTpZefl9TB2ucrJuL66ajzSCvO73bss__GnHmvNTOB0-xZRDC4PFtbOee_g-W6li7rpWRuc9O4B4__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" rel=\"stylesheet\" type=\"text/css\"/>\n",
              "<!--[if (lt IE 9)&(!IEMobile)]>\n",
              "<link type=\"text/css\" rel=\"stylesheet\" href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__XH6bpcI0f2dImc-p674DLCZtWBGb-QwxJK1YexVGtno__vUceGprdo5nIhV6DH93X7fI3r8RcTJbChbas9TQXeW4__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" />\n",
              "<![endif]-->\n",
              "<!--[if gte IE 9]><!-->\n",
              "<link href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__2WBMox6sOrN42ss5lCnH7WWVRdFdJCxtTKnQJYRwTE4__yqNvNYLvMpjy3ffuJrjjm9uW2i-Me1c23KLYuWHaqio__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" rel=\"stylesheet\" type=\"text/css\"/>\n",
              "<!--<![endif]-->\n",
              "<link href=\"https://www.biorxiv.org/sites/default/files/advagg_css/css__dMP0ngeKhq7e9DSwoh07jSjMdMBpvUxRs4IveUsPUNM__5c1ZiN_pM9tjxvpyA9VpEacScF0S_W4R222pC-s_-Pk__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.css\" media=\"all\" rel=\"stylesheet\" type=\"text/css\"/>\n",
              "<link href=\"https://d33xdlntwy0kbs.cloudfront.net/cshl_custom.css\" media=\"all\" rel=\"stylesheet\" type=\"text/css\"/>\n",
              "<script src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__-O5AqrChkaQBJNz3KVCNgPD7B2WnlmFaBWhKn3OLnSk__HPUPKsZe95KwbHSnKRhhlLxkHEchbpL1f2QRQnZQX38__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\" type=\"text/javascript\"></script>\n",
              "<script src=\"//cdn.jsdelivr.net/qtip2/2.2.1/jquery.qtip.min.js\" type=\"text/javascript\"></script>\n",
              "<script src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__49Ok9ZaeGdzHaYY0HeAxSZkJaw1DaCwfnGvLBVZ_w1Q__Tcognn9uta6IhrOugHP9-zOAzZitowtsZynB_751i8Q__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\" type=\"text/javascript\"></script>\n",
              "<script src=\"https://www.google.com/recaptcha/api.js?hl=en&amp;render=explicit&amp;onload=drupalRecaptchaOnload\" type=\"text/javascript\"></script>\n",
              "<script src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__6hIesE0QCBXoXlEIn01f-78o39DmDNEoInQ35EHxoXw__qAjUcZEMaB7nsfhw64tJNfG5xuXddaTuPmbIfa6fPMk__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\" type=\"text/javascript\"></script>\n",
              "<script async=\"async\" src=\"https://scholar.google.com/scholar_js/casa.js\" type=\"text/javascript\"></script>\n",
              "<script type=\"text/javascript\">\n",
              "<!--//--><![CDATA[//><!--\n",
              "/*!\n",
              " * yepnope1.5.4\n",
              " * (c) WTFPL, GPLv2\n",
              " */\n",
              "(function(a,b,c){function d(a){return\"[object Function]\"==o.call(a)}function e(a){return\"string\"==typeof a}function f(){}function g(a){return!a||\"loaded\"==a||\"complete\"==a||\"uninitialized\"==a}function h(){var a=p.shift();q=1,a?a.t?m(function(){(\"c\"==a.t?B.injectCss:B.injectJs)(a.s,0,a.a,a.x,a.e,1)},0):(a(),h()):q=0}function i(a,c,d,e,f,i,j){function k(b){if(!o&&g(l.readyState)&&(u.r=o=1,!q&&h(),l.onload=l.onreadystatechange=null,b)){\"img\"!=a&&m(function(){t.removeChild(l)},50);for(var d in y[c])y[c].hasOwnProperty(d)&&y[c][d].onload()}}var j=j||B.errorTimeout,l=b.createElement(a),o=0,r=0,u={t:d,s:c,e:f,a:i,x:j};1===y[c]&&(r=1,y[c]=[]),\"object\"==a?l.data=c:(l.src=c,l.type=a),l.width=l.height=\"0\",l.onerror=l.onload=l.onreadystatechange=function(){k.call(this,r)},p.splice(e,0,u),\"img\"!=a&&(r||2===y[c]?(t.insertBefore(l,s?null:n),m(k,j)):y[c].push(l))}function j(a,b,c,d,f){return q=0,b=b||\"j\",e(a)?i(\"c\"==b?v:u,a,b,this.i++,c,d,f):(p.splice(this.i++,0,a),1==p.length&&h()),this}function k(){var a=B;return a.loader={load:j,i:0},a}var l=b.documentElement,m=a.setTimeout,n=b.getElementsByTagName(\"script\")[0],o={}.toString,p=[],q=0,r=\"MozAppearance\"in l.style,s=r&&!!b.createRange().compareNode,t=s?l:n.parentNode,l=a.opera&&\"[object Opera]\"==o.call(a.opera),l=!!b.attachEvent&&!l,u=r?\"object\":l?\"script\":\"img\",v=l?\"script\":u,w=Array.isArray||function(a){return\"[object Array]\"==o.call(a)},x=[],y={},z={timeout:function(a,b){return b.length&&(a.timeout=b[0]),a}},A,B;B=function(a){function b(a){var a=a.split(\"!\"),b=x.length,c=a.pop(),d=a.length,c={url:c,origUrl:c,prefixes:a},e,f,g;for(f=0;f<d;f++)g=a[f].split(\"=\"),(e=z[g.shift()])&&(c=e(c,g));for(f=0;f<b;f++)c=x[f](c);return c}function g(a,e,f,g,h){var i=b(a),j=i.autoCallback;i.url.split(\".\").pop().split(\"?\").shift(),i.bypass||(e&&(e=d(e)?e:e[a]||e[g]||e[a.split(\"/\").pop().split(\"?\")[0]]),i.instead?i.instead(a,e,f,g,h):(y[i.url]?i.noexec=!0:y[i.url]=1,f.load(i.url,i.forceCSS||!i.forceJS&&\"css\"==i.url.split(\".\").pop().split(\"?\").shift()?\"c\":c,i.noexec,i.attrs,i.timeout),(d(e)||d(j))&&f.load(function(){k(),e&&e(i.origUrl,h,g),j&&j(i.origUrl,h,g),y[i.url]=2})))}function h(a,b){function c(a,c){if(a){if(e(a))c||(j=function(){var a=[].slice.call(arguments);k.apply(this,a),l()}),g(a,j,b,0,h);else if(Object(a)===a)for(n in m=function(){var b=0,c;for(c in a)a.hasOwnProperty(c)&&b++;return b}(),a)a.hasOwnProperty(n)&&(!c&&!--m&&(d(j)?j=function(){var a=[].slice.call(arguments);k.apply(this,a),l()}:j[n]=function(a){return function(){var b=[].slice.call(arguments);a&&a.apply(this,b),l()}}(k[n])),g(a[n],j,b,n,h))}else!c&&l()}var h=!!a.test,i=a.load||a.both,j=a.callback||f,k=j,l=a.complete||f,m,n;c(h?a.yep:a.nope,!!i),i&&c(i)}var i,j,l=this.yepnope.loader;if(e(a))g(a,0,l,0);else if(w(a))for(i=0;i<a.length;i++)j=a[i],e(j)?g(j,0,l,0):w(j)?B(j):Object(j)===j&&h(j,l);else Object(a)===a&&h(a,l)},B.addPrefix=function(a,b){z[a]=b},B.addFilter=function(a){x.push(a)},B.errorTimeout=1e4,null==b.readyState&&b.addEventListener&&(b.readyState=\"loading\",b.addEventListener(\"DOMContentLoaded\",A=function(){b.removeEventListener(\"DOMContentLoaded\",A,0),b.readyState=\"complete\"},0)),a.yepnope=k(),a.yepnope.executeStack=h,a.yepnope.injectJs=function(a,c,d,e,i,j){var k=b.createElement(\"script\"),l,o,e=e||B.errorTimeout;k.src=a;for(o in d)k.setAttribute(o,d[o]);c=j?h:c||f,k.onreadystatechange=k.onload=function(){!l&&g(k.readyState)&&(l=1,c(),k.onload=k.onreadystatechange=null)},m(function(){l||(l=1,c(1))},e),i?k.onload():n.parentNode.insertBefore(k,n)},a.yepnope.injectCss=function(a,c,d,e,g,i){var e=b.createElement(\"link\"),j,c=i?h:c||f;e.href=a,e.rel=\"stylesheet\",e.type=\"text/css\";for(j in d)e.setAttribute(j,d[j]);g||(n.parentNode.insertBefore(e,n),m(c,0))}})(this,document);\n",
              "\n",
              "//--><!]]>\n",
              "</script>\n",
              "<script type=\"text/javascript\">\n",
              "<!--//--><![CDATA[//><!--\n",
              "yepnope({\n",
              "  test: Modernizr.matchmedia,\n",
              "  nope: '/sites/all/libraries/media-match/media.match.min.js'\n",
              "});\n",
              "//--><!]]>\n",
              "</script>\n",
              "<script type=\"text/javascript\">\n",
              "<!--//--><![CDATA[//><!--\n",
              "var _prum=[['id', '612e3d94173350001100005d'], ['mark', 'firstbyte',\n",
              "      (new Date()).getTime()]]; (function() {\n",
              "      var s=document.getElementsByTagName('script')[0],\n",
              "      p=document.createElement('script');\n",
              "      p.async='async'; p.src='//rum-static.pingdom.net/prum.min.js';s.parentNode.insertBefore(p,s);})();\n",
              "//--><!]]>\n",
              "</script>\n",
              "<script type=\"text/javascript\">\n",
              "<!--//--><![CDATA[//><!--\n",
              "if(typeof window.MathJax === \"undefined\") window.MathJax = { menuSettings: { zoom: \"Click\" } };\n",
              "//--><!]]>\n",
              "</script>\n",
              "<script type=\"text/javascript\">\n",
              "<!--//--><![CDATA[//><!--\n",
              "(function(i,s,o,g,r,a,m){i[\"GoogleAnalyticsObject\"]=r;i[r]=i[r]||function(){(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)})(window,document,\"script\",\"//www.google-analytics.com/analytics.js\",\"ga\");ga(\"create\", \"UA-45638026-1\", {\"cookieDomain\":\"auto\"});ga(\"set\", \"page\", location.pathname + location.search + location.hash);ga(\"send\", \"pageview\");ga('create', 'UA-189672-38', 'auto', {'name': 'hwTracker'});\n",
              "ga('set', 'anonymizeIp', true);\n",
              "ga('hwTracker.send', 'pageview');\n",
              "//--><!]]>\n",
              "</script>\n",
              "<script type=\"text/javascript\">\n",
              "<!--//--><![CDATA[//><!--\n",
              "jQuery.extend(Drupal.settings,{\"basePath\":\"\\/\",\"pathPrefix\":\"\",\"ajaxPageState\":{\"theme\":\"jcore_1\",\"theme_token\":\"aTuGQhKIBxC958cyCtwunT2OOmpQqxP4UNn4e5Rh_Ak\"},\"colorbox\":{\"opacity\":\"0.85\",\"current\":\"{current} of {total}\",\"previous\":\"\\u00ab Prev\",\"next\":\"Next \\u00bb\",\"close\":\"Close\",\"maxWidth\":\"98%\",\"maxHeight\":\"98%\",\"fixed\":true,\"mobiledetect\":true,\"mobiledevicewidth\":\"480px\"},\"highwire\":{\"nid\":\"2125191\",\"apath\":\"\\/biorxiv\\/early\\/2021\\/08\\/26\\/2021.05.25.445601.atom\",\"pisa\":\"biorxiv;2021.05.25.445601v3\",\"processed\":[\"highwire_math\"],\"markup\":[{\"requested\":\"full-text\",\"variant\":\"full-text\",\"view\":\"full\",\"pisa\":\"biorxiv;2021.05.25.445601v3\"}],\"modal_window_width\":\"560\",\"share_modal_width\":\"560\",\"share_modal_title\":\"Share this Article\"},\"jcarousel\":{\"ajaxPath\":\"\\/jcarousel\\/ajax\\/views\"},\"instances\":\"{\\u0022highwire_abstract_tooltip\\u0022:{\\u0022content\\u0022:{\\u0022text\\u0022:\\u0022\\u0022},\\u0022style\\u0022:{\\u0022tip\\u0022:{\\u0022width\\u0022:20,\\u0022height\\u0022:20,\\u0022border\\u0022:1,\\u0022offset\\u0022:0,\\u0022corner\\u0022:true},\\u0022classes\\u0022:\\u0022qtip-custom hw-tooltip hw-abstract-tooltip qtip-shadow qtip-rounded\\u0022,\\u0022classes_custom\\u0022:\\u0022hw-tooltip hw-abstract-tooltip\\u0022},\\u0022position\\u0022:{\\u0022at\\u0022:\\u0022right center\\u0022,\\u0022my\\u0022:\\u0022left center\\u0022,\\u0022viewport\\u0022:true,\\u0022adjust\\u0022:{\\u0022method\\u0022:\\u0022shift\\u0022}},\\u0022show\\u0022:{\\u0022event\\u0022:\\u0022mouseenter click \\u0022,\\u0022solo\\u0022:true},\\u0022hide\\u0022:{\\u0022event\\u0022:\\u0022mouseleave \\u0022,\\u0022fixed\\u0022:1,\\u0022delay\\u0022:\\u0022100\\u0022}},\\u0022highwire_author_tooltip\\u0022:{\\u0022content\\u0022:{\\u0022text\\u0022:\\u0022\\u0022},\\u0022style\\u0022:{\\u0022tip\\u0022:{\\u0022width\\u0022:15,\\u0022height\\u0022:15,\\u0022border\\u0022:1,\\u0022offset\\u0022:0,\\u0022corner\\u0022:true},\\u0022classes\\u0022:\\u0022qtip-custom hw-tooltip hw-author-tooltip qtip-shadow qtip-rounded\\u0022,\\u0022classes_custom\\u0022:\\u0022hw-tooltip hw-author-tooltip\\u0022},\\u0022position\\u0022:{\\u0022at\\u0022:\\u0022top center\\u0022,\\u0022my\\u0022:\\u0022bottom center\\u0022,\\u0022viewport\\u0022:true,\\u0022adjust\\u0022:{\\u0022method\\u0022:\\u0022\\u0022}},\\u0022show\\u0022:{\\u0022event\\u0022:\\u0022mouseenter \\u0022,\\u0022solo\\u0022:true},\\u0022hide\\u0022:{\\u0022event\\u0022:\\u0022mouseleave \\u0022,\\u0022fixed\\u0022:1,\\u0022delay\\u0022:\\u0022100\\u0022}},\\u0022highwire_reflinks_tooltip\\u0022:{\\u0022content\\u0022:{\\u0022text\\u0022:\\u0022\\u0022},\\u0022style\\u0022:{\\u0022tip\\u0022:{\\u0022width\\u0022:15,\\u0022height\\u0022:15,\\u0022border\\u0022:1,\\u0022mimic\\u0022:\\u0022top center\\u0022,\\u0022offset\\u0022:0,\\u0022corner\\u0022:true},\\u0022classes\\u0022:\\u0022qtip-custom hw-tooltip hw-ref-link-tooltip qtip-shadow qtip-rounded\\u0022,\\u0022classes_custom\\u0022:\\u0022hw-tooltip hw-ref-link-tooltip\\u0022},\\u0022position\\u0022:{\\u0022at\\u0022:\\u0022bottom left\\u0022,\\u0022my\\u0022:\\u0022top left\\u0022,\\u0022viewport\\u0022:true,\\u0022adjust\\u0022:{\\u0022method\\u0022:\\u0022flip\\u0022}},\\u0022show\\u0022:{\\u0022event\\u0022:\\u0022mouseenter \\u0022,\\u0022solo\\u0022:true},\\u0022hide\\u0022:{\\u0022event\\u0022:\\u0022mouseleave \\u0022,\\u0022fixed\\u0022:1,\\u0022delay\\u0022:\\u0022100\\u0022}}}\",\"qtipDebug\":\"{\\u0022leaveElement\\u0022:0}\",\"panel_ajax_tab\":{\"path\":\"sites\\/all\\/modules\\/contrib\\/panels_ajax_tab\"},\"disqus\":{\"domain\":\"biorxivstage\",\"url\":\"https:\\/\\/www.biorxiv.org\\/content\\/10.1101\\/2021.05.25.445601v3\",\"title\":\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning\",\"identifier\":\"node\\/2125191\"},\"panels_ajax_pane\":{\"new-28877068-b9cb-4641-830b-b6b4638c98bb\":\"{\\u0022encrypted\\u0022:\\u0022{\\\\\\u0022encrypted\\\\\\u0022:\\\\\\u0022YDAGAQlyplmL1K84ipiv+SXHzUxOedjL3cmEwB3AME+J1bbYYBnU4YK+YvYm7FDYUdHVM0qny0j\\\\\\\\\\\\\\/O1vqi32LN6lWwmUEd1oM5D1iUwAI7pxoQdrwxZ0q4Ml0yGdKTGNtjbta92Ph5BwIWHGWnO40U9\\\\\\\\\\\\\\/ho449ON\\\\\\\\\\\\\\/1TjCtdVNnO\\\\\\\\\\\\\\/M0hsfBJzYVkxMJcuYuqcB2Z\\\\\\\\\\\\\\/GcsQlLy5rTTvLRZ7ulV5oBDMNyp+F29CIDdEdd7jA0zop5veEbcPLtQbrX6RUgwzR615KBDS6IWQoEOtQGjupavTozG2sJTqKTVtkfAj7E0FF2gBvjwE7Q0tUQIAIb6DRVJBxgcq85XtGhytKTgeaS5UJ30QiHaNagWrU8OVtovTaE\\\\\\\\\\\\\\/8FeceaO3HyL0x3+HbGDCgOviRAnyTblYSSlP4aWTbmhCvvSzM6lOflkwgeVHKWJcZ6D9K72SI0LVTN4K\\\\\\\\\\\\\\/nIXZs8bayjGS4AEbUrsUQHdSz6xW5tFt0O08hosNgV6I\\\\\\\\\\\\\\/wsQMcGpdsD68zHKQYBPDLC+wCElX5OMmcS0EycOzPG5084HCKuI5aaD+DmhrQbl3XfnqGwyiZvuGJLEJLMDVg5oGYH5lJoiK2kw2Zywc3frQizJU9NxOgpmYU2FYg\\\\\\\\\\\\\\/phlhk1DCLDFWmaJa9bjC4dyVzh8EnCrZedXq6iJv4whr4vgZdrzyALSIZisq\\\\\\\\\\\\\\/kNLVz7QxPyAGVyjk95tYxFliaBSbJGIxOQErmhsNjnuPknSFJhT\\\\\\\\\\\\\\/uPOxsAwNRaaZlNsyUm1rJLd38E3NyxcuuJ2eQiUgjv5Wy441syY9OZ9+MAA3wu9aK7+LfK\\\\\\\\\\\\\\/lx3A99cjmv4FlylCVr\\\\\\\\\\\\\\/u0HhqPtKNkw1EUmFyHyMxKBcpEtYmCW\\\\\\\\\\\\\\/6G\\\\\\\\\\\\\\/hDYzpYVbFgxsFYCDj5q244D14t+sCsiLgcAnRce3r1yxb8FfeDBQUA3poqxzMOVGnOjCem9+vYU3AV4ZtAUtW+\\\\\\\\\\\\\\/SccmOM6FLvIU1iRmx0mHMu4R9Gh1QuTDQmbsvtIuIlHe6e0LMWBIh0KPctQ6DuIiRT3rYSpmBQYSZ2JfISXBjkD0AH+epIiDC9IyODqtvuuuok75nk99DijUnS5Q85QGLfeCjhr\\\\\\\\\\\\\\/3WpnRi1sXb7trlwk6Dd3Ndc5dCTkGMTPwXbIbUBkHpt005IBBg+igPOHUvgILxsJm4kZT1pFDxg3L2yZ5gWiMBIHE7pcNnvu0+VE+vClDdB1Fzqk\\\\\\\\\\\\\\/bp++1Ii7yzTJ826jSj9ZAMXQzv90VUrUe3pO5B5IDpG4JwLptLi6eBhh5jXFN2fywYe3AazAk7olWMmd0aoJJbKski+Q9K2R5p0LvVcz5A8Nho3RL80kKUaKGJEsYGX4wxjPhW4m03YLcYQSKQu0dPCzRJ58254pUi+83wJguxinFcK0KTWJgjAkAUKtVSXA=\\\\\\u0022,\\\\\\u0022iv\\\\\\u0022:\\\\\\u0022+3OHMfVs2dwa72pzkvuKLA==\\\\\\u0022,\\\\\\u0022salt\\\\\\u0022:\\\\\\u0022f9d8845712f72c9fee5db7d9ff111eab\\\\\\u0022}\\u0022,\\u0022hmac\\u0022:\\u0022c07ae0a16c3ebb9397b9f2512d47f7b5d004b86cab2a3fb172d5d85ef6a94f72\\u0022}\"},\"urlIsAjaxTrusted\":{\"\\/content\\/10.1101\\/2021.05.25.445601v3.full\":true},\"ws_fl\":{\"width\":100,\"height\":21},\"ws_gpo\":{\"size\":\"\",\"annotation\":\"\",\"lang\":\"\",\"callback\":\"\",\"width\":300},\"color\":{\"logo\":\"https:\\/\\/www.biorxiv.org\\/sites\\/default\\/files\\/bioRxiv_article.jpg\"},\"highwire_list_expand\":{\"is_collapsed\":\"1\"},\"highwireResponsive\":{\"enquire_enabled\":1,\"breakpoints_configured\":1,\"breakpoints\":{\"zero\":\"all and (min-width: 0px)\",\"xsmall\":\"all and (min-width: 380px)\",\"narrow\":\"all and (min-width: 768px) and (min-device-width: 768px), (max-device-width: 800px) and (min-width: 768px) and (orientation:landscape)\",\"normal\":\"all and (min-width: 980px) and (min-device-width: 980px), all and (max-device-width: 1024px) and (min-width: 1024px) and (orientation:landscape)\",\"wide\":\"all and (min-width: 1220px)\"}},\"eu_cookie_compliance\":{\"popup_enabled\":1,\"popup_agreed_enabled\":0,\"popup_hide_agreed\":0,\"popup_clicking_confirmation\":1,\"popup_scrolling_confirmation\":false,\"popup_html_info\":\"\\u003Cdiv\\u003E\\n  \\u003Cdiv class =\\u0022popup-content info\\u0022\\u003E\\n    \\u003Cdiv id=\\u0022popup-text\\u0022\\u003E\\n      \\u003Cp\\u003EWe use cookies on this site to enhance your user experience. By clicking any link on this page you are giving your consent for us to set cookies.\\u003C\\/p\\u003E\\n    \\u003C\\/div\\u003E\\n    \\u003Cdiv id=\\u0022popup-buttons\\u0022\\u003E\\n      \\u003Cbutton type=\\u0022button\\u0022 class=\\u0022agree-button eu-cookie-compliance-default-button\\u0022\\u003EContinue\\u003C\\/button\\u003E\\n                    \\u003Cbutton type=\\u0022button\\u0022 class=\\u0022find-more-button eu-cookie-compliance-more-button\\u0022\\u003EFind out more\\u003C\\/button\\u003E\\n          \\u003C\\/div\\u003E\\n  \\u003C\\/div\\u003E\\n\\u003C\\/div\\u003E\",\"use_mobile_message\":false,\"mobile_popup_html_info\":\"\\u003Cdiv\\u003E\\n  \\u003Cdiv class =\\u0022popup-content info\\u0022\\u003E\\n    \\u003Cdiv id=\\u0022popup-text\\u0022\\u003E\\n          \\u003C\\/div\\u003E\\n    \\u003Cdiv id=\\u0022popup-buttons\\u0022\\u003E\\n      \\u003Cbutton type=\\u0022button\\u0022 class=\\u0022agree-button eu-cookie-compliance-default-button\\u0022\\u003EContinue\\u003C\\/button\\u003E\\n                    \\u003Cbutton type=\\u0022button\\u0022 class=\\u0022find-more-button eu-cookie-compliance-more-button\\u0022\\u003EFind out more\\u003C\\/button\\u003E\\n          \\u003C\\/div\\u003E\\n  \\u003C\\/div\\u003E\\n\\u003C\\/div\\u003E\\n\",\"mobile_breakpoint\":\"768\",\"popup_html_agreed\":\"\\u003Cdiv\\u003E\\n  \\u003Cdiv class=\\u0022popup-content agreed\\u0022\\u003E\\n    \\u003Cdiv id=\\u0022popup-text\\u0022\\u003E\\n      \\u003Ch2\\u003EThank you for accepting cookies\\u003C\\/h2\\u003E\\u003Cp\\u003EYou can now hide this message or find out more about cookies.\\u003C\\/p\\u003E    \\u003C\\/div\\u003E\\n    \\u003Cdiv id=\\u0022popup-buttons\\u0022\\u003E\\n      \\u003Cbutton type=\\u0022button\\u0022 class=\\u0022hide-popup-button eu-cookie-compliance-hide-button\\u0022\\u003EHide\\u003C\\/button\\u003E\\n              \\u003Cbutton type=\\u0022button\\u0022 class=\\u0022find-more-button eu-cookie-compliance-more-button-thank-you\\u0022 \\u003EMore info\\u003C\\/button\\u003E\\n          \\u003C\\/div\\u003E\\n  \\u003C\\/div\\u003E\\n\\u003C\\/div\\u003E\",\"popup_use_bare_css\":false,\"popup_height\":\"auto\",\"popup_width\":\"100%\",\"popup_delay\":1000,\"popup_link\":\"\\/help\\/cookie-policy\",\"popup_link_new_window\":1,\"popup_position\":null,\"popup_language\":\"en\",\"store_consent\":false,\"better_support_for_screen_readers\":0,\"reload_page\":0,\"domain\":\"\",\"popup_eu_only_js\":0,\"cookie_lifetime\":\"365\",\"cookie_session\":false,\"disagree_do_not_show_popup\":0,\"method\":\"default\",\"whitelisted_cookies\":\"\",\"withdraw_markup\":\"\\u003Cbutton type=\\u0022button\\u0022 class=\\u0022eu-cookie-withdraw-tab\\u0022\\u003EPrivacy settings\\u003C\\/button\\u003E\\n\\u003Cdiv class=\\u0022eu-cookie-withdraw-banner\\u0022\\u003E\\n  \\u003Cdiv class=\\u0022popup-content info\\u0022\\u003E\\n    \\u003Cdiv id=\\u0022popup-text\\u0022\\u003E\\n      \\u003Cp\\u003E\\u0026lt;h2\\u0026gt;We use cookies on this site to enhance your user experience\\u0026lt;\\/h2\\u0026gt;\\u0026lt;p\\u0026gt;You have given your consent for us to set cookies.\\u0026lt;\\/p\\u0026gt;\\u003C\\/p\\u003E\\n    \\u003C\\/div\\u003E\\n    \\u003Cdiv id=\\u0022popup-buttons\\u0022\\u003E\\n      \\u003Cbutton type=\\u0022button\\u0022 class=\\u0022eu-cookie-withdraw-button\\u0022\\u003EWithdraw consent\\u003C\\/button\\u003E\\n    \\u003C\\/div\\u003E\\n  \\u003C\\/div\\u003E\\n\\u003C\\/div\\u003E\\n\",\"withdraw_enabled\":false},\"googleanalytics\":{\"trackOutbound\":1,\"trackMailto\":1,\"trackDownload\":1,\"trackDownloadExtensions\":\"7z|aac|arc|arj|asf|asx|avi|bin|csv|doc(x|m)?|dot(x|m)?|exe|flv|gif|gz|gzip|hqx|jar|jpe?g|js|mp(2|3|4|e?g)|mov(ie)?|msi|msp|pdf|phps|png|ppt(x|m)?|pot(x|m)?|pps(x|m)?|ppam|sld(x|m)?|thmx|qtm?|ra(m|r)?|sea|sit|tar|tgz|torrent|txt|wav|wma|wmv|wpd|xls(x|m|b)?|xlt(x|m)|xlam|xml|z|zip\",\"trackColorbox\":1,\"trackUrlFragments\":1},\"jnl_biorxiv_styles\":{\"defaultJCode\":\"biorxiv\"},\"omega\":{\"layouts\":{\"primary\":\"normal\",\"order\":[\"narrow\",\"normal\",\"wide\"],\"queries\":{\"narrow\":\"all and (min-width: 768px) and (min-device-width: 768px), (max-device-width: 800px) and (min-width: 768px) and (orientation:landscape)\",\"normal\":\"all and (min-width: 980px) and (min-device-width: 980px), all and (max-device-width: 1024px) and (min-width: 1024px) and (orientation:landscape)\",\"wide\":\"all and (min-width: 1220px)\"}}}});\n",
              "//--><!]]>\n",
              "</script>\n",
              "<!--[if lt IE 9]><script src=\"http://html5shiv.googlecode.com/svn/trunk/html5.js\"></script><![endif]-->\n",
              "</head>\n",
              "<body class=\"html not-front not-logged-in page-node page-node- page-node-2125191 node-type-highwire-article context-content hw-default-jcode-biorxiv hw-article-type-article hw-article-category-new-results\">\n",
              "<!-- Google Tag Manager -->\n",
              "<noscript><iframe height=\"0\" src=\"//www.googletagmanager.com/ns.html?id=GTM-M677548\" style=\"display:none;visibility:hidden\" width=\"0\"></iframe></noscript>\n",
              "<script type=\"text/javascript\">(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0];var j=d.createElement(s);var dl=l!='dataLayer'?'&l='+l:'';j.src='//www.googletagmanager.com/gtm.js?id='+i+dl;j.type='text/javascript';j.async=true;f.parentNode.insertBefore(j,f);})(window,document,'script','dataLayer','GTM-M677548');</script>\n",
              "<!-- End Google Tag Manager -->\n",
              "<div id=\"skip-link\">\n",
              "<a class=\"element-invisible element-focusable\" href=\"#main-content\">Skip to main content</a>\n",
              "</div>\n",
              "<div class=\"page clearfix page-box-shadows footer-borders panels-page panels-layout-jcore_2col\" id=\"page\">\n",
              "<header class=\"section section-header\" id=\"section-header\">\n",
              "<div class=\"zone zone-branding clearfix print-display-block container-30\" id=\"zone-branding\">\n",
              "<div class=\"grid-15 prefix-1 region region-branding print-display-block\" id=\"region-branding\">\n",
              "<div class=\"region-inner region-branding-inner\">\n",
              "<div class=\"branding-data clearfix\">\n",
              "<div class=\"logo-img\">\n",
              "<a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/\" rel=\"home\"><img alt=\"bioRxiv\" src=\"https://www.biorxiv.org/sites/default/files/bioRxiv_article.jpg\"/></a> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div><div class=\"grid-11 suffix-1 region region-branding-second print-hidden\" id=\"region-branding-second\">\n",
              "<div class=\"region-inner region-branding-second-inner\">\n",
              "<div class=\"block block-system block-menu block-main-menu block-system-main-menu odd block-without-title\" id=\"block-system-main-menu\">\n",
              "<div class=\"block-inner clearfix\">\n",
              "<div class=\"content clearfix\">\n",
              "<nav class=\"menubar-nav\"><ul class=\"menu\" role=\"menu\"><li class=\"first leaf\" role=\"menuitem\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/\" title=\"\">Home</a></li>\n",
              "<li class=\"leaf\" role=\"menuitem\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/about-biorxiv\">About</a></li>\n",
              "<li class=\"leaf\" role=\"menuitem\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/submit-a-manuscript\">Submit</a></li>\n",
              "<li class=\"last leaf\" role=\"menuitem\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/content/alertsrss\" title=\"\">ALERTS / RSS</a></li>\n",
              "</ul></nav> </div>\n",
              "</div>\n",
              "</div><div class=\"block block-panels-mini block-biorxiv-search-box block-panels-mini-biorxiv-search-box even block-without-title\" id=\"block-panels-mini-biorxiv-search-box\">\n",
              "<div class=\"block-inner clearfix\">\n",
              "<div class=\"content clearfix\">\n",
              "<div class=\"panel-display panel-1col clearfix\" id=\"mini-panel-biorxiv_search_box\">\n",
              "<div class=\"panel-panel panel-col\">\n",
              "<div><div class=\"panel-pane pane-highwire-seach-quicksearch\">\n",
              "<div class=\"pane-content\">\n",
              "<form accept-charset=\"UTF-8\" action=\"/content/10.1101/2021.05.25.445601v3.full\" class=\"highwire-quicksearch button-style-mini button-style-mini\" id=\"highwire-search-quicksearch-form-0\" method=\"post\"><div><div class=\"form-item form-item-label-invisible form-type-textfield form-item-keywords\">\n",
              "<label class=\"element-invisible\" for=\"search_rightsidebar_keywords_1492764509\">Search for this keyword </label>\n",
              "<input class=\"form-text\" id=\"search_rightsidebar_keywords_1492764509\" maxlength=\"128\" name=\"keywords\" placeholder=\"Search\" size=\"60\" type=\"text\" value=\"\"/>\n",
              "</div>\n",
              "<div class=\"button-wrapper button-mini\"><span class=\"icon-search\"></span><input class=\"form-submit\" data-font-icon=\"icon-search\" data-icon-only=\"1\" data-icon-position=\"after\" id=\"search_rightsidebar_submit_30200201\" name=\"op\" type=\"submit\" value=\"Search\"/></div><input name=\"form_build_id\" type=\"hidden\" value=\"form-ynmanQ3JYnWCGvocrHvLLY0oOoZMiaiz_V5bR_hvUoU\"/>\n",
              "<input name=\"form_id\" type=\"hidden\" value=\"highwire_search_quicksearch_form_0\"/>\n",
              "</div></form> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-custom pane-2 advanced-search-link\">\n",
              "<div class=\"pane-content\">\n",
              "<a href=\"/search\">Advanced Search</a> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div> </div>\n",
              "</div> </div>\n",
              "<div class=\"zone zone-header clearfix container-30\" id=\"zone-header\">\n",
              "</div>\n",
              "</header>\n",
              "<section class=\"section section-content\" id=\"section-content\">\n",
              "<div class=\"zone zone-content clearfix container-30\" id=\"zone-content\">\n",
              "<div class=\"grid-28 suffix-1 prefix-1 region region-content\" id=\"region-content\">\n",
              "<div class=\"region-inner region-content-inner\">\n",
              "<a id=\"main-content\"></a>\n",
              "<div class=\"block block-system block-main block-system-main odd block-without-title\" id=\"block-system-main\">\n",
              "<div class=\"block-inner clearfix\">\n",
              "<div class=\"content clearfix\">\n",
              "<div class=\"panel-display panels-960-layout jcore-2col-layout\">\n",
              "<div class=\"panel-row-wrapper clearfix\">\n",
              "<div class=\"main-content-wrapper grid-17 suffix-1 alpha\">\n",
              "<div class=\"panel-panel panel-region-content\">\n",
              "<div class=\"inside\"><div class=\"panel-pane pane-highwire-article-citation\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"highwire-article-citation highwire-citation-type-highwire-article node2125191\" data-apath=\"/biorxiv/early/2021/08/26/2021.05.25.445601.atom\" data-hw-author-tooltip-instance=\"highwire_author_tooltip\" data-node-nid=\"2125191\" data-pisa=\"biorxiv;2021.05.25.445601v3\" data-pisa-master=\"biorxiv;2021.05.25.445601\" id=\"node2125191--21877820458\"><div class=\"highwire-cite highwire-cite-highwire-article highwire-citation-biorxiv-article-top clearfix has-author-tooltip\">\n",
              "<span class=\"biorxiv-article-type\">\n",
              "        New Results    </span>\n",
              "<h1 class=\"highwire-cite-title\" id=\"page-title\">Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning</h1>\n",
              "<div class=\"highwire-cite-authors\"><span class=\"highwire-citation-authors\"><span class=\"highwire-citation-author first hw-author-orcid-logo-wrapper\" data-delta=\"0\"><a class=\"hw-author-orcid-logo link-icon-only link-icon\" href=\"http://orcid.org/0000-0001-9936-7790\" target=\"_blank\"><span class=\"hw-icon-orcid hw-icon-color-orcid\"></span> <span class=\"title element-invisible\">View ORCID Profile</span></a><span class=\"nlm-given-names\">Sargun</span> <span class=\"nlm-surname\">Nagpal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"1\"><span class=\"nlm-given-names\">Ridam</span> <span class=\"nlm-surname\">Pal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"2\"><span class=\"nlm-surname\">Ashima</span></span>, <span class=\"highwire-citation-author\" data-delta=\"3\"><span class=\"nlm-given-names\">Ananya</span> <span class=\"nlm-surname\">Tyagi</span></span>, <span class=\"highwire-citation-author\" data-delta=\"4\"><span class=\"nlm-given-names\">Sadhana</span> <span class=\"nlm-surname\">Tripathi</span></span>, <span class=\"highwire-citation-author hw-author-orcid-logo-wrapper\" data-delta=\"5\"><a class=\"hw-author-orcid-logo link-icon-only link-icon\" href=\"http://orcid.org/0000-0002-6389-2179\" target=\"_blank\"><span class=\"hw-icon-orcid hw-icon-color-orcid\"></span> <span class=\"title element-invisible\">View ORCID Profile</span></a><span class=\"nlm-given-names\">Aditya</span> <span class=\"nlm-surname\">Nagori</span></span>, <span class=\"highwire-citation-author\" data-delta=\"6\"><span class=\"nlm-given-names\">Saad</span> <span class=\"nlm-surname\">Ahmad</span></span>, <span class=\"highwire-citation-author hw-author-orcid-logo-wrapper\" data-delta=\"7\"><a class=\"hw-author-orcid-logo link-icon-only link-icon\" href=\"http://orcid.org/0000-0001-9405-7822\" target=\"_blank\"><span class=\"hw-icon-orcid hw-icon-color-orcid\"></span> <span class=\"title element-invisible\">View ORCID Profile</span></a><span class=\"nlm-given-names\">Hara Prasad</span> <span class=\"nlm-surname\">Mishra</span></span>, <span class=\"highwire-citation-author hw-author-orcid-logo-wrapper\" data-delta=\"8\"><a class=\"hw-author-orcid-logo link-icon-only link-icon\" href=\"http://orcid.org/0000-0001-8667-6199\" target=\"_blank\"><span class=\"hw-icon-orcid hw-icon-color-orcid\"></span> <span class=\"title element-invisible\">View ORCID Profile</span></a><span class=\"nlm-given-names\">Rintu</span> <span class=\"nlm-surname\">Kutum</span></span>, <span class=\"highwire-citation-author hw-author-orcid-logo-wrapper\" data-delta=\"9\"><a class=\"hw-author-orcid-logo link-icon-only link-icon\" href=\"http://orcid.org/0000-0002-4776-7941\" target=\"_blank\"><span class=\"hw-icon-orcid hw-icon-color-orcid\"></span> <span class=\"title element-invisible\">View ORCID Profile</span></a><span class=\"nlm-given-names\">Tavpritesh</span> <span class=\"nlm-surname\">Sethi</span></span></span></div>\n",
              "<div class=\"highwire-cite-metadata\"><span class=\"highwire-cite-metadata-doi highwire-cite-metadata\"><span class=\"label\">doi:</span> https://doi.org/10.1101/2021.05.25.445601 </span></div>\n",
              "</div>\n",
              "<div id=\"hw-article-author-popups-node2125191--21877820458\" style=\"display: none;\"><div class=\"author-tooltip-0\"><div class=\"author-tooltip-name\">Sargun Nagpal </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\"author-affiliation\"><span class=\"nlm-sup\">1</span><span class=\"nlm-institution\">Indraprastha Institute of Information Technology Delhi</span>, <span class=\"nlm-country\">India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Sargun%2BNagpal%2B\" target=\"_blank\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/external-ref?access_num=Nagpal%20S&amp;link_type=AUTHORSEARCH\" target=\"_blank\">Find this author on PubMed</a></li><li class=\"author-site-search-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/search/author1%3ASargun%2BNagpal%2B\" rel=\"nofollow\">Search for this author on this site</a></li><li class=\"author-orcid-link last\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://orcid.org/0000-0001-9936-7790\" target=\"_blank\">ORCID record for Sargun Nagpal</a></li></ul></div><div class=\"author-tooltip-1\"><div class=\"author-tooltip-name\">Ridam Pal </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\"author-affiliation\"><span class=\"nlm-sup\">1</span><span class=\"nlm-institution\">Indraprastha Institute of Information Technology Delhi</span>, <span class=\"nlm-country\">India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Ridam%2BPal%2B\" target=\"_blank\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/external-ref?access_num=Pal%20R&amp;link_type=AUTHORSEARCH\" target=\"_blank\">Find this author on PubMed</a></li><li class=\"author-site-search-link last\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/search/author1%3ARidam%2BPal%2B\" rel=\"nofollow\">Search for this author on this site</a></li></ul></div><div class=\"author-tooltip-2\"><div class=\"author-tooltip-name\"> Ashima </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\"author-affiliation\"><span class=\"nlm-sup\">1</span><span class=\"nlm-institution\">Indraprastha Institute of Information Technology Delhi</span>, <span class=\"nlm-country\">India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=%2BAshima%2B\" target=\"_blank\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/external-ref?access_num=Ashima&amp;link_type=AUTHORSEARCH\" target=\"_blank\">Find this author on PubMed</a></li><li class=\"author-site-search-link last\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/search/author1%3A%2BAshima%2B\" rel=\"nofollow\">Search for this author on this site</a></li></ul></div><div class=\"author-tooltip-3\"><div class=\"author-tooltip-name\">Ananya Tyagi </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\"author-affiliation\"><span class=\"nlm-sup\">1</span><span class=\"nlm-institution\">Indraprastha Institute of Information Technology Delhi</span>, <span class=\"nlm-country\">India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Ananya%2BTyagi%2B\" target=\"_blank\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/external-ref?access_num=Tyagi%20A&amp;link_type=AUTHORSEARCH\" target=\"_blank\">Find this author on PubMed</a></li><li class=\"author-site-search-link last\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/search/author1%3AAnanya%2BTyagi%2B\" rel=\"nofollow\">Search for this author on this site</a></li></ul></div><div class=\"author-tooltip-4\"><div class=\"author-tooltip-name\">Sadhana Tripathi </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\"author-affiliation\"><span class=\"nlm-sup\">1</span><span class=\"nlm-institution\">Indraprastha Institute of Information Technology Delhi</span>, <span class=\"nlm-country\">India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Sadhana%2BTripathi%2B\" target=\"_blank\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/external-ref?access_num=Tripathi%20S&amp;link_type=AUTHORSEARCH\" target=\"_blank\">Find this author on PubMed</a></li><li class=\"author-site-search-link last\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/search/author1%3ASadhana%2BTripathi%2B\" rel=\"nofollow\">Search for this author on this site</a></li></ul></div><div class=\"author-tooltip-5\"><div class=\"author-tooltip-name\">Aditya Nagori </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\"author-affiliation\"><span class=\"nlm-sup\">1</span><span class=\"nlm-institution\">Indraprastha Institute of Information Technology Delhi</span>, <span class=\"nlm-country\">India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Aditya%2BNagori%2B\" target=\"_blank\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/external-ref?access_num=Nagori%20A&amp;link_type=AUTHORSEARCH\" target=\"_blank\">Find this author on PubMed</a></li><li class=\"author-site-search-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/search/author1%3AAditya%2BNagori%2B\" rel=\"nofollow\">Search for this author on this site</a></li><li class=\"author-orcid-link last\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://orcid.org/0000-0002-6389-2179\" target=\"_blank\">ORCID record for Aditya Nagori</a></li></ul></div><div class=\"author-tooltip-6\"><div class=\"author-tooltip-name\">Saad Ahmad </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\"author-affiliation\"><span class=\"nlm-sup\">1</span><span class=\"nlm-institution\">Indraprastha Institute of Information Technology Delhi</span>, <span class=\"nlm-country\">India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Saad%2BAhmad%2B\" target=\"_blank\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/external-ref?access_num=Ahmad%20S&amp;link_type=AUTHORSEARCH\" target=\"_blank\">Find this author on PubMed</a></li><li class=\"author-site-search-link last\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/search/author1%3ASaad%2BAhmad%2B\" rel=\"nofollow\">Search for this author on this site</a></li></ul></div><div class=\"author-tooltip-7\"><div class=\"author-tooltip-name\">Hara Prasad Mishra </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\"author-affiliation\"><span class=\"nlm-sup\">1</span><span class=\"nlm-institution\">Indraprastha Institute of Information Technology Delhi</span>, <span class=\"nlm-country\">India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Hara%2BPrasad%2BMishra%2B\" target=\"_blank\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/external-ref?access_num=Mishra%20HP&amp;link_type=AUTHORSEARCH\" target=\"_blank\">Find this author on PubMed</a></li><li class=\"author-site-search-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/search/author1%3AHara%2BPrasad%2BMishra%2B\" rel=\"nofollow\">Search for this author on this site</a></li><li class=\"author-orcid-link last\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://orcid.org/0000-0001-9405-7822\" target=\"_blank\">ORCID record for Hara Prasad Mishra</a></li></ul></div><div class=\"author-tooltip-8\"><div class=\"author-tooltip-name\">Rintu Kutum </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\"author-affiliation\"><span class=\"nlm-sup\">1</span><span class=\"nlm-institution\">Indraprastha Institute of Information Technology Delhi</span>, <span class=\"nlm-country\">India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Rintu%2BKutum%2B\" target=\"_blank\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/external-ref?access_num=Kutum%20R&amp;link_type=AUTHORSEARCH\" target=\"_blank\">Find this author on PubMed</a></li><li class=\"author-site-search-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/search/author1%3ARintu%2BKutum%2B\" rel=\"nofollow\">Search for this author on this site</a></li><li class=\"author-orcid-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://orcid.org/0000-0001-8667-6199\" target=\"_blank\">ORCID record for Rintu Kutum</a></li><li class=\"author-corresp-email-link last\"><span>For correspondence: \n",
              "<a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"mailto:tavpriteshsethi@iiitd.ac.in\">tavpriteshsethi@iiitd.ac.in</a>\n",
              "<a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"mailto:rintuk@iiitd.ac.in\">rintuk@iiitd.ac.in</a></span></li></ul></div><div class=\"author-tooltip-9\"><div class=\"author-tooltip-name\">Tavpritesh Sethi </div><div class=\"author-tooltip-affiliation\"><span class=\"author-tooltip-text\"><div class=\"author-affiliation\"><span class=\"nlm-sup\">1</span><span class=\"nlm-institution\">Indraprastha Institute of Information Technology Delhi</span>, <span class=\"nlm-country\">India</span></div><div class=\"author-affiliation\"><span class=\"nlm-sup\">2</span><span class=\"nlm-institution\">All India Institute of Medical Sciences</span>, New Delhi, <span class=\"nlm-country\">India</span></div></span></div><ul class=\"author-tooltip-find-more\"><li class=\"author-tooltip-gs-link first\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/google-scholar?link_type=googlescholar&amp;gs_type=author&amp;author%5B0%5D=Tavpritesh%2BSethi%2B\" target=\"_blank\">Find this author on Google Scholar</a></li><li class=\"author-tooltip-pubmed-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/lookup/external-ref?access_num=Sethi%20T&amp;link_type=AUTHORSEARCH\" target=\"_blank\">Find this author on PubMed</a></li><li class=\"author-site-search-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/search/author1%3ATavpritesh%2BSethi%2B\" rel=\"nofollow\">Search for this author on this site</a></li><li class=\"author-orcid-link\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://orcid.org/0000-0002-4776-7941\" target=\"_blank\">ORCID record for Tavpritesh Sethi</a></li><li class=\"author-corresp-email-link last\"><span>For correspondence: \n",
              "<a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"mailto:tavpriteshsethi@iiitd.ac.in\">tavpriteshsethi@iiitd.ac.in</a>\n",
              "<a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"mailto:rintuk@iiitd.ac.in\">rintuk@iiitd.ac.in</a></span></li></ul></div></div></div> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-panel-tabs pane-panels-ajax-tab-tabs\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"item-list\"><ul class=\"tabs inline panels-ajax-tab\"><li class=\"first\"><a class=\"panels-ajax-tab-tab\" data-entity-context=\"node:2125191\" data-panel-name=\"biorxiv_tab_art\" data-target-id=\"highwire_article_tabs\" data-trigger=\"\" data-url-enabled=\"1\" href=\"/content/10.1101/2021.05.25.445601v3\">Abstract</a><a class=\"js-crawler-link\" href=\"/panels_ajax_tab/biorxiv_tab_art/node:2125191/1\" rel=\"nofollow\" style=\"display:none\"></a></li><li><a class=\"panels-ajax-tab-tab\" data-entity-context=\"node:2125191\" data-panel-name=\"article_tab_full_text\" data-target-id=\"highwire_article_tabs\" data-trigger=\"full-text\" data-url-enabled=\"1\" href=\"/content/10.1101/2021.05.25.445601v3.full-text\">Full Text</a><a class=\"js-crawler-link\" href=\"/panels_ajax_tab/article_tab_full_text/node:2125191/1\" rel=\"nofollow\" style=\"display:none\"></a></li><li><a class=\"panels-ajax-tab-tab\" data-entity-context=\"node:2125191\" data-panel-name=\"biorxiv_tab_info\" data-target-id=\"highwire_article_tabs\" data-trigger=\"article-info\" data-url-enabled=\"1\" href=\"/content/10.1101/2021.05.25.445601v3.article-info\">Info/History</a><a class=\"js-crawler-link\" href=\"/panels_ajax_tab/biorxiv_tab_info/node:2125191/1\" rel=\"nofollow\" style=\"display:none\"></a></li><li><a class=\"panels-ajax-tab-tab\" data-entity-context=\"node:2125191\" data-panel-name=\"article_tab_metrics\" data-target-id=\"highwire_article_tabs\" data-trigger=\"article-metrics\" data-url-enabled=\"1\" href=\"/content/10.1101/2021.05.25.445601v3.article-metrics\">Metrics</a><a class=\"js-crawler-link\" href=\"/panels_ajax_tab/article_tab_metrics/node:2125191/1\" rel=\"nofollow\" style=\"display:none\"></a></li><li><a class=\"panels-ajax-tab-tab\" data-entity-context=\"node:2125191\" data-panel-name=\"biorxiv_tab_data\" data-target-id=\"highwire_article_tabs\" data-trigger=\"supplementary-material\" data-url-enabled=\"1\" href=\"/content/10.1101/2021.05.25.445601v3.supplementary-material\">Supplementary material</a><a class=\"js-crawler-link\" href=\"/panels_ajax_tab/biorxiv_tab_data/node:2125191/1\" rel=\"nofollow\" style=\"display:none\"></a></li><li class=\"last\"><a class=\"panels-ajax-tab-tab\" data-entity-context=\"node:2125191\" data-panel-name=\"biorxiv_tab_pdf\" data-target-id=\"highwire_article_tabs\" data-trigger=\"full.pdf+html\" data-url-enabled=\"1\" href=\"/content/10.1101/2021.05.25.445601v3.full.pdf+html\"><i class=\"icon-file-alt\"></i> Preview PDF</a><a class=\"js-crawler-link\" href=\"/panels_ajax_tab/biorxiv_tab_pdf/node:2125191/1\" rel=\"nofollow\" style=\"display:none\"></a></li></ul></div> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-panel-tabs-container\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"panels-ajax-tab-container\" data-panels-ajax-tab-preloaded=\"article_tab_full_text\" id=\"panels-ajax-tab-container-highwire_article_tabs\"><div class=\"panels-ajax-tab-loading\" style=\"display:none\"><img alt=\"Loading\" class=\"loading\" src=\"https://www.biorxiv.org/sites/all/modules/contrib/panels_ajax_tab/images/loading.gif\" title=\"Loading\"/></div><div class=\"panels-ajax-tab-wrap-article_tab_full_text\"><div class=\"panel-display panel-1col clearfix\">\n",
              "<div class=\"panel-panel panel-col\">\n",
              "<div><div class=\"panel-pane pane-highwire-markup\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"highwire-markup\"><div class=\"content-block-markup\" data-highwire-cite-ref-tooltip-instance=\"highwire_reflinks_tooltip\" xmlns=\"http://www.w3.org/1999/xhtml\" xmlns:xhtml=\"http://www.w3.org/1999/xhtml\"><div class=\"article fulltext-view \"><span class=\"highwire-journal-article-marker-start\"></span><div class=\"section abstract\" id=\"abstract-1\"><h2 class=\"\">Abstract</h2><p id=\"p-4\">The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed <em>Strainflow</em>, to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our <em>Strainflow</em> model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called <em>Dimensions of Concern</em> (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.</p><div class=\"subsection\" id=\"sec-1\"><p><strong>Highlights</strong></p><ul class=\"list-unord \" id=\"list-1\"><li id=\"list-item-1\"><p id=\"p-5\">We developed a genomic surveillance model for SARS-CoV-2 genome sequences, <em>Strainflow</em>, where sequences were treated as documents with words (codons) to learn the codon context of 0.9 million spike genes using the skip-gram algorithm.</p></li><li id=\"list-item-2\"><p id=\"p-6\">Time series analysis of the information content (Entropy) of the latent dimensions learned by <em>Strainflow</em> shows a leading relationship with the monthly COVID-19 cases for seven countries (e.g., USA, Japan, India, and others).</p></li><li id=\"list-item-3\"><p id=\"p-7\">Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads.</p></li><li id=\"list-item-4\"><p id=\"p-8\">The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants.</p></li></ul></div><div class=\"subsection\" id=\"sec-2\"><div class=\"fig pos-float type-figure odd\" id=\"F1\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a class=\"highwire-fragment fragment-images colorbox-load\" data-figure-caption='&lt;div class=\"highwire-markup\"&gt;&lt;/div&gt;' data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F1.large.jpg?width=800&amp;height=600&amp;carousel=1\" rel=\"gallery-fragment-images-1181327286\" title=\"\"><span class=\"hw-responsive-img\"><img alt=\"Figure\" class=\"highwire-fragment fragment-image lazyload\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F1.medium.gif\" height=\"440\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" width=\"434\"/><noscript><img alt=\"Figure\" class=\"highwire-fragment fragment-image\" height=\"440\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F1.medium.gif\" width=\"434\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a class=\"highwire-figure-link highwire-figure-link-download\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F1.large.jpg?download=true\" title=\"Download Figure1\">Download figure</a></li><li class=\"new-tab last\"><a class=\"highwire-figure-link highwire-figure-link-newtab\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F1.large.jpg\" target=\"_blank\">Open in new tab</a></li></ul></div></div></div></div><div class=\"section\" id=\"sec-3\"><h2 class=\"\">Introduction</h2><p id=\"p-11\">COVID-19 is reported to have claimed 4.37 million lives as of Aug 17, 2021 (WHO, 2021). A large number of these deaths are attributed to unexpected surges in infections caused by new strains of SARS-CoV-2, prompting international health organizations such as the CDC and WHO to declare these as variants of concerns (<a class=\"xref-bibr\" href=\"#ref-6\" id=\"xref-ref-6-1\">Khan et al., 2020</a>). Such emergence of new variants can seriously undermine the efficacy of global vaccination programs, the need for future booster doses, or cause multiple reinfection waves as new strains escape previously developed antibodies. Several initiatives have focused on providing high-quality tracking information for the strains and lineages as these emerge (<a class=\"xref-bibr\" href=\"#ref-4\" id=\"xref-ref-4-1\">Hadfield et al., 2018</a>). However, early prediction of emerging variants through genomic signals remains an open challenge. Although domain-based, expert-reasoning approaches are the mainstay of our understanding, which often yield retrospective insight rather than proactive predictions. On the other hand, machine learning approaches are likely to be biased by underlying data characteristics and do not explain the biological basis of predictions.</p><p id=\"p-12\">Here, we propose <strong><em>Strainflow</em></strong>, a hybrid architecture of machine learning and language modeling, along with empirical experiments to demonstrate explainable genomic signals for tracking and predicting the spread of SARS-CoV-2 across countries. Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences. Typically, in language models, word embeddings represent the latent space (dimensions) of a corpus of text (<a class=\"xref-bibr\" href=\"#ref-16\" id=\"xref-ref-16-1\">Mikolov et al., 2013</a>) and can capture highly nonlinear and contextual relationships. Codons (tri-nucleotides, 3-mers) translations represent a natural basis for word representations and have been utilized in the past for learning embedding models for modeling various outcomes such as mutation susceptibility (<a class=\"xref-bibr\" href=\"#ref-36\" id=\"xref-ref-36-1\">Yilmaz, 2020</a>) and gene sequence correlations (<a class=\"xref-bibr\" href=\"#ref-35\" id=\"xref-ref-35-1\">Wu et al., 2021</a>). Recently, Hie et. al (<a class=\"xref-bibr\" href=\"#ref-5\" id=\"xref-ref-5-1\">Hie et al., 2021</a>) used machine learning along with word embedding techniques to model the semantics and grammar of amino acids corresponding to antigenic change to predict the mutations which might lead to viral escape.</p><p id=\"p-13\">In this paper, we focus on the semantics of viral DNA sequences to derive Dimensions of Concern (DoCs) and demonstrate their causal potential for increasing epidemiological spread patterns across 7 countries. Our approach, Strainflow, is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance.</p></div><div class=\"section\" id=\"sec-4\"><h2 class=\"\">Results</h2><div class=\"subsection\" id=\"sec-5\"><h3>Genomic sequence-based language modeling captures emerging diversity in the SARS-CoV-2 spike gene</h3><p id=\"p-14\">Following the linguistic idea that the context of a word is characterized by its neighborhood, we leveraged a word2vec language model and developed “<em>Strainflow</em>” to learn the latent representations of codons in spike gene sequences of SARS-CoV-2. Strainflow is trained on 0.3 million high-quality genome sequences from March 1, 2020 to March 31, 2021, collected from the GISAID database (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-1\">Fig 1A</a></strong>). To understand the latent space representation (LR) learned by Strainflow, we performed qualitative analysis using tSNE for both global and local (country) levels on a monthly basis. The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-2\">Fig 1B</a></strong>) from September, 2020 to March, 2021, along with specific geographic locations (country-level) such as India, UK, USA, and China.</p><div class=\"fig pos-float type-figure odd\" id=\"F2\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a class=\"highwire-fragment fragment-images colorbox-load\" data-figure-caption='&lt;div class=\"highwire-markup\"&gt;&lt;div xmlns=\"http://www.w3.org/1999/xhtml\"&gt;&lt;span class=\"caption-title\"&gt;Latent space of spike gene derived using Strainflow preserves spatiotemporal information of SARS-CoV-2 spread.&lt;/span&gt; (&lt;strong&gt;A&lt;/strong&gt;) The implementation framework of Strainflow (details described in the method section) (&lt;strong&gt;B&lt;/strong&gt;) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China. (&lt;strong&gt;C&lt;/strong&gt;) Embeddings estimated or predicted from the Strainflow model for 0.36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021. &lt;strong&gt;(D)&lt;/strong&gt; Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021. For Pakistan, we couldn’t estimate the fast sample entropy due to fewer number of strain sequences (&lt;20).&lt;/div&gt;&lt;/div&gt;' data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F2.large.jpg?width=800&amp;height=600&amp;carousel=1\" rel=\"gallery-fragment-images-1181327286\" title=\"Latent space of spike gene derived using Strainflow preserves spatiotemporal information of SARS-CoV-2 spread. (A) The implementation framework of Strainflow (details described in the method section) (B) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China. (C) Embeddings estimated or predicted from the Strainflow model for 0.36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021. (D) Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021. For Pakistan, we couldn’t estimate the fast sample entropy due to fewer number of strain sequences (\"><span class=\"hw-responsive-img\"><img alt=\"Figure 1:\" class=\"highwire-fragment fragment-image lazyload\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F2.medium.gif\" height=\"440\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" width=\"317\"/><noscript><img alt=\"Figure 1:\" class=\"highwire-fragment fragment-image\" height=\"440\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F2.medium.gif\" width=\"317\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a class=\"highwire-figure-link highwire-figure-link-download\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F2.large.jpg?download=true\" title=\"Download Figure 1:\">Download figure</a></li><li class=\"new-tab last\"><a class=\"highwire-figure-link highwire-figure-link-newtab\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F2.large.jpg\" target=\"_blank\">Open in new tab</a></li></ul></div><div class=\"fig-caption\" xmlns:xhtml=\"http://www.w3.org/1999/xhtml\"><span class=\"fig-label\">Figure 1:</span> <span class=\"caption-title\">Latent space of spike gene derived using Strainflow preserves spatiotemporal information of SARS-CoV-2 spread.</span><p class=\"first-child\" id=\"p-15\">(<strong>A</strong>) The implementation framework of Strainflow (details described in the method section) (<strong>B</strong>) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China. (<strong>C</strong>) Embeddings estimated or predicted from the Strainflow model for 0.36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021. <strong>(D)</strong> Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021. For Pakistan, we couldn’t estimate the fast sample entropy due to fewer number of strain sequences (&lt;20).</p><div class=\"sb-div caption-clear\"></div></div></div><p id=\"p-16\">To investigate the information content in the latent dimensions or space (LD or LS) of the <em>Strainflow</em> model, we performed qualitative and quantitative analysis on 0.9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021. Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (<a class=\"xref-bibr\" href=\"#ref-12\" id=\"xref-ref-12-1\">Linderman et al., 2019</a>). We compared the 2D t-SNE plot of the world with four countries (India, UK, USA, China) from September, 2020 to June, 2021, which clearly highlights the dynamic changes in the spike genes across countries in different months (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-3\">Fig 1B</a></strong> and <strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-4\">Fig 1C</a></strong>). Additionally, the quantitative analysis of the latent space was performed by calculating the fast sample entropy (<a class=\"xref-bibr\" href=\"#ref-19\" id=\"xref-ref-19-1\">Pan et al., 2011</a>) for each LD. To compare the monthly entropy of the LS per country, mean entropy was calculated and normalized across the months for each country. We observed the highest entropy (information content) for India, UK, USA and China in the months of February-2021, July-2020, August-2020, and January-2021 respectively (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-5\">Fig 1D</a></strong>). Interestingly, we observed high entropy for 4 months from August, 2020 to November, 2020 in the USA (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-6\">Fig 1D</a></strong>). This highlights that the spike protein latent space representation learned by the Strainflow model could be used as a proxy to capture the spatiotemporal entropy or diversity in the emerging SARS-CoV-2 strains across different countries.</p></div><div class=\"subsection\" id=\"sec-6\"><h3>Preservation of spatiotemporal information for SARS-CoV-2 spread depicted through phylogenetic analysis</h3><p id=\"p-17\">Sequence-level embeddings were obtained from the codon embeddings and investigated for the presence of genomically meaningful characteristics. The phylogenetic tree derived from the embeddings for the United Kingdom (<strong><a class=\"xref-fig\" href=\"#F3\" id=\"xref-fig-3-1\">Fig 2A</a></strong>) shows two clear temporally split clusters for 2020 and 2021 sequences, which may be indicative of different strains in these time periods. The temporality of the collected sequences was found to be preserved in the two clusters, although the model was trained only on genome sequences.</p><div class=\"fig pos-float type-figure odd\" id=\"F3\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a class=\"highwire-fragment fragment-images colorbox-load\" data-figure-caption='&lt;div class=\"highwire-markup\"&gt;&lt;div xmlns=\"http://www.w3.org/1999/xhtml\"&gt;&lt;span class=\"caption-title\"&gt;Phylogenetic trees constructed using cosine similarities between 400 randomly sampled sequence embeddings.&lt;/span&gt; (&lt;strong&gt;A&lt;/strong&gt;) Dendrogram for strains from the U.K.: Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021. (&lt;strong&gt;B&lt;/strong&gt;) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.&lt;/div&gt;&lt;/div&gt;' data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F3.large.jpg?width=800&amp;height=600&amp;carousel=1\" rel=\"gallery-fragment-images-1181327286\" title=\"Phylogenetic trees constructed using cosine similarities between 400 randomly sampled sequence embeddings. (A) Dendrogram for strains from the U.K.: Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021. (B) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.\"><span class=\"hw-responsive-img\"><img alt=\"Figure 2:\" class=\"highwire-fragment fragment-image lazyload\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F3.medium.gif\" height=\"227\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" width=\"440\"/><noscript><img alt=\"Figure 2:\" class=\"highwire-fragment fragment-image\" height=\"227\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F3.medium.gif\" width=\"440\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a class=\"highwire-figure-link highwire-figure-link-download\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F3.large.jpg?download=true\" title=\"Download Figure 2:\">Download figure</a></li><li class=\"new-tab last\"><a class=\"highwire-figure-link highwire-figure-link-newtab\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F3.large.jpg\" target=\"_blank\">Open in new tab</a></li></ul></div><div class=\"fig-caption\"><span class=\"fig-label\">Figure 2:</span> <span class=\"caption-title\">Phylogenetic trees constructed using cosine similarities between 400 randomly sampled sequence embeddings.</span><p class=\"first-child\" id=\"p-18\">(<strong>A</strong>) Dendrogram for strains from the U.K.: Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021. (<strong>B</strong>) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.</p><div class=\"sb-div caption-clear\"></div></div></div><p id=\"p-19\">The phylogenetic tree with globally collected sequences (<strong><a class=\"xref-fig\" href=\"#F3\" id=\"xref-fig-3-2\">Fig 2B</a></strong>) demonstrates that geospatial information is also preserved in the sequence embeddings. Strains from countries like China, Australia, Mexico, and England were found in distinct clusters, while those from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India were found to be dispersed with other countries. This behavior is expected because viral strains spread from one country to another, and our samples had different collection dates for each country. Overall, Strainflow captures the temporal emergence of strains and geographic information in a country-specific manner.</p></div><div class=\"subsection\" id=\"sec-7\"><h3>Entropy in the latent space dimensions captures variability in the spike gene</h3><p id=\"p-20\">Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time. We attributed these changes to mutations in the spike protein of the genome sequence. This hypothesis was based on the fact that the embedding of a word (codon) represents the context it occurs in. A mutation results in a codon that is known to be found in a different context. This alters the semantic sense of the sequence, and this modification of meaning is reflected as a change in values across different latent dimensions. The changed values cause a change in the degree of disorder or the entropy of the latent space.</p><p id=\"p-21\">To compare different geographical regions, the sum of sample entropy was computed for each latent dimension across all the months. This revealed that certain geospatial regions such as France and Germany (<strong><a class=\"xref-fig\" href=\"#F4\" id=\"xref-fig-4-1\">Fig 3A</a></strong>) and USA and Canada (<strong><a class=\"xref-fig\" href=\"#F4\" id=\"xref-fig-4-2\">Fig 3B</a></strong>) have similar total entropies across the latent dimensions, indicating that these regions have been accumulating similar changes.</p><div class=\"fig pos-float type-figure odd\" id=\"F4\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a class=\"highwire-fragment fragment-images colorbox-load\" data-figure-caption='&lt;div class=\"highwire-markup\"&gt;&lt;div xmlns=\"http://www.w3.org/1999/xhtml\"&gt;&lt;span class=\"caption-title\"&gt;Sum of sample entropy for each latent dimension for different countries.&lt;/span&gt; Country pairs (&lt;strong&gt;A&lt;/strong&gt;) - France and Germany, (&lt;strong&gt;B&lt;/strong&gt;) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.&lt;/div&gt;&lt;/div&gt;' data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F4.large.jpg?width=800&amp;height=600&amp;carousel=1\" rel=\"gallery-fragment-images-1181327286\" title=\"Sum of sample entropy for each latent dimension for different countries. Country pairs (A) - France and Germany, (B) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.\"><span class=\"hw-responsive-img\"><img alt=\"Figure 3:\" class=\"highwire-fragment fragment-image lazyload\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F4.medium.gif\" height=\"136\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" width=\"440\"/><noscript><img alt=\"Figure 3:\" class=\"highwire-fragment fragment-image\" height=\"136\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F4.medium.gif\" width=\"440\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a class=\"highwire-figure-link highwire-figure-link-download\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F4.large.jpg?download=true\" title=\"Download Figure 3:\">Download figure</a></li><li class=\"new-tab last\"><a class=\"highwire-figure-link highwire-figure-link-newtab\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F4.large.jpg\" target=\"_blank\">Open in new tab</a></li></ul></div><div class=\"fig-caption\"><span class=\"fig-label\">Figure 3:</span> <span class=\"caption-title\">Sum of sample entropy for each latent dimension for different countries.</span><p class=\"first-child\" id=\"p-22\">Country pairs (<strong>A</strong>) - France and Germany, (<strong>B</strong>) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.</p><div class=\"sb-div caption-clear\"></div></div></div></div><div class=\"subsection\" id=\"sec-8\"><h3>Entropy dimensions are predictive of new COVID-19 caseloads</h3><p id=\"p-23\">We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries. Detrended cross-correlation coefficient was calculated at different lag values, which revealed that entropy dimensions have a leading relationship with new cases (<strong><a class=\"xref-fig\" href=\"#F5\" id=\"xref-fig-5-1\">Fig 4A, 4B</a></strong>). This suggests that the genome sequence data in a given month can be used to predict new cases in subsequent months. A lead period of two months was chosen and Boruta algorithm was employed to assign feature importance scores to different dimensions, which revealed that dimension 32 is the most significant predictor of new cases <strong>(<a class=\"xref-fig\" href=\"#F5\" id=\"xref-fig-5-2\">Fig 4C</a>)</strong>. Significant dimensions from Boruta analysis were termed Dimensions of Concern (DoCs). Random forest based regression modeling on the DoCs achieved a total R-squared of 73% on the test set. The predicted cases were found to be highly correlated with the actual cases (<strong><a class=\"xref-table\" href=\"#T1\" id=\"xref-table-wrap-1-1\">Table 1</a></strong>), which suggests that our model can indicate the directional change of cases for different countries. Further, the predicted relative change in cases between successive months was found to be correlated to the actual relative changes (<strong>Table S1</strong>), which suggests that our model can also indicate the magnitude of change that we expect to observe in the cases.</p><div class=\"table pos-float\" id=\"T1\"><div class=\"table-inline table-callout-links\"><div class=\"callout\"><span>View this table:</span><ul class=\"callout-links\"><li class=\"view-inline first\"><a class=\"table-expand-inline\" data-hide-link-title=\"0\" data-icon-position=\"\" data-table-url=\"/highwire/markup/2128021/expansion?postprocessors=highwire_tables%2Chighwire_reclass%2Chighwire_figures%2Chighwire_math%2Chighwire_inline_linked_media%2Chighwire_embed&amp;table-expand-inline=1\" href=\"##\">View inline</a></li><li class=\"view-popup\"><a class=\"colorbox colorbox-load table-expand-popup\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/markup/2128021/expansion?width=1000&amp;height=500&amp;iframe=true&amp;postprocessors=highwire_tables%2Chighwire_reclass%2Chighwire_figures%2Chighwire_math%2Chighwire_inline_linked_media%2Chighwire_embed\" rel=\"gallery-fragment-tables\">View popup</a></li><li class=\"download-ppt last\"><a class=\"highwire-figure-link highwire-figure-link-ppt\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/powerpoint/2128021\">Download powerpoint</a></li></ul></div></div><div class=\"table-caption\"><span class=\"table-label\">Table 1:</span> <span class=\"caption-title\">Pearson and Spearman’s Correlation coefficients between predicted and actual cases in different countries.</span><div class=\"sb-div caption-clear\"></div></div></div><div class=\"fig pos-float type-figure odd\" id=\"F5\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a class=\"highwire-fragment fragment-images colorbox-load\" data-figure-caption='&lt;div class=\"highwire-markup\"&gt;&lt;div xmlns=\"http://www.w3.org/1999/xhtml\"&gt;&lt;span class=\"caption-title\"&gt;Relationship of the entropy of latent space dimensions with COVID-19 caseloads.&lt;/span&gt; (&lt;strong&gt;A&lt;/strong&gt;) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA. High values are observed for a lead of 1 and 2 months. (&lt;strong&gt;B&lt;/strong&gt;) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases. (&lt;strong&gt;C&lt;/strong&gt;) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.&lt;/div&gt;&lt;/div&gt;' data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F5.large.jpg?width=800&amp;height=600&amp;carousel=1\" rel=\"gallery-fragment-images-1181327286\" title=\"Relationship of the entropy of latent space dimensions with COVID-19 caseloads. (A) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA. High values are observed for a lead of 1 and 2 months. (B) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases. (C) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.\"><span class=\"hw-responsive-img\"><img alt=\"Figure 4:\" class=\"highwire-fragment fragment-image lazyload\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F5.medium.gif\" height=\"378\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" width=\"440\"/><noscript><img alt=\"Figure 4:\" class=\"highwire-fragment fragment-image\" height=\"378\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F5.medium.gif\" width=\"440\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a class=\"highwire-figure-link highwire-figure-link-download\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F5.large.jpg?download=true\" title=\"Download Figure 4:\">Download figure</a></li><li class=\"new-tab last\"><a class=\"highwire-figure-link highwire-figure-link-newtab\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F5.large.jpg\" target=\"_blank\">Open in new tab</a></li></ul></div><div class=\"fig-caption\"><span class=\"fig-label\">Figure 4:</span> <span class=\"caption-title\">Relationship of the entropy of latent space dimensions with COVID-19 caseloads.</span><p class=\"first-child\" id=\"p-24\">(<strong>A</strong>) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA. High values are observed for a lead of 1 and 2 months. (<strong>B</strong>) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases. (<strong>C</strong>) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.</p><div class=\"sb-div caption-clear\"></div></div></div><p id=\"p-25\">Our model can be therefore used to predict the COVID-19 caseloads in several countries. Both USA (<strong><a class=\"xref-fig\" href=\"#F6\" id=\"xref-fig-6-1\">Fig 5A</a></strong>) and Japan (<strong><a class=\"xref-fig\" href=\"#F6\" id=\"xref-fig-6-2\">Fig 5C</a></strong>) show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries. Our model predicts new caseloads with a two-month lead time, which strongly predicts a spike in new cases both in USA (<strong><a class=\"xref-fig\" href=\"#F6\" id=\"xref-fig-6-3\">Fig 5B</a></strong>) and Japan (<strong><a class=\"xref-fig\" href=\"#F6\" id=\"xref-fig-6-4\">Fig 5D</a></strong>) in the months of July and August, 2021. Therefore our model may be used as an epidemiological early warning system to predict new caseloads.</p><div class=\"fig pos-float type-figure odd\" id=\"F6\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a class=\"highwire-fragment fragment-images colorbox-load\" data-figure-caption='&lt;div class=\"highwire-markup\"&gt;&lt;div xmlns=\"http://www.w3.org/1999/xhtml\"&gt;&lt;span class=\"caption-title\"&gt;Prediction of new COVID-19 cases with Sample Entropy values of the latent dimensions.&lt;/span&gt; (&lt;strong&gt;A&lt;/strong&gt;) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA. (&lt;strong&gt;B&lt;/strong&gt;) Actual and predicted cases based on the entropy values of DoCs for the USA. The model predicts a rise in cases for July and August 2021. (&lt;strong&gt;C&lt;/strong&gt;) Entropy of DoCs and new cases for Japan. (&lt;strong&gt;D&lt;/strong&gt;) Actual and predicted cases for Japan. A spike in cases is predicted for July and August 2021.&lt;/div&gt;&lt;/div&gt;' data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F6.large.jpg?width=800&amp;height=600&amp;carousel=1\" rel=\"gallery-fragment-images-1181327286\" title=\"Prediction of new COVID-19 cases with Sample Entropy values of the latent dimensions. (A) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA. (B) Actual and predicted cases based on the entropy values of DoCs for the USA. The model predicts a rise in cases for July and August 2021. (C) Entropy of DoCs and new cases for Japan. (D) Actual and predicted cases for Japan. A spike in cases is predicted for July and August 2021.\"><span class=\"hw-responsive-img\"><img alt=\"Figure 5:\" class=\"highwire-fragment fragment-image lazyload\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F6.medium.gif\" height=\"284\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" width=\"440\"/><noscript><img alt=\"Figure 5:\" class=\"highwire-fragment fragment-image\" height=\"284\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F6.medium.gif\" width=\"440\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a class=\"highwire-figure-link highwire-figure-link-download\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F6.large.jpg?download=true\" title=\"Download Figure 5:\">Download figure</a></li><li class=\"new-tab last\"><a class=\"highwire-figure-link highwire-figure-link-newtab\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F6.large.jpg\" target=\"_blank\">Open in new tab</a></li></ul></div><div class=\"fig-caption\"><span class=\"fig-label\">Figure 5:</span> <span class=\"caption-title\">Prediction of new COVID-19 cases with Sample Entropy values of the latent dimensions.</span><p class=\"first-child\" id=\"p-26\">(<strong>A</strong>) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA. (<strong>B</strong>) Actual and predicted cases based on the entropy values of DoCs for the USA. The model predicts a rise in cases for July and August 2021. (<strong>C</strong>) Entropy of DoCs and new cases for Japan. (<strong>D</strong>) Actual and predicted cases for Japan. A spike in cases is predicted for July and August 2021.</p><div class=\"sb-div caption-clear\"></div></div></div></div><div class=\"subsection\" id=\"sec-9\"><h3>Codons associated with Dimensions of Concerns (DoCs) could be linked to SARS-CoV-2 variants</h3><p id=\"p-27\">We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension (<strong>Table S2</strong>). The intuition behind this idea is that the codons with high weights in a given dimension, when mutated in the viral sequence, are likely to cause a significant change in the entropy of the associated dimensions. Therefore each DoC can be linked to codons, which can further be mapped to Variants of concern (VOCs) and Variants of Interest (VOIs) (<strong>Table S3</strong>). Despite the fact that our model cannot directly capture the SARS-CoV-2 variants, it was observed that dimension-32 captures the CTG, CGG codons (ranks 5 and 8 respectively), known to occur in the T19R variant. Similarly, dimension 3 captures three codons (ACG, CGG, CAC) that are associated with multiple variants such as K417T, L452R, and D1118H, causing increased infectivity, pathogenicity, and spread. Dimension 30 captures codons CAT and CAC associated with Δ69 and D1118H respectively which are linked to B.1.1.7 lineage.</p><p id=\"p-28\">Codon weights (<strong>Table S2</strong>) of a given DoC provide an opportunity to associate with specific Variants of concern (VOCs) and Variants of Interest (VOIs); and to predict emerging SARS-CoV-2 variants and take preventive measures in advance. Distinct dimensions capture country-specific changes and may be surveilled to monitor the spread of the pandemic. This approach was back-validated with several real-world examples. For instance, dimension 32 captures the codons CGG (R) and CAC (R), which are found in B.1.429 lineage (L425R mutation). Dimension 3 captures CGG which is seen in L452R (associated with lineage B.1.617.1), which was first observed in India in December 2020 and was found to have increased infectivity and transmissibility.</p></div><div class=\"subsection\" id=\"sec-10\"><h3>Temporal tracking of entropy in Dimensions of Concern (DoCs) may be used as a surrogate to track the spread of various SARS-CoV-2 variants</h3><p id=\"p-29\">To investigate the potential of DoCs to track the spread of SARS-CoV-2, we used the codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs. We selected DoCs (dimension 3, 4, 12, 13, 15, 16, 25, 28, 30, 32) with high absolute weights for the codons related to the delta variants (<strong><a class=\"xref-fig\" href=\"#F7\" id=\"xref-fig-7-1\">Fig 6A</a></strong>). The entropy of the DoCs was contrasted with the caseloads in England (<strong><a class=\"xref-fig\" href=\"#F7\" id=\"xref-fig-7-2\">Fig 6B</a></strong>), India (<strong><a class=\"xref-fig\" href=\"#F7\" id=\"xref-fig-7-3\">Fig 6C</a></strong>), and USA (<strong><a class=\"xref-fig\" href=\"#F7\" id=\"xref-fig-7-4\">Fig 6D</a></strong>). Overall, the temporal tracking of the entropy in Dimensions of Concern (DoCs) may be used as a surrogate to track the spread of various SAR-CoV-2 variants.</p><div class=\"fig pos-float type-figure odd\" id=\"F7\"><div class=\"highwire-figure\"><div class=\"fig-inline-img-wrapper\"><div class=\"fig-inline-img\"><a class=\"highwire-fragment fragment-images colorbox-load\" data-figure-caption='&lt;div class=\"highwire-markup\"&gt;&lt;div xmlns=\"http://www.w3.org/1999/xhtml\"&gt;&lt;span class=\"caption-title\"&gt;Potential association of codons observed in SARS-CoV-2 Delta variant (lineage B.&lt;strong&gt;1&lt;/strong&gt;.&lt;strong&gt;617&lt;/strong&gt;.&lt;strong&gt;2&lt;/strong&gt;) with their corresponding DoCs, and the trend of caseloads with the entropy of the DoCs.&lt;/span&gt; (&lt;strong&gt;A&lt;/strong&gt;) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling. Line plots showing the entropies of DoCs and cases in countries, (&lt;strong&gt;B&lt;/strong&gt;) England, (&lt;strong&gt;C&lt;/strong&gt;) India, and (&lt;strong&gt;D&lt;/strong&gt;) USA. The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.&lt;/div&gt;&lt;/div&gt;' data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F7.large.jpg?width=800&amp;height=600&amp;carousel=1\" rel=\"gallery-fragment-images-1181327286\" title=\"Potential association of codons observed in SARS-CoV-2 Delta variant (lineage B.1.617.2) with their corresponding DoCs, and the trend of caseloads with the entropy of the DoCs. (A) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling. Line plots showing the entropies of DoCs and cases in countries, (B) England, (C) India, and (D) USA. The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.\"><span class=\"hw-responsive-img\"><img alt=\"Fig 6.\" class=\"highwire-fragment fragment-image lazyload\" data-src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F7.medium.gif\" height=\"432\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\" width=\"440\"/><noscript><img alt=\"Fig 6.\" class=\"highwire-fragment fragment-image\" height=\"432\" src=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F7.medium.gif\" width=\"440\"/></noscript></span></a></div></div><ul class=\"highwire-figure-links inline\"><li class=\"download-fig first\"><a class=\"highwire-figure-link highwire-figure-link-download\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F7.large.jpg?download=true\" title=\"Download Fig 6.\">Download figure</a></li><li class=\"new-tab last\"><a class=\"highwire-figure-link highwire-figure-link-newtab\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/biorxiv/early/2021/08/26/2021.05.25.445601/F7.large.jpg\" target=\"_blank\">Open in new tab</a></li></ul></div><div class=\"fig-caption\"><span class=\"fig-label\">Fig 6.</span> <span class=\"caption-title\">Potential association of codons observed in SARS-CoV-2 Delta variant (lineage B.<strong>1</strong>.<strong>617</strong>.<strong>2</strong>) with their corresponding DoCs, and the trend of caseloads with the entropy of the DoCs.</span><p class=\"first-child\" id=\"p-30\">(<strong>A</strong>) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling. Line plots showing the entropies of DoCs and cases in countries, (<strong>B</strong>) England, (<strong>C</strong>) India, and (<strong>D</strong>) USA. The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.</p><div class=\"sb-div caption-clear\"></div></div></div></div></div><div class=\"section\" id=\"sec-11\"><h2 class=\"\">Discussion</h2><p id=\"p-31\">We have implemented an approach for analyzing the emerging strains based on the latent space of spike protein coding nucleotide sequences. We chose the nucleotide sequences instead of proteins in order to capture and track the variations that may not have immediate functional consequences. Our approach has two main underlying tenets: (i) long-range interactions are known to modulate the functional (<a class=\"xref-bibr\" href=\"#ref-17\" id=\"xref-ref-17-1\">Mugnai et al., 2020</a>) interaction between receptor binding domain and ACE2 receptors, hence may be captured in the NLP models that capture 3-mer changes and context, and (ii) latent dimensions may be differentially correlated with indicators of spread, thus providing a data-driven handle for tracking and predicting variants of concern and variants of interest. The pipeline takes advantage of temporal changes in the semantics of mutating sequences. Preservation of phylogenetic structure based upon the similarity matrix obtained using the embeddings validated that the latent dimensions capture spatio-temporal information. Analyzing the dynamic patterns and underlying correlations in the 30,000 base pair long sequence of SARS-CoV-2 (<a class=\"xref-bibr\" href=\"#ref-26\" id=\"xref-ref-26-1\">Shishir et al., 2021</a>) is important to highlight the mechanistic understanding of mutations. SARS-CoV-2 seems to show a particularly high frequency of recombinations arising due to the absence of a proof-reading mechanism and sequence diversity, which calls for urgency in studying its transmission pattern (<a class=\"xref-bibr\" href=\"#ref-15\" id=\"xref-ref-15-1\">Mandal et al., 2021</a>)(<a class=\"xref-bibr\" href=\"#ref-25\" id=\"xref-ref-25-1\">Rouchka et al., 2020</a>). Therefore predicting mutations in the spike protein, which binds to ACE2 receptors can help us estimate the spread of disease and the efficacy of therapeutic treatments and vaccines (<a class=\"xref-bibr\" href=\"#ref-28\" id=\"xref-ref-28-1\">Srivastava et al., 2021</a>)(<a class=\"xref-bibr\" href=\"#ref-11\" id=\"xref-ref-11-1\">Li et al., 2020</a>).</p><p id=\"p-32\">Entropy is a measure of the disorder of a system. We hypothesized that mutations increase the chaotic dynamics in the latent space of spike genes. To calculate entropy, we used the accelerated versions of the Approximate Entropy and Sample Entropy algorithms, called Fast Approximate Entropy and Fast Sample Entropy (<a class=\"xref-bibr\" href=\"#ref-19\" id=\"xref-ref-19-2\">Pan et al., 2011</a>). Both algorithms aim to quantify how often different patterns of data are found in a time series. Fast Approximate Entropy, however, is a biased statistic and depends on the length of the series. Since we could have different counts of genome sequences collected each month, we preferred Sample Entropy, which is independent of the length of the series. Entropy values were calculated for each latent dimension in each month. Thereafter, Detrended Cross-Correlation Analysis (DCCA) (<a class=\"xref-bibr\" href=\"#ref-22\" id=\"xref-ref-22-1\">Podobnik and Stanley, 2007</a>) was performed between the entropy dimensions and the new cases. DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series. High cross-correlation for different lead periods revealed that the entropy values in a given month could be used to predict the new cases in different countries in subsequent months. Different countries had different lead times at which the highest cross-correlation was observed between the entropy dimensions and the cases, ranging from 1-6 months. Overall, a lead time of two months was chosen to model the new cases.</p><p id=\"p-33\">A similar analysis was done with daily values of entropy and new cases. Entropy was calculated in rolling windows, and cross-correlation analysis was performed between entropy and new cases at different lead periods. Although the cross-correlation values were found to be significant, the values were low and ranged between -0.1 to 0.1. Therefore, we decided to use the monthly entropy values for the modeling exercise.</p><p id=\"p-34\">We also experimented with “blips” as a feature to predict the new cases. Blips are sudden changes in the values of the latent dimensions. These changes may be caused by a mutation, which changes the words (codons) in a given genome sequence. This hypothesis was validated in simulation experiments in synthetic datasets. Each dimension of the spike gene embeddings for a country was analyzed for the presence of temporal anomalies. Countries having a minimum of 20 samples in any given month were selected and the same number of records (minimum samples in any given month for that country) were sampled without replacement from each month. These records were used to define control limits of ±1 standard deviation from the mean value for each dimension, and all values in the full dataset outside those limits were categorized as ‘Blip’ points. Blip counts in each month were normalized by calculating the number of blips per sample collected in a month for each dimension (normalized blips). The embedding dimensions were then compared in terms of the total normalized blips for each country to observe the significant dimensions and dis(similarity) in trends among different countries. Cumulative counts of normalized blips were analyzed to understand the temporal accumulation of blips in each dimension. Similar to entropy, blips were found to have a leading relationship with the cases. However, regression modeling results with sample entropy were found to be better.</p><p id=\"p-35\">To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data. With sample entropy, the R-squared value achieved was 73%, while with approximate entropy, the value was only 10%. Therefore the model trained on sample entropy was selected. The predictions from the model were found to be highly correlated with the actual cases, indicating that our model can be used for preemptive warning signals for the rise in cases in different countries. Further, the actual and the predicted difference in the number of cases in consecutive months was found to be correlated, which suggests that the relative change in the cases in consecutive months predicted by our model is linked to the relative change in the number of cases. Overall, we recommend that our model be used to predict dangerous trends and not the actual number of cases. Further, the mapping from latent dimensions to Variants of Concern (VOCs) and Variants of Interest (VOIs) may help us track the country-specific spread of different variants.</p><p id=\"p-36\">Prediction models can be used for training the genomic sequence for predicting infection severity based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 (<a class=\"xref-bibr\" href=\"#ref-32\" id=\"xref-ref-32-1\">Wang et al., 2020</a>). The machine learning models can also be trained on genomic sequences for COVID-19 classification (<a class=\"xref-bibr\" href=\"#ref-2\" id=\"xref-ref-2-1\">Arslan, 2021</a>). Although the variance explained by our model is low, however, we were able to compute the variability associated with spike protein mutations. So our method showed a potential way to estimate the new cases variability associated with spike protein mutations. Our methods can be incorporated with the epidemic projections model to better predict the epidemic trajectories. The latent dimensions may further be employed to predict the clinical consequences of emerging strains. The currently available vaccines are intended for early SARS-CoV-2 strains, but with new emerging variants, immune responses triggered by these vaccines may be weaker and short-lived. As seen in the devastating second wave of the pandemic in India, newer SARS-CoV-2 variants have acquired an increased pathogenic potential resulting in rapid clinical progression and overwhelmed health systems. Mitigating such events in the future will require stronger surveillance systems in place. Our study offers a promising solution in this direction and lays the foundation for proactive genomic surveillance of COVID-19.</p><div class=\"subsection\" id=\"sec-12\"><h3>Limitations of our study</h3><p id=\"p-37\">Our approach of codon (3-mers) embeddings does not indicate the position where the codon change may have happened in the spike gene. This is because low-dimensional embeddings do not preserve the positional encoding of words. However, we are investigating advanced approaches such as complex-valued word embeddings with positional encodings (<a class=\"xref-bibr\" href=\"#ref-31\" id=\"xref-ref-31-1\">Wang et al., 2019</a>) and transformer models such as BERT (<a class=\"xref-bibr\" href=\"#ref-34\" id=\"xref-ref-34-1\">Wolf et al., 2020</a>)(<a class=\"xref-bibr\" href=\"#ref-9\" id=\"xref-ref-9-1\">Lee et al., 2020</a>) to overcome our current limitation. The latter are considered expensive and data-hungry models and it will remain to be evaluated if the gain of positional information may be countered by the loss of prediction accuracy for forecasting new cases in the future. However, we believe that the availability of sequences for a wide variety of viral pathogens presents an exciting opportunity to train data-hungry models that may be able to transfer insights across pathogens and yet remain interpretable.</p><p id=\"p-38\">Further, our Strainflow model is trained only on the spike gene of the viral genome, which does not represent the complete variation spectrum of the virus. To mitigate this shortcoming, we are currently developing a genome-level Strainflow model (version 2.0) for SARS-CoV-2.</p><p id=\"p-39\">Another limitation of our study is the relatively small number of samples that were used to construct the supervised predictive model for case prediction. As more data becomes available in subsequent months, we can produce more confident case predictions. Nonetheless, the unsupervised embeddings and temporal cross-correlations were learned upon the full datasets and these presented clear patterns in DoCs and significant cross-correlations with caseloads. However, it is important for our models to receive at least 30 samples per month for getting reliable prediction results for the future months. This also underscores the need for a more reliable and agile approach to deposit country-level datasets on repositories such as GISAID. We make an appeal to the countries to facilitate the sharing of such data in order to be prepared for any future waves of the current pandemic and for preventing the new emergence of strains. We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance.</p></div></div><div class=\"section\" id=\"sec-13\"><h2 class=\"\">Methods</h2><div class=\"subsection\" id=\"sec-14\"><h3>Key resources table</h3><div class=\"table pos-float\" id=\"T2\"><div class=\"table-inline table-callout-links\"><div class=\"callout\"><span>View this table:</span><ul class=\"callout-links\"><li class=\"view-inline first\"><a class=\"table-expand-inline\" data-hide-link-title=\"0\" data-icon-position=\"\" data-table-url=\"/highwire/markup/2128022/expansion?postprocessors=highwire_tables%2Chighwire_reclass%2Chighwire_figures%2Chighwire_math%2Chighwire_inline_linked_media%2Chighwire_embed&amp;table-expand-inline=1\" href=\"##\">View inline</a></li><li class=\"view-popup\"><a class=\"colorbox colorbox-load table-expand-popup\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/markup/2128022/expansion?width=1000&amp;height=500&amp;iframe=true&amp;postprocessors=highwire_tables%2Chighwire_reclass%2Chighwire_figures%2Chighwire_math%2Chighwire_inline_linked_media%2Chighwire_embed\" rel=\"gallery-fragment-tables\">View popup</a></li><li class=\"download-ppt last\"><a class=\"highwire-figure-link highwire-figure-link-ppt\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/powerpoint/2128022\">Download powerpoint</a></li></ul></div></div></div></div><div class=\"subsection\" id=\"sec-15\"><h3>Datasets</h3><div class=\"subsection\" id=\"sec-16\"><h4>Training dataset</h4><p id=\"p-40\">The dataset was downloaded from GISAID EpiCoV (<a class=\"xref-bibr\" href=\"#ref-27\" id=\"xref-ref-27-1\">Shu and McCauley, 2017</a>) (April 8, 2021 release). 0.63 million genome sequences with high nucleotide completeness, coverage, complete temporal information, and presence of less than 5% non-identified nucleotide bases (N) were downloaded. The sequences included 63 countries, including India, United Kingdom, USA, Australia, New Zealand, Germany, Russia, Italy, France, Mexico, Canada, China, Japan, Pakistan, Bangladesh, Iran, Iraq, the continent of South America, and Africa. Duplicate samples were removed, and whole genome sequences were parsed using CoV-Seq (<a class=\"xref-bibr\" href=\"#ref-13\" id=\"xref-ref-13-1\">Liu et al., 2020</a>) to extract nucleotide sequences corresponding to each of the 12 CDS. Accession IDs that did not cover 12 coding regions were discarded, yielding 0.31 million high-quality SARS-CoV-2 genome sequences for language modelling. We downloaded country-wise COVID-19 data for new cases from a publicly available repository maintained by Johns Hopkins University Center for Systems Science and Engineering (JHU CSSE).</p></div><div class=\"subsection\" id=\"sec-17\"><h4>Test dataset</h4><p id=\"p-41\">We downloaded around 0.6 million genome sequences submitted to GISAID from April 2021 to June 2021. We have used 0.36 million sequences to predict their latent space using the <em>Strainflow</em> model for the months of April 2021 to June 2021.</p></div></div><div class=\"subsection\" id=\"sec-18\"><h3>Word Embeddings in Strainflow model</h3><p id=\"p-42\">In our Strainflow model, we have adopted a word2vec model (<a class=\"xref-bibr\" href=\"#ref-16\" id=\"xref-ref-16-2\">Mikolov et al., 2013</a>). A low dimensional representation for the genome sequences was learned using the word2vec model. Non-overlapping sequences of 3-mers (codons) were considered as words for training the word2vec model, implemented in Gensim (<a class=\"xref-bibr\" href=\"#ref-24\" id=\"xref-ref-24-1\">Radim Rehurek, 2010</a>). The word2vec model was trained using the skip-gram algorithm, with a fixed window size of twenty and vector size of thirty-six. For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise. The mean was calculated by summing across all the k-mers over each dimension and then dividing it by the total number of codons present in the sequence. For selecting the vector size of the word2vec model, we calculated PIP (Pairwise Inner Product) loss (<a class=\"xref-bibr\" href=\"#ref-37\" id=\"xref-ref-37-1\">Yin and Shen, 2018</a>). PIP loss is a metric used for calculating the dissimilarity between two word embedding matrices. For the embedding matrix of strains (E), the PIP matrix is defined as the dot product of the embedding matrix with its transpose (<em>E. E</em> <sup><em>T</em></sup>). The PIP loss between two embedding matrices is defined as the norm of the difference between their PIP matrices.\n",
              "<span class=\"disp-formula\" id=\"disp-formula-1\"><span class=\"highwire-responsive-lazyload\"><img alt=\"Embedded Image\" class=\"highwire-embed lazyload\" data-src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-10.gif\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\"/><noscript><img alt=\"Embedded Image\" class=\"highwire-embed\" src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-10.gif\"/></noscript></span>\n",
              "</span>\n",
              "</p><p id=\"p-43\">Various word2vec models were trained on the dataset with different vector sizes varying in multiples of three. Based on the PIP loss calculations, we found out that word embeddings with 36 dimensions showed a differential dent in the curve (change in straight line), due to which we selected this to be the dimension of the word embeddings (<strong>Fig. S1</strong>).</p></div><div class=\"subsection\" id=\"sec-19\"><h3>Phylogenetic analysis using the latent dimensions of the spike genes</h3><p id=\"p-44\">To evaluate the phylogenetic properties based on the latent dimensions of the spike gene, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions. The pairwise distance was further used for hierarchical clustering using the ‘hclust’ function in R statistical programming language. This analysis was performed using 400 random sequences of spike genes from 16 countries. The visualization of the phylogenetic tree derived based on the latent dimensions was done using ‘iTOL’ (Letunic) software (<strong><a class=\"xref-fig\" href=\"#F3\" id=\"xref-fig-3-3\">Fig 2</a></strong>).</p></div><div class=\"subsection\" id=\"sec-20\"><h3>Entropy of the latent dimensions</h3><p id=\"p-45\">To quantify the properties of latent dimensions, we have used a well-known information theory based algorithm suitable for time series datasets, called ‘Fast Sample Entropy’ (<a class=\"xref-bibr\" href=\"#ref-30\" id=\"xref-ref-30-1\">Tomčala, 2020</a>). To compute Fast Sample Entropy, we have used the ‘FastSampEn’ function in the ‘TSEntropies’ package in R (<a class=\"xref-bibr\" href=\"#ref-29\" id=\"xref-ref-29-1\">Tomcala, 2018</a>). Fast Sample Entropy can be computed as follows.\n",
              "<span class=\"disp-formula\" id=\"disp-formula-2\"><span class=\"highwire-responsive-lazyload\"><img alt=\"Embedded Image\" class=\"highwire-embed lazyload\" data-src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-11.gif\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\"/><noscript><img alt=\"Embedded Image\" class=\"highwire-embed\" src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-11.gif\"/></noscript></span>\n",
              "</span>\n",
              "where,\n",
              "<span class=\"disp-formula\" id=\"disp-formula-3\"><span class=\"highwire-responsive-lazyload\"><img alt=\"Embedded Image\" class=\"highwire-embed lazyload\" data-src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-12.gif\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\"/><noscript><img alt=\"Embedded Image\" class=\"highwire-embed\" src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-12.gif\"/></noscript></span>\n",
              "</span>\n",
              "</p><p id=\"p-46\"><em>S</em><sub><em>i,mm</em></sub> is a set of sub-sequences of length m belonging to the i-th neighborhood, and <em>N</em><sub>m</sub> is the number of these neighborhoods.</p><p id=\"p-47\">In our case, ‘x’ is the latent dimension of the spike genes of the SARS-CoV-2 strains per month for a given country with default values of ‘m’ and ‘r’. Entropy was computed for each latent dimension on a monthly basis for each country. The minimum number of SARS-CoV-2 spike genes was 20. To compare across months, we used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-7\">Fig 1D</a></strong>).</p><p id=\"p-48\">To compare the entropy of the latent dimensions among countries, we used the total entropy of the country for each dimension and visualized it with line graphs (<strong><a class=\"xref-fig\" href=\"#F4\" id=\"xref-fig-4-3\">Fig 3</a></strong>).</p></div><div class=\"subsection\" id=\"sec-21\"><h3>Detrended Cross Correlations Analysis (DCCA)</h3><p id=\"p-49\">To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used the Detrended Cross Correlation Analysis (<a class=\"xref-bibr\" href=\"#ref-23\" id=\"xref-ref-23-1\">Prass and Pumi, 2019</a>). Here, DCCA captures the long-range cross correlation between time series (entropy of the months and caseloads for a given country). We tested both time series for stationarity using Augmented Dickey-Fuller (ADF) test (<a class=\"xref-bibr\" href=\"#ref-18\" id=\"xref-ref-18-1\">Mushtaq, 2011</a>). The ADF test was implemented using the function ‘adf.test’ available in the ‘tseries’ (2020a) package in R. Due to the non-stationary distribution of the estimated entropies and the caseloads for a given country, we used the ‘DCCA’ R package (2020b). Cross-correlation was calculated between the entropy dimensions at time t+h and new cases at time t, where h = 0, ±1, ±2, ±3 … ±10.</p></div><div class=\"subsection\" id=\"sec-22\"><h3>Machine Learning based identification of Dimensions of Concern (DoCs)</h3><p id=\"p-50\"><em>Dimensions of Concern (DoCs)</em> were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis and predictive of new case count in the respective countries. Country-wise monthly total new cases data was taken at the end of each month. Total new cases data for each month was merged to monthly entropy dimensions data from March, 2020 to June, 2021. We used a regression based machine learning approach called ‘Boruta’ (<a class=\"xref-bibr\" href=\"#ref-7\" id=\"xref-ref-7-1\">Kursa and Rudnicki, 2010</a>), a wrapper algorithm around random forest algorithm (<a class=\"xref-bibr\" href=\"#ref-8\" id=\"xref-ref-8-1\">Kursa et al., 2010</a>) to select the most relevant entropy dimensions for the prediction of subsequent two months’ new cases. We used the default parameters with the modification of the maximum runs as 1000. We selected the confirmed entropy dimensions as the most relevant features for the prediction of new cases.</p></div><div class=\"subsection\" id=\"sec-23\"><h3>Model development and evaluation for prediction of new cases in subsequent months</h3><p id=\"p-51\">To predict the new cases in the next to next months, we used a regression based random forest model (Breiman 2001) using the most relevant DoCs based on Boruta. The model training was performed using the data from March, 2020 to April, 2021; and the fitted model was tested on data from May, 2021 to June, 2021. The regression modeling was performed using 1000 decision trees using the ‘randomForest’ package in R (A. Liaw et al., 2002).</p></div><div class=\"subsection\" id=\"sec-24\"><h3>Top codons associated with DoCs</h3><p id=\"p-52\">To find the top codons associated with the DoCs, we extracted the absolute weights of each codon for a given DoC. The top 10 codons having the highest absolute weights (contribution) were identified corresponding to each of the DoCs to compare our DoCs with SARS-CoV-2 variants. We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (<a class=\"xref-bibr\" href=\"#ref-3\" id=\"xref-ref-3-1\">CDC, 2021</a>); and a list of codons associated with VOIs and VOCs was curated <strong>(Table S4)</strong> (Lopez-Rincon et al.; <a class=\"xref-bibr\" href=\"#ref-20\" id=\"xref-ref-20-1\">paola, 2021</a>; <a class=\"xref-bibr\" href=\"#ref-21\" id=\"xref-ref-21-1\">Peacock et al., 2021</a>; <a class=\"xref-bibr\" href=\"#ref-28\" id=\"xref-ref-28-2\">Srivastava et al., 2021</a>). The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant (<a class=\"xref-bibr\" href=\"#ref-3\" id=\"xref-ref-3-2\">CDC, 2021</a>)</p></div></div><div class=\"section ack\" id=\"ack-1\"><h2 class=\"\">Acknowledgements</h2><p id=\"p-53\">This work was supported by the Delhi Cluster-Delhi Research Implementation and Innovation (DRIIV) Project supported by the Principal Scientific Advisor Office, Prn.SA/Delhi/ Hub/2018(C) and the Center of Excellence in Healthcare supported by Delhi Knowledge Development Foundation (DKDF) at IIIT-Delhi. We also thank Dr. Chitra Pattabiraman (NIMHANS) for her valuable inputs on viral sequences, and Harleen Kaur, Nishkarsh Saxena, and Anjali for their contribution to the dashboard visualizations.</p></div><div class=\"section fn-group\" id=\"fn-group-1\"><h2>Footnotes</h2><ul><li class=\"fn-update fn-group-summary-of-updates\" id=\"fn-3\"><p id=\"p-10\">In this revised version, we have implemented fast tSNE for qualitative inspection of the latent dimensions (LD) of 0.9 million SARS-CoV-2 spike genes. Quantitative analyses of the LDs were performed using the fast sample entropy method. Also, we have used fast sample entropy instead of 'blips' to model COVID-19 caseloads.</p></li></ul></div><div class=\"section ref-list\" id=\"ref-list-1\"><h2 class=\"\">References</h2><ol class=\"cit-list ref-use-labels\"><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.1\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"> <span class=\"cit-name-given-names\">A.</span> <span class=\"cit-name-surname\">Liaw</span></span> and <span class=\"cit-auth\"> <span class=\"cit-name-given-names\">M.</span> <span class=\"cit-name-surname\">Wiener</span></span> (<span class=\"cit-pub-date\">2002</span>). <span class=\"cit-article-title\">Classification and Regression by randomForest</span>. <abbr class=\"cit-jnl-abbrev\">R News</abbr> <span class=\"cit-vol\">2</span>(<span class=\"cit-issue\">3</span>), <span class=\"cit-fpage\">18</span>--22.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DR%2BNews%26rft.volume%253D2%26rft.spage%253D18%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-2-1\" id=\"ref-2\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.2\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Arslan</span>,  <span class=\"cit-name-given-names\">H.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">Machine Learning Methods for COVID-19 Prediction Using Human Genomic Data</span>. <abbr class=\"cit-jnl-abbrev\">Proceedings</abbr> <span class=\"cit-vol\">74</span>, <span class=\"cit-fpage\">20</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DProceedings%26rft.volume%253D74%26rft.spage%253D20%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-3-1\" id=\"ref-3\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.3\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth cit-collab\">CDC</span> (<span class=\"cit-pub-date\">2021</span>). <abbr class=\"cit-jnl-abbrev\">SARS-CoV-2 Variant Classifications and Definitions</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-4-1\" id=\"ref-4\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" data-doi=\"10/gdkbqx\" id=\"cit-2021.05.25.445601v3.4\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Hadfield</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Megill</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Bell</span>,  <span class=\"cit-name-given-names\">S.M.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Huddleston</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Potter</span>,  <span class=\"cit-name-given-names\">B.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Callender</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Sagulenko</span>,  <span class=\"cit-name-given-names\">P.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Bedford</span>,  <span class=\"cit-name-given-names\">T.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Neher</span>,  <span class=\"cit-name-given-names\">R.A.</span></span> (<span class=\"cit-pub-date\">2018</span>). <span class=\"cit-article-title\">Nextstrain: real-time tracking of pathogen evolution</span>. <abbr class=\"cit-jnl-abbrev\">Bioinformatics</abbr> <span class=\"cit-vol\">34</span>, <span class=\"cit-fpage\">4121</span>–<span class=\"cit-lpage\">4123</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DBioinformatics%26rft.volume%253D34%26rft.spage%253D4121%26rft_id%253Dinfo%253Adoi%252F10%252Fgdkbqx%26rft_id%253Dinfo%253Apmid%252Fhttp%253A%252F%252Fwww.n%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\" href=\"/lookup/external-ref?access_num=10/gdkbqx&amp;link_type=DOI\"><span>CrossRef</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-medline\" href=\"/lookup/external-ref?access_num=http://www.n&amp;link_type=MED&amp;atom=%2Fbiorxiv%2Fearly%2F2021%2F08%2F26%2F2021.05.25.445601.atom\"><span>PubMed</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-5-1\" id=\"ref-5\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" data-doi=\"10.1126/science.abd7331\" id=\"cit-2021.05.25.445601v3.5\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Hie</span>,  <span class=\"cit-name-given-names\">B.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhong</span>,  <span class=\"cit-name-given-names\">E.D.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Berger</span>,  <span class=\"cit-name-given-names\">B.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Bryson</span>,  <span class=\"cit-name-given-names\">B.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">Learning the language of viral evolution and escape</span>. <abbr class=\"cit-jnl-abbrev\">Science</abbr> <span class=\"cit-vol\">371</span>, <span class=\"cit-fpage\">284</span>–<span class=\"cit-lpage\">288</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DScience%26rft.stitle%253DScience%26rft.aulast%253DHie%26rft.auinit1%253DB.%26rft.volume%253D371%26rft.issue%253D6526%26rft.spage%253D284%26rft.epage%253D288%26rft.atitle%253DLearning%2Bthe%2Blanguage%2Bof%2Bviral%2Bevolution%2Band%2Bescape%26rft_id%253Dinfo%253Adoi%252F10.1126%252Fscience.abd7331%26rft_id%253Dinfo%253Apmid%252F33446556%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-ijlink\" href=\"/lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6Mzoic2NpIjtzOjU6InJlc2lkIjtzOjEyOiIzNzEvNjUyNi8yODQiO3M6NDoiYXRvbSI7czo0ODoiL2Jpb3J4aXYvZWFybHkvMjAyMS8wOC8yNi8yMDIxLjA1LjI1LjQ0NTYwMS5hdG9tIjt9czo4OiJmcmFnbWVudCI7czowOiIiO30=\"><span><span class=\"cit-reflinks-abstract\">Abstract</span><span class=\"cit-sep cit-reflinks-variant-name-sep\">/</span><span class=\"cit-reflinks-full-text\"><span class=\"free-full-text\">FREE </span>Full Text</span></span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-6-1\" id=\"ref-6\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" data-doi=\"10.1371/journal.pone.0238344\" id=\"cit-2021.05.25.445601v3.6\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Khan</span>,  <span class=\"cit-name-given-names\">M.I.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Khan</span>,  <span class=\"cit-name-given-names\">Z.A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Baig</span>,  <span class=\"cit-name-given-names\">M.H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Ahmad</span>,  <span class=\"cit-name-given-names\">I.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Farouk</span>,  <span class=\"cit-name-given-names\">A.-E.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Song</span>,  <span class=\"cit-name-given-names\">Y.G.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Dong</span>,  <span class=\"cit-name-given-names\">J.-J.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Comparative genome analysis of novel coronavirus (SARS-CoV-2) from different geographical locations and the effect of mutations on major target proteins: An in silico insight</span>. <abbr class=\"cit-jnl-abbrev\">PLoS One</abbr> <span class=\"cit-vol\">15</span>, <span class=\"cit-fpage\">e0238344</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DPLoS%2BOne%26rft.volume%253D15%26rft.spage%253De0238344%26rft_id%253Dinfo%253Adoi%252F10.1371%252Fjournal.pone.0238344%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\" href=\"/lookup/external-ref?access_num=10.1371/journal.pone.0238344&amp;link_type=DOI\"><span>CrossRef</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-7-1\" id=\"ref-7\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.7\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Kursa</span>,  <span class=\"cit-name-given-names\">M.B.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Rudnicki</span>,  <span class=\"cit-name-given-names\">W.R.</span></span> (<span class=\"cit-pub-date\">2010</span>). <span class=\"cit-article-title\">Feature Selection with theBorutaPackage</span>. <abbr class=\"cit-jnl-abbrev\">Journal of Statistical Software</abbr> <span class=\"cit-vol\">36</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-8-1\" id=\"ref-8\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.8\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Kursa</span>,  <span class=\"cit-name-given-names\">M.B.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Jankowski</span>,  <span class=\"cit-name-given-names\">A.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Rudnicki</span>,  <span class=\"cit-name-given-names\">W.R.</span></span> (<span class=\"cit-pub-date\">2010</span>). <span class=\"cit-article-title\">Boruta - A System for Feature Selection</span>. <abbr class=\"cit-jnl-abbrev\">Fund. Inform</abbr>. <span class=\"cit-vol\">101</span>, <span class=\"cit-fpage\">271</span>–<span class=\"cit-lpage\">285</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DFund.%2BInform%26rft.volume%253D101%26rft.spage%253D271%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-9-1\" id=\"ref-9\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" data-doi=\"10.1093/bioinformatics/btz682\" id=\"cit-2021.05.25.445601v3.9\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Lee</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Yoon</span>,  <span class=\"cit-name-given-names\">W.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kim</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kim</span>,  <span class=\"cit-name-given-names\">D.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kim</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">So</span>,  <span class=\"cit-name-given-names\">C.H.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kang</span>,  <span class=\"cit-name-given-names\">J.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">BioBERT: a pre-trained biomedical language representation model for biomedical text mining</span>. <abbr class=\"cit-jnl-abbrev\">Bioinformatics</abbr> <span class=\"cit-vol\">36</span>, <span class=\"cit-fpage\">1234</span>–<span class=\"cit-lpage\">1240</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DBioinformatics%26rft.volume%253D36%26rft.spage%253D1234%26rft_id%253Dinfo%253Adoi%252F10.1093%252Fbioinformatics%252Fbtz682%26rft_id%253Dinfo%253Apmid%252F31501885%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\" href=\"/lookup/external-ref?access_num=10.1093/bioinformatics/btz682&amp;link_type=DOI\"><span>CrossRef</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-medline\" href=\"/lookup/external-ref?access_num=31501885&amp;link_type=MED&amp;atom=%2Fbiorxiv%2Fearly%2F2021%2F08%2F26%2F2021.05.25.445601.atom\"><span>PubMed</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.10\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Letunic</span>,  <span class=\"cit-name-given-names\">I.</span></span> <abbr class=\"cit-jnl-abbrev\">iTOL: Interactive Tree Of Life</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-11-1\" id=\"ref-11\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" data-doi=\"S0092-8674(20)30877-1\" id=\"cit-2021.05.25.445601v3.11\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Li</span>,  <span class=\"cit-name-given-names\">Q.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Wu</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Nie</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">L.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Hao</span>,  <span class=\"cit-name-given-names\">H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Liu</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhao</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">Q.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Liu</span>,  <span class=\"cit-name-given-names\">H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Nie</span>,  <span class=\"cit-name-given-names\">L.</span></span>, <span class=\"cit-etal\">et al.</span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">The Impact of Mutations in SARS-CoV-2 Spike on Viral Infectivity and Antigenicity</span>. <abbr class=\"cit-jnl-abbrev\">Cell</abbr> <span class=\"cit-vol\">182</span>, <span class=\"cit-fpage\">1284</span>–<span class=\"cit-lpage\">1294</span>.e9.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DCell%26rft.volume%253D182%26rft.spage%253D1284%26rft_id%253Dinfo%253Adoi%252FS0092-8674%252820%252930877-1%26rft_id%253Dinfo%253Apmid%252FWOS%253A00057144%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\" href=\"/lookup/external-ref?access_num=S0092-8674(20)30877-1&amp;link_type=DOI\"><span>CrossRef</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-medline\" href=\"/lookup/external-ref?access_num=WOS:00057144&amp;link_type=MED&amp;atom=%2Fbiorxiv%2Fearly%2F2021%2F08%2F26%2F2021.05.25.445601.atom\"><span>PubMed</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-12-1\" id=\"ref-12\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" data-doi=\"10.1038/s41592-018-0308-4\" id=\"cit-2021.05.25.445601v3.12\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Linderman</span>,  <span class=\"cit-name-given-names\">G.C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Rachh</span>,  <span class=\"cit-name-given-names\">M.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Hoskins</span>,  <span class=\"cit-name-given-names\">J.G.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Steinerberger</span>,  <span class=\"cit-name-given-names\">S.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kluger</span>,  <span class=\"cit-name-given-names\">Y.</span></span> (<span class=\"cit-pub-date\">2019</span>). <span class=\"cit-article-title\">Fast interpolation-based t-SNE for improved visualization of single-cell RNA-seq data</span>. <abbr class=\"cit-jnl-abbrev\">Nat. Methods</abbr> <span class=\"cit-vol\">16</span>, <span class=\"cit-fpage\">243</span>–<span class=\"cit-lpage\">245</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DNat.%2BMethods%26rft.volume%253D16%26rft.spage%253D243%26rft_id%253Dinfo%253Adoi%252F10.1038%252Fs41592-018-0308-4%26rft_id%253Dinfo%253Apmid%252F30742040%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\" href=\"/lookup/external-ref?access_num=10.1038/s41592-018-0308-4&amp;link_type=DOI\"><span>CrossRef</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-medline\" href=\"/lookup/external-ref?access_num=30742040&amp;link_type=MED&amp;atom=%2Fbiorxiv%2Fearly%2F2021%2F08%2F26%2F2021.05.25.445601.atom\"><span>PubMed</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-13-1\" id=\"ref-13\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.13\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Liu</span>,  <span class=\"cit-name-given-names\">B.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Liu</span>,  <span class=\"cit-name-given-names\">K.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">L.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Bian</span>,  <span class=\"cit-name-given-names\">Y.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Huang</span>,  <span class=\"cit-name-given-names\">L.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">CoV-Seq, a New Tool for SARS-CoV-2 Genome Analysis and Visualization: Development and Usability Study</span>. <abbr class=\"cit-jnl-abbrev\">J. Med. Internet Res</abbr>. <span class=\"cit-vol\">22</span>, <span class=\"cit-fpage\">e22299</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DJ.%2BMed.%2BInternet%2BRes%26rft.volume%253D22%26rft.spage%253De22299%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.14\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Lopez-Rincon</span>,  <span class=\"cit-name-given-names\">A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Perez-Romero</span>,  <span class=\"cit-name-given-names\">C.A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Tonda</span>,  <span class=\"cit-name-given-names\">A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Mendoza-Maldonado</span>,  <span class=\"cit-name-given-names\">L.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Claassen</span>,  <span class=\"cit-name-given-names\">E.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Garssen</span>,  <span class=\"cit-name-given-names\">J.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Kraneveld</span>,  <span class=\"cit-name-given-names\">A.D.</span></span> <abbr class=\"cit-jnl-abbrev\">Design of Specific Primer Sets for the Detection of B.1.1.7, B.1.351, P.1, B.1.617.2 and B.1.1.519 Variants of SARS-CoV-2 using Artificial Intelligence</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-15-1\" id=\"ref-15\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.15\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Mandal</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Roychowdhury</span>,  <span class=\"cit-name-given-names\">T.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Bhattacharya</span>,  <span class=\"cit-name-given-names\">A.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">Pattern of genomic variation in SARS-CoV-2 (COVID-19) suggests restricted nonrandom changes: Analysis using Shewhart control charts</span>. <abbr class=\"cit-jnl-abbrev\">J. Biosci</abbr>. <span class=\"cit-vol\">46</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-16-1\" id=\"ref-16\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.16\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Mikolov</span>,  <span class=\"cit-name-given-names\">T.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Chen</span>,  <span class=\"cit-name-given-names\">K.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Corrado</span>,  <span class=\"cit-name-given-names\">G.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Dean</span>,  <span class=\"cit-name-given-names\">J.</span></span> (<span class=\"cit-pub-date\">2013</span>). <abbr class=\"cit-jnl-abbrev\">Efficient Estimation of Word Representations in Vector Space</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-17-1\" id=\"ref-17\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.17\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Mugnai</span>,  <span class=\"cit-name-given-names\">M.L.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Templeton</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Elber</span>,  <span class=\"cit-name-given-names\">R.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Thirumalai</span>,  <span class=\"cit-name-given-names\">D.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Role of Long-range Allosteric Communication in Determining the Stability and Disassembly of SARS-COV-2 in Complex with ACE2</span>. <abbr class=\"cit-jnl-abbrev\">bioRxiv</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-18-1\" id=\"ref-18\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.18\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Mushtaq</span>,  <span class=\"cit-name-given-names\">R.</span></span> (<span class=\"cit-pub-date\">2011</span>). <span class=\"cit-article-title\">Augmented Dickey Fuller Test</span>. <abbr class=\"cit-jnl-abbrev\">SSRN Electronic Journal</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-19-1\" id=\"ref-19\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.19\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Pan</span>,  <span class=\"cit-name-given-names\">Y.H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Wang</span>,  <span class=\"cit-name-given-names\">Y.H.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Liang</span>,  <span class=\"cit-name-given-names\">S.F.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Lee</span>,  <span class=\"cit-name-given-names\">K.T.</span></span> (<span class=\"cit-pub-date\">2011</span>). <span class=\"cit-article-title\">Fast computation of sample entropy and approximate entropy in biomedicine</span>. <abbr class=\"cit-jnl-abbrev\">Comput. Methods Programs Biomed</abbr>. <span class=\"cit-vol\">104</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-20-1\" id=\"ref-20\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.20\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth cit-collab\">paola</span> (<span class=\"cit-pub-date\">2021</span>). <abbr class=\"cit-jnl-abbrev\">Phylogenetic relationship of SARS-CoV-2 sequences from Amazonas with emerging Brazilian variants harboring mutations E484K and N501Y in the Spike protein</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-21-1\" id=\"ref-21\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.21\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Peacock</span>,  <span class=\"cit-name-given-names\">T.P.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Penrice-Randal</span>,  <span class=\"cit-name-given-names\">R.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Hiscox</span>,  <span class=\"cit-name-given-names\">J.A.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Barclay</span>,  <span class=\"cit-name-given-names\">W.S.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">SARS-CoV-2 one year on: evidence for ongoing viral adaptation</span>. <abbr class=\"cit-jnl-abbrev\">J. Gen. Virol</abbr>. <span class=\"cit-vol\">102</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-22-1\" id=\"ref-22\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.22\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Podobnik</span>,  <span class=\"cit-name-given-names\">B.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Stanley</span>,  <span class=\"cit-name-given-names\">H.E.</span></span> (<span class=\"cit-pub-date\">2007</span>). <abbr class=\"cit-jnl-abbrev\">Detrended Cross-Correlation Analysis: A New Method for Analyzing Two Non-stationary Time Series</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-23-1\" id=\"ref-23\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.23\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Prass</span>,  <span class=\"cit-name-given-names\">T.S.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Pumi</span>,  <span class=\"cit-name-given-names\">G.</span></span> (<span class=\"cit-pub-date\">2019</span>). <abbr class=\"cit-jnl-abbrev\">On the behavior of the DFA and DCCA in trend-stationary processes</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-24-1\" id=\"ref-24\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.24\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Radim Rehurek</span>,  <span class=\"cit-name-given-names\">P.S.</span></span> (<span class=\"cit-pub-date\">2010</span>). <span class=\"cit-article-title\">Software Framework for Topic Modelling with Large Corpora</span>. <abbr class=\"cit-jnl-abbrev\">In IN PROCEEDINGS OF THE LREC 2010 WORKSHOP ON NEW CHALLENGES FOR NLP FRAMEWORKS</abbr>,.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-25-1\" id=\"ref-25\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" data-doi=\"10.1371/journal.pone.0241535\" id=\"cit-2021.05.25.445601v3.25\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Rouchka</span>,  <span class=\"cit-name-given-names\">E.C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Chariker</span>,  <span class=\"cit-name-given-names\">J.H.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Chung</span>,  <span class=\"cit-name-given-names\">D.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Variant analysis of 1,040 SARS-CoV-2 genomes</span>. <abbr class=\"cit-jnl-abbrev\">PLoS One</abbr> <span class=\"cit-vol\">15</span>, <span class=\"cit-fpage\">e0241535</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DPLoS%2BOne%26rft.volume%253D15%26rft.spage%253De0241535%26rft_id%253Dinfo%253Adoi%252F10.1371%252Fjournal.pone.0241535%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\" href=\"/lookup/external-ref?access_num=10.1371/journal.pone.0241535&amp;link_type=DOI\"><span>CrossRef</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-26-1\" id=\"ref-26\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" data-doi=\"10.1371/journal.pone.0245584\" id=\"cit-2021.05.25.445601v3.26\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Shishir</span>,  <span class=\"cit-name-given-names\">T.A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Naser</span>,  <span class=\"cit-name-given-names\">I.B.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Faruque</span>,  <span class=\"cit-name-given-names\">S.M.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">In silico comparative genomics of SARS-CoV-2 to determine the source and diversity of the pathogen in Bangladesh</span>. <abbr class=\"cit-jnl-abbrev\">PLoS One</abbr> <span class=\"cit-vol\">16</span>, <span class=\"cit-fpage\">e0245584</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DPLoS%2BOne%26rft.volume%253D16%26rft.spage%253De0245584%26rft_id%253Dinfo%253Adoi%252F10.1371%252Fjournal.pone.0245584%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a><a class=\"cit-ref-sprinkles cit-ref-sprinkles-doi cit-ref-sprinkles-crossref\" href=\"/lookup/external-ref?access_num=10.1371/journal.pone.0245584&amp;link_type=DOI\"><span>CrossRef</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-27-1\" id=\"ref-27\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.27\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Shu</span>,  <span class=\"cit-name-given-names\">Y.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">McCauley</span>,  <span class=\"cit-name-given-names\">J.</span></span> (<span class=\"cit-pub-date\">2017</span>). <span class=\"cit-article-title\">GISAID: Global initiative on sharing all influenza data - from vision to reality</span>. <abbr class=\"cit-jnl-abbrev\">Euro Surveill</abbr>. <span class=\"cit-vol\">22</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-28-1\" id=\"ref-28\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.28\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Srivastava</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Banu</span>,  <span class=\"cit-name-given-names\">S.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Singh</span>,  <span class=\"cit-name-given-names\">P.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Sowpati</span>,  <span class=\"cit-name-given-names\">D.T.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Mishra</span>,  <span class=\"cit-name-given-names\">R.K.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">SARS-CoV-2 genomics: An Indian perspective on sequencing viral variants</span>. <abbr class=\"cit-jnl-abbrev\">J. Biosci</abbr>. <span class=\"cit-vol\">46</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-29-1\" id=\"ref-29\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.29\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Tomcala</span>,  <span class=\"cit-name-given-names\">J.</span></span> (<span class=\"cit-pub-date\">2018</span>). <span class=\"cit-article-title\">Time Series Entropies</span> <abbr class=\"cit-jnl-abbrev\">[R package TSEntropies version 0.9]</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-30-1\" id=\"ref-30\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.30\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Tomčala</span>,  <span class=\"cit-name-given-names\">J.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">New Fast ApEn and SampEn Entropy Algorithms Implementation and Their Application to Supercomputer Power Consumption</span>. <abbr class=\"cit-jnl-abbrev\">Entropy</abbr> <span class=\"cit-vol\">22</span>, <span class=\"cit-fpage\">863</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DEntropy%26rft.volume%253D22%26rft.spage%253D863%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-31-1\" id=\"ref-31\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.31\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Wang</span>,  <span class=\"cit-name-given-names\">B.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhao</span>,  <span class=\"cit-name-given-names\">D.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Lioma</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Li</span>,  <span class=\"cit-name-given-names\">Q.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">P.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Simonsen</span>,  <span class=\"cit-name-given-names\">J.G.</span></span> (<span class=\"cit-pub-date\">2019</span>). <abbr class=\"cit-jnl-abbrev\">Encoding word order in complex embeddings</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-32-1\" id=\"ref-32\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.32\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Wang</span>,  <span class=\"cit-name-given-names\">R.Y.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Guo</span>,  <span class=\"cit-name-given-names\">T.Q.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Li</span>,  <span class=\"cit-name-given-names\">L.G.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Jiao</span>,  <span class=\"cit-name-given-names\">J.Y.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Wang</span>,  <span class=\"cit-name-given-names\">L.Y.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Predictions of COVID-19 Infection Severity Based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 through Machine Learning of Genetic Data</span>. <abbr class=\"cit-jnl-abbrev\">2020 IEEE 8th International Conference on Computer Science and Network Technology (ICCSNT)</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.33\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth cit-collab\">WHO Coronavirus (COVID-19) Dashboard</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-34-1\" id=\"ref-34\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.34\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Wolf</span>,  <span class=\"cit-name-given-names\">T.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Debut</span>,  <span class=\"cit-name-given-names\">L.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Sanh</span>,  <span class=\"cit-name-given-names\">V.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Chaumond</span>,  <span class=\"cit-name-given-names\">J.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Delangue</span>,  <span class=\"cit-name-given-names\">C.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Moi</span>,  <span class=\"cit-name-given-names\">A.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Cistac</span>,  <span class=\"cit-name-given-names\">P.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Rault</span>,  <span class=\"cit-name-given-names\">T.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Louf</span>,  <span class=\"cit-name-given-names\">R.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Funtowicz</span>,  <span class=\"cit-name-given-names\">M.</span></span>, <span class=\"cit-etal\">et al.</span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Transformers: State-of-the-Art Natural Language Processing</span>. <abbr class=\"cit-jnl-abbrev\">Proceedings of the 2020 Conference on Empirical Methods in Natural Language Processing: System Demonstrations</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-35-1\" id=\"ref-35\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.35\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Wu</span>,  <span class=\"cit-name-given-names\">F.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Yang</span>,  <span class=\"cit-name-given-names\">R.</span></span>, <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">C.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Zhang</span>,  <span class=\"cit-name-given-names\">L.</span></span> (<span class=\"cit-pub-date\">2021</span>). <span class=\"cit-article-title\">A deep learning framework combined with word embedding to identify DNA replication origins</span>. <abbr class=\"cit-jnl-abbrev\">Sci. Rep</abbr>. <span class=\"cit-vol\">11</span>, <span class=\"cit-fpage\">844</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DSci.%2BRep%26rft.volume%253D11%26rft.spage%253D844%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-36-1\" id=\"ref-36\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.36\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Yilmaz</span>,  <span class=\"cit-name-given-names\">A.</span></span> (<span class=\"cit-pub-date\">2020</span>). <span class=\"cit-article-title\">Assessment of Mutation Susceptibility in DNA Sequences with Word Vectors</span>. <abbr class=\"cit-jnl-abbrev\">Journal of Intelligent Systems: Theory and Applications</abbr> <span class=\"cit-vol\">3</span>, <span class=\"cit-fpage\">1</span>–<span class=\"cit-lpage\">6</span>.</cite></div><div class=\"cit-extra\"><a class=\"cit-ref-sprinkles cit-ref-sprinkles-openurl cit-ref-sprinkles-open-url\" href=\"{openurl}?query=rft.jtitle%253DJournal%2Bof%2BIntelligent%2BSystems%253A%2BTheory%2Band%2BApplications%26rft.volume%253D3%26rft.spage%253D1%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx\"><span>OpenUrl</span></a></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><a class=\"rev-xref-ref\" href=\"#xref-ref-37-1\" id=\"ref-37\" title=\"View reference  in text\">↵</a><div class=\"cit ref-cit ref-journal\" id=\"cit-2021.05.25.445601v3.37\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth\"><span class=\"cit-name-surname\">Yin</span>,  <span class=\"cit-name-given-names\">Z.</span></span>, and <span class=\"cit-auth\"><span class=\"cit-name-surname\">Shen</span>,  <span class=\"cit-name-given-names\">Y.</span></span> (<span class=\"cit-pub-date\">2018</span>). <span class=\"cit-article-title\">On the Dimensionality of Word Embedding</span>. <abbr class=\"cit-jnl-abbrev\">Adv. Neural Inf. Process. Syst</abbr>. <span class=\"cit-vol\">31</span>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.38\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth cit-collab\">(2020a)</span>. <span class=\"cit-article-title\">Time Series Analysis and Computational Finance</span> <abbr class=\"cit-jnl-abbrev\">[R package tseries version 0.10-48]</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li><li><span class=\"ref-label ref-label-empty\"></span><div class=\"cit ref-cit ref-journal no-rev-xref\" id=\"cit-2021.05.25.445601v3.39\"><div class=\"cit-metadata\"><cite><span class=\"cit-auth cit-collab\">(2020b)</span>. <span class=\"cit-article-title\">Detrended Fluctuation and Detrended Cross-Correlation Analysis</span> <abbr class=\"cit-jnl-abbrev\">[R package DCCA version 0.1.1]</abbr>.</cite></div><div class=\"cit-extra\"></div></div></li></ol></div><span class=\"highwire-journal-article-marker-end\"></span></div><span class=\"related-urls\"></span></div></div> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div></div> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-disqus-comment\">\n",
              "<div class=\"pane-content\">\n",
              "<div id=\"disqus_thread\"><noscript><p><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://biorxivstage.disqus.com/?url=https%3A%2F%2Fwww.biorxiv.org%2Fcontent%2F10.1101%2F2021.05.25.445601v3\">View the discussion thread.</a></p></noscript></div> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-back-to-top\">\n",
              "<div class=\"pane-content\">\n",
              "<a class=\"back-to-top\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"#page\"><span class=\"icon-chevron-up\"></span> Back to top</a> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "<div class=\"sidebar-right-wrapper grid-10 omega\">\n",
              "<div class=\"panel-panel panel-region-sidebar-right\">\n",
              "<div class=\"inside\"><div class=\"panel-pane pane-highwire-node-pager\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"pager highwire-pager pager-mini clearfix highwire-node-pager highwire-article-pager\"><span class=\"pager-prev\"><a class=\"pager-link-prev link-icon\" href=\"/content/10.1101/2021.05.25.445523v2\" rel=\"prev\" title=\"Impact of glycosylation on a broad-spectrum vaccine against SARS-CoV-2\"><span class=\"icon-circle-arrow-left\"></span> <span class=\"title\">Previous</span></a></span><span class=\"pager-next\"><a class=\"pager-link-next link-icon-right link-icon\" href=\"/content/10.1101/2021.08.25.457533v1\" rel=\"next\" title=\"The ultrafast snap of a finger is mediated by skin friction\"><span class=\"title\">Next</span> <span class=\"icon-circle-arrow-right\"></span></a></span></div> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-custom pane-1\">\n",
              "<div class=\"pane-content\">\n",
              "    Posted August 26, 2021.  </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-panels-mini pane-biorxiv-art-tools\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"highwire-2col-stacked panel-display\" id=\"mini-panel-biorxiv_art_tools\">\n",
              "<div class=\"panel-row-wrapper clearfix\">\n",
              "<div class=\"content-left-wrapper content-column\">\n",
              "<div class=\"panel-panel panel-region-content-left\">\n",
              "<div class=\"inside\"><div class=\"panel-pane pane-highwire-variant-link\">\n",
              "<div class=\"pane-content\">\n",
              "<a class=\"article-dl-pdf-link link-icon\" href=\"/content/10.1101/2021.05.25.445601v3.full.pdf\" target=\"_self\"><span class=\"icon-external-link-sign\"></span> <span class=\"title\">Download PDF</span></a> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-biorxiv-supplementary-fragment\">\n",
              "<div class=\"pane-content\">\n",
              "<p><a href=\"/content/10.1101/2021.05.25.445601v3.supplementary-material?versioned=true\"><i class=\"icon-file\"></i>Supplementary Material </a></p> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "<div class=\"content-right-wrapper content-column\">\n",
              "<div class=\"panel-panel panel-region-content-right\">\n",
              "<div class=\"inside\"><div class=\"panel-pane pane-minipanel-dialog-link pane-biorxiv-art-email\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"minipanel-dialog-wrapper\"><div class=\"minipanel-dialog-link-link\"><a class=\"minipanel-dialog-link-trigger\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/\" oncontextmenu=\"javascript: return false;\" title=\"Email this Article\"><i class=\"icon-envelope\"></i> Email</a></div><div class=\"minipanel-dialog-link-mini\" style=\"display:none\"><div class=\"panel-display panel-1col clearfix\" id=\"mini-panel-biorxiv_art_email\">\n",
              "<div class=\"panel-panel panel-col\">\n",
              "<div><div class=\"panel-pane pane-block pane-forward-form pane-forward\">\n",
              "<div class=\"pane-content\">\n",
              "<form accept-charset=\"UTF-8\" action=\"/content/10.1101/2021.05.25.445601v3.full\" id=\"forward-form\" method=\"post\"><div><div class=\"form-item form-item-label-before form-type-item\" id=\"edit-instructions\">\n",
              "<p>Thank you for your interest in spreading the word about bioRxiv.</p><p>NOTE: Your email address is requested solely to identify you as the sender of this article.</p>\n",
              "</div>\n",
              "<div class=\"form-item form-item-label-before form-type-textfield form-item-email\">\n",
              "<label for=\"edit-email\">Your Email <span class=\"form-required\" title=\"This field is required.\">*</span></label>\n",
              "<input class=\"form-text required\" id=\"edit-email\" maxlength=\"256\" name=\"email\" size=\"58\" type=\"text\" value=\"\"/>\n",
              "</div>\n",
              "<div class=\"form-item form-item-label-before form-type-textfield form-item-name\">\n",
              "<label for=\"edit-name\">Your Name <span class=\"form-required\" title=\"This field is required.\">*</span></label>\n",
              "<input class=\"form-text required\" id=\"edit-name\" maxlength=\"256\" name=\"name\" size=\"58\" type=\"text\" value=\"\"/>\n",
              "</div>\n",
              "<div class=\"form-item form-item-label-before form-type-textarea form-item-recipients\">\n",
              "<label for=\"edit-recipients\">Send To <span class=\"form-required\" title=\"This field is required.\">*</span></label>\n",
              "<div class=\"form-textarea-wrapper resizable\"><textarea class=\"form-textarea required\" cols=\"50\" id=\"edit-recipients\" name=\"recipients\" rows=\"5\"></textarea></div>\n",
              "<div class=\"description\">Enter multiple addresses on separate lines or separate them with commas.</div>\n",
              "</div>\n",
              "<div class=\"form-item form-item-label-before form-type-item\" id=\"edit-page\">\n",
              "<label for=\"edit-page\">You are going to email the following </label>\n",
              "<a class=\"active\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/content/10.1101/2021.05.25.445601v3\">Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning</a>\n",
              "</div>\n",
              "<div class=\"form-item form-item-label-before form-type-item\" id=\"edit-subject\">\n",
              "<label for=\"edit-subject\">Message Subject </label>\n",
              " (Your Name) has forwarded a page to you from bioRxiv\n",
              "</div>\n",
              "<div class=\"form-item form-item-label-before form-type-item\" id=\"edit-body\">\n",
              "<label for=\"edit-body\">Message Body </label>\n",
              " (Your Name) thought you would like to see this page from the bioRxiv website.\n",
              "</div>\n",
              "<div class=\"form-item form-item-label-before form-type-textarea form-item-message\">\n",
              "<label for=\"edit-message--2\">Your Personal Message </label>\n",
              "<div class=\"form-textarea-wrapper resizable\"><textarea class=\"form-textarea\" cols=\"50\" id=\"edit-message--2\" name=\"message\" rows=\"10\"></textarea></div>\n",
              "</div>\n",
              "<input name=\"path\" type=\"hidden\" value=\"node/2125191\"/>\n",
              "<input name=\"path_cid\" type=\"hidden\" value=\"\"/>\n",
              "<input name=\"forward_footer\" type=\"hidden\" value=\" \"/>\n",
              "<input name=\"form_build_id\" type=\"hidden\" value=\"form-M42UdW_KPHtrmp0JJEJt9TMaOECXs8X-JzC357FpLlA\"/>\n",
              "<input name=\"form_id\" type=\"hidden\" value=\"forward_form\"/>\n",
              "<fieldset class=\"captcha form-wrapper\"><legend><span class=\"fieldset-legend\">CAPTCHA</span></legend><div class=\"fieldset-wrapper\"><div class=\"fieldset-description\">This question is for testing whether or not you are a human visitor and to prevent automated spam submissions.</div><input name=\"captcha_sid\" type=\"hidden\" value=\"345930021\"/>\n",
              "<input name=\"captcha_token\" type=\"hidden\" value=\"be6f1827934e713c99e2ba0c3456dccf\"/>\n",
              "<input name=\"captcha_response\" type=\"hidden\" value=\"Google no captcha\"/>\n",
              "<div class=\"g-recaptcha\" data-sitekey=\"6LfnJVIUAAAAAE-bUOMg0MJGki4lqSvDmhJp19fN\" data-theme=\"light\" data-type=\"image\"></div></div></fieldset>\n",
              "<div class=\"form-actions form-wrapper\" id=\"edit-actions\"><input class=\"form-submit\" id=\"edit-submit\" name=\"op\" type=\"submit\" value=\"Send Message\"/></div></div></form> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div></div> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-share-link highwire_clipboard_link_ajax\" id=\"shareit\">\n",
              "<div class=\"pane-content\">\n",
              "<a class=\"link-icon\" href=\"/\"><span class=\"icon-share-alt\"></span> <span class=\"title\">Share</span></a> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-panels-mini pane-biorxiv-share highwire_clipboard_form_ajax_shareit\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"panel-display omega-12-onecol\" id=\"mini-panel-biorxiv_share\">\n",
              "<div class=\"panel-panel grid-12 panel-region-preface\">\n",
              "<div class=\"inside\"><div class=\"panel-pane pane-highwire-article-citation\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"highwire-article-citation highwire-citation-type-highwire-article node2125191--3\" data-apath=\"/biorxiv/early/2021/08/26/2021.05.25.445601.atom\" data-node-nid=\"2125191\" data-pisa=\"biorxiv;2021.05.25.445601v3\" data-pisa-master=\"biorxiv;2021.05.25.445601\" data-seqnum=\"2125191\" id=\"node2125191--4826694706\"><div class=\"highwire-cite highwire-cite-highwire-article highwire-citation-biorxiv-article-pap-list clearfix\">\n",
              "<div class=\"highwire-cite-title\">\n",
              "<div class=\"highwire-cite-title\">Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning</div> </div>\n",
              "<div class=\"highwire-cite-authors\"><span class=\"highwire-citation-authors\"><span class=\"highwire-citation-author first\" data-delta=\"0\"><span class=\"nlm-given-names\">Sargun</span> <span class=\"nlm-surname\">Nagpal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"1\"><span class=\"nlm-given-names\">Ridam</span> <span class=\"nlm-surname\">Pal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"2\"><span class=\"nlm-surname\">Ashima</span></span>, <span class=\"highwire-citation-author\" data-delta=\"3\"><span class=\"nlm-given-names\">Ananya</span> <span class=\"nlm-surname\">Tyagi</span></span>, <span class=\"highwire-citation-author\" data-delta=\"4\"><span class=\"nlm-given-names\">Sadhana</span> <span class=\"nlm-surname\">Tripathi</span></span>, <span class=\"highwire-citation-author\" data-delta=\"5\"><span class=\"nlm-given-names\">Aditya</span> <span class=\"nlm-surname\">Nagori</span></span>, <span class=\"highwire-citation-author\" data-delta=\"6\"><span class=\"nlm-given-names\">Saad</span> <span class=\"nlm-surname\">Ahmad</span></span>, <span class=\"highwire-citation-author\" data-delta=\"7\"><span class=\"nlm-given-names\">Hara Prasad</span> <span class=\"nlm-surname\">Mishra</span></span>, <span class=\"highwire-citation-author\" data-delta=\"8\"><span class=\"nlm-given-names\">Rintu</span> <span class=\"nlm-surname\">Kutum</span></span>, <span class=\"highwire-citation-author\" data-delta=\"9\"><span class=\"nlm-given-names\">Tavpritesh</span> <span class=\"nlm-surname\">Sethi</span></span></span></div>\n",
              "<div class=\"highwire-cite-metadata\"><span class=\"highwire-cite-metadata-journal highwire-cite-metadata\">bioRxiv </span><span class=\"highwire-cite-metadata-pages highwire-cite-metadata\">2021.05.25.445601; </span><span class=\"highwire-cite-metadata-doi highwire-cite-metadata\"><span class=\"doi_label\">doi:</span> https://doi.org/10.1101/2021.05.25.445601 </span></div>\n",
              "</div>\n",
              "</div> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "<div class=\"panel-panel grid-12 panel-region-content\">\n",
              "<div class=\"inside\"><div class=\"panel-pane pane-highwire-article-clipboard-copy\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"clipboard-copy\">\n",
              "<span class=\"label-url\">\n",
              "<label for=\"dynamic\">Share This Article:</label>\n",
              "</span>\n",
              "<span class=\"input-text-url\">\n",
              "<input id=\"dynamic\" size=\"50\" type=\"text\" value=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3\"/>\n",
              "</span>\n",
              "<span class=\"copy-button button\">\n",
              "<button class=\"clipboardjs-button\" data-clipboard-alert-style=\"tooltip\" data-clipboard-alert-text=\"Copied!\" data-clipboard-target=\"#dynamic\" id=\"copy-dynamic\">Copy</button>\n",
              "</span>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "<div class=\"panel-panel grid-12 panel-region-postscript\">\n",
              "<div class=\"inside\"><div class=\"panel-pane pane-service-links\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"service-links\"><a class=\"service-links-digg\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://digg.com/submit?phase=2&amp;url=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;title=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"digg\" rel=\"nofollow\" title=\"Digg this post on digg.com\"><img alt=\"Digg logo\" src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/digg.png\"/></a> <a class=\"service-links-reddit\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://reddit.com/submit?url=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;title=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"reddit\" rel=\"nofollow\" title=\"Submit this post on reddit.com\"><img alt=\"Reddit logo\" src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/reddit.png\"/></a> <a class=\"service-links-twitter\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://twitter.com/share?url=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;text=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"twitter\" rel=\"nofollow\" title=\"Share this on Twitter\"><img alt=\"Twitter logo\" src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/twitter.png\"/></a> <a class=\"service-links-facebook\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://www.facebook.com/sharer.php?u=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;t=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"facebook\" rel=\"nofollow\" title=\"Share on Facebook\"><img alt=\"Facebook logo\" src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/fb-blue.png\"/></a> <a class=\"service-links-google\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://www.google.com/bookmarks/mark?op=add&amp;bkmk=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;title=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"google\" rel=\"nofollow\" title=\"Bookmark this post on Google\"><img alt=\"Google logo\" src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/google-32.png\"/></a> <a class=\"service-links-mendeley\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://www.mendeley.com/import/?url=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;title=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning\" id=\"mendeley\" rel=\"nofollow\" title=\"Share on Mendeley\"><img alt=\"Mendeley logo\" src=\"https://www.biorxiv.org/sites/all/modules/highwire/highwire/images/mendeley.png\"/></a></div> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-minipanel-dialog-link pane-biorxiv-cite-tool\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"minipanel-dialog-wrapper\"><div class=\"minipanel-dialog-link-link\"><a class=\"minipanel-dialog-link-trigger link-icon\" href=\"/\" oncontextmenu=\"javascript: return false;\" title=\"Citation Tools\"><span class=\"icon-globe\"></span> <span class=\"title\">Citation Tools</span></a></div><div class=\"minipanel-dialog-link-mini\" style=\"display:none\"><div class=\"panel-display panel-1col clearfix\" id=\"mini-panel-biorxiv_cite_tool\">\n",
              "<div class=\"panel-panel panel-col\">\n",
              "<div><div class=\"panel-pane pane-highwire-citation-export\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"highwire-citation-export\">\n",
              "<div class=\"highwire-citation-info\">\n",
              "<div class=\"highwire-article-citation highwire-citation-type-highwire-article cite-tool-node2125191--5\" data-apath=\"/biorxiv/early/2021/08/26/2021.05.25.445601.atom\" data-node-nid=\"2125191\" data-pisa=\"biorxiv;2021.05.25.445601v3\" data-pisa-master=\"biorxiv;2021.05.25.445601\" data-seqnum=\"2125191\" id=\"cite-tool-node2125191--6600659653\"><div class=\"highwire-cite highwire-cite-highwire-article highwire-citation-biorxiv-article-pap-list clearfix\">\n",
              "<div class=\"highwire-cite-title\">\n",
              "<div class=\"highwire-cite-title\">Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning</div> </div>\n",
              "<div class=\"highwire-cite-authors\"><span class=\"highwire-citation-authors\"><span class=\"highwire-citation-author first\" data-delta=\"0\"><span class=\"nlm-given-names\">Sargun</span> <span class=\"nlm-surname\">Nagpal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"1\"><span class=\"nlm-given-names\">Ridam</span> <span class=\"nlm-surname\">Pal</span></span>, <span class=\"highwire-citation-author\" data-delta=\"2\"><span class=\"nlm-surname\">Ashima</span></span>, <span class=\"highwire-citation-author\" data-delta=\"3\"><span class=\"nlm-given-names\">Ananya</span> <span class=\"nlm-surname\">Tyagi</span></span>, <span class=\"highwire-citation-author\" data-delta=\"4\"><span class=\"nlm-given-names\">Sadhana</span> <span class=\"nlm-surname\">Tripathi</span></span>, <span class=\"highwire-citation-author\" data-delta=\"5\"><span class=\"nlm-given-names\">Aditya</span> <span class=\"nlm-surname\">Nagori</span></span>, <span class=\"highwire-citation-author\" data-delta=\"6\"><span class=\"nlm-given-names\">Saad</span> <span class=\"nlm-surname\">Ahmad</span></span>, <span class=\"highwire-citation-author\" data-delta=\"7\"><span class=\"nlm-given-names\">Hara Prasad</span> <span class=\"nlm-surname\">Mishra</span></span>, <span class=\"highwire-citation-author\" data-delta=\"8\"><span class=\"nlm-given-names\">Rintu</span> <span class=\"nlm-surname\">Kutum</span></span>, <span class=\"highwire-citation-author\" data-delta=\"9\"><span class=\"nlm-given-names\">Tavpritesh</span> <span class=\"nlm-surname\">Sethi</span></span></span></div>\n",
              "<div class=\"highwire-cite-metadata\"><span class=\"highwire-cite-metadata-journal highwire-cite-metadata\">bioRxiv </span><span class=\"highwire-cite-metadata-pages highwire-cite-metadata\">2021.05.25.445601; </span><span class=\"highwire-cite-metadata-doi highwire-cite-metadata\"><span class=\"doi_label\">doi:</span> https://doi.org/10.1101/2021.05.25.445601 </span></div>\n",
              "</div>\n",
              "</div> </div>\n",
              "<div class=\"highwire-citation-formats\">\n",
              "<h2>Citation Manager Formats</h2>\n",
              "<div class=\"highwire-citation-formats-links\">\n",
              "<span class=\"Z3988\" title=\"ctx_ver=Z39.88-2004&amp;rft_val_fmt=info%3Aofi%2Ffmt%3Akev%3Amtx%3Ajournal&amp;rft.spage&amp;rft.epage&amp;rft.atitle=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning&amp;rft.volume&amp;rft.issue&amp;rft.date=2021-01-01%2000%3A00%3A00&amp;rft.stitle&amp;rft.jtitle=bioRxiv&amp;rft.au=Nagpal%2C+Sargun&amp;rft.au=Pal%2C+Ridam&amp;rft.au=Ashima%2C+&amp;rft.au=Tyagi%2C+Ananya&amp;rft.au=Tripathi%2C+Sadhana&amp;rft.au=Nagori%2C+Aditya&amp;rft.au=Ahmad%2C+Saad&amp;rft.au=Mishra%2C+Hara+Prasad&amp;rft.au=Kutum%2C+Rintu&amp;rft.au=Sethi%2C+Tavpritesh\"></span><ul class=\"hw-citation-links inline button button-alt button-grid clearfix\"><li class=\"bibtext first\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/bibtext\">BibTeX</a></li><li class=\"bookends\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/bookends\">Bookends</a></li><li class=\"easybib\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/easybib\">EasyBib</a></li><li class=\"endnote-tagged\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/endnote-tagged\">EndNote (tagged)</a></li><li class=\"endnote-8-xml\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/endnote-8-xml\">EndNote 8 (xml)</a></li><li class=\"medlars\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/medlars\">Medlars</a></li><li class=\"mendeley\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/mendeley\">Mendeley</a></li><li class=\"papers\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/papers\">Papers</a></li><li class=\"refworks-tagged\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/refworks-tagged\">RefWorks Tagged</a></li><li class=\"reference-manager\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/reference-manager\">Ref Manager</a></li><li class=\"ris\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/ris\">RIS</a></li><li class=\"zotero last\"><a class=\"hw-download-citation-link\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/highwire/citation/2125191/zotero\">Zotero</a></li></ul> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div></div> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div> <!-- /.panel-row-wrapper -->\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-service-links\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"service-links\"><div class=\"item-list\"><ul><li class=\"first\"><a class=\"twitter-share-button service-links-twitter-widget\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://twitter.com/share?url=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;count=horizontal&amp;via=&amp;text=Genomic%20Surveillance%20of%20COVID-19%20Variants%20with%20Language%20Models%20and%20Machine%20Learning&amp;counturl=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" id=\"twitter_widget\" rel=\"nofollow\" title=\"Tweet This\"><span class=\"element-invisible\">Tweet Widget</span></a></li><li><a class=\"service-links-facebook-like\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://www.facebook.com/plugins/like.php?href=https%3A//www.biorxiv.org/content/10.1101/2021.05.25.445601v3&amp;layout=button_count&amp;show_faces=false&amp;action=like&amp;colorscheme=light&amp;width=100&amp;height=21&amp;font=&amp;locale=\" id=\"facebook_like\" rel=\"nofollow\" title=\"I Like it\"><span class=\"element-invisible\">Facebook Like</span></a></li><li class=\"last\"><a class=\"service-links-google-plus-one\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"https://www.biorxiv.org/content/10.1101/2021.05.25.445601v3\" id=\"google_plus_one\" rel=\"nofollow\" title=\"Plus it\"><span class=\"element-invisible\">Google Plus One</span></a></li></ul></div></div> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-article-collections\">\n",
              "<h2 class=\"pane-title\">Subject Area</h2>\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"highwire-list-wrapper highwire-article-collections\"><div class=\"highwire-list\"><ul class=\"highwire-article-collection-term-list\"><li class=\"first last odd\"><span class=\"highwire-article-collection-term\"><a class=\"highlight\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/bioinformatics\">Bioinformatics<i class=\"icon-caret-right\"></i>\n",
              "</a></span></li></ul></div></div> </div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-panels-mini pane-biorxiv-subject-collections block-style-col2\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"panel-flexible panels-flexible-new clearfix\" id=\"mini-panel-biorxiv_subject_collections\">\n",
              "<div class=\"panel-flexible-inside panels-flexible-new-inside\">\n",
              "<div class=\"panels-flexible-region panels-flexible-region-new-center panels-flexible-region-first panels-flexible-region-last\">\n",
              "<div class=\"inside panels-flexible-region-inside panels-flexible-region-new-center-inside panels-flexible-region-inside-first panels-flexible-region-inside-last\">\n",
              "<div class=\"panel-pane pane-snippet\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"snippet biorxiv-subject-areas-table-title\" id=\"biorxiv-subject-areas-table-title\">\n",
              "<div class=\"snippet-content\">\n",
              "<b>Subject Areas</b> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-snippet\">\n",
              "<div class=\"pane-content\">\n",
              "<div class=\"snippet biorxiv-subject-areas-view-papers\" id=\"biorxiv-subject-areas-view-papers\">\n",
              "<div class=\"snippet-content\">\n",
              "<a href=\"/content/early/recent\"><strong>All Articles</strong></a> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "<div class=\"panel-separator\"></div><div class=\"panel-pane pane-highwire-subject-collections\">\n",
              "<div class=\"pane-content\">\n",
              "<ul class=\"collection highwire-list-expand\" id=\"collection\"><li class=\"outer collection depth-2 child first\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/animal-behavior-and-cognition\">Animal Behavior and Cognition</a> <span class=\"article-count\">(3426)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/biochemistry\">Biochemistry</a> <span class=\"article-count\">(7191)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/bioengineering\">Bioengineering</a> <span class=\"article-count\">(5204)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/bioinformatics\">Bioinformatics</a> <span class=\"article-count\">(19894)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/biophysics\">Biophysics</a> <span class=\"article-count\">(9797)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/cancer-biology\">Cancer Biology</a> <span class=\"article-count\">(7530)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/cell-biology\">Cell Biology</a> <span class=\"article-count\">(11031)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/clinical-trials\">Clinical Trials</a> <span class=\"article-count\">(138)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/developmental-biology\">Developmental Biology</a> <span class=\"article-count\">(6299)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/ecology\">Ecology</a> <span class=\"article-count\">(9765)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/epidemiology\">Epidemiology</a> <span class=\"article-count\">(2064)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/evolutionary-biology\">Evolutionary Biology</a> <span class=\"article-count\">(13113)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/genetics\">Genetics</a> <span class=\"article-count\">(9270)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/genomics\">Genomics</a> <span class=\"article-count\">(12382)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/immunology\">Immunology</a> <span class=\"article-count\">(7523)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/microbiology\">Microbiology</a> <span class=\"article-count\">(18640)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/molecular-biology\">Molecular Biology</a> <span class=\"article-count\">(7295)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/neuroscience\">Neuroscience</a> <span class=\"article-count\">(40254)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/paleontology\">Paleontology</a> <span class=\"article-count\">(292)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/pathology\">Pathology</a> <span class=\"article-count\">(1209)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/pharmacology-and-toxicology\">Pharmacology and Toxicology</a> <span class=\"article-count\">(2093)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/physiology\">Physiology</a> <span class=\"article-count\">(3089)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/plant-biology\">Plant Biology</a> <span class=\"article-count\">(6723)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/scientific-communication-and-education\">Scientific Communication and Education</a> <span class=\"article-count\">(1255)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/synthetic-biology\">Synthetic Biology</a> <span class=\"article-count\">(1865)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/systems-biology\">Systems Biology</a> <span class=\"article-count\">(5235)</span></div></li>\n",
              "<li class=\"outer collection depth-2 child last\"><div class=\"data-wrapper\"><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"/collection/zoology\">Zoology</a> <span class=\"article-count\">(1068)</span></div></li>\n",
              "</ul> </div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div> <!-- /.panel-row-wrapper -->\n",
              "</div>\n",
              "</div>\n",
              "</div>\n",
              "</div> </div>\n",
              "</div> </div>\n",
              "</section>\n",
              "</div> <div class=\"region region-page-bottom\" id=\"region-page-bottom\">\n",
              "<div class=\"region-inner region-page-bottom-inner\">\n",
              "</div>\n",
              "</div><script src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__VNH5GD8Zz7g6_hGCOZjVjdMndxxC6naiExSSJKX3k_A__kzmEdtbqCsJx5Z9yg2Qy0PptB__ufXm-TxUOl0KZzUw__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\" type=\"text/javascript\"></script>\n",
              "<script defer=\"defer\" src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__2WRbxlwOW0MEUc_hSWU5MBepQg6Lch6O5SZwefpJ6IE__HCL0YQJqLkOhrLPZZYGqosGvtFsEHMGghHIkSx4y9vA__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\" type=\"text/javascript\"></script>\n",
              "<script src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__N7ERJBYsOWyRnJgyoM125_Aiez2MOJGaUofG1JdWWBg__2cpwCQ7-xzTVeVvg_KOzwA1jka23oWApDPpgjoZKDCY__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\" type=\"text/javascript\"></script>\n",
              "<script src=\"//d33xdlntwy0kbs.cloudfront.net/cshl_custom.js\" type=\"text/javascript\"></script>\n",
              "<script src=\"https://www.biorxiv.org/sites/default/files/advagg_js/js__BmqjBnkz3MgYCAoc25s1lDRMEjLhC3mEPVonUFIHi08__Unwv5-ZIuHBfFwytsjEx1niBVJ7n1T4lPws7VrkHXM4__qq3UvOzgsLmitgSBu1FkrqF2RTzrXSZPGDY9ypzjud4.js\" type=\"text/javascript\"></script>\n",
              "<script type=\"text/javascript\">\n",
              "<!--//--><![CDATA[//><!--\n",
              "function euCookieComplianceLoadScripts() {}\n",
              "//--><!]]>\n",
              "</script>\n",
              "<script type=\"text/javascript\">\n",
              "<!--//--><![CDATA[//><!--\n",
              "var eu_cookie_compliance_cookie_name = \"\";\n",
              "//--><!]]>\n",
              "</script>\n",
              "</body>\n",
              "</html>"
            ]
          },
          "metadata": {},
          "execution_count": 41
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "results = soup.find_all(['h1', 'p'])\n",
        "results"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_aDycgQ3W0b7",
        "outputId": "e97c317c-dcd5-48fa-aabb-6144e366a37a"
      },
      "execution_count": 42,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<h1 class=\"highwire-cite-title\" id=\"page-title\">Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning</h1>,\n",
              " <p id=\"p-4\">The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed <em>Strainflow</em>, to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our <em>Strainflow</em> model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called <em>Dimensions of Concern</em> (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.</p>,\n",
              " <p><strong>Highlights</strong></p>,\n",
              " <p id=\"p-5\">We developed a genomic surveillance model for SARS-CoV-2 genome sequences, <em>Strainflow</em>, where sequences were treated as documents with words (codons) to learn the codon context of 0.9 million spike genes using the skip-gram algorithm.</p>,\n",
              " <p id=\"p-6\">Time series analysis of the information content (Entropy) of the latent dimensions learned by <em>Strainflow</em> shows a leading relationship with the monthly COVID-19 cases for seven countries (e.g., USA, Japan, India, and others).</p>,\n",
              " <p id=\"p-7\">Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads.</p>,\n",
              " <p id=\"p-8\">The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants.</p>,\n",
              " <p id=\"p-11\">COVID-19 is reported to have claimed 4.37 million lives as of Aug 17, 2021 (WHO, 2021). A large number of these deaths are attributed to unexpected surges in infections caused by new strains of SARS-CoV-2, prompting international health organizations such as the CDC and WHO to declare these as variants of concerns (<a class=\"xref-bibr\" href=\"#ref-6\" id=\"xref-ref-6-1\">Khan et al., 2020</a>). Such emergence of new variants can seriously undermine the efficacy of global vaccination programs, the need for future booster doses, or cause multiple reinfection waves as new strains escape previously developed antibodies. Several initiatives have focused on providing high-quality tracking information for the strains and lineages as these emerge (<a class=\"xref-bibr\" href=\"#ref-4\" id=\"xref-ref-4-1\">Hadfield et al., 2018</a>). However, early prediction of emerging variants through genomic signals remains an open challenge. Although domain-based, expert-reasoning approaches are the mainstay of our understanding, which often yield retrospective insight rather than proactive predictions. On the other hand, machine learning approaches are likely to be biased by underlying data characteristics and do not explain the biological basis of predictions.</p>,\n",
              " <p id=\"p-12\">Here, we propose <strong><em>Strainflow</em></strong>, a hybrid architecture of machine learning and language modeling, along with empirical experiments to demonstrate explainable genomic signals for tracking and predicting the spread of SARS-CoV-2 across countries. Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences. Typically, in language models, word embeddings represent the latent space (dimensions) of a corpus of text (<a class=\"xref-bibr\" href=\"#ref-16\" id=\"xref-ref-16-1\">Mikolov et al., 2013</a>) and can capture highly nonlinear and contextual relationships. Codons (tri-nucleotides, 3-mers) translations represent a natural basis for word representations and have been utilized in the past for learning embedding models for modeling various outcomes such as mutation susceptibility (<a class=\"xref-bibr\" href=\"#ref-36\" id=\"xref-ref-36-1\">Yilmaz, 2020</a>) and gene sequence correlations (<a class=\"xref-bibr\" href=\"#ref-35\" id=\"xref-ref-35-1\">Wu et al., 2021</a>). Recently, Hie et. al (<a class=\"xref-bibr\" href=\"#ref-5\" id=\"xref-ref-5-1\">Hie et al., 2021</a>) used machine learning along with word embedding techniques to model the semantics and grammar of amino acids corresponding to antigenic change to predict the mutations which might lead to viral escape.</p>,\n",
              " <p id=\"p-13\">In this paper, we focus on the semantics of viral DNA sequences to derive Dimensions of Concern (DoCs) and demonstrate their causal potential for increasing epidemiological spread patterns across 7 countries. Our approach, Strainflow, is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance.</p>,\n",
              " <p id=\"p-14\">Following the linguistic idea that the context of a word is characterized by its neighborhood, we leveraged a word2vec language model and developed “<em>Strainflow</em>” to learn the latent representations of codons in spike gene sequences of SARS-CoV-2. Strainflow is trained on 0.3 million high-quality genome sequences from March 1, 2020 to March 31, 2021, collected from the GISAID database (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-1\">Fig 1A</a></strong>). To understand the latent space representation (LR) learned by Strainflow, we performed qualitative analysis using tSNE for both global and local (country) levels on a monthly basis. The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-2\">Fig 1B</a></strong>) from September, 2020 to March, 2021, along with specific geographic locations (country-level) such as India, UK, USA, and China.</p>,\n",
              " <p class=\"first-child\" id=\"p-15\">(<strong>A</strong>) The implementation framework of Strainflow (details described in the method section) (<strong>B</strong>) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China. (<strong>C</strong>) Embeddings estimated or predicted from the Strainflow model for 0.36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021. <strong>(D)</strong> Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021. For Pakistan, we couldn’t estimate the fast sample entropy due to fewer number of strain sequences (&lt;20).</p>,\n",
              " <p id=\"p-16\">To investigate the information content in the latent dimensions or space (LD or LS) of the <em>Strainflow</em> model, we performed qualitative and quantitative analysis on 0.9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021. Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (<a class=\"xref-bibr\" href=\"#ref-12\" id=\"xref-ref-12-1\">Linderman et al., 2019</a>). We compared the 2D t-SNE plot of the world with four countries (India, UK, USA, China) from September, 2020 to June, 2021, which clearly highlights the dynamic changes in the spike genes across countries in different months (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-3\">Fig 1B</a></strong> and <strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-4\">Fig 1C</a></strong>). Additionally, the quantitative analysis of the latent space was performed by calculating the fast sample entropy (<a class=\"xref-bibr\" href=\"#ref-19\" id=\"xref-ref-19-1\">Pan et al., 2011</a>) for each LD. To compare the monthly entropy of the LS per country, mean entropy was calculated and normalized across the months for each country. We observed the highest entropy (information content) for India, UK, USA and China in the months of February-2021, July-2020, August-2020, and January-2021 respectively (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-5\">Fig 1D</a></strong>). Interestingly, we observed high entropy for 4 months from August, 2020 to November, 2020 in the USA (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-6\">Fig 1D</a></strong>). This highlights that the spike protein latent space representation learned by the Strainflow model could be used as a proxy to capture the spatiotemporal entropy or diversity in the emerging SARS-CoV-2 strains across different countries.</p>,\n",
              " <p id=\"p-17\">Sequence-level embeddings were obtained from the codon embeddings and investigated for the presence of genomically meaningful characteristics. The phylogenetic tree derived from the embeddings for the United Kingdom (<strong><a class=\"xref-fig\" href=\"#F3\" id=\"xref-fig-3-1\">Fig 2A</a></strong>) shows two clear temporally split clusters for 2020 and 2021 sequences, which may be indicative of different strains in these time periods. The temporality of the collected sequences was found to be preserved in the two clusters, although the model was trained only on genome sequences.</p>,\n",
              " <p class=\"first-child\" id=\"p-18\">(<strong>A</strong>) Dendrogram for strains from the U.K.: Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021. (<strong>B</strong>) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.</p>,\n",
              " <p id=\"p-19\">The phylogenetic tree with globally collected sequences (<strong><a class=\"xref-fig\" href=\"#F3\" id=\"xref-fig-3-2\">Fig 2B</a></strong>) demonstrates that geospatial information is also preserved in the sequence embeddings. Strains from countries like China, Australia, Mexico, and England were found in distinct clusters, while those from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India were found to be dispersed with other countries. This behavior is expected because viral strains spread from one country to another, and our samples had different collection dates for each country. Overall, Strainflow captures the temporal emergence of strains and geographic information in a country-specific manner.</p>,\n",
              " <p id=\"p-20\">Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time. We attributed these changes to mutations in the spike protein of the genome sequence. This hypothesis was based on the fact that the embedding of a word (codon) represents the context it occurs in. A mutation results in a codon that is known to be found in a different context. This alters the semantic sense of the sequence, and this modification of meaning is reflected as a change in values across different latent dimensions. The changed values cause a change in the degree of disorder or the entropy of the latent space.</p>,\n",
              " <p id=\"p-21\">To compare different geographical regions, the sum of sample entropy was computed for each latent dimension across all the months. This revealed that certain geospatial regions such as France and Germany (<strong><a class=\"xref-fig\" href=\"#F4\" id=\"xref-fig-4-1\">Fig 3A</a></strong>) and USA and Canada (<strong><a class=\"xref-fig\" href=\"#F4\" id=\"xref-fig-4-2\">Fig 3B</a></strong>) have similar total entropies across the latent dimensions, indicating that these regions have been accumulating similar changes.</p>,\n",
              " <p class=\"first-child\" id=\"p-22\">Country pairs (<strong>A</strong>) - France and Germany, (<strong>B</strong>) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.</p>,\n",
              " <p id=\"p-23\">We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries. Detrended cross-correlation coefficient was calculated at different lag values, which revealed that entropy dimensions have a leading relationship with new cases (<strong><a class=\"xref-fig\" href=\"#F5\" id=\"xref-fig-5-1\">Fig 4A, 4B</a></strong>). This suggests that the genome sequence data in a given month can be used to predict new cases in subsequent months. A lead period of two months was chosen and Boruta algorithm was employed to assign feature importance scores to different dimensions, which revealed that dimension 32 is the most significant predictor of new cases <strong>(<a class=\"xref-fig\" href=\"#F5\" id=\"xref-fig-5-2\">Fig 4C</a>)</strong>. Significant dimensions from Boruta analysis were termed Dimensions of Concern (DoCs). Random forest based regression modeling on the DoCs achieved a total R-squared of 73% on the test set. The predicted cases were found to be highly correlated with the actual cases (<strong><a class=\"xref-table\" href=\"#T1\" id=\"xref-table-wrap-1-1\">Table 1</a></strong>), which suggests that our model can indicate the directional change of cases for different countries. Further, the predicted relative change in cases between successive months was found to be correlated to the actual relative changes (<strong>Table S1</strong>), which suggests that our model can also indicate the magnitude of change that we expect to observe in the cases.</p>,\n",
              " <p class=\"first-child\" id=\"p-24\">(<strong>A</strong>) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA. High values are observed for a lead of 1 and 2 months. (<strong>B</strong>) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases. (<strong>C</strong>) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.</p>,\n",
              " <p id=\"p-25\">Our model can be therefore used to predict the COVID-19 caseloads in several countries. Both USA (<strong><a class=\"xref-fig\" href=\"#F6\" id=\"xref-fig-6-1\">Fig 5A</a></strong>) and Japan (<strong><a class=\"xref-fig\" href=\"#F6\" id=\"xref-fig-6-2\">Fig 5C</a></strong>) show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries. Our model predicts new caseloads with a two-month lead time, which strongly predicts a spike in new cases both in USA (<strong><a class=\"xref-fig\" href=\"#F6\" id=\"xref-fig-6-3\">Fig 5B</a></strong>) and Japan (<strong><a class=\"xref-fig\" href=\"#F6\" id=\"xref-fig-6-4\">Fig 5D</a></strong>) in the months of July and August, 2021. Therefore our model may be used as an epidemiological early warning system to predict new caseloads.</p>,\n",
              " <p class=\"first-child\" id=\"p-26\">(<strong>A</strong>) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA. (<strong>B</strong>) Actual and predicted cases based on the entropy values of DoCs for the USA. The model predicts a rise in cases for July and August 2021. (<strong>C</strong>) Entropy of DoCs and new cases for Japan. (<strong>D</strong>) Actual and predicted cases for Japan. A spike in cases is predicted for July and August 2021.</p>,\n",
              " <p id=\"p-27\">We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension (<strong>Table S2</strong>). The intuition behind this idea is that the codons with high weights in a given dimension, when mutated in the viral sequence, are likely to cause a significant change in the entropy of the associated dimensions. Therefore each DoC can be linked to codons, which can further be mapped to Variants of concern (VOCs) and Variants of Interest (VOIs) (<strong>Table S3</strong>). Despite the fact that our model cannot directly capture the SARS-CoV-2 variants, it was observed that dimension-32 captures the CTG, CGG codons (ranks 5 and 8 respectively), known to occur in the T19R variant. Similarly, dimension 3 captures three codons (ACG, CGG, CAC) that are associated with multiple variants such as K417T, L452R, and D1118H, causing increased infectivity, pathogenicity, and spread. Dimension 30 captures codons CAT and CAC associated with Δ69 and D1118H respectively which are linked to B.1.1.7 lineage.</p>,\n",
              " <p id=\"p-28\">Codon weights (<strong>Table S2</strong>) of a given DoC provide an opportunity to associate with specific Variants of concern (VOCs) and Variants of Interest (VOIs); and to predict emerging SARS-CoV-2 variants and take preventive measures in advance. Distinct dimensions capture country-specific changes and may be surveilled to monitor the spread of the pandemic. This approach was back-validated with several real-world examples. For instance, dimension 32 captures the codons CGG (R) and CAC (R), which are found in B.1.429 lineage (L425R mutation). Dimension 3 captures CGG which is seen in L452R (associated with lineage B.1.617.1), which was first observed in India in December 2020 and was found to have increased infectivity and transmissibility.</p>,\n",
              " <p id=\"p-29\">To investigate the potential of DoCs to track the spread of SARS-CoV-2, we used the codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs. We selected DoCs (dimension 3, 4, 12, 13, 15, 16, 25, 28, 30, 32) with high absolute weights for the codons related to the delta variants (<strong><a class=\"xref-fig\" href=\"#F7\" id=\"xref-fig-7-1\">Fig 6A</a></strong>). The entropy of the DoCs was contrasted with the caseloads in England (<strong><a class=\"xref-fig\" href=\"#F7\" id=\"xref-fig-7-2\">Fig 6B</a></strong>), India (<strong><a class=\"xref-fig\" href=\"#F7\" id=\"xref-fig-7-3\">Fig 6C</a></strong>), and USA (<strong><a class=\"xref-fig\" href=\"#F7\" id=\"xref-fig-7-4\">Fig 6D</a></strong>). Overall, the temporal tracking of the entropy in Dimensions of Concern (DoCs) may be used as a surrogate to track the spread of various SAR-CoV-2 variants.</p>,\n",
              " <p class=\"first-child\" id=\"p-30\">(<strong>A</strong>) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling. Line plots showing the entropies of DoCs and cases in countries, (<strong>B</strong>) England, (<strong>C</strong>) India, and (<strong>D</strong>) USA. The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.</p>,\n",
              " <p id=\"p-31\">We have implemented an approach for analyzing the emerging strains based on the latent space of spike protein coding nucleotide sequences. We chose the nucleotide sequences instead of proteins in order to capture and track the variations that may not have immediate functional consequences. Our approach has two main underlying tenets: (i) long-range interactions are known to modulate the functional (<a class=\"xref-bibr\" href=\"#ref-17\" id=\"xref-ref-17-1\">Mugnai et al., 2020</a>) interaction between receptor binding domain and ACE2 receptors, hence may be captured in the NLP models that capture 3-mer changes and context, and (ii) latent dimensions may be differentially correlated with indicators of spread, thus providing a data-driven handle for tracking and predicting variants of concern and variants of interest. The pipeline takes advantage of temporal changes in the semantics of mutating sequences. Preservation of phylogenetic structure based upon the similarity matrix obtained using the embeddings validated that the latent dimensions capture spatio-temporal information. Analyzing the dynamic patterns and underlying correlations in the 30,000 base pair long sequence of SARS-CoV-2 (<a class=\"xref-bibr\" href=\"#ref-26\" id=\"xref-ref-26-1\">Shishir et al., 2021</a>) is important to highlight the mechanistic understanding of mutations. SARS-CoV-2 seems to show a particularly high frequency of recombinations arising due to the absence of a proof-reading mechanism and sequence diversity, which calls for urgency in studying its transmission pattern (<a class=\"xref-bibr\" href=\"#ref-15\" id=\"xref-ref-15-1\">Mandal et al., 2021</a>)(<a class=\"xref-bibr\" href=\"#ref-25\" id=\"xref-ref-25-1\">Rouchka et al., 2020</a>). Therefore predicting mutations in the spike protein, which binds to ACE2 receptors can help us estimate the spread of disease and the efficacy of therapeutic treatments and vaccines (<a class=\"xref-bibr\" href=\"#ref-28\" id=\"xref-ref-28-1\">Srivastava et al., 2021</a>)(<a class=\"xref-bibr\" href=\"#ref-11\" id=\"xref-ref-11-1\">Li et al., 2020</a>).</p>,\n",
              " <p id=\"p-32\">Entropy is a measure of the disorder of a system. We hypothesized that mutations increase the chaotic dynamics in the latent space of spike genes. To calculate entropy, we used the accelerated versions of the Approximate Entropy and Sample Entropy algorithms, called Fast Approximate Entropy and Fast Sample Entropy (<a class=\"xref-bibr\" href=\"#ref-19\" id=\"xref-ref-19-2\">Pan et al., 2011</a>). Both algorithms aim to quantify how often different patterns of data are found in a time series. Fast Approximate Entropy, however, is a biased statistic and depends on the length of the series. Since we could have different counts of genome sequences collected each month, we preferred Sample Entropy, which is independent of the length of the series. Entropy values were calculated for each latent dimension in each month. Thereafter, Detrended Cross-Correlation Analysis (DCCA) (<a class=\"xref-bibr\" href=\"#ref-22\" id=\"xref-ref-22-1\">Podobnik and Stanley, 2007</a>) was performed between the entropy dimensions and the new cases. DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series. High cross-correlation for different lead periods revealed that the entropy values in a given month could be used to predict the new cases in different countries in subsequent months. Different countries had different lead times at which the highest cross-correlation was observed between the entropy dimensions and the cases, ranging from 1-6 months. Overall, a lead time of two months was chosen to model the new cases.</p>,\n",
              " <p id=\"p-33\">A similar analysis was done with daily values of entropy and new cases. Entropy was calculated in rolling windows, and cross-correlation analysis was performed between entropy and new cases at different lead periods. Although the cross-correlation values were found to be significant, the values were low and ranged between -0.1 to 0.1. Therefore, we decided to use the monthly entropy values for the modeling exercise.</p>,\n",
              " <p id=\"p-34\">We also experimented with “blips” as a feature to predict the new cases. Blips are sudden changes in the values of the latent dimensions. These changes may be caused by a mutation, which changes the words (codons) in a given genome sequence. This hypothesis was validated in simulation experiments in synthetic datasets. Each dimension of the spike gene embeddings for a country was analyzed for the presence of temporal anomalies. Countries having a minimum of 20 samples in any given month were selected and the same number of records (minimum samples in any given month for that country) were sampled without replacement from each month. These records were used to define control limits of ±1 standard deviation from the mean value for each dimension, and all values in the full dataset outside those limits were categorized as ‘Blip’ points. Blip counts in each month were normalized by calculating the number of blips per sample collected in a month for each dimension (normalized blips). The embedding dimensions were then compared in terms of the total normalized blips for each country to observe the significant dimensions and dis(similarity) in trends among different countries. Cumulative counts of normalized blips were analyzed to understand the temporal accumulation of blips in each dimension. Similar to entropy, blips were found to have a leading relationship with the cases. However, regression modeling results with sample entropy were found to be better.</p>,\n",
              " <p id=\"p-35\">To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data. With sample entropy, the R-squared value achieved was 73%, while with approximate entropy, the value was only 10%. Therefore the model trained on sample entropy was selected. The predictions from the model were found to be highly correlated with the actual cases, indicating that our model can be used for preemptive warning signals for the rise in cases in different countries. Further, the actual and the predicted difference in the number of cases in consecutive months was found to be correlated, which suggests that the relative change in the cases in consecutive months predicted by our model is linked to the relative change in the number of cases. Overall, we recommend that our model be used to predict dangerous trends and not the actual number of cases. Further, the mapping from latent dimensions to Variants of Concern (VOCs) and Variants of Interest (VOIs) may help us track the country-specific spread of different variants.</p>,\n",
              " <p id=\"p-36\">Prediction models can be used for training the genomic sequence for predicting infection severity based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 (<a class=\"xref-bibr\" href=\"#ref-32\" id=\"xref-ref-32-1\">Wang et al., 2020</a>). The machine learning models can also be trained on genomic sequences for COVID-19 classification (<a class=\"xref-bibr\" href=\"#ref-2\" id=\"xref-ref-2-1\">Arslan, 2021</a>). Although the variance explained by our model is low, however, we were able to compute the variability associated with spike protein mutations. So our method showed a potential way to estimate the new cases variability associated with spike protein mutations. Our methods can be incorporated with the epidemic projections model to better predict the epidemic trajectories. The latent dimensions may further be employed to predict the clinical consequences of emerging strains. The currently available vaccines are intended for early SARS-CoV-2 strains, but with new emerging variants, immune responses triggered by these vaccines may be weaker and short-lived. As seen in the devastating second wave of the pandemic in India, newer SARS-CoV-2 variants have acquired an increased pathogenic potential resulting in rapid clinical progression and overwhelmed health systems. Mitigating such events in the future will require stronger surveillance systems in place. Our study offers a promising solution in this direction and lays the foundation for proactive genomic surveillance of COVID-19.</p>,\n",
              " <p id=\"p-37\">Our approach of codon (3-mers) embeddings does not indicate the position where the codon change may have happened in the spike gene. This is because low-dimensional embeddings do not preserve the positional encoding of words. However, we are investigating advanced approaches such as complex-valued word embeddings with positional encodings (<a class=\"xref-bibr\" href=\"#ref-31\" id=\"xref-ref-31-1\">Wang et al., 2019</a>) and transformer models such as BERT (<a class=\"xref-bibr\" href=\"#ref-34\" id=\"xref-ref-34-1\">Wolf et al., 2020</a>)(<a class=\"xref-bibr\" href=\"#ref-9\" id=\"xref-ref-9-1\">Lee et al., 2020</a>) to overcome our current limitation. The latter are considered expensive and data-hungry models and it will remain to be evaluated if the gain of positional information may be countered by the loss of prediction accuracy for forecasting new cases in the future. However, we believe that the availability of sequences for a wide variety of viral pathogens presents an exciting opportunity to train data-hungry models that may be able to transfer insights across pathogens and yet remain interpretable.</p>,\n",
              " <p id=\"p-38\">Further, our Strainflow model is trained only on the spike gene of the viral genome, which does not represent the complete variation spectrum of the virus. To mitigate this shortcoming, we are currently developing a genome-level Strainflow model (version 2.0) for SARS-CoV-2.</p>,\n",
              " <p id=\"p-39\">Another limitation of our study is the relatively small number of samples that were used to construct the supervised predictive model for case prediction. As more data becomes available in subsequent months, we can produce more confident case predictions. Nonetheless, the unsupervised embeddings and temporal cross-correlations were learned upon the full datasets and these presented clear patterns in DoCs and significant cross-correlations with caseloads. However, it is important for our models to receive at least 30 samples per month for getting reliable prediction results for the future months. This also underscores the need for a more reliable and agile approach to deposit country-level datasets on repositories such as GISAID. We make an appeal to the countries to facilitate the sharing of such data in order to be prepared for any future waves of the current pandemic and for preventing the new emergence of strains. We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance.</p>,\n",
              " <p id=\"p-40\">The dataset was downloaded from GISAID EpiCoV (<a class=\"xref-bibr\" href=\"#ref-27\" id=\"xref-ref-27-1\">Shu and McCauley, 2017</a>) (April 8, 2021 release). 0.63 million genome sequences with high nucleotide completeness, coverage, complete temporal information, and presence of less than 5% non-identified nucleotide bases (N) were downloaded. The sequences included 63 countries, including India, United Kingdom, USA, Australia, New Zealand, Germany, Russia, Italy, France, Mexico, Canada, China, Japan, Pakistan, Bangladesh, Iran, Iraq, the continent of South America, and Africa. Duplicate samples were removed, and whole genome sequences were parsed using CoV-Seq (<a class=\"xref-bibr\" href=\"#ref-13\" id=\"xref-ref-13-1\">Liu et al., 2020</a>) to extract nucleotide sequences corresponding to each of the 12 CDS. Accession IDs that did not cover 12 coding regions were discarded, yielding 0.31 million high-quality SARS-CoV-2 genome sequences for language modelling. We downloaded country-wise COVID-19 data for new cases from a publicly available repository maintained by Johns Hopkins University Center for Systems Science and Engineering (JHU CSSE).</p>,\n",
              " <p id=\"p-41\">We downloaded around 0.6 million genome sequences submitted to GISAID from April 2021 to June 2021. We have used 0.36 million sequences to predict their latent space using the <em>Strainflow</em> model for the months of April 2021 to June 2021.</p>,\n",
              " <p id=\"p-42\">In our Strainflow model, we have adopted a word2vec model (<a class=\"xref-bibr\" href=\"#ref-16\" id=\"xref-ref-16-2\">Mikolov et al., 2013</a>). A low dimensional representation for the genome sequences was learned using the word2vec model. Non-overlapping sequences of 3-mers (codons) were considered as words for training the word2vec model, implemented in Gensim (<a class=\"xref-bibr\" href=\"#ref-24\" id=\"xref-ref-24-1\">Radim Rehurek, 2010</a>). The word2vec model was trained using the skip-gram algorithm, with a fixed window size of twenty and vector size of thirty-six. For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise. The mean was calculated by summing across all the k-mers over each dimension and then dividing it by the total number of codons present in the sequence. For selecting the vector size of the word2vec model, we calculated PIP (Pairwise Inner Product) loss (<a class=\"xref-bibr\" href=\"#ref-37\" id=\"xref-ref-37-1\">Yin and Shen, 2018</a>). PIP loss is a metric used for calculating the dissimilarity between two word embedding matrices. For the embedding matrix of strains (E), the PIP matrix is defined as the dot product of the embedding matrix with its transpose (<em>E. E</em> <sup><em>T</em></sup>). The PIP loss between two embedding matrices is defined as the norm of the difference between their PIP matrices.\n",
              " <span class=\"disp-formula\" id=\"disp-formula-1\"><span class=\"highwire-responsive-lazyload\"><img alt=\"Embedded Image\" class=\"highwire-embed lazyload\" data-src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-10.gif\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\"/><noscript><img alt=\"Embedded Image\" class=\"highwire-embed\" src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-10.gif\"/></noscript></span>\n",
              " </span>\n",
              " </p>,\n",
              " <p id=\"p-43\">Various word2vec models were trained on the dataset with different vector sizes varying in multiples of three. Based on the PIP loss calculations, we found out that word embeddings with 36 dimensions showed a differential dent in the curve (change in straight line), due to which we selected this to be the dimension of the word embeddings (<strong>Fig. S1</strong>).</p>,\n",
              " <p id=\"p-44\">To evaluate the phylogenetic properties based on the latent dimensions of the spike gene, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions. The pairwise distance was further used for hierarchical clustering using the ‘hclust’ function in R statistical programming language. This analysis was performed using 400 random sequences of spike genes from 16 countries. The visualization of the phylogenetic tree derived based on the latent dimensions was done using ‘iTOL’ (Letunic) software (<strong><a class=\"xref-fig\" href=\"#F3\" id=\"xref-fig-3-3\">Fig 2</a></strong>).</p>,\n",
              " <p id=\"p-45\">To quantify the properties of latent dimensions, we have used a well-known information theory based algorithm suitable for time series datasets, called ‘Fast Sample Entropy’ (<a class=\"xref-bibr\" href=\"#ref-30\" id=\"xref-ref-30-1\">Tomčala, 2020</a>). To compute Fast Sample Entropy, we have used the ‘FastSampEn’ function in the ‘TSEntropies’ package in R (<a class=\"xref-bibr\" href=\"#ref-29\" id=\"xref-ref-29-1\">Tomcala, 2018</a>). Fast Sample Entropy can be computed as follows.\n",
              " <span class=\"disp-formula\" id=\"disp-formula-2\"><span class=\"highwire-responsive-lazyload\"><img alt=\"Embedded Image\" class=\"highwire-embed lazyload\" data-src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-11.gif\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\"/><noscript><img alt=\"Embedded Image\" class=\"highwire-embed\" src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-11.gif\"/></noscript></span>\n",
              " </span>\n",
              " where,\n",
              " <span class=\"disp-formula\" id=\"disp-formula-3\"><span class=\"highwire-responsive-lazyload\"><img alt=\"Embedded Image\" class=\"highwire-embed lazyload\" data-src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-12.gif\" src=\"data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7\"/><noscript><img alt=\"Embedded Image\" class=\"highwire-embed\" src=\"https://www.biorxiv.org/sites/default/files/highwire/biorxiv/early/2021/08/26/2021.05.25.445601/embed/graphic-12.gif\"/></noscript></span>\n",
              " </span>\n",
              " </p>,\n",
              " <p id=\"p-46\"><em>S</em><sub><em>i,mm</em></sub> is a set of sub-sequences of length m belonging to the i-th neighborhood, and <em>N</em><sub>m</sub> is the number of these neighborhoods.</p>,\n",
              " <p id=\"p-47\">In our case, ‘x’ is the latent dimension of the spike genes of the SARS-CoV-2 strains per month for a given country with default values of ‘m’ and ‘r’. Entropy was computed for each latent dimension on a monthly basis for each country. The minimum number of SARS-CoV-2 spike genes was 20. To compare across months, we used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country (<strong><a class=\"xref-fig\" href=\"#F2\" id=\"xref-fig-2-7\">Fig 1D</a></strong>).</p>,\n",
              " <p id=\"p-48\">To compare the entropy of the latent dimensions among countries, we used the total entropy of the country for each dimension and visualized it with line graphs (<strong><a class=\"xref-fig\" href=\"#F4\" id=\"xref-fig-4-3\">Fig 3</a></strong>).</p>,\n",
              " <p id=\"p-49\">To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used the Detrended Cross Correlation Analysis (<a class=\"xref-bibr\" href=\"#ref-23\" id=\"xref-ref-23-1\">Prass and Pumi, 2019</a>). Here, DCCA captures the long-range cross correlation between time series (entropy of the months and caseloads for a given country). We tested both time series for stationarity using Augmented Dickey-Fuller (ADF) test (<a class=\"xref-bibr\" href=\"#ref-18\" id=\"xref-ref-18-1\">Mushtaq, 2011</a>). The ADF test was implemented using the function ‘adf.test’ available in the ‘tseries’ (2020a) package in R. Due to the non-stationary distribution of the estimated entropies and the caseloads for a given country, we used the ‘DCCA’ R package (2020b). Cross-correlation was calculated between the entropy dimensions at time t+h and new cases at time t, where h = 0, ±1, ±2, ±3 … ±10.</p>,\n",
              " <p id=\"p-50\"><em>Dimensions of Concern (DoCs)</em> were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis and predictive of new case count in the respective countries. Country-wise monthly total new cases data was taken at the end of each month. Total new cases data for each month was merged to monthly entropy dimensions data from March, 2020 to June, 2021. We used a regression based machine learning approach called ‘Boruta’ (<a class=\"xref-bibr\" href=\"#ref-7\" id=\"xref-ref-7-1\">Kursa and Rudnicki, 2010</a>), a wrapper algorithm around random forest algorithm (<a class=\"xref-bibr\" href=\"#ref-8\" id=\"xref-ref-8-1\">Kursa et al., 2010</a>) to select the most relevant entropy dimensions for the prediction of subsequent two months’ new cases. We used the default parameters with the modification of the maximum runs as 1000. We selected the confirmed entropy dimensions as the most relevant features for the prediction of new cases.</p>,\n",
              " <p id=\"p-51\">To predict the new cases in the next to next months, we used a regression based random forest model (Breiman 2001) using the most relevant DoCs based on Boruta. The model training was performed using the data from March, 2020 to April, 2021; and the fitted model was tested on data from May, 2021 to June, 2021. The regression modeling was performed using 1000 decision trees using the ‘randomForest’ package in R (A. Liaw et al., 2002).</p>,\n",
              " <p id=\"p-52\">To find the top codons associated with the DoCs, we extracted the absolute weights of each codon for a given DoC. The top 10 codons having the highest absolute weights (contribution) were identified corresponding to each of the DoCs to compare our DoCs with SARS-CoV-2 variants. We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (<a class=\"xref-bibr\" href=\"#ref-3\" id=\"xref-ref-3-1\">CDC, 2021</a>); and a list of codons associated with VOIs and VOCs was curated <strong>(Table S4)</strong> (Lopez-Rincon et al.; <a class=\"xref-bibr\" href=\"#ref-20\" id=\"xref-ref-20-1\">paola, 2021</a>; <a class=\"xref-bibr\" href=\"#ref-21\" id=\"xref-ref-21-1\">Peacock et al., 2021</a>; <a class=\"xref-bibr\" href=\"#ref-28\" id=\"xref-ref-28-2\">Srivastava et al., 2021</a>). The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant (<a class=\"xref-bibr\" href=\"#ref-3\" id=\"xref-ref-3-2\">CDC, 2021</a>)</p>,\n",
              " <p id=\"p-53\">This work was supported by the Delhi Cluster-Delhi Research Implementation and Innovation (DRIIV) Project supported by the Principal Scientific Advisor Office, Prn.SA/Delhi/ Hub/2018(C) and the Center of Excellence in Healthcare supported by Delhi Knowledge Development Foundation (DKDF) at IIIT-Delhi. We also thank Dr. Chitra Pattabiraman (NIMHANS) for her valuable inputs on viral sequences, and Harleen Kaur, Nishkarsh Saxena, and Anjali for their contribution to the dashboard visualizations.</p>,\n",
              " <p id=\"p-10\">In this revised version, we have implemented fast tSNE for qualitative inspection of the latent dimensions (LD) of 0.9 million SARS-CoV-2 spike genes. Quantitative analyses of the LDs were performed using the fast sample entropy method. Also, we have used fast sample entropy instead of 'blips' to model COVID-19 caseloads.</p>,\n",
              " <p><a class=\"\" data-hide-link-title=\"0\" data-icon-position=\"\" href=\"http://biorxivstage.disqus.com/?url=https%3A%2F%2Fwww.biorxiv.org%2Fcontent%2F10.1101%2F2021.05.25.445601v3\">View the discussion thread.</a></p>,\n",
              " <p><a href=\"/content/10.1101/2021.05.25.445601v3.supplementary-material?versioned=true\"><i class=\"icon-file\"></i>Supplementary Material </a></p>,\n",
              " <p>Thank you for your interest in spreading the word about bioRxiv.</p>,\n",
              " <p>NOTE: Your email address is requested solely to identify you as the sender of this article.</p>]"
            ]
          },
          "metadata": {},
          "execution_count": 42
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "text = [result.text for result in results]\n",
        "text\n",
        "# for result in results:\n",
        "#   text=result.text\n",
        "#   print(text)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "cpwepY1TW0hU",
        "outputId": "60541be3-0924-4ae7-b1ac-fcab4b8fe337"
      },
      "execution_count": 43,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning',\n",
              " 'The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed Strainflow, to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.',\n",
              " 'Highlights',\n",
              " 'We developed a genomic surveillance model for SARS-CoV-2 genome sequences, Strainflow, where sequences were treated as documents with words (codons) to learn the codon context of 0.9 million spike genes using the skip-gram algorithm.',\n",
              " 'Time series analysis of the information content (Entropy) of the latent dimensions learned by Strainflow shows a leading relationship with the monthly COVID-19 cases for seven countries (e.g., USA, Japan, India, and others).',\n",
              " 'Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads.',\n",
              " 'The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants.',\n",
              " 'COVID-19 is reported to have claimed 4.37 million lives as of Aug 17, 2021 (WHO, 2021). A large number of these deaths are attributed to unexpected surges in infections caused by new strains of SARS-CoV-2, prompting international health organizations such as the CDC and WHO to declare these as variants of concerns (Khan et al., 2020). Such emergence of new variants can seriously undermine the efficacy of global vaccination programs, the need for future booster doses, or cause multiple reinfection waves as new strains escape previously developed antibodies. Several initiatives have focused on providing high-quality tracking information for the strains and lineages as these emerge (Hadfield et al., 2018). However, early prediction of emerging variants through genomic signals remains an open challenge. Although domain-based, expert-reasoning approaches are the mainstay of our understanding, which often yield retrospective insight rather than proactive predictions. On the other hand, machine learning approaches are likely to be biased by underlying data characteristics and do not explain the biological basis of predictions.',\n",
              " 'Here, we propose Strainflow, a hybrid architecture of machine learning and language modeling, along with empirical experiments to demonstrate explainable genomic signals for tracking and predicting the spread of SARS-CoV-2 across countries. Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences. Typically, in language models, word embeddings represent the latent space (dimensions) of a corpus of text (Mikolov et al., 2013) and can capture highly nonlinear and contextual relationships. Codons (tri-nucleotides, 3-mers) translations represent a natural basis for word representations and have been utilized in the past for learning embedding models for modeling various outcomes such as mutation susceptibility (Yilmaz, 2020) and gene sequence correlations (Wu et al., 2021). Recently, Hie et. al (Hie et al., 2021) used machine learning along with word embedding techniques to model the semantics and grammar of amino acids corresponding to antigenic change to predict the mutations which might lead to viral escape.',\n",
              " 'In this paper, we focus on the semantics of viral DNA sequences to derive Dimensions of Concern (DoCs) and demonstrate their causal potential for increasing epidemiological spread patterns across 7 countries. Our approach, Strainflow, is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance.',\n",
              " 'Following the linguistic idea that the context of a word is characterized by its neighborhood, we leveraged a word2vec language model and developed “Strainflow” to learn the latent representations of codons in spike gene sequences of SARS-CoV-2. Strainflow is trained on 0.3 million high-quality genome sequences from March 1, 2020 to March 31, 2021, collected from the GISAID database (Fig 1A). To understand the latent space representation (LR) learned by Strainflow, we performed qualitative analysis using tSNE for both global and local (country) levels on a monthly basis. The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 (Fig 1B) from September, 2020 to March, 2021, along with specific geographic locations (country-level) such as India, UK, USA, and China.',\n",
              " '(A) The implementation framework of Strainflow (details described in the method section) (B) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China. (C) Embeddings estimated or predicted from the Strainflow model for 0.36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021. (D) Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021. For Pakistan, we couldn’t estimate the fast sample entropy due to fewer number of strain sequences (<20).',\n",
              " 'To investigate the information content in the latent dimensions or space (LD or LS) of the Strainflow model, we performed qualitative and quantitative analysis on 0.9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021. Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (Linderman et al., 2019). We compared the 2D t-SNE plot of the world with four countries (India, UK, USA, China) from September, 2020 to June, 2021, which clearly highlights the dynamic changes in the spike genes across countries in different months (Fig 1B and Fig 1C). Additionally, the quantitative analysis of the latent space was performed by calculating the fast sample entropy (Pan et al., 2011) for each LD. To compare the monthly entropy of the LS per country, mean entropy was calculated and normalized across the months for each country. We observed the highest entropy (information content) for India, UK, USA and China in the months of February-2021, July-2020, August-2020, and January-2021 respectively (Fig 1D). Interestingly, we observed high entropy for 4 months from August, 2020 to November, 2020 in the USA (Fig 1D). This highlights that the spike protein latent space representation learned by the Strainflow model could be used as a proxy to capture the spatiotemporal entropy or diversity in the emerging SARS-CoV-2 strains across different countries.',\n",
              " 'Sequence-level embeddings were obtained from the codon embeddings and investigated for the presence of genomically meaningful characteristics. The phylogenetic tree derived from the embeddings for the United Kingdom (Fig 2A) shows two clear temporally split clusters for 2020 and 2021 sequences, which may be indicative of different strains in these time periods. The temporality of the collected sequences was found to be preserved in the two clusters, although the model was trained only on genome sequences.',\n",
              " '(A) Dendrogram for strains from the U.K.: Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021. (B) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.',\n",
              " 'The phylogenetic tree with globally collected sequences (Fig 2B) demonstrates that geospatial information is also preserved in the sequence embeddings. Strains from countries like China, Australia, Mexico, and England were found in distinct clusters, while those from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India were found to be dispersed with other countries. This behavior is expected because viral strains spread from one country to another, and our samples had different collection dates for each country. Overall, Strainflow captures the temporal emergence of strains and geographic information in a country-specific manner.',\n",
              " 'Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time. We attributed these changes to mutations in the spike protein of the genome sequence. This hypothesis was based on the fact that the embedding of a word (codon) represents the context it occurs in. A mutation results in a codon that is known to be found in a different context. This alters the semantic sense of the sequence, and this modification of meaning is reflected as a change in values across different latent dimensions. The changed values cause a change in the degree of disorder or the entropy of the latent space.',\n",
              " 'To compare different geographical regions, the sum of sample entropy was computed for each latent dimension across all the months. This revealed that certain geospatial regions such as France and Germany (Fig 3A) and USA and Canada (Fig 3B) have similar total entropies across the latent dimensions, indicating that these regions have been accumulating similar changes.',\n",
              " 'Country pairs (A) - France and Germany, (B) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.',\n",
              " 'We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries. Detrended cross-correlation coefficient was calculated at different lag values, which revealed that entropy dimensions have a leading relationship with new cases (Fig 4A, 4B). This suggests that the genome sequence data in a given month can be used to predict new cases in subsequent months. A lead period of two months was chosen and Boruta algorithm was employed to assign feature importance scores to different dimensions, which revealed that dimension 32 is the most significant predictor of new cases (Fig 4C). Significant dimensions from Boruta analysis were termed Dimensions of Concern (DoCs). Random forest based regression modeling on the DoCs achieved a total R-squared of 73% on the test set. The predicted cases were found to be highly correlated with the actual cases (Table 1), which suggests that our model can indicate the directional change of cases for different countries. Further, the predicted relative change in cases between successive months was found to be correlated to the actual relative changes (Table S1), which suggests that our model can also indicate the magnitude of change that we expect to observe in the cases.',\n",
              " '(A) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA. High values are observed for a lead of 1 and 2 months. (B) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases. (C) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.',\n",
              " 'Our model can be therefore used to predict the COVID-19 caseloads in several countries. Both USA (Fig 5A) and Japan (Fig 5C) show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries. Our model predicts new caseloads with a two-month lead time, which strongly predicts a spike in new cases both in USA (Fig 5B) and Japan (Fig 5D) in the months of July and August, 2021. Therefore our model may be used as an epidemiological early warning system to predict new caseloads.',\n",
              " '(A) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA. (B) Actual and predicted cases based on the entropy values of DoCs for the USA. The model predicts a rise in cases for July and August 2021. (C) Entropy of DoCs and new cases for Japan. (D) Actual and predicted cases for Japan. A spike in cases is predicted for July and August 2021.',\n",
              " 'We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension (Table S2). The intuition behind this idea is that the codons with high weights in a given dimension, when mutated in the viral sequence, are likely to cause a significant change in the entropy of the associated dimensions. Therefore each DoC can be linked to codons, which can further be mapped to Variants of concern (VOCs) and Variants of Interest (VOIs) (Table S3). Despite the fact that our model cannot directly capture the SARS-CoV-2 variants, it was observed that dimension-32 captures the CTG, CGG codons (ranks 5 and 8 respectively), known to occur in the T19R variant. Similarly, dimension 3 captures three codons (ACG, CGG, CAC) that are associated with multiple variants such as K417T, L452R, and D1118H, causing increased infectivity, pathogenicity, and spread. Dimension 30 captures codons CAT and CAC associated with Δ69 and D1118H respectively which are linked to B.1.1.7 lineage.',\n",
              " 'Codon weights (Table S2) of a given DoC provide an opportunity to associate with specific Variants of concern (VOCs) and Variants of Interest (VOIs); and to predict emerging SARS-CoV-2 variants and take preventive measures in advance. Distinct dimensions capture country-specific changes and may be surveilled to monitor the spread of the pandemic. This approach was back-validated with several real-world examples. For instance, dimension 32 captures the codons CGG (R) and CAC (R), which are found in B.1.429 lineage (L425R mutation). Dimension 3 captures CGG which is seen in L452R (associated with lineage B.1.617.1), which was first observed in India in December 2020 and was found to have increased infectivity and transmissibility.',\n",
              " 'To investigate the potential of DoCs to track the spread of SARS-CoV-2, we used the codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs. We selected DoCs (dimension 3, 4, 12, 13, 15, 16, 25, 28, 30, 32) with high absolute weights for the codons related to the delta variants (Fig 6A). The entropy of the DoCs was contrasted with the caseloads in England (Fig 6B), India (Fig 6C), and USA (Fig 6D). Overall, the temporal tracking of the entropy in Dimensions of Concern (DoCs) may be used as a surrogate to track the spread of various SAR-CoV-2 variants.',\n",
              " '(A) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling. Line plots showing the entropies of DoCs and cases in countries, (B) England, (C) India, and (D) USA. The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.',\n",
              " 'We have implemented an approach for analyzing the emerging strains based on the latent space of spike protein coding nucleotide sequences. We chose the nucleotide sequences instead of proteins in order to capture and track the variations that may not have immediate functional consequences. Our approach has two main underlying tenets: (i) long-range interactions are known to modulate the functional (Mugnai et al., 2020) interaction between receptor binding domain and ACE2 receptors, hence may be captured in the NLP models that capture 3-mer changes and context, and (ii) latent dimensions may be differentially correlated with indicators of spread, thus providing a data-driven handle for tracking and predicting variants of concern and variants of interest. The pipeline takes advantage of temporal changes in the semantics of mutating sequences. Preservation of phylogenetic structure based upon the similarity matrix obtained using the embeddings validated that the latent dimensions capture spatio-temporal information. Analyzing the dynamic patterns and underlying correlations in the 30,000 base pair long sequence of SARS-CoV-2 (Shishir et al., 2021) is important to highlight the mechanistic understanding of mutations. SARS-CoV-2 seems to show a particularly high frequency of recombinations arising due to the absence of a proof-reading mechanism and sequence diversity, which calls for urgency in studying its transmission pattern (Mandal et al., 2021)(Rouchka et al., 2020). Therefore predicting mutations in the spike protein, which binds to ACE2 receptors can help us estimate the spread of disease and the efficacy of therapeutic treatments and vaccines (Srivastava et al., 2021)(Li et al., 2020).',\n",
              " 'Entropy is a measure of the disorder of a system. We hypothesized that mutations increase the chaotic dynamics in the latent space of spike genes. To calculate entropy, we used the accelerated versions of the Approximate Entropy and Sample Entropy algorithms, called Fast Approximate Entropy and Fast Sample Entropy (Pan et al., 2011). Both algorithms aim to quantify how often different patterns of data are found in a time series. Fast Approximate Entropy, however, is a biased statistic and depends on the length of the series. Since we could have different counts of genome sequences collected each month, we preferred Sample Entropy, which is independent of the length of the series. Entropy values were calculated for each latent dimension in each month. Thereafter, Detrended Cross-Correlation Analysis (DCCA) (Podobnik and Stanley, 2007) was performed between the entropy dimensions and the new cases. DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series. High cross-correlation for different lead periods revealed that the entropy values in a given month could be used to predict the new cases in different countries in subsequent months. Different countries had different lead times at which the highest cross-correlation was observed between the entropy dimensions and the cases, ranging from 1-6 months. Overall, a lead time of two months was chosen to model the new cases.',\n",
              " 'A similar analysis was done with daily values of entropy and new cases. Entropy was calculated in rolling windows, and cross-correlation analysis was performed between entropy and new cases at different lead periods. Although the cross-correlation values were found to be significant, the values were low and ranged between -0.1 to 0.1. Therefore, we decided to use the monthly entropy values for the modeling exercise.',\n",
              " 'We also experimented with “blips” as a feature to predict the new cases. Blips are sudden changes in the values of the latent dimensions. These changes may be caused by a mutation, which changes the words (codons) in a given genome sequence. This hypothesis was validated in simulation experiments in synthetic datasets. Each dimension of the spike gene embeddings for a country was analyzed for the presence of temporal anomalies. Countries having a minimum of 20 samples in any given month were selected and the same number of records (minimum samples in any given month for that country) were sampled without replacement from each month. These records were used to define control limits of ±1 standard deviation from the mean value for each dimension, and all values in the full dataset outside those limits were categorized as ‘Blip’ points. Blip counts in each month were normalized by calculating the number of blips per sample collected in a month for each dimension (normalized blips). The embedding dimensions were then compared in terms of the total normalized blips for each country to observe the significant dimensions and dis(similarity) in trends among different countries. Cumulative counts of normalized blips were analyzed to understand the temporal accumulation of blips in each dimension. Similar to entropy, blips were found to have a leading relationship with the cases. However, regression modeling results with sample entropy were found to be better.',\n",
              " 'To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data. With sample entropy, the R-squared value achieved was 73%, while with approximate entropy, the value was only 10%. Therefore the model trained on sample entropy was selected. The predictions from the model were found to be highly correlated with the actual cases, indicating that our model can be used for preemptive warning signals for the rise in cases in different countries. Further, the actual and the predicted difference in the number of cases in consecutive months was found to be correlated, which suggests that the relative change in the cases in consecutive months predicted by our model is linked to the relative change in the number of cases. Overall, we recommend that our model be used to predict dangerous trends and not the actual number of cases. Further, the mapping from latent dimensions to Variants of Concern (VOCs) and Variants of Interest (VOIs) may help us track the country-specific spread of different variants.',\n",
              " 'Prediction models can be used for training the genomic sequence for predicting infection severity based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 (Wang et al., 2020). The machine learning models can also be trained on genomic sequences for COVID-19 classification (Arslan, 2021). Although the variance explained by our model is low, however, we were able to compute the variability associated with spike protein mutations. So our method showed a potential way to estimate the new cases variability associated with spike protein mutations. Our methods can be incorporated with the epidemic projections model to better predict the epidemic trajectories. The latent dimensions may further be employed to predict the clinical consequences of emerging strains. The currently available vaccines are intended for early SARS-CoV-2 strains, but with new emerging variants, immune responses triggered by these vaccines may be weaker and short-lived. As seen in the devastating second wave of the pandemic in India, newer SARS-CoV-2 variants have acquired an increased pathogenic potential resulting in rapid clinical progression and overwhelmed health systems. Mitigating such events in the future will require stronger surveillance systems in place. Our study offers a promising solution in this direction and lays the foundation for proactive genomic surveillance of COVID-19.',\n",
              " 'Our approach of codon (3-mers) embeddings does not indicate the position where the codon change may have happened in the spike gene. This is because low-dimensional embeddings do not preserve the positional encoding of words. However, we are investigating advanced approaches such as complex-valued word embeddings with positional encodings (Wang et al., 2019) and transformer models such as BERT (Wolf et al., 2020)(Lee et al., 2020) to overcome our current limitation. The latter are considered expensive and data-hungry models and it will remain to be evaluated if the gain of positional information may be countered by the loss of prediction accuracy for forecasting new cases in the future. However, we believe that the availability of sequences for a wide variety of viral pathogens presents an exciting opportunity to train data-hungry models that may be able to transfer insights across pathogens and yet remain interpretable.',\n",
              " 'Further, our Strainflow model is trained only on the spike gene of the viral genome, which does not represent the complete variation spectrum of the virus. To mitigate this shortcoming, we are currently developing a genome-level Strainflow model (version 2.0) for SARS-CoV-2.',\n",
              " 'Another limitation of our study is the relatively small number of samples that were used to construct the supervised predictive model for case prediction. As more data becomes available in subsequent months, we can produce more confident case predictions. Nonetheless, the unsupervised embeddings and temporal cross-correlations were learned upon the full datasets and these presented clear patterns in DoCs and significant cross-correlations with caseloads. However, it is important for our models to receive at least 30 samples per month for getting reliable prediction results for the future months. This also underscores the need for a more reliable and agile approach to deposit country-level datasets on repositories such as GISAID. We make an appeal to the countries to facilitate the sharing of such data in order to be prepared for any future waves of the current pandemic and for preventing the new emergence of strains. We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance.',\n",
              " 'The dataset was downloaded from GISAID EpiCoV (Shu and McCauley, 2017) (April 8, 2021 release). 0.63 million genome sequences with high nucleotide completeness, coverage, complete temporal information, and presence of less than 5% non-identified nucleotide bases (N) were downloaded. The sequences included 63 countries, including India, United Kingdom, USA, Australia, New Zealand, Germany, Russia, Italy, France, Mexico, Canada, China, Japan, Pakistan, Bangladesh, Iran, Iraq, the continent of South America, and Africa. Duplicate samples were removed, and whole genome sequences were parsed using CoV-Seq (Liu et al., 2020) to extract nucleotide sequences corresponding to each of the 12 CDS. Accession IDs that did not cover 12 coding regions were discarded, yielding 0.31 million high-quality SARS-CoV-2 genome sequences for language modelling. We downloaded country-wise COVID-19 data for new cases from a publicly available repository maintained by Johns Hopkins University Center for Systems Science and Engineering (JHU CSSE).',\n",
              " 'We downloaded around 0.6 million genome sequences submitted to GISAID from April 2021 to June 2021. We have used 0.36 million sequences to predict their latent space using the Strainflow model for the months of April 2021 to June 2021.',\n",
              " 'In our Strainflow model, we have adopted a word2vec model (Mikolov et al., 2013). A low dimensional representation for the genome sequences was learned using the word2vec model. Non-overlapping sequences of 3-mers (codons) were considered as words for training the word2vec model, implemented in Gensim (Radim Rehurek, 2010). The word2vec model was trained using the skip-gram algorithm, with a fixed window size of twenty and vector size of thirty-six. For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise. The mean was calculated by summing across all the k-mers over each dimension and then dividing it by the total number of codons present in the sequence. For selecting the vector size of the word2vec model, we calculated PIP (Pairwise Inner Product) loss (Yin and Shen, 2018). PIP loss is a metric used for calculating the dissimilarity between two word embedding matrices. For the embedding matrix of strains (E), the PIP matrix is defined as the dot product of the embedding matrix with its transpose (E. E T). The PIP loss between two embedding matrices is defined as the norm of the difference between their PIP matrices.\\n\\n\\n',\n",
              " 'Various word2vec models were trained on the dataset with different vector sizes varying in multiples of three. Based on the PIP loss calculations, we found out that word embeddings with 36 dimensions showed a differential dent in the curve (change in straight line), due to which we selected this to be the dimension of the word embeddings (Fig. S1).',\n",
              " 'To evaluate the phylogenetic properties based on the latent dimensions of the spike gene, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions. The pairwise distance was further used for hierarchical clustering using the ‘hclust’ function in R statistical programming language. This analysis was performed using 400 random sequences of spike genes from 16 countries. The visualization of the phylogenetic tree derived based on the latent dimensions was done using ‘iTOL’ (Letunic) software (Fig 2).',\n",
              " 'To quantify the properties of latent dimensions, we have used a well-known information theory based algorithm suitable for time series datasets, called ‘Fast Sample Entropy’ (Tomčala, 2020). To compute Fast Sample Entropy, we have used the ‘FastSampEn’ function in the ‘TSEntropies’ package in R (Tomcala, 2018). Fast Sample Entropy can be computed as follows.\\n\\n\\nwhere,\\n\\n\\n',\n",
              " 'Si,mm is a set of sub-sequences of length m belonging to the i-th neighborhood, and Nm is the number of these neighborhoods.',\n",
              " 'In our case, ‘x’ is the latent dimension of the spike genes of the SARS-CoV-2 strains per month for a given country with default values of ‘m’ and ‘r’. Entropy was computed for each latent dimension on a monthly basis for each country. The minimum number of SARS-CoV-2 spike genes was 20. To compare across months, we used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country (Fig 1D).',\n",
              " 'To compare the entropy of the latent dimensions among countries, we used the total entropy of the country for each dimension and visualized it with line graphs (Fig 3).',\n",
              " 'To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used the Detrended Cross Correlation Analysis (Prass and Pumi, 2019). Here, DCCA captures the long-range cross correlation between time series (entropy of the months and caseloads for a given country). We tested both time series for stationarity using Augmented Dickey-Fuller (ADF) test (Mushtaq, 2011). The ADF test was implemented using the function ‘adf.test’ available in the ‘tseries’ (2020a) package in R. Due to the non-stationary distribution of the estimated entropies and the caseloads for a given country, we used the ‘DCCA’ R package (2020b). Cross-correlation was calculated between the entropy dimensions at time t+h and new cases at time t, where h = 0, ±1, ±2, ±3 … ±10.',\n",
              " 'Dimensions of Concern (DoCs) were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis and predictive of new case count in the respective countries. Country-wise monthly total new cases data was taken at the end of each month. Total new cases data for each month was merged to monthly entropy dimensions data from March, 2020 to June, 2021. We used a regression based machine learning approach called ‘Boruta’ (Kursa and Rudnicki, 2010), a wrapper algorithm around random forest algorithm (Kursa et al., 2010) to select the most relevant entropy dimensions for the prediction of subsequent two months’ new cases. We used the default parameters with the modification of the maximum runs as 1000. We selected the confirmed entropy dimensions as the most relevant features for the prediction of new cases.',\n",
              " 'To predict the new cases in the next to next months, we used a regression based random forest model (Breiman 2001) using the most relevant DoCs based on Boruta. The model training was performed using the data from March, 2020 to April, 2021; and the fitted model was tested on data from May, 2021 to June, 2021. The regression modeling was performed using 1000 decision trees using the ‘randomForest’ package in R (A. Liaw et al., 2002).',\n",
              " 'To find the top codons associated with the DoCs, we extracted the absolute weights of each codon for a given DoC. The top 10 codons having the highest absolute weights (contribution) were identified corresponding to each of the DoCs to compare our DoCs with SARS-CoV-2 variants. We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (CDC, 2021); and a list of codons associated with VOIs and VOCs was curated (Table S4) (Lopez-Rincon et al.; paola, 2021; Peacock et al., 2021; Srivastava et al., 2021). The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant (CDC, 2021)',\n",
              " 'This work was supported by the Delhi Cluster-Delhi Research Implementation and Innovation (DRIIV) Project supported by the Principal Scientific Advisor Office, Prn.SA/Delhi/ Hub/2018(C) and the Center of Excellence in Healthcare supported by Delhi Knowledge Development Foundation (DKDF) at IIIT-Delhi. We also thank Dr. Chitra Pattabiraman (NIMHANS) for her valuable inputs on viral sequences, and Harleen Kaur, Nishkarsh Saxena, and Anjali for their contribution to the dashboard visualizations.',\n",
              " \"In this revised version, we have implemented fast tSNE for qualitative inspection of the latent dimensions (LD) of 0.9 million SARS-CoV-2 spike genes. Quantitative analyses of the LDs were performed using the fast sample entropy method. Also, we have used fast sample entropy instead of 'blips' to model COVID-19 caseloads.\",\n",
              " 'View the discussion thread.',\n",
              " 'Supplementary Material ',\n",
              " 'Thank you for your interest in spreading the word about bioRxiv.',\n",
              " 'NOTE: Your email address is requested solely to identify you as the sender of this article.']"
            ]
          },
          "metadata": {},
          "execution_count": 43
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "Article_data = ' '.join(text)\n",
        "Article_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 122
        },
        "id": "f1JlE4qaXYlt",
        "outputId": "6d186410-b445-4302-af91-08fc36e2aeb6"
      },
      "execution_count": 44,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity. Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge. Here, we developed Strainflow, to learn the latent dimensions of 0.9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2. In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions). We discovered that codon-level changes lead to a change in the entropy of the latent dimensions. We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries. The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants. Highlights We developed a genomic surveillance model for SARS-CoV-2 genome sequences, Strainflow, where sequences were treated as documents with words (codons) to learn the codon context of 0.9 million spike genes using the skip-gram algorithm. Time series analysis of the information content (Entropy) of the latent dimensions learned by Strainflow shows a leading relationship with the monthly COVID-19 cases for seven countries (e.g., USA, Japan, India, and others). Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads. The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants. COVID-19 is reported to have claimed 4.37 million lives as of Aug 17, 2021 (WHO, 2021). A large number of these deaths are attributed to unexpected surges in infections caused by new strains of SARS-CoV-2, prompting international health organizations such as the CDC and WHO to declare these as variants of concerns (Khan et al., 2020). Such emergence of new variants can seriously undermine the efficacy of global vaccination programs, the need for future booster doses, or cause multiple reinfection waves as new strains escape previously developed antibodies. Several initiatives have focused on providing high-quality tracking information for the strains and lineages as these emerge (Hadfield et al., 2018). However, early prediction of emerging variants through genomic signals remains an open challenge. Although domain-based, expert-reasoning approaches are the mainstay of our understanding, which often yield retrospective insight rather than proactive predictions. On the other hand, machine learning approaches are likely to be biased by underlying data characteristics and do not explain the biological basis of predictions. Here, we propose Strainflow, a hybrid architecture of machine learning and language modeling, along with empirical experiments to demonstrate explainable genomic signals for tracking and predicting the spread of SARS-CoV-2 across countries. Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences. Typically, in language models, word embeddings represent the latent space (dimensions) of a corpus of text (Mikolov et al., 2013) and can capture highly nonlinear and contextual relationships. Codons (tri-nucleotides, 3-mers) translations represent a natural basis for word representations and have been utilized in the past for learning embedding models for modeling various outcomes such as mutation susceptibility (Yilmaz, 2020) and gene sequence correlations (Wu et al., 2021). Recently, Hie et. al (Hie et al., 2021) used machine learning along with word embedding techniques to model the semantics and grammar of amino acids corresponding to antigenic change to predict the mutations which might lead to viral escape. In this paper, we focus on the semantics of viral DNA sequences to derive Dimensions of Concern (DoCs) and demonstrate their causal potential for increasing epidemiological spread patterns across 7 countries. Our approach, Strainflow, is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance. Following the linguistic idea that the context of a word is characterized by its neighborhood, we leveraged a word2vec language model and developed “Strainflow” to learn the latent representations of codons in spike gene sequences of SARS-CoV-2. Strainflow is trained on 0.3 million high-quality genome sequences from March 1, 2020 to March 31, 2021, collected from the GISAID database (Fig 1A). To understand the latent space representation (LR) learned by Strainflow, we performed qualitative analysis using tSNE for both global and local (country) levels on a monthly basis. The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 (Fig 1B) from September, 2020 to March, 2021, along with specific geographic locations (country-level) such as India, UK, USA, and China. (A) The implementation framework of Strainflow (details described in the method section) (B) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China. (C) Embeddings estimated or predicted from the Strainflow model for 0.36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021. (D) Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021. For Pakistan, we couldn’t estimate the fast sample entropy due to fewer number of strain sequences (<20). To investigate the information content in the latent dimensions or space (LD or LS) of the Strainflow model, we performed qualitative and quantitative analysis on 0.9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021. Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (Linderman et al., 2019). We compared the 2D t-SNE plot of the world with four countries (India, UK, USA, China) from September, 2020 to June, 2021, which clearly highlights the dynamic changes in the spike genes across countries in different months (Fig 1B and Fig 1C). Additionally, the quantitative analysis of the latent space was performed by calculating the fast sample entropy (Pan et al., 2011) for each LD. To compare the monthly entropy of the LS per country, mean entropy was calculated and normalized across the months for each country. We observed the highest entropy (information content) for India, UK, USA and China in the months of February-2021, July-2020, August-2020, and January-2021 respectively (Fig 1D). Interestingly, we observed high entropy for 4 months from August, 2020 to November, 2020 in the USA (Fig 1D). This highlights that the spike protein latent space representation learned by the Strainflow model could be used as a proxy to capture the spatiotemporal entropy or diversity in the emerging SARS-CoV-2 strains across different countries. Sequence-level embeddings were obtained from the codon embeddings and investigated for the presence of genomically meaningful characteristics. The phylogenetic tree derived from the embeddings for the United Kingdom (Fig 2A) shows two clear temporally split clusters for 2020 and 2021 sequences, which may be indicative of different strains in these time periods. The temporality of the collected sequences was found to be preserved in the two clusters, although the model was trained only on genome sequences. (A) Dendrogram for strains from the U.K.: Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021. (B) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries. The phylogenetic tree with globally collected sequences (Fig 2B) demonstrates that geospatial information is also preserved in the sequence embeddings. Strains from countries like China, Australia, Mexico, and England were found in distinct clusters, while those from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India were found to be dispersed with other countries. This behavior is expected because viral strains spread from one country to another, and our samples had different collection dates for each country. Overall, Strainflow captures the temporal emergence of strains and geographic information in a country-specific manner. Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time. We attributed these changes to mutations in the spike protein of the genome sequence. This hypothesis was based on the fact that the embedding of a word (codon) represents the context it occurs in. A mutation results in a codon that is known to be found in a different context. This alters the semantic sense of the sequence, and this modification of meaning is reflected as a change in values across different latent dimensions. The changed values cause a change in the degree of disorder or the entropy of the latent space. To compare different geographical regions, the sum of sample entropy was computed for each latent dimension across all the months. This revealed that certain geospatial regions such as France and Germany (Fig 3A) and USA and Canada (Fig 3B) have similar total entropies across the latent dimensions, indicating that these regions have been accumulating similar changes. Country pairs (A) - France and Germany, (B) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other. We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries. Detrended cross-correlation coefficient was calculated at different lag values, which revealed that entropy dimensions have a leading relationship with new cases (Fig 4A, 4B). This suggests that the genome sequence data in a given month can be used to predict new cases in subsequent months. A lead period of two months was chosen and Boruta algorithm was employed to assign feature importance scores to different dimensions, which revealed that dimension 32 is the most significant predictor of new cases (Fig 4C). Significant dimensions from Boruta analysis were termed Dimensions of Concern (DoCs). Random forest based regression modeling on the DoCs achieved a total R-squared of 73% on the test set. The predicted cases were found to be highly correlated with the actual cases (Table 1), which suggests that our model can indicate the directional change of cases for different countries. Further, the predicted relative change in cases between successive months was found to be correlated to the actual relative changes (Table S1), which suggests that our model can also indicate the magnitude of change that we expect to observe in the cases. (A) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA. High values are observed for a lead of 1 and 2 months. (B) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases. (C) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month. Our model can be therefore used to predict the COVID-19 caseloads in several countries. Both USA (Fig 5A) and Japan (Fig 5C) show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries. Our model predicts new caseloads with a two-month lead time, which strongly predicts a spike in new cases both in USA (Fig 5B) and Japan (Fig 5D) in the months of July and August, 2021. Therefore our model may be used as an epidemiological early warning system to predict new caseloads. (A) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA. (B) Actual and predicted cases based on the entropy values of DoCs for the USA. The model predicts a rise in cases for July and August 2021. (C) Entropy of DoCs and new cases for Japan. (D) Actual and predicted cases for Japan. A spike in cases is predicted for July and August 2021. We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension (Table S2). The intuition behind this idea is that the codons with high weights in a given dimension, when mutated in the viral sequence, are likely to cause a significant change in the entropy of the associated dimensions. Therefore each DoC can be linked to codons, which can further be mapped to Variants of concern (VOCs) and Variants of Interest (VOIs) (Table S3). Despite the fact that our model cannot directly capture the SARS-CoV-2 variants, it was observed that dimension-32 captures the CTG, CGG codons (ranks 5 and 8 respectively), known to occur in the T19R variant. Similarly, dimension 3 captures three codons (ACG, CGG, CAC) that are associated with multiple variants such as K417T, L452R, and D1118H, causing increased infectivity, pathogenicity, and spread. Dimension 30 captures codons CAT and CAC associated with Δ69 and D1118H respectively which are linked to B.1.1.7 lineage. Codon weights (Table S2) of a given DoC provide an opportunity to associate with specific Variants of concern (VOCs) and Variants of Interest (VOIs); and to predict emerging SARS-CoV-2 variants and take preventive measures in advance. Distinct dimensions capture country-specific changes and may be surveilled to monitor the spread of the pandemic. This approach was back-validated with several real-world examples. For instance, dimension 32 captures the codons CGG (R) and CAC (R), which are found in B.1.429 lineage (L425R mutation). Dimension 3 captures CGG which is seen in L452R (associated with lineage B.1.617.1), which was first observed in India in December 2020 and was found to have increased infectivity and transmissibility. To investigate the potential of DoCs to track the spread of SARS-CoV-2, we used the codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs. We selected DoCs (dimension 3, 4, 12, 13, 15, 16, 25, 28, 30, 32) with high absolute weights for the codons related to the delta variants (Fig 6A). The entropy of the DoCs was contrasted with the caseloads in England (Fig 6B), India (Fig 6C), and USA (Fig 6D). Overall, the temporal tracking of the entropy in Dimensions of Concern (DoCs) may be used as a surrogate to track the spread of various SAR-CoV-2 variants. (A) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling. Line plots showing the entropies of DoCs and cases in countries, (B) England, (C) India, and (D) USA. The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries. We have implemented an approach for analyzing the emerging strains based on the latent space of spike protein coding nucleotide sequences. We chose the nucleotide sequences instead of proteins in order to capture and track the variations that may not have immediate functional consequences. Our approach has two main underlying tenets: (i) long-range interactions are known to modulate the functional (Mugnai et al., 2020) interaction between receptor binding domain and ACE2 receptors, hence may be captured in the NLP models that capture 3-mer changes and context, and (ii) latent dimensions may be differentially correlated with indicators of spread, thus providing a data-driven handle for tracking and predicting variants of concern and variants of interest. The pipeline takes advantage of temporal changes in the semantics of mutating sequences. Preservation of phylogenetic structure based upon the similarity matrix obtained using the embeddings validated that the latent dimensions capture spatio-temporal information. Analyzing the dynamic patterns and underlying correlations in the 30,000 base pair long sequence of SARS-CoV-2 (Shishir et al., 2021) is important to highlight the mechanistic understanding of mutations. SARS-CoV-2 seems to show a particularly high frequency of recombinations arising due to the absence of a proof-reading mechanism and sequence diversity, which calls for urgency in studying its transmission pattern (Mandal et al., 2021)(Rouchka et al., 2020). Therefore predicting mutations in the spike protein, which binds to ACE2 receptors can help us estimate the spread of disease and the efficacy of therapeutic treatments and vaccines (Srivastava et al., 2021)(Li et al., 2020). Entropy is a measure of the disorder of a system. We hypothesized that mutations increase the chaotic dynamics in the latent space of spike genes. To calculate entropy, we used the accelerated versions of the Approximate Entropy and Sample Entropy algorithms, called Fast Approximate Entropy and Fast Sample Entropy (Pan et al., 2011). Both algorithms aim to quantify how often different patterns of data are found in a time series. Fast Approximate Entropy, however, is a biased statistic and depends on the length of the series. Since we could have different counts of genome sequences collected each month, we preferred Sample Entropy, which is independent of the length of the series. Entropy values were calculated for each latent dimension in each month. Thereafter, Detrended Cross-Correlation Analysis (DCCA) (Podobnik and Stanley, 2007) was performed between the entropy dimensions and the new cases. DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series. High cross-correlation for different lead periods revealed that the entropy values in a given month could be used to predict the new cases in different countries in subsequent months. Different countries had different lead times at which the highest cross-correlation was observed between the entropy dimensions and the cases, ranging from 1-6 months. Overall, a lead time of two months was chosen to model the new cases. A similar analysis was done with daily values of entropy and new cases. Entropy was calculated in rolling windows, and cross-correlation analysis was performed between entropy and new cases at different lead periods. Although the cross-correlation values were found to be significant, the values were low and ranged between -0.1 to 0.1. Therefore, we decided to use the monthly entropy values for the modeling exercise. We also experimented with “blips” as a feature to predict the new cases. Blips are sudden changes in the values of the latent dimensions. These changes may be caused by a mutation, which changes the words (codons) in a given genome sequence. This hypothesis was validated in simulation experiments in synthetic datasets. Each dimension of the spike gene embeddings for a country was analyzed for the presence of temporal anomalies. Countries having a minimum of 20 samples in any given month were selected and the same number of records (minimum samples in any given month for that country) were sampled without replacement from each month. These records were used to define control limits of ±1 standard deviation from the mean value for each dimension, and all values in the full dataset outside those limits were categorized as ‘Blip’ points. Blip counts in each month were normalized by calculating the number of blips per sample collected in a month for each dimension (normalized blips). The embedding dimensions were then compared in terms of the total normalized blips for each country to observe the significant dimensions and dis(similarity) in trends among different countries. Cumulative counts of normalized blips were analyzed to understand the temporal accumulation of blips in each dimension. Similar to entropy, blips were found to have a leading relationship with the cases. However, regression modeling results with sample entropy were found to be better. To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data. With sample entropy, the R-squared value achieved was 73%, while with approximate entropy, the value was only 10%. Therefore the model trained on sample entropy was selected. The predictions from the model were found to be highly correlated with the actual cases, indicating that our model can be used for preemptive warning signals for the rise in cases in different countries. Further, the actual and the predicted difference in the number of cases in consecutive months was found to be correlated, which suggests that the relative change in the cases in consecutive months predicted by our model is linked to the relative change in the number of cases. Overall, we recommend that our model be used to predict dangerous trends and not the actual number of cases. Further, the mapping from latent dimensions to Variants of Concern (VOCs) and Variants of Interest (VOIs) may help us track the country-specific spread of different variants. Prediction models can be used for training the genomic sequence for predicting infection severity based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 (Wang et al., 2020). The machine learning models can also be trained on genomic sequences for COVID-19 classification (Arslan, 2021). Although the variance explained by our model is low, however, we were able to compute the variability associated with spike protein mutations. So our method showed a potential way to estimate the new cases variability associated with spike protein mutations. Our methods can be incorporated with the epidemic projections model to better predict the epidemic trajectories. The latent dimensions may further be employed to predict the clinical consequences of emerging strains. The currently available vaccines are intended for early SARS-CoV-2 strains, but with new emerging variants, immune responses triggered by these vaccines may be weaker and short-lived. As seen in the devastating second wave of the pandemic in India, newer SARS-CoV-2 variants have acquired an increased pathogenic potential resulting in rapid clinical progression and overwhelmed health systems. Mitigating such events in the future will require stronger surveillance systems in place. Our study offers a promising solution in this direction and lays the foundation for proactive genomic surveillance of COVID-19. Our approach of codon (3-mers) embeddings does not indicate the position where the codon change may have happened in the spike gene. This is because low-dimensional embeddings do not preserve the positional encoding of words. However, we are investigating advanced approaches such as complex-valued word embeddings with positional encodings (Wang et al., 2019) and transformer models such as BERT (Wolf et al., 2020)(Lee et al., 2020) to overcome our current limitation. The latter are considered expensive and data-hungry models and it will remain to be evaluated if the gain of positional information may be countered by the loss of prediction accuracy for forecasting new cases in the future. However, we believe that the availability of sequences for a wide variety of viral pathogens presents an exciting opportunity to train data-hungry models that may be able to transfer insights across pathogens and yet remain interpretable. Further, our Strainflow model is trained only on the spike gene of the viral genome, which does not represent the complete variation spectrum of the virus. To mitigate this shortcoming, we are currently developing a genome-level Strainflow model (version 2.0) for SARS-CoV-2. Another limitation of our study is the relatively small number of samples that were used to construct the supervised predictive model for case prediction. As more data becomes available in subsequent months, we can produce more confident case predictions. Nonetheless, the unsupervised embeddings and temporal cross-correlations were learned upon the full datasets and these presented clear patterns in DoCs and significant cross-correlations with caseloads. However, it is important for our models to receive at least 30 samples per month for getting reliable prediction results for the future months. This also underscores the need for a more reliable and agile approach to deposit country-level datasets on repositories such as GISAID. We make an appeal to the countries to facilitate the sharing of such data in order to be prepared for any future waves of the current pandemic and for preventing the new emergence of strains. We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance. The dataset was downloaded from GISAID EpiCoV (Shu and McCauley, 2017) (April 8, 2021 release). 0.63 million genome sequences with high nucleotide completeness, coverage, complete temporal information, and presence of less than 5% non-identified nucleotide bases (N) were downloaded. The sequences included 63 countries, including India, United Kingdom, USA, Australia, New Zealand, Germany, Russia, Italy, France, Mexico, Canada, China, Japan, Pakistan, Bangladesh, Iran, Iraq, the continent of South America, and Africa. Duplicate samples were removed, and whole genome sequences were parsed using CoV-Seq (Liu et al., 2020) to extract nucleotide sequences corresponding to each of the 12 CDS. Accession IDs that did not cover 12 coding regions were discarded, yielding 0.31 million high-quality SARS-CoV-2 genome sequences for language modelling. We downloaded country-wise COVID-19 data for new cases from a publicly available repository maintained by Johns Hopkins University Center for Systems Science and Engineering (JHU CSSE). We downloaded around 0.6 million genome sequences submitted to GISAID from April 2021 to June 2021. We have used 0.36 million sequences to predict their latent space using the Strainflow model for the months of April 2021 to June 2021. In our Strainflow model, we have adopted a word2vec model (Mikolov et al., 2013). A low dimensional representation for the genome sequences was learned using the word2vec model. Non-overlapping sequences of 3-mers (codons) were considered as words for training the word2vec model, implemented in Gensim (Radim Rehurek, 2010). The word2vec model was trained using the skip-gram algorithm, with a fixed window size of twenty and vector size of thirty-six. For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise. The mean was calculated by summing across all the k-mers over each dimension and then dividing it by the total number of codons present in the sequence. For selecting the vector size of the word2vec model, we calculated PIP (Pairwise Inner Product) loss (Yin and Shen, 2018). PIP loss is a metric used for calculating the dissimilarity between two word embedding matrices. For the embedding matrix of strains (E), the PIP matrix is defined as the dot product of the embedding matrix with its transpose (E. E T). The PIP loss between two embedding matrices is defined as the norm of the difference between their PIP matrices.\\n\\n\\n Various word2vec models were trained on the dataset with different vector sizes varying in multiples of three. Based on the PIP loss calculations, we found out that word embeddings with 36 dimensions showed a differential dent in the curve (change in straight line), due to which we selected this to be the dimension of the word embeddings (Fig. S1). To evaluate the phylogenetic properties based on the latent dimensions of the spike gene, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions. The pairwise distance was further used for hierarchical clustering using the ‘hclust’ function in R statistical programming language. This analysis was performed using 400 random sequences of spike genes from 16 countries. The visualization of the phylogenetic tree derived based on the latent dimensions was done using ‘iTOL’ (Letunic) software (Fig 2). To quantify the properties of latent dimensions, we have used a well-known information theory based algorithm suitable for time series datasets, called ‘Fast Sample Entropy’ (Tomčala, 2020). To compute Fast Sample Entropy, we have used the ‘FastSampEn’ function in the ‘TSEntropies’ package in R (Tomcala, 2018). Fast Sample Entropy can be computed as follows.\\n\\n\\nwhere,\\n\\n\\n Si,mm is a set of sub-sequences of length m belonging to the i-th neighborhood, and Nm is the number of these neighborhoods. In our case, ‘x’ is the latent dimension of the spike genes of the SARS-CoV-2 strains per month for a given country with default values of ‘m’ and ‘r’. Entropy was computed for each latent dimension on a monthly basis for each country. The minimum number of SARS-CoV-2 spike genes was 20. To compare across months, we used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country (Fig 1D). To compare the entropy of the latent dimensions among countries, we used the total entropy of the country for each dimension and visualized it with line graphs (Fig 3). To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used the Detrended Cross Correlation Analysis (Prass and Pumi, 2019). Here, DCCA captures the long-range cross correlation between time series (entropy of the months and caseloads for a given country). We tested both time series for stationarity using Augmented Dickey-Fuller (ADF) test (Mushtaq, 2011). The ADF test was implemented using the function ‘adf.test’ available in the ‘tseries’ (2020a) package in R. Due to the non-stationary distribution of the estimated entropies and the caseloads for a given country, we used the ‘DCCA’ R package (2020b). Cross-correlation was calculated between the entropy dimensions at time t+h and new cases at time t, where h = 0, ±1, ±2, ±3 … ±10. Dimensions of Concern (DoCs) were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis and predictive of new case count in the respective countries. Country-wise monthly total new cases data was taken at the end of each month. Total new cases data for each month was merged to monthly entropy dimensions data from March, 2020 to June, 2021. We used a regression based machine learning approach called ‘Boruta’ (Kursa and Rudnicki, 2010), a wrapper algorithm around random forest algorithm (Kursa et al., 2010) to select the most relevant entropy dimensions for the prediction of subsequent two months’ new cases. We used the default parameters with the modification of the maximum runs as 1000. We selected the confirmed entropy dimensions as the most relevant features for the prediction of new cases. To predict the new cases in the next to next months, we used a regression based random forest model (Breiman 2001) using the most relevant DoCs based on Boruta. The model training was performed using the data from March, 2020 to April, 2021; and the fitted model was tested on data from May, 2021 to June, 2021. The regression modeling was performed using 1000 decision trees using the ‘randomForest’ package in R (A. Liaw et al., 2002). To find the top codons associated with the DoCs, we extracted the absolute weights of each codon for a given DoC. The top 10 codons having the highest absolute weights (contribution) were identified corresponding to each of the DoCs to compare our DoCs with SARS-CoV-2 variants. We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (CDC, 2021); and a list of codons associated with VOIs and VOCs was curated (Table S4) (Lopez-Rincon et al.; paola, 2021; Peacock et al., 2021; Srivastava et al., 2021). The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant (CDC, 2021) This work was supported by the Delhi Cluster-Delhi Research Implementation and Innovation (DRIIV) Project supported by the Principal Scientific Advisor Office, Prn.SA/Delhi/ Hub/2018(C) and the Center of Excellence in Healthcare supported by Delhi Knowledge Development Foundation (DKDF) at IIIT-Delhi. We also thank Dr. Chitra Pattabiraman (NIMHANS) for her valuable inputs on viral sequences, and Harleen Kaur, Nishkarsh Saxena, and Anjali for their contribution to the dashboard visualizations. In this revised version, we have implemented fast tSNE for qualitative inspection of the latent dimensions (LD) of 0.9 million SARS-CoV-2 spike genes. Quantitative analyses of the LDs were performed using the fast sample entropy method. Also, we have used fast sample entropy instead of 'blips' to model COVID-19 caseloads. View the discussion thread. Supplementary Material  Thank you for your interest in spreading the word about bioRxiv. NOTE: Your email address is requested solely to identify you as the sender of this article.\""
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 44
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Split article into individual sentences"
      ],
      "metadata": {
        "id": "X30Do5fIezeM"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "#Replacing .?! into \"end of sentence\"\n",
        "Article_data = Article_data.replace('.', '.<eos>') \n",
        "Article_data = Article_data.replace('?', '?<eos>')\n",
        "Article_data = Article_data.replace('!', '!<eos>')\n",
        "Article_data"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 122
        },
        "id": "vJLdZJD4XYok",
        "outputId": "f4a3335e-620e-4c2f-d3cd-39fbd17a88b8"
      },
      "execution_count": 45,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "\"Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity.<eos> Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge.<eos> Here, we developed Strainflow, to learn the latent dimensions of 0.<eos>9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2.<eos> In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions).<eos> We discovered that codon-level changes lead to a change in the entropy of the latent dimensions.<eos> We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries.<eos> The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.<eos> Highlights We developed a genomic surveillance model for SARS-CoV-2 genome sequences, Strainflow, where sequences were treated as documents with words (codons) to learn the codon context of 0.<eos>9 million spike genes using the skip-gram algorithm.<eos> Time series analysis of the information content (Entropy) of the latent dimensions learned by Strainflow shows a leading relationship with the monthly COVID-19 cases for seven countries (e.<eos>g.<eos>, USA, Japan, India, and others).<eos> Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads.<eos> The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants.<eos> COVID-19 is reported to have claimed 4.<eos>37 million lives as of Aug 17, 2021 (WHO, 2021).<eos> A large number of these deaths are attributed to unexpected surges in infections caused by new strains of SARS-CoV-2, prompting international health organizations such as the CDC and WHO to declare these as variants of concerns (Khan et al.<eos>, 2020).<eos> Such emergence of new variants can seriously undermine the efficacy of global vaccination programs, the need for future booster doses, or cause multiple reinfection waves as new strains escape previously developed antibodies.<eos> Several initiatives have focused on providing high-quality tracking information for the strains and lineages as these emerge (Hadfield et al.<eos>, 2018).<eos> However, early prediction of emerging variants through genomic signals remains an open challenge.<eos> Although domain-based, expert-reasoning approaches are the mainstay of our understanding, which often yield retrospective insight rather than proactive predictions.<eos> On the other hand, machine learning approaches are likely to be biased by underlying data characteristics and do not explain the biological basis of predictions.<eos> Here, we propose Strainflow, a hybrid architecture of machine learning and language modeling, along with empirical experiments to demonstrate explainable genomic signals for tracking and predicting the spread of SARS-CoV-2 across countries.<eos> Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences.<eos> Typically, in language models, word embeddings represent the latent space (dimensions) of a corpus of text (Mikolov et al.<eos>, 2013) and can capture highly nonlinear and contextual relationships.<eos> Codons (tri-nucleotides, 3-mers) translations represent a natural basis for word representations and have been utilized in the past for learning embedding models for modeling various outcomes such as mutation susceptibility (Yilmaz, 2020) and gene sequence correlations (Wu et al.<eos>, 2021).<eos> Recently, Hie et.<eos> al (Hie et al.<eos>, 2021) used machine learning along with word embedding techniques to model the semantics and grammar of amino acids corresponding to antigenic change to predict the mutations which might lead to viral escape.<eos> In this paper, we focus on the semantics of viral DNA sequences to derive Dimensions of Concern (DoCs) and demonstrate their causal potential for increasing epidemiological spread patterns across 7 countries.<eos> Our approach, Strainflow, is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance.<eos> Following the linguistic idea that the context of a word is characterized by its neighborhood, we leveraged a word2vec language model and developed “Strainflow” to learn the latent representations of codons in spike gene sequences of SARS-CoV-2.<eos> Strainflow is trained on 0.<eos>3 million high-quality genome sequences from March 1, 2020 to March 31, 2021, collected from the GISAID database (Fig 1A).<eos> To understand the latent space representation (LR) learned by Strainflow, we performed qualitative analysis using tSNE for both global and local (country) levels on a monthly basis.<eos> The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 (Fig 1B) from September, 2020 to March, 2021, along with specific geographic locations (country-level) such as India, UK, USA, and China.<eos> (A) The implementation framework of Strainflow (details described in the method section) (B) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.<eos>308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China.<eos> (C) Embeddings estimated or predicted from the Strainflow model for 0.<eos>36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021.<eos> (D) Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021.<eos> For Pakistan, we couldn’t estimate the fast sample entropy due to fewer number of strain sequences (<20).<eos> To investigate the information content in the latent dimensions or space (LD or LS) of the Strainflow model, we performed qualitative and quantitative analysis on 0.<eos>9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021.<eos> Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (Linderman et al.<eos>, 2019).<eos> We compared the 2D t-SNE plot of the world with four countries (India, UK, USA, China) from September, 2020 to June, 2021, which clearly highlights the dynamic changes in the spike genes across countries in different months (Fig 1B and Fig 1C).<eos> Additionally, the quantitative analysis of the latent space was performed by calculating the fast sample entropy (Pan et al.<eos>, 2011) for each LD.<eos> To compare the monthly entropy of the LS per country, mean entropy was calculated and normalized across the months for each country.<eos> We observed the highest entropy (information content) for India, UK, USA and China in the months of February-2021, July-2020, August-2020, and January-2021 respectively (Fig 1D).<eos> Interestingly, we observed high entropy for 4 months from August, 2020 to November, 2020 in the USA (Fig 1D).<eos> This highlights that the spike protein latent space representation learned by the Strainflow model could be used as a proxy to capture the spatiotemporal entropy or diversity in the emerging SARS-CoV-2 strains across different countries.<eos> Sequence-level embeddings were obtained from the codon embeddings and investigated for the presence of genomically meaningful characteristics.<eos> The phylogenetic tree derived from the embeddings for the United Kingdom (Fig 2A) shows two clear temporally split clusters for 2020 and 2021 sequences, which may be indicative of different strains in these time periods.<eos> The temporality of the collected sequences was found to be preserved in the two clusters, although the model was trained only on genome sequences.<eos> (A) Dendrogram for strains from the U.<eos>K.<eos>: Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021.<eos> (B) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.<eos> The phylogenetic tree with globally collected sequences (Fig 2B) demonstrates that geospatial information is also preserved in the sequence embeddings.<eos> Strains from countries like China, Australia, Mexico, and England were found in distinct clusters, while those from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India were found to be dispersed with other countries.<eos> This behavior is expected because viral strains spread from one country to another, and our samples had different collection dates for each country.<eos> Overall, Strainflow captures the temporal emergence of strains and geographic information in a country-specific manner.<eos> Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time.<eos> We attributed these changes to mutations in the spike protein of the genome sequence.<eos> This hypothesis was based on the fact that the embedding of a word (codon) represents the context it occurs in.<eos> A mutation results in a codon that is known to be found in a different context.<eos> This alters the semantic sense of the sequence, and this modification of meaning is reflected as a change in values across different latent dimensions.<eos> The changed values cause a change in the degree of disorder or the entropy of the latent space.<eos> To compare different geographical regions, the sum of sample entropy was computed for each latent dimension across all the months.<eos> This revealed that certain geospatial regions such as France and Germany (Fig 3A) and USA and Canada (Fig 3B) have similar total entropies across the latent dimensions, indicating that these regions have been accumulating similar changes.<eos> Country pairs (A) - France and Germany, (B) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.<eos> We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries.<eos> Detrended cross-correlation coefficient was calculated at different lag values, which revealed that entropy dimensions have a leading relationship with new cases (Fig 4A, 4B).<eos> This suggests that the genome sequence data in a given month can be used to predict new cases in subsequent months.<eos> A lead period of two months was chosen and Boruta algorithm was employed to assign feature importance scores to different dimensions, which revealed that dimension 32 is the most significant predictor of new cases (Fig 4C).<eos> Significant dimensions from Boruta analysis were termed Dimensions of Concern (DoCs).<eos> Random forest based regression modeling on the DoCs achieved a total R-squared of 73% on the test set.<eos> The predicted cases were found to be highly correlated with the actual cases (Table 1), which suggests that our model can indicate the directional change of cases for different countries.<eos> Further, the predicted relative change in cases between successive months was found to be correlated to the actual relative changes (Table S1), which suggests that our model can also indicate the magnitude of change that we expect to observe in the cases.<eos> (A) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA.<eos> High values are observed for a lead of 1 and 2 months.<eos> (B) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases.<eos> (C) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.<eos> Our model can be therefore used to predict the COVID-19 caseloads in several countries.<eos> Both USA (Fig 5A) and Japan (Fig 5C) show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries.<eos> Our model predicts new caseloads with a two-month lead time, which strongly predicts a spike in new cases both in USA (Fig 5B) and Japan (Fig 5D) in the months of July and August, 2021.<eos> Therefore our model may be used as an epidemiological early warning system to predict new caseloads.<eos> (A) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA.<eos> (B) Actual and predicted cases based on the entropy values of DoCs for the USA.<eos> The model predicts a rise in cases for July and August 2021.<eos> (C) Entropy of DoCs and new cases for Japan.<eos> (D) Actual and predicted cases for Japan.<eos> A spike in cases is predicted for July and August 2021.<eos> We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension (Table S2).<eos> The intuition behind this idea is that the codons with high weights in a given dimension, when mutated in the viral sequence, are likely to cause a significant change in the entropy of the associated dimensions.<eos> Therefore each DoC can be linked to codons, which can further be mapped to Variants of concern (VOCs) and Variants of Interest (VOIs) (Table S3).<eos> Despite the fact that our model cannot directly capture the SARS-CoV-2 variants, it was observed that dimension-32 captures the CTG, CGG codons (ranks 5 and 8 respectively), known to occur in the T19R variant.<eos> Similarly, dimension 3 captures three codons (ACG, CGG, CAC) that are associated with multiple variants such as K417T, L452R, and D1118H, causing increased infectivity, pathogenicity, and spread.<eos> Dimension 30 captures codons CAT and CAC associated with Δ69 and D1118H respectively which are linked to B.<eos>1.<eos>1.<eos>7 lineage.<eos> Codon weights (Table S2) of a given DoC provide an opportunity to associate with specific Variants of concern (VOCs) and Variants of Interest (VOIs); and to predict emerging SARS-CoV-2 variants and take preventive measures in advance.<eos> Distinct dimensions capture country-specific changes and may be surveilled to monitor the spread of the pandemic.<eos> This approach was back-validated with several real-world examples.<eos> For instance, dimension 32 captures the codons CGG (R) and CAC (R), which are found in B.<eos>1.<eos>429 lineage (L425R mutation).<eos> Dimension 3 captures CGG which is seen in L452R (associated with lineage B.<eos>1.<eos>617.<eos>1), which was first observed in India in December 2020 and was found to have increased infectivity and transmissibility.<eos> To investigate the potential of DoCs to track the spread of SARS-CoV-2, we used the codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs.<eos> We selected DoCs (dimension 3, 4, 12, 13, 15, 16, 25, 28, 30, 32) with high absolute weights for the codons related to the delta variants (Fig 6A).<eos> The entropy of the DoCs was contrasted with the caseloads in England (Fig 6B), India (Fig 6C), and USA (Fig 6D).<eos> Overall, the temporal tracking of the entropy in Dimensions of Concern (DoCs) may be used as a surrogate to track the spread of various SAR-CoV-2 variants.<eos> (A) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling.<eos> Line plots showing the entropies of DoCs and cases in countries, (B) England, (C) India, and (D) USA.<eos> The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.<eos> We have implemented an approach for analyzing the emerging strains based on the latent space of spike protein coding nucleotide sequences.<eos> We chose the nucleotide sequences instead of proteins in order to capture and track the variations that may not have immediate functional consequences.<eos> Our approach has two main underlying tenets: (i) long-range interactions are known to modulate the functional (Mugnai et al.<eos>, 2020) interaction between receptor binding domain and ACE2 receptors, hence may be captured in the NLP models that capture 3-mer changes and context, and (ii) latent dimensions may be differentially correlated with indicators of spread, thus providing a data-driven handle for tracking and predicting variants of concern and variants of interest.<eos> The pipeline takes advantage of temporal changes in the semantics of mutating sequences.<eos> Preservation of phylogenetic structure based upon the similarity matrix obtained using the embeddings validated that the latent dimensions capture spatio-temporal information.<eos> Analyzing the dynamic patterns and underlying correlations in the 30,000 base pair long sequence of SARS-CoV-2 (Shishir et al.<eos>, 2021) is important to highlight the mechanistic understanding of mutations.<eos> SARS-CoV-2 seems to show a particularly high frequency of recombinations arising due to the absence of a proof-reading mechanism and sequence diversity, which calls for urgency in studying its transmission pattern (Mandal et al.<eos>, 2021)(Rouchka et al.<eos>, 2020).<eos> Therefore predicting mutations in the spike protein, which binds to ACE2 receptors can help us estimate the spread of disease and the efficacy of therapeutic treatments and vaccines (Srivastava et al.<eos>, 2021)(Li et al.<eos>, 2020).<eos> Entropy is a measure of the disorder of a system.<eos> We hypothesized that mutations increase the chaotic dynamics in the latent space of spike genes.<eos> To calculate entropy, we used the accelerated versions of the Approximate Entropy and Sample Entropy algorithms, called Fast Approximate Entropy and Fast Sample Entropy (Pan et al.<eos>, 2011).<eos> Both algorithms aim to quantify how often different patterns of data are found in a time series.<eos> Fast Approximate Entropy, however, is a biased statistic and depends on the length of the series.<eos> Since we could have different counts of genome sequences collected each month, we preferred Sample Entropy, which is independent of the length of the series.<eos> Entropy values were calculated for each latent dimension in each month.<eos> Thereafter, Detrended Cross-Correlation Analysis (DCCA) (Podobnik and Stanley, 2007) was performed between the entropy dimensions and the new cases.<eos> DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series.<eos> High cross-correlation for different lead periods revealed that the entropy values in a given month could be used to predict the new cases in different countries in subsequent months.<eos> Different countries had different lead times at which the highest cross-correlation was observed between the entropy dimensions and the cases, ranging from 1-6 months.<eos> Overall, a lead time of two months was chosen to model the new cases.<eos> A similar analysis was done with daily values of entropy and new cases.<eos> Entropy was calculated in rolling windows, and cross-correlation analysis was performed between entropy and new cases at different lead periods.<eos> Although the cross-correlation values were found to be significant, the values were low and ranged between -0.<eos>1 to 0.<eos>1.<eos> Therefore, we decided to use the monthly entropy values for the modeling exercise.<eos> We also experimented with “blips” as a feature to predict the new cases.<eos> Blips are sudden changes in the values of the latent dimensions.<eos> These changes may be caused by a mutation, which changes the words (codons) in a given genome sequence.<eos> This hypothesis was validated in simulation experiments in synthetic datasets.<eos> Each dimension of the spike gene embeddings for a country was analyzed for the presence of temporal anomalies.<eos> Countries having a minimum of 20 samples in any given month were selected and the same number of records (minimum samples in any given month for that country) were sampled without replacement from each month.<eos> These records were used to define control limits of ±1 standard deviation from the mean value for each dimension, and all values in the full dataset outside those limits were categorized as ‘Blip’ points.<eos> Blip counts in each month were normalized by calculating the number of blips per sample collected in a month for each dimension (normalized blips).<eos> The embedding dimensions were then compared in terms of the total normalized blips for each country to observe the significant dimensions and dis(similarity) in trends among different countries.<eos> Cumulative counts of normalized blips were analyzed to understand the temporal accumulation of blips in each dimension.<eos> Similar to entropy, blips were found to have a leading relationship with the cases.<eos> However, regression modeling results with sample entropy were found to be better.<eos> To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data.<eos> With sample entropy, the R-squared value achieved was 73%, while with approximate entropy, the value was only 10%.<eos> Therefore the model trained on sample entropy was selected.<eos> The predictions from the model were found to be highly correlated with the actual cases, indicating that our model can be used for preemptive warning signals for the rise in cases in different countries.<eos> Further, the actual and the predicted difference in the number of cases in consecutive months was found to be correlated, which suggests that the relative change in the cases in consecutive months predicted by our model is linked to the relative change in the number of cases.<eos> Overall, we recommend that our model be used to predict dangerous trends and not the actual number of cases.<eos> Further, the mapping from latent dimensions to Variants of Concern (VOCs) and Variants of Interest (VOIs) may help us track the country-specific spread of different variants.<eos> Prediction models can be used for training the genomic sequence for predicting infection severity based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 (Wang et al.<eos>, 2020).<eos> The machine learning models can also be trained on genomic sequences for COVID-19 classification (Arslan, 2021).<eos> Although the variance explained by our model is low, however, we were able to compute the variability associated with spike protein mutations.<eos> So our method showed a potential way to estimate the new cases variability associated with spike protein mutations.<eos> Our methods can be incorporated with the epidemic projections model to better predict the epidemic trajectories.<eos> The latent dimensions may further be employed to predict the clinical consequences of emerging strains.<eos> The currently available vaccines are intended for early SARS-CoV-2 strains, but with new emerging variants, immune responses triggered by these vaccines may be weaker and short-lived.<eos> As seen in the devastating second wave of the pandemic in India, newer SARS-CoV-2 variants have acquired an increased pathogenic potential resulting in rapid clinical progression and overwhelmed health systems.<eos> Mitigating such events in the future will require stronger surveillance systems in place.<eos> Our study offers a promising solution in this direction and lays the foundation for proactive genomic surveillance of COVID-19.<eos> Our approach of codon (3-mers) embeddings does not indicate the position where the codon change may have happened in the spike gene.<eos> This is because low-dimensional embeddings do not preserve the positional encoding of words.<eos> However, we are investigating advanced approaches such as complex-valued word embeddings with positional encodings (Wang et al.<eos>, 2019) and transformer models such as BERT (Wolf et al.<eos>, 2020)(Lee et al.<eos>, 2020) to overcome our current limitation.<eos> The latter are considered expensive and data-hungry models and it will remain to be evaluated if the gain of positional information may be countered by the loss of prediction accuracy for forecasting new cases in the future.<eos> However, we believe that the availability of sequences for a wide variety of viral pathogens presents an exciting opportunity to train data-hungry models that may be able to transfer insights across pathogens and yet remain interpretable.<eos> Further, our Strainflow model is trained only on the spike gene of the viral genome, which does not represent the complete variation spectrum of the virus.<eos> To mitigate this shortcoming, we are currently developing a genome-level Strainflow model (version 2.<eos>0) for SARS-CoV-2.<eos> Another limitation of our study is the relatively small number of samples that were used to construct the supervised predictive model for case prediction.<eos> As more data becomes available in subsequent months, we can produce more confident case predictions.<eos> Nonetheless, the unsupervised embeddings and temporal cross-correlations were learned upon the full datasets and these presented clear patterns in DoCs and significant cross-correlations with caseloads.<eos> However, it is important for our models to receive at least 30 samples per month for getting reliable prediction results for the future months.<eos> This also underscores the need for a more reliable and agile approach to deposit country-level datasets on repositories such as GISAID.<eos> We make an appeal to the countries to facilitate the sharing of such data in order to be prepared for any future waves of the current pandemic and for preventing the new emergence of strains.<eos> We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance.<eos> The dataset was downloaded from GISAID EpiCoV (Shu and McCauley, 2017) (April 8, 2021 release).<eos> 0.<eos>63 million genome sequences with high nucleotide completeness, coverage, complete temporal information, and presence of less than 5% non-identified nucleotide bases (N) were downloaded.<eos> The sequences included 63 countries, including India, United Kingdom, USA, Australia, New Zealand, Germany, Russia, Italy, France, Mexico, Canada, China, Japan, Pakistan, Bangladesh, Iran, Iraq, the continent of South America, and Africa.<eos> Duplicate samples were removed, and whole genome sequences were parsed using CoV-Seq (Liu et al.<eos>, 2020) to extract nucleotide sequences corresponding to each of the 12 CDS.<eos> Accession IDs that did not cover 12 coding regions were discarded, yielding 0.<eos>31 million high-quality SARS-CoV-2 genome sequences for language modelling.<eos> We downloaded country-wise COVID-19 data for new cases from a publicly available repository maintained by Johns Hopkins University Center for Systems Science and Engineering (JHU CSSE).<eos> We downloaded around 0.<eos>6 million genome sequences submitted to GISAID from April 2021 to June 2021.<eos> We have used 0.<eos>36 million sequences to predict their latent space using the Strainflow model for the months of April 2021 to June 2021.<eos> In our Strainflow model, we have adopted a word2vec model (Mikolov et al.<eos>, 2013).<eos> A low dimensional representation for the genome sequences was learned using the word2vec model.<eos> Non-overlapping sequences of 3-mers (codons) were considered as words for training the word2vec model, implemented in Gensim (Radim Rehurek, 2010).<eos> The word2vec model was trained using the skip-gram algorithm, with a fixed window size of twenty and vector size of thirty-six.<eos> For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise.<eos> The mean was calculated by summing across all the k-mers over each dimension and then dividing it by the total number of codons present in the sequence.<eos> For selecting the vector size of the word2vec model, we calculated PIP (Pairwise Inner Product) loss (Yin and Shen, 2018).<eos> PIP loss is a metric used for calculating the dissimilarity between two word embedding matrices.<eos> For the embedding matrix of strains (E), the PIP matrix is defined as the dot product of the embedding matrix with its transpose (E.<eos> E T).<eos> The PIP loss between two embedding matrices is defined as the norm of the difference between their PIP matrices.<eos>\\n\\n\\n Various word2vec models were trained on the dataset with different vector sizes varying in multiples of three.<eos> Based on the PIP loss calculations, we found out that word embeddings with 36 dimensions showed a differential dent in the curve (change in straight line), due to which we selected this to be the dimension of the word embeddings (Fig.<eos> S1).<eos> To evaluate the phylogenetic properties based on the latent dimensions of the spike gene, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions.<eos> The pairwise distance was further used for hierarchical clustering using the ‘hclust’ function in R statistical programming language.<eos> This analysis was performed using 400 random sequences of spike genes from 16 countries.<eos> The visualization of the phylogenetic tree derived based on the latent dimensions was done using ‘iTOL’ (Letunic) software (Fig 2).<eos> To quantify the properties of latent dimensions, we have used a well-known information theory based algorithm suitable for time series datasets, called ‘Fast Sample Entropy’ (Tomčala, 2020).<eos> To compute Fast Sample Entropy, we have used the ‘FastSampEn’ function in the ‘TSEntropies’ package in R (Tomcala, 2018).<eos> Fast Sample Entropy can be computed as follows.<eos>\\n\\n\\nwhere,\\n\\n\\n Si,mm is a set of sub-sequences of length m belonging to the i-th neighborhood, and Nm is the number of these neighborhoods.<eos> In our case, ‘x’ is the latent dimension of the spike genes of the SARS-CoV-2 strains per month for a given country with default values of ‘m’ and ‘r’.<eos> Entropy was computed for each latent dimension on a monthly basis for each country.<eos> The minimum number of SARS-CoV-2 spike genes was 20.<eos> To compare across months, we used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country (Fig 1D).<eos> To compare the entropy of the latent dimensions among countries, we used the total entropy of the country for each dimension and visualized it with line graphs (Fig 3).<eos> To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used the Detrended Cross Correlation Analysis (Prass and Pumi, 2019).<eos> Here, DCCA captures the long-range cross correlation between time series (entropy of the months and caseloads for a given country).<eos> We tested both time series for stationarity using Augmented Dickey-Fuller (ADF) test (Mushtaq, 2011).<eos> The ADF test was implemented using the function ‘adf.<eos>test’ available in the ‘tseries’ (2020a) package in R.<eos> Due to the non-stationary distribution of the estimated entropies and the caseloads for a given country, we used the ‘DCCA’ R package (2020b).<eos> Cross-correlation was calculated between the entropy dimensions at time t+h and new cases at time t, where h = 0, ±1, ±2, ±3 … ±10.<eos> Dimensions of Concern (DoCs) were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis and predictive of new case count in the respective countries.<eos> Country-wise monthly total new cases data was taken at the end of each month.<eos> Total new cases data for each month was merged to monthly entropy dimensions data from March, 2020 to June, 2021.<eos> We used a regression based machine learning approach called ‘Boruta’ (Kursa and Rudnicki, 2010), a wrapper algorithm around random forest algorithm (Kursa et al.<eos>, 2010) to select the most relevant entropy dimensions for the prediction of subsequent two months’ new cases.<eos> We used the default parameters with the modification of the maximum runs as 1000.<eos> We selected the confirmed entropy dimensions as the most relevant features for the prediction of new cases.<eos> To predict the new cases in the next to next months, we used a regression based random forest model (Breiman 2001) using the most relevant DoCs based on Boruta.<eos> The model training was performed using the data from March, 2020 to April, 2021; and the fitted model was tested on data from May, 2021 to June, 2021.<eos> The regression modeling was performed using 1000 decision trees using the ‘randomForest’ package in R (A.<eos> Liaw et al.<eos>, 2002).<eos> To find the top codons associated with the DoCs, we extracted the absolute weights of each codon for a given DoC.<eos> The top 10 codons having the highest absolute weights (contribution) were identified corresponding to each of the DoCs to compare our DoCs with SARS-CoV-2 variants.<eos> We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (CDC, 2021); and a list of codons associated with VOIs and VOCs was curated (Table S4) (Lopez-Rincon et al.<eos>; paola, 2021; Peacock et al.<eos>, 2021; Srivastava et al.<eos>, 2021).<eos> The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant (CDC, 2021) This work was supported by the Delhi Cluster-Delhi Research Implementation and Innovation (DRIIV) Project supported by the Principal Scientific Advisor Office, Prn.<eos>SA/Delhi/ Hub/2018(C) and the Center of Excellence in Healthcare supported by Delhi Knowledge Development Foundation (DKDF) at IIIT-Delhi.<eos> We also thank Dr.<eos> Chitra Pattabiraman (NIMHANS) for her valuable inputs on viral sequences, and Harleen Kaur, Nishkarsh Saxena, and Anjali for their contribution to the dashboard visualizations.<eos> In this revised version, we have implemented fast tSNE for qualitative inspection of the latent dimensions (LD) of 0.<eos>9 million SARS-CoV-2 spike genes.<eos> Quantitative analyses of the LDs were performed using the fast sample entropy method.<eos> Also, we have used fast sample entropy instead of 'blips' to model COVID-19 caseloads.<eos> View the discussion thread.<eos> Supplementary Material  Thank you for your interest in spreading the word about bioRxiv.<eos> NOTE: Your email address is requested solely to identify you as the sender of this article.<eos>\""
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 45
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "sentences=Article_data.split('<eos>')\n",
        "sentences"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YjJhdQhvXYrV",
        "outputId": "112524c5-22f7-45ab-abfc-b7d402b152ab"
      },
      "execution_count": 46,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity.',\n",
              " ' Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge.',\n",
              " ' Here, we developed Strainflow, to learn the latent dimensions of 0.',\n",
              " '9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2.',\n",
              " ' In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions).',\n",
              " ' We discovered that codon-level changes lead to a change in the entropy of the latent dimensions.',\n",
              " ' We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries.',\n",
              " ' The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.',\n",
              " ' Highlights We developed a genomic surveillance model for SARS-CoV-2 genome sequences, Strainflow, where sequences were treated as documents with words (codons) to learn the codon context of 0.',\n",
              " '9 million spike genes using the skip-gram algorithm.',\n",
              " ' Time series analysis of the information content (Entropy) of the latent dimensions learned by Strainflow shows a leading relationship with the monthly COVID-19 cases for seven countries (e.',\n",
              " 'g.',\n",
              " ', USA, Japan, India, and others).',\n",
              " ' Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads.',\n",
              " ' The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants.',\n",
              " ' COVID-19 is reported to have claimed 4.',\n",
              " '37 million lives as of Aug 17, 2021 (WHO, 2021).',\n",
              " ' A large number of these deaths are attributed to unexpected surges in infections caused by new strains of SARS-CoV-2, prompting international health organizations such as the CDC and WHO to declare these as variants of concerns (Khan et al.',\n",
              " ', 2020).',\n",
              " ' Such emergence of new variants can seriously undermine the efficacy of global vaccination programs, the need for future booster doses, or cause multiple reinfection waves as new strains escape previously developed antibodies.',\n",
              " ' Several initiatives have focused on providing high-quality tracking information for the strains and lineages as these emerge (Hadfield et al.',\n",
              " ', 2018).',\n",
              " ' However, early prediction of emerging variants through genomic signals remains an open challenge.',\n",
              " ' Although domain-based, expert-reasoning approaches are the mainstay of our understanding, which often yield retrospective insight rather than proactive predictions.',\n",
              " ' On the other hand, machine learning approaches are likely to be biased by underlying data characteristics and do not explain the biological basis of predictions.',\n",
              " ' Here, we propose Strainflow, a hybrid architecture of machine learning and language modeling, along with empirical experiments to demonstrate explainable genomic signals for tracking and predicting the spread of SARS-CoV-2 across countries.',\n",
              " ' Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences.',\n",
              " ' Typically, in language models, word embeddings represent the latent space (dimensions) of a corpus of text (Mikolov et al.',\n",
              " ', 2013) and can capture highly nonlinear and contextual relationships.',\n",
              " ' Codons (tri-nucleotides, 3-mers) translations represent a natural basis for word representations and have been utilized in the past for learning embedding models for modeling various outcomes such as mutation susceptibility (Yilmaz, 2020) and gene sequence correlations (Wu et al.',\n",
              " ', 2021).',\n",
              " ' Recently, Hie et.',\n",
              " ' al (Hie et al.',\n",
              " ', 2021) used machine learning along with word embedding techniques to model the semantics and grammar of amino acids corresponding to antigenic change to predict the mutations which might lead to viral escape.',\n",
              " ' In this paper, we focus on the semantics of viral DNA sequences to derive Dimensions of Concern (DoCs) and demonstrate their causal potential for increasing epidemiological spread patterns across 7 countries.',\n",
              " ' Our approach, Strainflow, is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance.',\n",
              " ' Following the linguistic idea that the context of a word is characterized by its neighborhood, we leveraged a word2vec language model and developed “Strainflow” to learn the latent representations of codons in spike gene sequences of SARS-CoV-2.',\n",
              " ' Strainflow is trained on 0.',\n",
              " '3 million high-quality genome sequences from March 1, 2020 to March 31, 2021, collected from the GISAID database (Fig 1A).',\n",
              " ' To understand the latent space representation (LR) learned by Strainflow, we performed qualitative analysis using tSNE for both global and local (country) levels on a monthly basis.',\n",
              " ' The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 (Fig 1B) from September, 2020 to March, 2021, along with specific geographic locations (country-level) such as India, UK, USA, and China.',\n",
              " ' (A) The implementation framework of Strainflow (details described in the method section) (B) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0.',\n",
              " '308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China.',\n",
              " ' (C) Embeddings estimated or predicted from the Strainflow model for 0.',\n",
              " '36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021.',\n",
              " ' (D) Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021.',\n",
              " ' For Pakistan, we couldn’t estimate the fast sample entropy due to fewer number of strain sequences (<20).',\n",
              " ' To investigate the information content in the latent dimensions or space (LD or LS) of the Strainflow model, we performed qualitative and quantitative analysis on 0.',\n",
              " '9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021.',\n",
              " ' Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (Linderman et al.',\n",
              " ', 2019).',\n",
              " ' We compared the 2D t-SNE plot of the world with four countries (India, UK, USA, China) from September, 2020 to June, 2021, which clearly highlights the dynamic changes in the spike genes across countries in different months (Fig 1B and Fig 1C).',\n",
              " ' Additionally, the quantitative analysis of the latent space was performed by calculating the fast sample entropy (Pan et al.',\n",
              " ', 2011) for each LD.',\n",
              " ' To compare the monthly entropy of the LS per country, mean entropy was calculated and normalized across the months for each country.',\n",
              " ' We observed the highest entropy (information content) for India, UK, USA and China in the months of February-2021, July-2020, August-2020, and January-2021 respectively (Fig 1D).',\n",
              " ' Interestingly, we observed high entropy for 4 months from August, 2020 to November, 2020 in the USA (Fig 1D).',\n",
              " ' This highlights that the spike protein latent space representation learned by the Strainflow model could be used as a proxy to capture the spatiotemporal entropy or diversity in the emerging SARS-CoV-2 strains across different countries.',\n",
              " ' Sequence-level embeddings were obtained from the codon embeddings and investigated for the presence of genomically meaningful characteristics.',\n",
              " ' The phylogenetic tree derived from the embeddings for the United Kingdom (Fig 2A) shows two clear temporally split clusters for 2020 and 2021 sequences, which may be indicative of different strains in these time periods.',\n",
              " ' The temporality of the collected sequences was found to be preserved in the two clusters, although the model was trained only on genome sequences.',\n",
              " ' (A) Dendrogram for strains from the U.',\n",
              " 'K.',\n",
              " ': Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021.',\n",
              " ' (B) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.',\n",
              " ' The phylogenetic tree with globally collected sequences (Fig 2B) demonstrates that geospatial information is also preserved in the sequence embeddings.',\n",
              " ' Strains from countries like China, Australia, Mexico, and England were found in distinct clusters, while those from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India were found to be dispersed with other countries.',\n",
              " ' This behavior is expected because viral strains spread from one country to another, and our samples had different collection dates for each country.',\n",
              " ' Overall, Strainflow captures the temporal emergence of strains and geographic information in a country-specific manner.',\n",
              " ' Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time.',\n",
              " ' We attributed these changes to mutations in the spike protein of the genome sequence.',\n",
              " ' This hypothesis was based on the fact that the embedding of a word (codon) represents the context it occurs in.',\n",
              " ' A mutation results in a codon that is known to be found in a different context.',\n",
              " ' This alters the semantic sense of the sequence, and this modification of meaning is reflected as a change in values across different latent dimensions.',\n",
              " ' The changed values cause a change in the degree of disorder or the entropy of the latent space.',\n",
              " ' To compare different geographical regions, the sum of sample entropy was computed for each latent dimension across all the months.',\n",
              " ' This revealed that certain geospatial regions such as France and Germany (Fig 3A) and USA and Canada (Fig 3B) have similar total entropies across the latent dimensions, indicating that these regions have been accumulating similar changes.',\n",
              " ' Country pairs (A) - France and Germany, (B) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.',\n",
              " ' We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries.',\n",
              " ' Detrended cross-correlation coefficient was calculated at different lag values, which revealed that entropy dimensions have a leading relationship with new cases (Fig 4A, 4B).',\n",
              " ' This suggests that the genome sequence data in a given month can be used to predict new cases in subsequent months.',\n",
              " ' A lead period of two months was chosen and Boruta algorithm was employed to assign feature importance scores to different dimensions, which revealed that dimension 32 is the most significant predictor of new cases (Fig 4C).',\n",
              " ' Significant dimensions from Boruta analysis were termed Dimensions of Concern (DoCs).',\n",
              " ' Random forest based regression modeling on the DoCs achieved a total R-squared of 73% on the test set.',\n",
              " ' The predicted cases were found to be highly correlated with the actual cases (Table 1), which suggests that our model can indicate the directional change of cases for different countries.',\n",
              " ' Further, the predicted relative change in cases between successive months was found to be correlated to the actual relative changes (Table S1), which suggests that our model can also indicate the magnitude of change that we expect to observe in the cases.',\n",
              " ' (A) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA.',\n",
              " ' High values are observed for a lead of 1 and 2 months.',\n",
              " ' (B) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases.',\n",
              " ' (C) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.',\n",
              " ' Our model can be therefore used to predict the COVID-19 caseloads in several countries.',\n",
              " ' Both USA (Fig 5A) and Japan (Fig 5C) show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries.',\n",
              " ' Our model predicts new caseloads with a two-month lead time, which strongly predicts a spike in new cases both in USA (Fig 5B) and Japan (Fig 5D) in the months of July and August, 2021.',\n",
              " ' Therefore our model may be used as an epidemiological early warning system to predict new caseloads.',\n",
              " ' (A) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA.',\n",
              " ' (B) Actual and predicted cases based on the entropy values of DoCs for the USA.',\n",
              " ' The model predicts a rise in cases for July and August 2021.',\n",
              " ' (C) Entropy of DoCs and new cases for Japan.',\n",
              " ' (D) Actual and predicted cases for Japan.',\n",
              " ' A spike in cases is predicted for July and August 2021.',\n",
              " ' We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension (Table S2).',\n",
              " ' The intuition behind this idea is that the codons with high weights in a given dimension, when mutated in the viral sequence, are likely to cause a significant change in the entropy of the associated dimensions.',\n",
              " ' Therefore each DoC can be linked to codons, which can further be mapped to Variants of concern (VOCs) and Variants of Interest (VOIs) (Table S3).',\n",
              " ' Despite the fact that our model cannot directly capture the SARS-CoV-2 variants, it was observed that dimension-32 captures the CTG, CGG codons (ranks 5 and 8 respectively), known to occur in the T19R variant.',\n",
              " ' Similarly, dimension 3 captures three codons (ACG, CGG, CAC) that are associated with multiple variants such as K417T, L452R, and D1118H, causing increased infectivity, pathogenicity, and spread.',\n",
              " ' Dimension 30 captures codons CAT and CAC associated with Δ69 and D1118H respectively which are linked to B.',\n",
              " '1.',\n",
              " '1.',\n",
              " '7 lineage.',\n",
              " ' Codon weights (Table S2) of a given DoC provide an opportunity to associate with specific Variants of concern (VOCs) and Variants of Interest (VOIs); and to predict emerging SARS-CoV-2 variants and take preventive measures in advance.',\n",
              " ' Distinct dimensions capture country-specific changes and may be surveilled to monitor the spread of the pandemic.',\n",
              " ' This approach was back-validated with several real-world examples.',\n",
              " ' For instance, dimension 32 captures the codons CGG (R) and CAC (R), which are found in B.',\n",
              " '1.',\n",
              " '429 lineage (L425R mutation).',\n",
              " ' Dimension 3 captures CGG which is seen in L452R (associated with lineage B.',\n",
              " '1.',\n",
              " '617.',\n",
              " '1), which was first observed in India in December 2020 and was found to have increased infectivity and transmissibility.',\n",
              " ' To investigate the potential of DoCs to track the spread of SARS-CoV-2, we used the codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs.',\n",
              " ' We selected DoCs (dimension 3, 4, 12, 13, 15, 16, 25, 28, 30, 32) with high absolute weights for the codons related to the delta variants (Fig 6A).',\n",
              " ' The entropy of the DoCs was contrasted with the caseloads in England (Fig 6B), India (Fig 6C), and USA (Fig 6D).',\n",
              " ' Overall, the temporal tracking of the entropy in Dimensions of Concern (DoCs) may be used as a surrogate to track the spread of various SAR-CoV-2 variants.',\n",
              " ' (A) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling.',\n",
              " ' Line plots showing the entropies of DoCs and cases in countries, (B) England, (C) India, and (D) USA.',\n",
              " ' The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.',\n",
              " ' We have implemented an approach for analyzing the emerging strains based on the latent space of spike protein coding nucleotide sequences.',\n",
              " ' We chose the nucleotide sequences instead of proteins in order to capture and track the variations that may not have immediate functional consequences.',\n",
              " ' Our approach has two main underlying tenets: (i) long-range interactions are known to modulate the functional (Mugnai et al.',\n",
              " ', 2020) interaction between receptor binding domain and ACE2 receptors, hence may be captured in the NLP models that capture 3-mer changes and context, and (ii) latent dimensions may be differentially correlated with indicators of spread, thus providing a data-driven handle for tracking and predicting variants of concern and variants of interest.',\n",
              " ' The pipeline takes advantage of temporal changes in the semantics of mutating sequences.',\n",
              " ' Preservation of phylogenetic structure based upon the similarity matrix obtained using the embeddings validated that the latent dimensions capture spatio-temporal information.',\n",
              " ' Analyzing the dynamic patterns and underlying correlations in the 30,000 base pair long sequence of SARS-CoV-2 (Shishir et al.',\n",
              " ', 2021) is important to highlight the mechanistic understanding of mutations.',\n",
              " ' SARS-CoV-2 seems to show a particularly high frequency of recombinations arising due to the absence of a proof-reading mechanism and sequence diversity, which calls for urgency in studying its transmission pattern (Mandal et al.',\n",
              " ', 2021)(Rouchka et al.',\n",
              " ', 2020).',\n",
              " ' Therefore predicting mutations in the spike protein, which binds to ACE2 receptors can help us estimate the spread of disease and the efficacy of therapeutic treatments and vaccines (Srivastava et al.',\n",
              " ', 2021)(Li et al.',\n",
              " ', 2020).',\n",
              " ' Entropy is a measure of the disorder of a system.',\n",
              " ' We hypothesized that mutations increase the chaotic dynamics in the latent space of spike genes.',\n",
              " ' To calculate entropy, we used the accelerated versions of the Approximate Entropy and Sample Entropy algorithms, called Fast Approximate Entropy and Fast Sample Entropy (Pan et al.',\n",
              " ', 2011).',\n",
              " ' Both algorithms aim to quantify how often different patterns of data are found in a time series.',\n",
              " ' Fast Approximate Entropy, however, is a biased statistic and depends on the length of the series.',\n",
              " ' Since we could have different counts of genome sequences collected each month, we preferred Sample Entropy, which is independent of the length of the series.',\n",
              " ' Entropy values were calculated for each latent dimension in each month.',\n",
              " ' Thereafter, Detrended Cross-Correlation Analysis (DCCA) (Podobnik and Stanley, 2007) was performed between the entropy dimensions and the new cases.',\n",
              " ' DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series.',\n",
              " ' High cross-correlation for different lead periods revealed that the entropy values in a given month could be used to predict the new cases in different countries in subsequent months.',\n",
              " ' Different countries had different lead times at which the highest cross-correlation was observed between the entropy dimensions and the cases, ranging from 1-6 months.',\n",
              " ' Overall, a lead time of two months was chosen to model the new cases.',\n",
              " ' A similar analysis was done with daily values of entropy and new cases.',\n",
              " ' Entropy was calculated in rolling windows, and cross-correlation analysis was performed between entropy and new cases at different lead periods.',\n",
              " ' Although the cross-correlation values were found to be significant, the values were low and ranged between -0.',\n",
              " '1 to 0.',\n",
              " '1.',\n",
              " ' Therefore, we decided to use the monthly entropy values for the modeling exercise.',\n",
              " ' We also experimented with “blips” as a feature to predict the new cases.',\n",
              " ' Blips are sudden changes in the values of the latent dimensions.',\n",
              " ' These changes may be caused by a mutation, which changes the words (codons) in a given genome sequence.',\n",
              " ' This hypothesis was validated in simulation experiments in synthetic datasets.',\n",
              " ' Each dimension of the spike gene embeddings for a country was analyzed for the presence of temporal anomalies.',\n",
              " ' Countries having a minimum of 20 samples in any given month were selected and the same number of records (minimum samples in any given month for that country) were sampled without replacement from each month.',\n",
              " ' These records were used to define control limits of ±1 standard deviation from the mean value for each dimension, and all values in the full dataset outside those limits were categorized as ‘Blip’ points.',\n",
              " ' Blip counts in each month were normalized by calculating the number of blips per sample collected in a month for each dimension (normalized blips).',\n",
              " ' The embedding dimensions were then compared in terms of the total normalized blips for each country to observe the significant dimensions and dis(similarity) in trends among different countries.',\n",
              " ' Cumulative counts of normalized blips were analyzed to understand the temporal accumulation of blips in each dimension.',\n",
              " ' Similar to entropy, blips were found to have a leading relationship with the cases.',\n",
              " ' However, regression modeling results with sample entropy were found to be better.',\n",
              " ' To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data.',\n",
              " ' With sample entropy, the R-squared value achieved was 73%, while with approximate entropy, the value was only 10%.',\n",
              " ' Therefore the model trained on sample entropy was selected.',\n",
              " ' The predictions from the model were found to be highly correlated with the actual cases, indicating that our model can be used for preemptive warning signals for the rise in cases in different countries.',\n",
              " ' Further, the actual and the predicted difference in the number of cases in consecutive months was found to be correlated, which suggests that the relative change in the cases in consecutive months predicted by our model is linked to the relative change in the number of cases.',\n",
              " ' Overall, we recommend that our model be used to predict dangerous trends and not the actual number of cases.',\n",
              " ' Further, the mapping from latent dimensions to Variants of Concern (VOCs) and Variants of Interest (VOIs) may help us track the country-specific spread of different variants.',\n",
              " ' Prediction models can be used for training the genomic sequence for predicting infection severity based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 (Wang et al.',\n",
              " ', 2020).',\n",
              " ' The machine learning models can also be trained on genomic sequences for COVID-19 classification (Arslan, 2021).',\n",
              " ' Although the variance explained by our model is low, however, we were able to compute the variability associated with spike protein mutations.',\n",
              " ' So our method showed a potential way to estimate the new cases variability associated with spike protein mutations.',\n",
              " ' Our methods can be incorporated with the epidemic projections model to better predict the epidemic trajectories.',\n",
              " ' The latent dimensions may further be employed to predict the clinical consequences of emerging strains.',\n",
              " ' The currently available vaccines are intended for early SARS-CoV-2 strains, but with new emerging variants, immune responses triggered by these vaccines may be weaker and short-lived.',\n",
              " ' As seen in the devastating second wave of the pandemic in India, newer SARS-CoV-2 variants have acquired an increased pathogenic potential resulting in rapid clinical progression and overwhelmed health systems.',\n",
              " ' Mitigating such events in the future will require stronger surveillance systems in place.',\n",
              " ' Our study offers a promising solution in this direction and lays the foundation for proactive genomic surveillance of COVID-19.',\n",
              " ' Our approach of codon (3-mers) embeddings does not indicate the position where the codon change may have happened in the spike gene.',\n",
              " ' This is because low-dimensional embeddings do not preserve the positional encoding of words.',\n",
              " ' However, we are investigating advanced approaches such as complex-valued word embeddings with positional encodings (Wang et al.',\n",
              " ', 2019) and transformer models such as BERT (Wolf et al.',\n",
              " ', 2020)(Lee et al.',\n",
              " ', 2020) to overcome our current limitation.',\n",
              " ' The latter are considered expensive and data-hungry models and it will remain to be evaluated if the gain of positional information may be countered by the loss of prediction accuracy for forecasting new cases in the future.',\n",
              " ' However, we believe that the availability of sequences for a wide variety of viral pathogens presents an exciting opportunity to train data-hungry models that may be able to transfer insights across pathogens and yet remain interpretable.',\n",
              " ' Further, our Strainflow model is trained only on the spike gene of the viral genome, which does not represent the complete variation spectrum of the virus.',\n",
              " ' To mitigate this shortcoming, we are currently developing a genome-level Strainflow model (version 2.',\n",
              " '0) for SARS-CoV-2.',\n",
              " ' Another limitation of our study is the relatively small number of samples that were used to construct the supervised predictive model for case prediction.',\n",
              " ' As more data becomes available in subsequent months, we can produce more confident case predictions.',\n",
              " ' Nonetheless, the unsupervised embeddings and temporal cross-correlations were learned upon the full datasets and these presented clear patterns in DoCs and significant cross-correlations with caseloads.',\n",
              " ' However, it is important for our models to receive at least 30 samples per month for getting reliable prediction results for the future months.',\n",
              " ' This also underscores the need for a more reliable and agile approach to deposit country-level datasets on repositories such as GISAID.',\n",
              " ' We make an appeal to the countries to facilitate the sharing of such data in order to be prepared for any future waves of the current pandemic and for preventing the new emergence of strains.',\n",
              " ' We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance.',\n",
              " ' The dataset was downloaded from GISAID EpiCoV (Shu and McCauley, 2017) (April 8, 2021 release).',\n",
              " ' 0.',\n",
              " '63 million genome sequences with high nucleotide completeness, coverage, complete temporal information, and presence of less than 5% non-identified nucleotide bases (N) were downloaded.',\n",
              " ' The sequences included 63 countries, including India, United Kingdom, USA, Australia, New Zealand, Germany, Russia, Italy, France, Mexico, Canada, China, Japan, Pakistan, Bangladesh, Iran, Iraq, the continent of South America, and Africa.',\n",
              " ' Duplicate samples were removed, and whole genome sequences were parsed using CoV-Seq (Liu et al.',\n",
              " ', 2020) to extract nucleotide sequences corresponding to each of the 12 CDS.',\n",
              " ' Accession IDs that did not cover 12 coding regions were discarded, yielding 0.',\n",
              " '31 million high-quality SARS-CoV-2 genome sequences for language modelling.',\n",
              " ' We downloaded country-wise COVID-19 data for new cases from a publicly available repository maintained by Johns Hopkins University Center for Systems Science and Engineering (JHU CSSE).',\n",
              " ' We downloaded around 0.',\n",
              " '6 million genome sequences submitted to GISAID from April 2021 to June 2021.',\n",
              " ' We have used 0.',\n",
              " '36 million sequences to predict their latent space using the Strainflow model for the months of April 2021 to June 2021.',\n",
              " ' In our Strainflow model, we have adopted a word2vec model (Mikolov et al.',\n",
              " ', 2013).',\n",
              " ' A low dimensional representation for the genome sequences was learned using the word2vec model.',\n",
              " ' Non-overlapping sequences of 3-mers (codons) were considered as words for training the word2vec model, implemented in Gensim (Radim Rehurek, 2010).',\n",
              " ' The word2vec model was trained using the skip-gram algorithm, with a fixed window size of twenty and vector size of thirty-six.',\n",
              " ' For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise.',\n",
              " ' The mean was calculated by summing across all the k-mers over each dimension and then dividing it by the total number of codons present in the sequence.',\n",
              " ' For selecting the vector size of the word2vec model, we calculated PIP (Pairwise Inner Product) loss (Yin and Shen, 2018).',\n",
              " ' PIP loss is a metric used for calculating the dissimilarity between two word embedding matrices.',\n",
              " ' For the embedding matrix of strains (E), the PIP matrix is defined as the dot product of the embedding matrix with its transpose (E.',\n",
              " ' E T).',\n",
              " ' The PIP loss between two embedding matrices is defined as the norm of the difference between their PIP matrices.',\n",
              " '\\n\\n\\n Various word2vec models were trained on the dataset with different vector sizes varying in multiples of three.',\n",
              " ' Based on the PIP loss calculations, we found out that word embeddings with 36 dimensions showed a differential dent in the curve (change in straight line), due to which we selected this to be the dimension of the word embeddings (Fig.',\n",
              " ' S1).',\n",
              " ' To evaluate the phylogenetic properties based on the latent dimensions of the spike gene, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions.',\n",
              " ' The pairwise distance was further used for hierarchical clustering using the ‘hclust’ function in R statistical programming language.',\n",
              " ' This analysis was performed using 400 random sequences of spike genes from 16 countries.',\n",
              " ' The visualization of the phylogenetic tree derived based on the latent dimensions was done using ‘iTOL’ (Letunic) software (Fig 2).',\n",
              " ' To quantify the properties of latent dimensions, we have used a well-known information theory based algorithm suitable for time series datasets, called ‘Fast Sample Entropy’ (Tomčala, 2020).',\n",
              " ' To compute Fast Sample Entropy, we have used the ‘FastSampEn’ function in the ‘TSEntropies’ package in R (Tomcala, 2018).',\n",
              " ' Fast Sample Entropy can be computed as follows.',\n",
              " '\\n\\n\\nwhere,\\n\\n\\n Si,mm is a set of sub-sequences of length m belonging to the i-th neighborhood, and Nm is the number of these neighborhoods.',\n",
              " ' In our case, ‘x’ is the latent dimension of the spike genes of the SARS-CoV-2 strains per month for a given country with default values of ‘m’ and ‘r’.',\n",
              " ' Entropy was computed for each latent dimension on a monthly basis for each country.',\n",
              " ' The minimum number of SARS-CoV-2 spike genes was 20.',\n",
              " ' To compare across months, we used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country (Fig 1D).',\n",
              " ' To compare the entropy of the latent dimensions among countries, we used the total entropy of the country for each dimension and visualized it with line graphs (Fig 3).',\n",
              " ' To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used the Detrended Cross Correlation Analysis (Prass and Pumi, 2019).',\n",
              " ' Here, DCCA captures the long-range cross correlation between time series (entropy of the months and caseloads for a given country).',\n",
              " ' We tested both time series for stationarity using Augmented Dickey-Fuller (ADF) test (Mushtaq, 2011).',\n",
              " ' The ADF test was implemented using the function ‘adf.',\n",
              " 'test’ available in the ‘tseries’ (2020a) package in R.',\n",
              " ' Due to the non-stationary distribution of the estimated entropies and the caseloads for a given country, we used the ‘DCCA’ R package (2020b).',\n",
              " ' Cross-correlation was calculated between the entropy dimensions at time t+h and new cases at time t, where h = 0, ±1, ±2, ±3 … ±10.',\n",
              " ' Dimensions of Concern (DoCs) were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis and predictive of new case count in the respective countries.',\n",
              " ' Country-wise monthly total new cases data was taken at the end of each month.',\n",
              " ' Total new cases data for each month was merged to monthly entropy dimensions data from March, 2020 to June, 2021.',\n",
              " ' We used a regression based machine learning approach called ‘Boruta’ (Kursa and Rudnicki, 2010), a wrapper algorithm around random forest algorithm (Kursa et al.',\n",
              " ', 2010) to select the most relevant entropy dimensions for the prediction of subsequent two months’ new cases.',\n",
              " ' We used the default parameters with the modification of the maximum runs as 1000.',\n",
              " ' We selected the confirmed entropy dimensions as the most relevant features for the prediction of new cases.',\n",
              " ' To predict the new cases in the next to next months, we used a regression based random forest model (Breiman 2001) using the most relevant DoCs based on Boruta.',\n",
              " ' The model training was performed using the data from March, 2020 to April, 2021; and the fitted model was tested on data from May, 2021 to June, 2021.',\n",
              " ' The regression modeling was performed using 1000 decision trees using the ‘randomForest’ package in R (A.',\n",
              " ' Liaw et al.',\n",
              " ', 2002).',\n",
              " ' To find the top codons associated with the DoCs, we extracted the absolute weights of each codon for a given DoC.',\n",
              " ' The top 10 codons having the highest absolute weights (contribution) were identified corresponding to each of the DoCs to compare our DoCs with SARS-CoV-2 variants.',\n",
              " ' We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (CDC, 2021); and a list of codons associated with VOIs and VOCs was curated (Table S4) (Lopez-Rincon et al.',\n",
              " '; paola, 2021; Peacock et al.',\n",
              " ', 2021; Srivastava et al.',\n",
              " ', 2021).',\n",
              " ' The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant (CDC, 2021) This work was supported by the Delhi Cluster-Delhi Research Implementation and Innovation (DRIIV) Project supported by the Principal Scientific Advisor Office, Prn.',\n",
              " 'SA/Delhi/ Hub/2018(C) and the Center of Excellence in Healthcare supported by Delhi Knowledge Development Foundation (DKDF) at IIIT-Delhi.',\n",
              " ' We also thank Dr.',\n",
              " ' Chitra Pattabiraman (NIMHANS) for her valuable inputs on viral sequences, and Harleen Kaur, Nishkarsh Saxena, and Anjali for their contribution to the dashboard visualizations.',\n",
              " ' In this revised version, we have implemented fast tSNE for qualitative inspection of the latent dimensions (LD) of 0.',\n",
              " '9 million SARS-CoV-2 spike genes.',\n",
              " ' Quantitative analyses of the LDs were performed using the fast sample entropy method.',\n",
              " \" Also, we have used fast sample entropy instead of 'blips' to model COVID-19 caseloads.\",\n",
              " ' View the discussion thread.',\n",
              " ' Supplementary Material  Thank you for your interest in spreading the word about bioRxiv.',\n",
              " ' NOTE: Your email address is requested solely to identify you as the sender of this article.',\n",
              " '']"
            ]
          },
          "metadata": {},
          "execution_count": 46
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "sentences[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 53
        },
        "id": "TMSpWry4XYuO",
        "outputId": "b14f6d15-d096-4ac6-b6b4-7a800271fdb8"
      },
      "execution_count": 47,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity.'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 47
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Converging the sentences into chunks of 500 words"
      ],
      "metadata": {
        "id": "-qYJZ6Xggjw1"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "max_chunk = 500\n",
        "current_chunk = 0 \n",
        "chunks = []\n",
        "for sentence in sentences:\n",
        "    if len(chunks) == current_chunk + 1: \n",
        "        if len(chunks[current_chunk]) + len(sentence.split(' ')) <= max_chunk:\n",
        "            chunks[current_chunk].extend(sentence.split(' '))\n",
        "        else:\n",
        "            current_chunk += 1\n",
        "            chunks.append(sentence.split(' '))\n",
        "    else:\n",
        "        print(current_chunk)\n",
        "        chunks.append(sentence.split(' '))\n",
        "\n",
        "for chunk_id in range(len(chunks)):\n",
        "    chunks[chunk_id] = ' '.join(chunks[chunk_id])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CXrnp4Y9W0lk",
        "outputId": "8cdf3e3f-8609-476d-afd7-cc3c5ec449a5"
      },
      "execution_count": 48,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "0\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "chunks"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "v5EgZydtg9js",
        "outputId": "24fa908c-c56e-4659-ffbc-3317027e1b91"
      },
      "execution_count": 49,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity.  Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge.  Here, we developed Strainflow, to learn the latent dimensions of 0. 9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2.  In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions).  We discovered that codon-level changes lead to a change in the entropy of the latent dimensions.  We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries.  The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.  Highlights We developed a genomic surveillance model for SARS-CoV-2 genome sequences, Strainflow, where sequences were treated as documents with words (codons) to learn the codon context of 0. 9 million spike genes using the skip-gram algorithm.  Time series analysis of the information content (Entropy) of the latent dimensions learned by Strainflow shows a leading relationship with the monthly COVID-19 cases for seven countries (e. g. , USA, Japan, India, and others).  Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads.  The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants.  COVID-19 is reported to have claimed 4. 37 million lives as of Aug 17, 2021 (WHO, 2021).  A large number of these deaths are attributed to unexpected surges in infections caused by new strains of SARS-CoV-2, prompting international health organizations such as the CDC and WHO to declare these as variants of concerns (Khan et al. , 2020).  Such emergence of new variants can seriously undermine the efficacy of global vaccination programs, the need for future booster doses, or cause multiple reinfection waves as new strains escape previously developed antibodies.  Several initiatives have focused on providing high-quality tracking information for the strains and lineages as these emerge (Hadfield et al. , 2018).  However, early prediction of emerging variants through genomic signals remains an open challenge.  Although domain-based, expert-reasoning approaches are the mainstay of our understanding, which often yield retrospective insight rather than proactive predictions.',\n",
              " ' On the other hand, machine learning approaches are likely to be biased by underlying data characteristics and do not explain the biological basis of predictions.  Here, we propose Strainflow, a hybrid architecture of machine learning and language modeling, along with empirical experiments to demonstrate explainable genomic signals for tracking and predicting the spread of SARS-CoV-2 across countries.  Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences.  Typically, in language models, word embeddings represent the latent space (dimensions) of a corpus of text (Mikolov et al. , 2013) and can capture highly nonlinear and contextual relationships.  Codons (tri-nucleotides, 3-mers) translations represent a natural basis for word representations and have been utilized in the past for learning embedding models for modeling various outcomes such as mutation susceptibility (Yilmaz, 2020) and gene sequence correlations (Wu et al. , 2021).  Recently, Hie et.  al (Hie et al. , 2021) used machine learning along with word embedding techniques to model the semantics and grammar of amino acids corresponding to antigenic change to predict the mutations which might lead to viral escape.  In this paper, we focus on the semantics of viral DNA sequences to derive Dimensions of Concern (DoCs) and demonstrate their causal potential for increasing epidemiological spread patterns across 7 countries.  Our approach, Strainflow, is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance.  Following the linguistic idea that the context of a word is characterized by its neighborhood, we leveraged a word2vec language model and developed “Strainflow” to learn the latent representations of codons in spike gene sequences of SARS-CoV-2.  Strainflow is trained on 0. 3 million high-quality genome sequences from March 1, 2020 to March 31, 2021, collected from the GISAID database (Fig 1A).  To understand the latent space representation (LR) learned by Strainflow, we performed qualitative analysis using tSNE for both global and local (country) levels on a monthly basis.  The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 (Fig 1B) from September, 2020 to March, 2021, along with specific geographic locations (country-level) such as India, UK, USA, and China.  (A) The implementation framework of Strainflow (details described in the method section) (B) tSNE plot showing distinct spatio-temporal relationship based on the latent space learned from the spike gene of 0. 308 million SARS-CoV-2 genomes collected till 31 March 2021 (world), India, UK, USA, and China.  (C) Embeddings estimated or predicted from the Strainflow model for 0. 36 million SARS-CoV-2 spike genes from the month of April, 2021 to June, 2021.  (D) Heatmap showing the normalized entropy for 20 countries from March, 2020 to June, 2021.  For Pakistan, we couldn’t estimate the fast sample entropy due to fewer number of strain sequences (<20).',\n",
              " ' To investigate the information content in the latent dimensions or space (LD or LS) of the Strainflow model, we performed qualitative and quantitative analysis on 0. 9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021.  Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (Linderman et al. , 2019).  We compared the 2D t-SNE plot of the world with four countries (India, UK, USA, China) from September, 2020 to June, 2021, which clearly highlights the dynamic changes in the spike genes across countries in different months (Fig 1B and Fig 1C).  Additionally, the quantitative analysis of the latent space was performed by calculating the fast sample entropy (Pan et al. , 2011) for each LD.  To compare the monthly entropy of the LS per country, mean entropy was calculated and normalized across the months for each country.  We observed the highest entropy (information content) for India, UK, USA and China in the months of February-2021, July-2020, August-2020, and January-2021 respectively (Fig 1D).  Interestingly, we observed high entropy for 4 months from August, 2020 to November, 2020 in the USA (Fig 1D).  This highlights that the spike protein latent space representation learned by the Strainflow model could be used as a proxy to capture the spatiotemporal entropy or diversity in the emerging SARS-CoV-2 strains across different countries.  Sequence-level embeddings were obtained from the codon embeddings and investigated for the presence of genomically meaningful characteristics.  The phylogenetic tree derived from the embeddings for the United Kingdom (Fig 2A) shows two clear temporally split clusters for 2020 and 2021 sequences, which may be indicative of different strains in these time periods.  The temporality of the collected sequences was found to be preserved in the two clusters, although the model was trained only on genome sequences.  (A) Dendrogram for strains from the U. K. : Cluster 1 (blue) contains strains from the period Oct 2020 - Dec 2020, while Cluster 2 (orange) contains strains collected between Jan 2021 - Mar 2021.  (B) Dendrogram for 16 countries across the globe: Chinese, Australian, Mexican, and England strains form tight clusters (marked in green, purple, and magenta), while strains from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India are dispersed with other countries.  The phylogenetic tree with globally collected sequences (Fig 2B) demonstrates that geospatial information is also preserved in the sequence embeddings.  Strains from countries like China, Australia, Mexico, and England were found in distinct clusters, while those from Italy, France, Brazil, Japan, Canada, USA, Scotland, and India were found to be dispersed with other countries.  This behavior is expected because viral strains spread from one country to another, and our samples had different collection dates for each country.  Overall, Strainflow captures the temporal emergence of strains and geographic information in a country-specific manner.',\n",
              " ' Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time.  We attributed these changes to mutations in the spike protein of the genome sequence.  This hypothesis was based on the fact that the embedding of a word (codon) represents the context it occurs in.  A mutation results in a codon that is known to be found in a different context.  This alters the semantic sense of the sequence, and this modification of meaning is reflected as a change in values across different latent dimensions.  The changed values cause a change in the degree of disorder or the entropy of the latent space.  To compare different geographical regions, the sum of sample entropy was computed for each latent dimension across all the months.  This revealed that certain geospatial regions such as France and Germany (Fig 3A) and USA and Canada (Fig 3B) have similar total entropies across the latent dimensions, indicating that these regions have been accumulating similar changes.  Country pairs (A) - France and Germany, (B) USA and Canada show a similar distribution of total sample entropy across dimensions, while each pair differs from the other.  We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries.  Detrended cross-correlation coefficient was calculated at different lag values, which revealed that entropy dimensions have a leading relationship with new cases (Fig 4A, 4B).  This suggests that the genome sequence data in a given month can be used to predict new cases in subsequent months.  A lead period of two months was chosen and Boruta algorithm was employed to assign feature importance scores to different dimensions, which revealed that dimension 32 is the most significant predictor of new cases (Fig 4C).  Significant dimensions from Boruta analysis were termed Dimensions of Concern (DoCs).  Random forest based regression modeling on the DoCs achieved a total R-squared of 73% on the test set.  The predicted cases were found to be highly correlated with the actual cases (Table 1), which suggests that our model can indicate the directional change of cases for different countries.  Further, the predicted relative change in cases between successive months was found to be correlated to the actual relative changes (Table S1), which suggests that our model can also indicate the magnitude of change that we expect to observe in the cases.  (A) Detrended Cross-correlation coefficient values for different lags between Entropy dimension 32 and new cases for USA.  High values are observed for a lead of 1 and 2 months.  (B) Line plot for Sample Entropy dimension 32 and monthly new cases for USA, indicating that the entropy in dimension 32 has a leading relationship with the cases.  (C) Feature importance scores from the Boruta algorithm for predicting cases in the month following the next month.',\n",
              " ' Our model can be therefore used to predict the COVID-19 caseloads in several countries.  Both USA (Fig 5A) and Japan (Fig 5C) show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries.  Our model predicts new caseloads with a two-month lead time, which strongly predicts a spike in new cases both in USA (Fig 5B) and Japan (Fig 5D) in the months of July and August, 2021.  Therefore our model may be used as an epidemiological early warning system to predict new caseloads.  (A) Line plot showing the Entropy values of DoCs and new COVID-19 cases for the USA.  (B) Actual and predicted cases based on the entropy values of DoCs for the USA.  The model predicts a rise in cases for July and August 2021.  (C) Entropy of DoCs and new cases for Japan.  (D) Actual and predicted cases for Japan.  A spike in cases is predicted for July and August 2021.  We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension (Table S2).  The intuition behind this idea is that the codons with high weights in a given dimension, when mutated in the viral sequence, are likely to cause a significant change in the entropy of the associated dimensions.  Therefore each DoC can be linked to codons, which can further be mapped to Variants of concern (VOCs) and Variants of Interest (VOIs) (Table S3).  Despite the fact that our model cannot directly capture the SARS-CoV-2 variants, it was observed that dimension-32 captures the CTG, CGG codons (ranks 5 and 8 respectively), known to occur in the T19R variant.  Similarly, dimension 3 captures three codons (ACG, CGG, CAC) that are associated with multiple variants such as K417T, L452R, and D1118H, causing increased infectivity, pathogenicity, and spread.  Dimension 30 captures codons CAT and CAC associated with Δ69 and D1118H respectively which are linked to B. 1. 1. 7 lineage.  Codon weights (Table S2) of a given DoC provide an opportunity to associate with specific Variants of concern (VOCs) and Variants of Interest (VOIs); and to predict emerging SARS-CoV-2 variants and take preventive measures in advance.  Distinct dimensions capture country-specific changes and may be surveilled to monitor the spread of the pandemic.  This approach was back-validated with several real-world examples.  For instance, dimension 32 captures the codons CGG (R) and CAC (R), which are found in B. 1. 429 lineage (L425R mutation).  Dimension 3 captures CGG which is seen in L452R (associated with lineage B. 1. 617. 1), which was first observed in India in December 2020 and was found to have increased infectivity and transmissibility.',\n",
              " ' To investigate the potential of DoCs to track the spread of SARS-CoV-2, we used the codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs.  We selected DoCs (dimension 3, 4, 12, 13, 15, 16, 25, 28, 30, 32) with high absolute weights for the codons related to the delta variants (Fig 6A).  The entropy of the DoCs was contrasted with the caseloads in England (Fig 6B), India (Fig 6C), and USA (Fig 6D).  Overall, the temporal tracking of the entropy in Dimensions of Concern (DoCs) may be used as a surrogate to track the spread of various SAR-CoV-2 variants.  (A) Heat map showing the absolute latent space weights of the codons in the DoCs used for modeling.  Line plots showing the entropies of DoCs and cases in countries, (B) England, (C) India, and (D) USA.  The entropies of the DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant in these countries.  We have implemented an approach for analyzing the emerging strains based on the latent space of spike protein coding nucleotide sequences.  We chose the nucleotide sequences instead of proteins in order to capture and track the variations that may not have immediate functional consequences.  Our approach has two main underlying tenets: (i) long-range interactions are known to modulate the functional (Mugnai et al. , 2020) interaction between receptor binding domain and ACE2 receptors, hence may be captured in the NLP models that capture 3-mer changes and context, and (ii) latent dimensions may be differentially correlated with indicators of spread, thus providing a data-driven handle for tracking and predicting variants of concern and variants of interest.  The pipeline takes advantage of temporal changes in the semantics of mutating sequences.  Preservation of phylogenetic structure based upon the similarity matrix obtained using the embeddings validated that the latent dimensions capture spatio-temporal information.  Analyzing the dynamic patterns and underlying correlations in the 30,000 base pair long sequence of SARS-CoV-2 (Shishir et al. , 2021) is important to highlight the mechanistic understanding of mutations.  SARS-CoV-2 seems to show a particularly high frequency of recombinations arising due to the absence of a proof-reading mechanism and sequence diversity, which calls for urgency in studying its transmission pattern (Mandal et al. , 2021)(Rouchka et al. , 2020).  Therefore predicting mutations in the spike protein, which binds to ACE2 receptors can help us estimate the spread of disease and the efficacy of therapeutic treatments and vaccines (Srivastava et al. , 2021)(Li et al. , 2020).  Entropy is a measure of the disorder of a system.  We hypothesized that mutations increase the chaotic dynamics in the latent space of spike genes.',\n",
              " ' To calculate entropy, we used the accelerated versions of the Approximate Entropy and Sample Entropy algorithms, called Fast Approximate Entropy and Fast Sample Entropy (Pan et al. , 2011).  Both algorithms aim to quantify how often different patterns of data are found in a time series.  Fast Approximate Entropy, however, is a biased statistic and depends on the length of the series.  Since we could have different counts of genome sequences collected each month, we preferred Sample Entropy, which is independent of the length of the series.  Entropy values were calculated for each latent dimension in each month.  Thereafter, Detrended Cross-Correlation Analysis (DCCA) (Podobnik and Stanley, 2007) was performed between the entropy dimensions and the new cases.  DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series.  High cross-correlation for different lead periods revealed that the entropy values in a given month could be used to predict the new cases in different countries in subsequent months.  Different countries had different lead times at which the highest cross-correlation was observed between the entropy dimensions and the cases, ranging from 1-6 months.  Overall, a lead time of two months was chosen to model the new cases.  A similar analysis was done with daily values of entropy and new cases.  Entropy was calculated in rolling windows, and cross-correlation analysis was performed between entropy and new cases at different lead periods.  Although the cross-correlation values were found to be significant, the values were low and ranged between -0. 1 to 0. 1.  Therefore, we decided to use the monthly entropy values for the modeling exercise.  We also experimented with “blips” as a feature to predict the new cases.  Blips are sudden changes in the values of the latent dimensions.  These changes may be caused by a mutation, which changes the words (codons) in a given genome sequence.  This hypothesis was validated in simulation experiments in synthetic datasets.  Each dimension of the spike gene embeddings for a country was analyzed for the presence of temporal anomalies.  Countries having a minimum of 20 samples in any given month were selected and the same number of records (minimum samples in any given month for that country) were sampled without replacement from each month.  These records were used to define control limits of ±1 standard deviation from the mean value for each dimension, and all values in the full dataset outside those limits were categorized as ‘Blip’ points.  Blip counts in each month were normalized by calculating the number of blips per sample collected in a month for each dimension (normalized blips).  The embedding dimensions were then compared in terms of the total normalized blips for each country to observe the significant dimensions and dis(similarity) in trends among different countries.  Cumulative counts of normalized blips were analyzed to understand the temporal accumulation of blips in each dimension.',\n",
              " ' Similar to entropy, blips were found to have a leading relationship with the cases.  However, regression modeling results with sample entropy were found to be better.  To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data.  With sample entropy, the R-squared value achieved was 73%, while with approximate entropy, the value was only 10%.  Therefore the model trained on sample entropy was selected.  The predictions from the model were found to be highly correlated with the actual cases, indicating that our model can be used for preemptive warning signals for the rise in cases in different countries.  Further, the actual and the predicted difference in the number of cases in consecutive months was found to be correlated, which suggests that the relative change in the cases in consecutive months predicted by our model is linked to the relative change in the number of cases.  Overall, we recommend that our model be used to predict dangerous trends and not the actual number of cases.  Further, the mapping from latent dimensions to Variants of Concern (VOCs) and Variants of Interest (VOIs) may help us track the country-specific spread of different variants.  Prediction models can be used for training the genomic sequence for predicting infection severity based on Co-associations between the SNPs of Co-morbid Diseases and COVID-19 (Wang et al. , 2020).  The machine learning models can also be trained on genomic sequences for COVID-19 classification (Arslan, 2021).  Although the variance explained by our model is low, however, we were able to compute the variability associated with spike protein mutations.  So our method showed a potential way to estimate the new cases variability associated with spike protein mutations.  Our methods can be incorporated with the epidemic projections model to better predict the epidemic trajectories.  The latent dimensions may further be employed to predict the clinical consequences of emerging strains.  The currently available vaccines are intended for early SARS-CoV-2 strains, but with new emerging variants, immune responses triggered by these vaccines may be weaker and short-lived.  As seen in the devastating second wave of the pandemic in India, newer SARS-CoV-2 variants have acquired an increased pathogenic potential resulting in rapid clinical progression and overwhelmed health systems.  Mitigating such events in the future will require stronger surveillance systems in place.  Our study offers a promising solution in this direction and lays the foundation for proactive genomic surveillance of COVID-19.  Our approach of codon (3-mers) embeddings does not indicate the position where the codon change may have happened in the spike gene.  This is because low-dimensional embeddings do not preserve the positional encoding of words.  However, we are investigating advanced approaches such as complex-valued word embeddings with positional encodings (Wang et al. , 2019) and transformer models such as BERT (Wolf et al. , 2020)(Lee et al. , 2020) to overcome our current limitation.',\n",
              " ' The latter are considered expensive and data-hungry models and it will remain to be evaluated if the gain of positional information may be countered by the loss of prediction accuracy for forecasting new cases in the future.  However, we believe that the availability of sequences for a wide variety of viral pathogens presents an exciting opportunity to train data-hungry models that may be able to transfer insights across pathogens and yet remain interpretable.  Further, our Strainflow model is trained only on the spike gene of the viral genome, which does not represent the complete variation spectrum of the virus.  To mitigate this shortcoming, we are currently developing a genome-level Strainflow model (version 2. 0) for SARS-CoV-2.  Another limitation of our study is the relatively small number of samples that were used to construct the supervised predictive model for case prediction.  As more data becomes available in subsequent months, we can produce more confident case predictions.  Nonetheless, the unsupervised embeddings and temporal cross-correlations were learned upon the full datasets and these presented clear patterns in DoCs and significant cross-correlations with caseloads.  However, it is important for our models to receive at least 30 samples per month for getting reliable prediction results for the future months.  This also underscores the need for a more reliable and agile approach to deposit country-level datasets on repositories such as GISAID.  We make an appeal to the countries to facilitate the sharing of such data in order to be prepared for any future waves of the current pandemic and for preventing the new emergence of strains.  We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance.  The dataset was downloaded from GISAID EpiCoV (Shu and McCauley, 2017) (April 8, 2021 release).  0. 63 million genome sequences with high nucleotide completeness, coverage, complete temporal information, and presence of less than 5% non-identified nucleotide bases (N) were downloaded.  The sequences included 63 countries, including India, United Kingdom, USA, Australia, New Zealand, Germany, Russia, Italy, France, Mexico, Canada, China, Japan, Pakistan, Bangladesh, Iran, Iraq, the continent of South America, and Africa.  Duplicate samples were removed, and whole genome sequences were parsed using CoV-Seq (Liu et al. , 2020) to extract nucleotide sequences corresponding to each of the 12 CDS.  Accession IDs that did not cover 12 coding regions were discarded, yielding 0. 31 million high-quality SARS-CoV-2 genome sequences for language modelling.  We downloaded country-wise COVID-19 data for new cases from a publicly available repository maintained by Johns Hopkins University Center for Systems Science and Engineering (JHU CSSE).  We downloaded around 0. 6 million genome sequences submitted to GISAID from April 2021 to June 2021.  We have used 0. 36 million sequences to predict their latent space using the Strainflow model for the months of April 2021 to June 2021.',\n",
              " ' In our Strainflow model, we have adopted a word2vec model (Mikolov et al. , 2013).  A low dimensional representation for the genome sequences was learned using the word2vec model.  Non-overlapping sequences of 3-mers (codons) were considered as words for training the word2vec model, implemented in Gensim (Radim Rehurek, 2010).  The word2vec model was trained using the skip-gram algorithm, with a fixed window size of twenty and vector size of thirty-six.  For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise.  The mean was calculated by summing across all the k-mers over each dimension and then dividing it by the total number of codons present in the sequence.  For selecting the vector size of the word2vec model, we calculated PIP (Pairwise Inner Product) loss (Yin and Shen, 2018).  PIP loss is a metric used for calculating the dissimilarity between two word embedding matrices.  For the embedding matrix of strains (E), the PIP matrix is defined as the dot product of the embedding matrix with its transpose (E.  E T).  The PIP loss between two embedding matrices is defined as the norm of the difference between their PIP matrices. \\n\\n\\n Various word2vec models were trained on the dataset with different vector sizes varying in multiples of three.  Based on the PIP loss calculations, we found out that word embeddings with 36 dimensions showed a differential dent in the curve (change in straight line), due to which we selected this to be the dimension of the word embeddings (Fig.  S1).  To evaluate the phylogenetic properties based on the latent dimensions of the spike gene, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions.  The pairwise distance was further used for hierarchical clustering using the ‘hclust’ function in R statistical programming language.  This analysis was performed using 400 random sequences of spike genes from 16 countries.  The visualization of the phylogenetic tree derived based on the latent dimensions was done using ‘iTOL’ (Letunic) software (Fig 2).  To quantify the properties of latent dimensions, we have used a well-known information theory based algorithm suitable for time series datasets, called ‘Fast Sample Entropy’ (Tomčala, 2020).  To compute Fast Sample Entropy, we have used the ‘FastSampEn’ function in the ‘TSEntropies’ package in R (Tomcala, 2018).  Fast Sample Entropy can be computed as follows. \\n\\n\\nwhere,\\n\\n\\n Si,mm is a set of sub-sequences of length m belonging to the i-th neighborhood, and Nm is the number of these neighborhoods.  In our case, ‘x’ is the latent dimension of the spike genes of the SARS-CoV-2 strains per month for a given country with default values of ‘m’ and ‘r’.  Entropy was computed for each latent dimension on a monthly basis for each country.  The minimum number of SARS-CoV-2 spike genes was 20.',\n",
              " ' To compare across months, we used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country (Fig 1D).  To compare the entropy of the latent dimensions among countries, we used the total entropy of the country for each dimension and visualized it with line graphs (Fig 3).  To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used the Detrended Cross Correlation Analysis (Prass and Pumi, 2019).  Here, DCCA captures the long-range cross correlation between time series (entropy of the months and caseloads for a given country).  We tested both time series for stationarity using Augmented Dickey-Fuller (ADF) test (Mushtaq, 2011).  The ADF test was implemented using the function ‘adf. test’ available in the ‘tseries’ (2020a) package in R.  Due to the non-stationary distribution of the estimated entropies and the caseloads for a given country, we used the ‘DCCA’ R package (2020b).  Cross-correlation was calculated between the entropy dimensions at time t+h and new cases at time t, where h = 0, ±1, ±2, ±3 … ±10.  Dimensions of Concern (DoCs) were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis and predictive of new case count in the respective countries.  Country-wise monthly total new cases data was taken at the end of each month.  Total new cases data for each month was merged to monthly entropy dimensions data from March, 2020 to June, 2021.  We used a regression based machine learning approach called ‘Boruta’ (Kursa and Rudnicki, 2010), a wrapper algorithm around random forest algorithm (Kursa et al. , 2010) to select the most relevant entropy dimensions for the prediction of subsequent two months’ new cases.  We used the default parameters with the modification of the maximum runs as 1000.  We selected the confirmed entropy dimensions as the most relevant features for the prediction of new cases.  To predict the new cases in the next to next months, we used a regression based random forest model (Breiman 2001) using the most relevant DoCs based on Boruta.  The model training was performed using the data from March, 2020 to April, 2021; and the fitted model was tested on data from May, 2021 to June, 2021.  The regression modeling was performed using 1000 decision trees using the ‘randomForest’ package in R (A.  Liaw et al. , 2002).  To find the top codons associated with the DoCs, we extracted the absolute weights of each codon for a given DoC.  The top 10 codons having the highest absolute weights (contribution) were identified corresponding to each of the DoCs to compare our DoCs with SARS-CoV-2 variants.',\n",
              " \" We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (CDC, 2021); and a list of codons associated with VOIs and VOCs was curated (Table S4) (Lopez-Rincon et al. ; paola, 2021; Peacock et al. , 2021; Srivastava et al. , 2021).  The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant (CDC, 2021) This work was supported by the Delhi Cluster-Delhi Research Implementation and Innovation (DRIIV) Project supported by the Principal Scientific Advisor Office, Prn. SA/Delhi/ Hub/2018(C) and the Center of Excellence in Healthcare supported by Delhi Knowledge Development Foundation (DKDF) at IIIT-Delhi.  We also thank Dr.  Chitra Pattabiraman (NIMHANS) for her valuable inputs on viral sequences, and Harleen Kaur, Nishkarsh Saxena, and Anjali for their contribution to the dashboard visualizations.  In this revised version, we have implemented fast tSNE for qualitative inspection of the latent dimensions (LD) of 0. 9 million SARS-CoV-2 spike genes.  Quantitative analyses of the LDs were performed using the fast sample entropy method.  Also, we have used fast sample entropy instead of 'blips' to model COVID-19 caseloads.  View the discussion thread.  Supplementary Material  Thank you for your interest in spreading the word about bioRxiv.  NOTE: Your email address is requested solely to identify you as the sender of this article. \"]"
            ]
          },
          "metadata": {},
          "execution_count": 49
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(chunks) #number of 500-word chunks"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RadCBQMMg-SI",
        "outputId": "3fcbdd63-6a07-4d0a-81e9-a017e6927965"
      },
      "execution_count": 50,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "12"
            ]
          },
          "metadata": {},
          "execution_count": 50
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "chunks[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 122
        },
        "id": "cw0DZLWyhAk8",
        "outputId": "da6108e1-9bab-47f2-dff7-ba09e09578cd"
      },
      "execution_count": 51,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Genomic Surveillance of COVID-19 Variants with Language Models and Machine Learning The global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity.  Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge.  Here, we developed Strainflow, to learn the latent dimensions of 0. 9 million high-quality SARS-CoV-2 genome sequences, and used machine learning algorithms to predict upcoming caseloads of SARS-CoV-2.  In our Strainflow model, SARS-CoV-2 genome sequences were treated as documents, and codons as words to learn unsupervised codon embeddings (latent dimensions).  We discovered that codon-level changes lead to a change in the entropy of the latent dimensions.  We used a machine learning algorithm to find the most relevant latent dimensions called Dimensions of Concern (DoCs) of SARS-CoV-2 spike genes, and demonstrate their potential to provide a lead time for predicting new caseloads in several countries.  The DoCs capture codons associated with global Variants of Concern (VOCs) and Variants of Interest (VOIs), and may be surveilled to predict country-specific emergence and spread of SARS-CoV-2 variants.  Highlights We developed a genomic surveillance model for SARS-CoV-2 genome sequences, Strainflow, where sequences were treated as documents with words (codons) to learn the codon context of 0. 9 million spike genes using the skip-gram algorithm.  Time series analysis of the information content (Entropy) of the latent dimensions learned by Strainflow shows a leading relationship with the monthly COVID-19 cases for seven countries (e. g. , USA, Japan, India, and others).  Machine Learning modeling of the entropy of the latent dimensions helped us to develop an epidemiological early warning system for the COVID-19 caseloads.  The top codons associated with the most relevant latent dimensions (DoCs) were linked to SARS-CoV-2 variants, and these DoCs may be used as a surrogate to track the country-specific spread of the variants.  COVID-19 is reported to have claimed 4. 37 million lives as of Aug 17, 2021 (WHO, 2021).  A large number of these deaths are attributed to unexpected surges in infections caused by new strains of SARS-CoV-2, prompting international health organizations such as the CDC and WHO to declare these as variants of concerns (Khan et al. , 2020).  Such emergence of new variants can seriously undermine the efficacy of global vaccination programs, the need for future booster doses, or cause multiple reinfection waves as new strains escape previously developed antibodies.  Several initiatives have focused on providing high-quality tracking information for the strains and lineages as these emerge (Hadfield et al. , 2018).  However, early prediction of emerging variants through genomic signals remains an open challenge.  Although domain-based, expert-reasoning approaches are the mainstay of our understanding, which often yield retrospective insight rather than proactive predictions.'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 51
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "#Summarizing each chunk"
      ],
      "metadata": {
        "id": "BQxaxKTzhuaG"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "chunk_summary = summarizer(chunks, max_length=200, min_length=50, do_sample=False)\n",
        "chunk_summary"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "b2KtW9yehI8W",
        "outputId": "e9a7c3b1-f4a9-44ba-8057-81d7bcb8efe8"
      },
      "execution_count": 52,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[{'summary_text': ' Global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity . Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge .'},\n",
              " {'summary_text': ' Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences . The approach is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance . The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 .'},\n",
              " {'summary_text': ' To investigate the information content in the latent dimensions or space (LD or LS) of the Strainflow model, we performed qualitative and quantitative analysis on 0. 9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021 . Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (Linderman et al. , 2019) We compared the world with four countries (India, UK, USA, China) from September 2020 to June 2021 .'},\n",
              " {'summary_text': ' Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time . We attributed these changes to mutations in the spike protein of the genome sequence . We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries . Certain geospatial regions such as France and Germany have similar total entropies across the latent dimensions .'},\n",
              " {'summary_text': ' Both USA (Fig 5A) and Japan show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries . We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension .'},\n",
              " {'summary_text': ' We used codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs . We selected DoCs with high absolute weights for the codons related to the delta variants . Entropies of DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant .'},\n",
              " {'summary_text': ' Entropy values were calculated for each latent dimension in each month . Detrended Cross-Correlation Analysis (DCCA) was performed between the entropy dimensions and the new cases . DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series . Countries having a minimum of 20 samples in any given month were selected .'},\n",
              " {'summary_text': ' To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data . The predictions from the model were found to be highly correlated with the actual cases . This suggests that our model can be used for preemptive warning signals for the rise in cases in different countries .'},\n",
              " {'summary_text': ' We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance . The dataset was downloaded from GISAID EpiCoV (Shu and McCauley, 2017) (April 8, 2021 release)'},\n",
              " {'summary_text': ' In our Strainflow model, we have adopted a word2vec model (Mikolov et al. , 2013) A low dimensional representation for the genome sequences was learned using the word1vec model . For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise . To evaluate the phylogenetic properties based on the latent dimensions of the spike genes, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions .'},\n",
              " {'summary_text': ' To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used Detrended Cross Correlation Analysis (Prass and Pumi, 2019) We used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country . Dimensions of Concern (DoCs) were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis .'},\n",
              " {'summary_text': ' We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (CDC, 2021) A list of codons associated with VOIs and VOCs was curated (Table S4) The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant . Quantitative analyses of the LDs were performed using the fast sample entropy method .'}]"
            ]
          },
          "metadata": {},
          "execution_count": 52
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "chunk_summary[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6WIQPQ5-iIb8",
        "outputId": "10c9ae3a-e0ca-4929-c165-13ea5404ccab"
      },
      "execution_count": 53,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'summary_text': ' Global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity . Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge .'}"
            ]
          },
          "metadata": {},
          "execution_count": 53
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "' '.join([summ['summary_text'] for summ in chunk_summary])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 122
        },
        "id": "xme3JQ4RiIjD",
        "outputId": "75ecc612-94ac-4fc1-db75-f52835b593e3"
      },
      "execution_count": 54,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "' Global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity . Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge .  Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences . The approach is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance . The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 .  To investigate the information content in the latent dimensions or space (LD or LS) of the Strainflow model, we performed qualitative and quantitative analysis on 0. 9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021 . Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (Linderman et al. , 2019) We compared the world with four countries (India, UK, USA, China) from September 2020 to June 2021 .  Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time . We attributed these changes to mutations in the spike protein of the genome sequence . We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries . Certain geospatial regions such as France and Germany have similar total entropies across the latent dimensions .  Both USA (Fig 5A) and Japan show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries . We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension .  We used codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs . We selected DoCs with high absolute weights for the codons related to the delta variants . Entropies of DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant .  Entropy values were calculated for each latent dimension in each month . Detrended Cross-Correlation Analysis (DCCA) was performed between the entropy dimensions and the new cases . DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series . Countries having a minimum of 20 samples in any given month were selected .  To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data . The predictions from the model were found to be highly correlated with the actual cases . This suggests that our model can be used for preemptive warning signals for the rise in cases in different countries .  We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance . The dataset was downloaded from GISAID EpiCoV (Shu and McCauley, 2017) (April 8, 2021 release)  In our Strainflow model, we have adopted a word2vec model (Mikolov et al. , 2013) A low dimensional representation for the genome sequences was learned using the word1vec model . For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise . To evaluate the phylogenetic properties based on the latent dimensions of the spike genes, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions .  To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used Detrended Cross Correlation Analysis (Prass and Pumi, 2019) We used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country . Dimensions of Concern (DoCs) were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis .  We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (CDC, 2021) A list of codons associated with VOIs and VOCs was curated (Table S4) The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant . Quantitative analyses of the LDs were performed using the fast sample entropy method .'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 54
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "text = ' '.join([summ['summary_text'] for summ in chunk_summary])\n",
        "text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 487
        },
        "id": "9ouxFVsMiItX",
        "outputId": "08188495-ec99-4456-f60b-e8e2361b9db1"
      },
      "execution_count": 55,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "' Global efforts to control COVID-19 are threatened by the rapid emergence of novel SARS-CoV-2 variants that may display undesirable characteristics such as immune escape or increased pathogenicity . Early prediction of emerging strains could be vital to pandemic preparedness but remains an open challenge .  Strainflow is rooted in language models for generating sequence embeddings that have recently shown promise for capturing biological insights from DNA sequences . The approach is extensible to global threats in pathogen surveillance such as emerging infections, pandemics, and antibiotic resistance . The global tSNE highlights dynamic emerging patterns derived from LR of spike genes of SARS-CoV-2 .  To investigate the information content in the latent dimensions or space (LD or LS) of the Strainflow model, we performed qualitative and quantitative analysis on 0. 9 million SARS-CoV-2 spike genes collected from December, 2019 to June, 2021 . Qualitative analysis was performed by performing dimensionality reduction with a fast tSNE method called Flt-SNE (Linderman et al. , 2019) We compared the world with four countries (India, UK, USA, China) from September 2020 to June 2021 .  Analysis of the sequence embeddings for different time periods revealed that the value of each latent dimension changes with time . We attributed these changes to mutations in the spike protein of the genome sequence . We then attempted to decipher the relationship between monthly sample entropy and monthly new COVID-19 cases in different countries . Certain geospatial regions such as France and Germany have similar total entropies across the latent dimensions .  Both USA (Fig 5A) and Japan show an increase in sample entropy in all the DoCs across the time period April - June 2021, concurrent with the respective spreads in these countries . We further assessed the potential link of DoCs with SARS-COV-2 by extracting the top 10 contributing codons and their associated weights for each dimension .  We used codon level information of the SARS-CoV-2 delta variant for the spike gene and extracted the weights of these codons specific to the DoCs . We selected DoCs with high absolute weights for the codons related to the delta variants . Entropies of DoCs show an increasing trend in the months April - June 2021 for India and USA, indicating a possible surge in the delta variant .  Entropy values were calculated for each latent dimension in each month . Detrended Cross-Correlation Analysis (DCCA) was performed between the entropy dimensions and the new cases . DCCA is a modification of the standard cross-correlation analysis for finding relationships between non-stationary time series . Countries having a minimum of 20 samples in any given month were selected .  To predict new COVID-19 cases, a Random Forest regression model was trained on the monthly entropy data . The predictions from the model were found to be highly correlated with the actual cases . This suggests that our model can be used for preemptive warning signals for the rise in cases in different countries .  We believe our study is an instance of the new paradigm of pathogen surveillance using a novel language modeling approach that is potentially scalable to infectious disease surveillance and antimicrobial resistance . The dataset was downloaded from GISAID EpiCoV (Shu and McCauley, 2017) (April 8, 2021 release)  In our Strainflow model, we have adopted a word2vec model (Mikolov et al. , 2013) A low dimensional representation for the genome sequences was learned using the word1vec model . For generating a consensus embedding for a particular strain, genomic sequences were represented by taking the mean of each codon occurring in the sequence dimension-wise . To evaluate the phylogenetic properties based on the latent dimensions of the spike genes, we computed the cosine distances among spike genes of SARS-CoV-2 with the 36 latent dimensions .  To investigate the information content (entropy) of the latent dimensions with the new cases observed for COVID-19, we used Detrended Cross Correlation Analysis (Prass and Pumi, 2019) We used average entropy derived from 36 latent dimensions, followed by normalization using all the monthly entropies for a given country . Dimensions of Concern (DoCs) were defined as those dimensions which were statistically significant, leading the case count in cross-correlation analysis .  We collected the SARS-CoV-2 variants and their associated genetic variations at the codon level linked to the spike gene (CDC, 2021) A list of codons associated with VOIs and VOCs was curated (Table S4) The curated list is based on the CDC guidelines, and we are consistent with their definition of lineage and variant . Quantitative analyses of the LDs were performed using the fast sample entropy method .'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 55
        }
      ]
    }
  ]
}